{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "56a63c23-3e1f-4b7c-bc15-3e385781e080",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# Liftover and genotype data creation\n",
    "\n",
    "Create genotype array data and recreate all the genotype files for White European, Asian and African to be used in the burden test of different traits in REGENIE"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39463091-9e2b-4dcf-a292-9f8065adc113",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "Original file \n",
    "`~/UKBiobank_Yale_transfer/pleiotropy_geneticfiles/UKB_originalgenotypefilesdownloaded083019/UKB_genotypedatadownloaded083019.bed`\n",
    "\n",
    "Starting file for the liftover. After doing the first part of variant_qc and sample_qc in this notebook(project/UKBB_GWAS_dev/workflow/UKBB_QC_genoarray.ipynb)\n",
    "`~/UKBiobank/genotype_files_processed/082621_sampleqc_call90/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.082621_sampleqc_call90.filtered.bed`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d9177fca-0624-43bd-aac2-11ecc8ea02b9",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "liftover_sos=~/project/bioworkflows/GWAS/liftover.ipynb\n",
    "input_file=/mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/082621_sampleqc_call90/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.082621_sampleqc_call90.filtered.bim \n",
    "output_file=UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples \n",
    "cwd=/mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/\n",
    "fr=hg19 \n",
    "to=hg38 \n",
    "tpl_file=~/project/bioworkflows/admin/csg.yml\n",
    "liftover_sbatch=/mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/liftover_hg19tohg38_$(date +\"%Y-%m-%d\").sbatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "9cb15b1d-9c2b-4edb-9a1e-1c8b4c6b4bd6",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/liftover_hg19tohg38_2023-01-20.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wc6ae0c5fa24262a9) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "liftover_args=\"\"\"default\n",
    "    --cwd $cwd\n",
    "    --input_file $input_file\n",
    "    --output_file $output_file\n",
    "    --fr $fr \n",
    "    --to $to\n",
    "    --remove-missing\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $liftover_sos \\\n",
    "    --to-script $liftover_sbatch \\\n",
    "    --args \"$liftover_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5762a49c-1a2e-4f8c-a276-cd44261b7d4c",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "\n",
    "**I had to run by login into the node**\n",
    "\n",
    "```\n",
    "qrsh -l h_vmem=100G -l h_rt=200:00:00 -q csg2.q -l t_pri\n",
    "module load Singularity/3.5.3\n",
    " sos run /home/dmc2245/project/bioworkflows/GWAS/liftover.ipynb     default    --cwd /mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/    --input_file /mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/082621_sampleqc_call90/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.082621_sampleqc_call90.filtered.bim    --output_file UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples    --fr hg19     --to hg38    --remove-missing\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f883b24-11c6-42a4-a153-b881a314bac2",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "I've got an error in plink \n",
    "```\n",
    "Error:\n",
    "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.bim\n",
    "has a split chromosome. Use --make-pgen + --sort-vars to remedy this.\n",
    "```\n",
    "\n",
    "`plink --bfile [old name] --make-bed --out [new name]`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "52afe709-3e3b-40b2-b72d-b66adbf727ca",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Sort the bed file to eliminate split chromosomes error"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91199340-07e8-4c19-aa5c-04a67154512f",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "```\n",
    "qrsh -l h_vmem=100G -l h_rt=200:00:00 -q csg2.q -l t_pri\n",
    "module load PLINK/1.9.10\n",
    "plink --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples \\\n",
    "--make-bed --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "ddecd583-d8cf-4776-8c21-2859065bd92b",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/home/dmc2245/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array.sh\" target=\"_blank\">/home/dmc2245/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /home/dmc2245/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save -f ~/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=200:00:00\n",
    "#$ -l h_vmem=100G\n",
    "#$ -N plink_sorting_hg38_geno_array\n",
    "#$ -o /home/dmc2245/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array-$JOB_ID.out\n",
    "#$ -e /home/dmc2245/project/UKBB_GWAS_dev/output/plink_sorting_hg38_geno_array-$JOB_ID.err\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "module load PLINK/1.9\n",
    "plink --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples \\\n",
    "--make-bed --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9de6eab-acd9-455a-bcc9-4d8bc672c377",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Generate the phenotype files for the different ancestries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "45444572-6bf4-4a78-8ea9-7accc0e67133",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9a25c1d2-bde0-493c-9a6e-acc493e89946",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Read-in database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a38bf6c1-3a49-4194-b0fe-3bc1e3ebfaf1",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# collect the necessary column names of the database for our analysis\n",
    "\n",
    "with open(\"/mnt/mfs/statgen/UKBiobank/data/ukbb_databases/ukb47922_updatedAug2021/ukb47922.tab\") as fp:\n",
    "    line = fp.readline() # header\n",
    "    header = line.split(\"\\t\")\n",
    "    \n",
    "    indiv = [\"f.eid\"]\n",
    "    reported_sex = [\"f.31.0.0\"]\n",
    "    genetic_sex = [\"f.22001.0.0\"]\n",
    "    white_british = [\"f.22006.0.0\"]\n",
    "    ethnicity = [col.strip('\"') for col in header if \"f.21000.\" in col]\n",
    "    year_of_birth = [col.strip('\"') for col in header if \"f.34.\" in col]\n",
    "    month_of_birth = [col.strip('\"') for col in header if \"f.52.\" in col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "32baadce-4f76-4376-b3fb-c8c818a3ac98",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "combined_cols = indiv  + ethnicity + reported_sex + genetic_sex +  year_of_birth + month_of_birth + white_british"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "71d33e19-dbbd-4cee-bb8b-b89bf87f796b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-01-23 10:12:13.311799\n"
     ]
    }
   ],
   "source": [
    "print(datetime.now())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "65664d61-3255-4e46-ace0-c848b7972a44",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1</td>\n",
       "      <td>1954</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1</td>\n",
       "      <td>1944</td>\n",
       "      <td>5</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502456</th>\n",
       "      <td>6025409</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502457</th>\n",
       "      <td>6025411</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502458</th>\n",
       "      <td>6025425</td>\n",
       "      <td>0</td>\n",
       "      <td>1963</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502459</th>\n",
       "      <td>6025438</td>\n",
       "      <td>1</td>\n",
       "      <td>1952</td>\n",
       "      <td>9</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502460</th>\n",
       "      <td>6025442</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>502461 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          f.eid f.31.0.0 f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0  \\\n",
       "0       1000019        0     1960       11        1001        <NA>   \n",
       "1       1000022        1     1954        8        1001        <NA>   \n",
       "2       1000035        1     1944        5        1001        <NA>   \n",
       "3       1000046        0     1946        3        1001        <NA>   \n",
       "4       1000054        0     1942        1        1001        <NA>   \n",
       "...         ...      ...      ...      ...         ...         ...   \n",
       "502456  6025409        0     1946       11        1001        1001   \n",
       "502457  6025411        0     1960       11        1001        <NA>   \n",
       "502458  6025425        0     1963        8        1001        <NA>   \n",
       "502459  6025438        1     1952        9        1001        <NA>   \n",
       "502460  6025442     <NA>     <NA>     <NA>        <NA>        <NA>   \n",
       "\n",
       "       f.21000.2.0 f.22001.0.0 f.22006.0.0  \n",
       "0             <NA>           0           1  \n",
       "1             <NA>           1           1  \n",
       "2             <NA>           1           1  \n",
       "3             <NA>           0           1  \n",
       "4             <NA>           0           1  \n",
       "...            ...         ...         ...  \n",
       "502456        <NA>           0           1  \n",
       "502457        <NA>           0           1  \n",
       "502458        <NA>           0           1  \n",
       "502459        <NA>           1           1  \n",
       "502460        <NA>        <NA>        <NA>  \n",
       "\n",
       "[502461 rows x 9 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/data/ukbb_databases/ukb47922_updatedAug2021/ukb47922.tab\", dtype=\"string\", sep='\\t', usecols=combined_cols)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8d2d1e0d-ffa9-43a0-bab8-b544851fe0fc",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-01-23 12:08:16.569455\n"
     ]
    }
   ],
   "source": [
    "print(datetime.now())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76a98173-fd69-43cb-b44d-242ed9a93f81",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Count ancestry of individuals in the full database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "89b5f849-ab79-4ac6-a615-9a5270bc32da",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'-1',\n",
       " '-3',\n",
       " '1',\n",
       " '1001',\n",
       " '1002',\n",
       " '1003',\n",
       " '2',\n",
       " '2001',\n",
       " '2002',\n",
       " '2003',\n",
       " '2004',\n",
       " '3',\n",
       " '3001',\n",
       " '3002',\n",
       " '3003',\n",
       " '3004',\n",
       " '4',\n",
       " '4001',\n",
       " '4002',\n",
       " '4003',\n",
       " '5',\n",
       " '6',\n",
       " <NA>}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set of answers for the ethnicity question\n",
    "set(df[ethnicity[0]].to_list()).union( set(df[ethnicity[1]].to_list()) , set(df[ethnicity[2]].to_list()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "62974b23-b3aa-4818-965b-43f7de7806af",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# these should align with all possible options for ethnicity answers except for <NA>, Do not know, and Prefer not to answer\n",
    "white = ['1001', '1002', '1','1003']\n",
    "african = ['4001','2001', '4002', '2002', '4', '4003' ]\n",
    "asian = ['3001', '3002', '2003', '3004', '3003', '3']\n",
    "mixed = ['2', '2004']\n",
    "chinese = ['5']\n",
    "other = ['6']\n",
    "\n",
    "# figure out the ancestry of each individual\n",
    "def ancestry(row):\n",
    "    temp = [x for x in row[ethnicity] if not pd.isna(x) and x != \"-3\" and x != \"-1\"]\n",
    "    if len(temp) == 0:\n",
    "        return \"Unknown\"\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in white: # if we have only one unique answer and the answer is in the white variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len([x for x in temp if x in white]) == len(temp):\n",
    "        return \"Inconsistent_white\"\n",
    "    if len([x for x in temp if x in asian]) == len(temp):\n",
    "        return \"Asian\"\n",
    "    if len([x for x in temp if x in african]) == len(temp):\n",
    "        return \"African\"\n",
    "    if len([x for x in temp if x in mixed]) == len(temp):\n",
    "        return \"Mixed\"\n",
    "    if len([x for x in temp if x in chinese]) == len(temp):\n",
    "        return \"Chinese\"\n",
    "    if len([x for x in temp if x in other]) == len(temp):\n",
    "        return \"Other\"\n",
    "    return \"Inconsistent\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "64baddd1-3036-40c4-b1a5-a0f61c8bda25",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_white = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "03bb17e2-74cd-40d6-acd6-119ccf682ddb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# This part was to get a better count of the individuals by their ancestry. For practical reasons I'll use the ancestry function in general\n",
    "df_white[\"ethnicity\"] = df_white[ethnicity].apply(ancestry_all, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "d1cf4807-21a4-47c7-95ec-4ffac748f36b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>553</td>\n",
       "      <td>553</td>\n",
       "      <td>553</td>\n",
       "      <td>553</td>\n",
       "      <td>553</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>529</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1001</th>\n",
       "      <td>442342</td>\n",
       "      <td>442342</td>\n",
       "      <td>442342</td>\n",
       "      <td>442342</td>\n",
       "      <td>442337</td>\n",
       "      <td>18619</td>\n",
       "      <td>10593</td>\n",
       "      <td>430782</td>\n",
       "      <td>409433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002</th>\n",
       "      <td>13021</td>\n",
       "      <td>13021</td>\n",
       "      <td>13021</td>\n",
       "      <td>13021</td>\n",
       "      <td>13021</td>\n",
       "      <td>395</td>\n",
       "      <td>229</td>\n",
       "      <td>12575</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>16149</td>\n",
       "      <td>16149</td>\n",
       "      <td>16149</td>\n",
       "      <td>16149</td>\n",
       "      <td>16148</td>\n",
       "      <td>377</td>\n",
       "      <td>199</td>\n",
       "      <td>15636</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>617</td>\n",
       "      <td>617</td>\n",
       "      <td>617</td>\n",
       "      <td>617</td>\n",
       "      <td>617</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>594</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>421</td>\n",
       "      <td>421</td>\n",
       "      <td>421</td>\n",
       "      <td>421</td>\n",
       "      <td>421</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>398</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>825</td>\n",
       "      <td>825</td>\n",
       "      <td>825</td>\n",
       "      <td>825</td>\n",
       "      <td>825</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>796</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2004</th>\n",
       "      <td>1019</td>\n",
       "      <td>1019</td>\n",
       "      <td>1019</td>\n",
       "      <td>1019</td>\n",
       "      <td>1019</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>982</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3001</th>\n",
       "      <td>5948</td>\n",
       "      <td>5948</td>\n",
       "      <td>5948</td>\n",
       "      <td>5948</td>\n",
       "      <td>5948</td>\n",
       "      <td>71</td>\n",
       "      <td>84</td>\n",
       "      <td>5713</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3002</th>\n",
       "      <td>1834</td>\n",
       "      <td>1834</td>\n",
       "      <td>1834</td>\n",
       "      <td>1834</td>\n",
       "      <td>1834</td>\n",
       "      <td>40</td>\n",
       "      <td>26</td>\n",
       "      <td>1745</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3003</th>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>221</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3004</th>\n",
       "      <td>1800</td>\n",
       "      <td>1800</td>\n",
       "      <td>1800</td>\n",
       "      <td>1800</td>\n",
       "      <td>1800</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>1732</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>26</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4001</th>\n",
       "      <td>4514</td>\n",
       "      <td>4514</td>\n",
       "      <td>4514</td>\n",
       "      <td>4514</td>\n",
       "      <td>4514</td>\n",
       "      <td>45</td>\n",
       "      <td>42</td>\n",
       "      <td>4294</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4002</th>\n",
       "      <td>3391</td>\n",
       "      <td>3391</td>\n",
       "      <td>3391</td>\n",
       "      <td>3391</td>\n",
       "      <td>3391</td>\n",
       "      <td>38</td>\n",
       "      <td>22</td>\n",
       "      <td>3201</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4003</th>\n",
       "      <td>121</td>\n",
       "      <td>121</td>\n",
       "      <td>121</td>\n",
       "      <td>121</td>\n",
       "      <td>121</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chinese</th>\n",
       "      <td>1571</td>\n",
       "      <td>1571</td>\n",
       "      <td>1571</td>\n",
       "      <td>1571</td>\n",
       "      <td>1571</td>\n",
       "      <td>45</td>\n",
       "      <td>34</td>\n",
       "      <td>1501</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent</th>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>148</td>\n",
       "      <td>104</td>\n",
       "      <td>68</td>\n",
       "      <td>145</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent_white</th>\n",
       "      <td>607</td>\n",
       "      <td>607</td>\n",
       "      <td>607</td>\n",
       "      <td>607</td>\n",
       "      <td>607</td>\n",
       "      <td>448</td>\n",
       "      <td>271</td>\n",
       "      <td>590</td>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>4508</td>\n",
       "      <td>4508</td>\n",
       "      <td>4508</td>\n",
       "      <td>4508</td>\n",
       "      <td>4508</td>\n",
       "      <td>55</td>\n",
       "      <td>35</td>\n",
       "      <td>4308</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unknown</th>\n",
       "      <td>2717</td>\n",
       "      <td>2716</td>\n",
       "      <td>2716</td>\n",
       "      <td>2716</td>\n",
       "      <td>1824</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>2249</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     f.eid  f.31.0.0  f.34.0.0  f.52.0.0  f.21000.0.0  \\\n",
       "ethnicity                                                               \n",
       "1                      553       553       553       553          553   \n",
       "1001                442342    442342    442342    442342       442337   \n",
       "1002                 13021     13021     13021     13021        13021   \n",
       "1003                 16149     16149     16149     16149        16148   \n",
       "2                       49        49        49        49           49   \n",
       "2001                   617       617       617       617          617   \n",
       "2002                   421       421       421       421          421   \n",
       "2003                   825       825       825       825          825   \n",
       "2004                  1019      1019      1019      1019         1019   \n",
       "3                       44        44        44        44           44   \n",
       "3001                  5948      5948      5948      5948         5948   \n",
       "3002                  1834      1834      1834      1834         1834   \n",
       "3003                   236       236       236       236          236   \n",
       "3004                  1800      1800      1800      1800         1800   \n",
       "4                       26        26        26        26           26   \n",
       "4001                  4514      4514      4514      4514         4514   \n",
       "4002                  3391      3391      3391      3391         3391   \n",
       "4003                   121       121       121       121          121   \n",
       "Chinese               1571      1571      1571      1571         1571   \n",
       "Inconsistent           148       148       148       148          148   \n",
       "Inconsistent_white     607       607       607       607          607   \n",
       "Other                 4508      4508      4508      4508         4508   \n",
       "Unknown               2717      2716      2716      2716         1824   \n",
       "\n",
       "                    f.21000.1.0  f.21000.2.0  f.22001.0.0  f.22006.0.0  \n",
       "ethnicity                                                               \n",
       "1                             5            1          529            0  \n",
       "1001                      18619        10593       430782       409433  \n",
       "1002                        395          229        12575            0  \n",
       "1003                        377          199        15636            0  \n",
       "2                             0            0           46            0  \n",
       "2001                          9           18          594            0  \n",
       "2002                         15            6          398            0  \n",
       "2003                         18           10          796            0  \n",
       "2004                         14            7          982            0  \n",
       "3                             1            1           43            0  \n",
       "3001                         71           84         5713            0  \n",
       "3002                         40           26         1745            0  \n",
       "3003                          1            2          221            0  \n",
       "3004                         18           11         1732            0  \n",
       "4                             0            0           25            0  \n",
       "4001                         45           42         4294            0  \n",
       "4002                         38           22         3201            0  \n",
       "4003                          2            0          116            0  \n",
       "Chinese                      45           34         1501            0  \n",
       "Inconsistent                104           68          145            5  \n",
       "Inconsistent_white          448          271          590          147  \n",
       "Other                        55           35         4308            0  \n",
       "Unknown                      15            9         2249            0  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_white.groupby(['ethnicity']).count()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d5a3e74-51b7-4f98-839b-fbc2afa9fc79",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## Subset for the invididuals present in the QCed sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9a075587-3e45-4aa6-9bed-3ec793b7b6bb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "qc_ind = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/genotype_files_processed/082621_sampleqc_call90/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.082621_sampleqc_call90.filtered.fam', dtype=\"string\", sep='\\t', names=[\"f.eid\", \"IID\", \"father\",\"mother\",\"sex\", \"pheno\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3607f56f-fa20-44aa-81f3-d00f2175b257",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>IID</th>\n",
       "      <th>father</th>\n",
       "      <th>mother</th>\n",
       "      <th>sex</th>\n",
       "      <th>pheno</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>1000019</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1000022</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1000035</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>1000046</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>1000054</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486411</th>\n",
       "      <td>6025390</td>\n",
       "      <td>6025390</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486412</th>\n",
       "      <td>6025409</td>\n",
       "      <td>6025409</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486413</th>\n",
       "      <td>6025411</td>\n",
       "      <td>6025411</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486414</th>\n",
       "      <td>6025425</td>\n",
       "      <td>6025425</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486415</th>\n",
       "      <td>6025438</td>\n",
       "      <td>6025438</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>486416 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          f.eid      IID father mother sex pheno\n",
       "0       1000019  1000019      0      0   2    -9\n",
       "1       1000022  1000022      0      0   1    -9\n",
       "2       1000035  1000035      0      0   1    -9\n",
       "3       1000046  1000046      0      0   2    -9\n",
       "4       1000054  1000054      0      0   2    -9\n",
       "...         ...      ...    ...    ...  ..   ...\n",
       "486411  6025390  6025390      0      0   2    -9\n",
       "486412  6025409  6025409      0      0   2    -9\n",
       "486413  6025411  6025411      0      0   2    -9\n",
       "486414  6025425  6025425      0      0   2    -9\n",
       "486415  6025438  6025438      0      0   1    -9\n",
       "\n",
       "[486416 rows x 6 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qc_ind"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e0006204-2f31-4a93-8ed2-b057ccb05d1e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "qc_list = set([str(i) for i in qc_ind['f.eid'].to_list()])\n",
    "\n",
    "def matches_qc_individuals(row):\n",
    "    return row[\"f.eid\"] in qc_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1f72263a-6afa-4b82-afff-f05d9a9c6754",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "filtered = df[df[[\"f.eid\"]].apply(matches_qc_individuals, axis=1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "347c0bc8-774f-4fb5-b59a-ec3dcb739df0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1</td>\n",
       "      <td>1954</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1</td>\n",
       "      <td>1944</td>\n",
       "      <td>5</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502455</th>\n",
       "      <td>6025390</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502456</th>\n",
       "      <td>6025409</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502457</th>\n",
       "      <td>6025411</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502458</th>\n",
       "      <td>6025425</td>\n",
       "      <td>0</td>\n",
       "      <td>1963</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502459</th>\n",
       "      <td>6025438</td>\n",
       "      <td>1</td>\n",
       "      <td>1952</td>\n",
       "      <td>9</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>486416 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          f.eid f.31.0.0 f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0  \\\n",
       "0       1000019        0     1960       11        1001        <NA>   \n",
       "1       1000022        1     1954        8        1001        <NA>   \n",
       "2       1000035        1     1944        5        1001        <NA>   \n",
       "3       1000046        0     1946        3        1001        <NA>   \n",
       "4       1000054        0     1942        1        1001        <NA>   \n",
       "...         ...      ...      ...      ...         ...         ...   \n",
       "502455  6025390        0     1942        3        1001        <NA>   \n",
       "502456  6025409        0     1946       11        1001        1001   \n",
       "502457  6025411        0     1960       11        1001        <NA>   \n",
       "502458  6025425        0     1963        8        1001        <NA>   \n",
       "502459  6025438        1     1952        9        1001        <NA>   \n",
       "\n",
       "       f.21000.2.0 f.22001.0.0 f.22006.0.0  \n",
       "0             <NA>           0           1  \n",
       "1             <NA>           1           1  \n",
       "2             <NA>           1           1  \n",
       "3             <NA>           0           1  \n",
       "4             <NA>           0           1  \n",
       "...            ...         ...         ...  \n",
       "502455        <NA>           0        <NA>  \n",
       "502456        <NA>           0           1  \n",
       "502457        <NA>           0           1  \n",
       "502458        <NA>           0           1  \n",
       "502459        <NA>           1           1  \n",
       "\n",
       "[486416 rows x 9 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8367d297-77bc-4443-859d-79c202c01d7b",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Count ancestry in the qc'ed database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b3b10e99-2884-422c-8834-75c7934bb811",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# these should align with all possible options for ethnicity answers except for <NA>, Do not know, and Prefer not to answer\n",
    "white = ['1001', '1002', '1','1003']\n",
    "african = ['4001','2001', '4002', '2002', '4', '4003' ]\n",
    "asian = ['3001', '3002', '2003', '3004', '3003', '3']\n",
    "mixed = ['2', '2004']\n",
    "chinese = ['5']\n",
    "other = ['6']\n",
    "\n",
    "# figure out the ancestry of each individual\n",
    "def ancestry_all(row):\n",
    "    temp = [x for x in row[ethnicity] if not pd.isna(x) and x != \"-3\" and x != \"-1\"]\n",
    "    if len(temp) == 0:\n",
    "        return \"Unknown\"\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in white: # if we have only one unique answer and the answer is in the white variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len([x for x in temp if x in white]) == len(temp):\n",
    "        return \"Inconsistent_white\"\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in asian: # if we have only one unique answer and the answer is in the asian variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in african: # if we have only one unique answer and the answer is in the white variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in mixed: # if we have only one unique answer and the answer is in the white variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len([x for x in temp if x in chinese]) == len(temp):\n",
    "        return \"Chinese\"\n",
    "    if len([x for x in temp if x in other]) == len(temp):\n",
    "        return \"Other\"\n",
    "    return \"Inconsistent\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "deeb2a8a-64d7-4ba8-8d08-6fc3bac7552d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc = filtered.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "fc149ccc-0ad3-4f36-889c-18f66743f334",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc[\"ethnicity\"] = df_qc[ethnicity].apply(ancestry_all, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "76bc0665-b29e-4317-8dce-ef8b0d750ac9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>526</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1001</th>\n",
       "      <td>429231</td>\n",
       "      <td>429231</td>\n",
       "      <td>429231</td>\n",
       "      <td>429231</td>\n",
       "      <td>429226</td>\n",
       "      <td>18462</td>\n",
       "      <td>10350</td>\n",
       "      <td>429231</td>\n",
       "      <td>407993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1002</th>\n",
       "      <td>12518</td>\n",
       "      <td>12518</td>\n",
       "      <td>12518</td>\n",
       "      <td>12518</td>\n",
       "      <td>12518</td>\n",
       "      <td>385</td>\n",
       "      <td>216</td>\n",
       "      <td>12518</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1003</th>\n",
       "      <td>15543</td>\n",
       "      <td>15543</td>\n",
       "      <td>15543</td>\n",
       "      <td>15543</td>\n",
       "      <td>15542</td>\n",
       "      <td>374</td>\n",
       "      <td>192</td>\n",
       "      <td>15543</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>593</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>396</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>796</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2004</th>\n",
       "      <td>981</td>\n",
       "      <td>981</td>\n",
       "      <td>981</td>\n",
       "      <td>981</td>\n",
       "      <td>981</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>981</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3001</th>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>70</td>\n",
       "      <td>82</td>\n",
       "      <td>5651</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3002</th>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>39</td>\n",
       "      <td>26</td>\n",
       "      <td>1739</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3003</th>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>221</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3004</th>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>1732</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4001</th>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>44</td>\n",
       "      <td>40</td>\n",
       "      <td>4286</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4002</th>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>3199</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4003</th>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Chinese</th>\n",
       "      <td>1498</td>\n",
       "      <td>1498</td>\n",
       "      <td>1498</td>\n",
       "      <td>1498</td>\n",
       "      <td>1498</td>\n",
       "      <td>45</td>\n",
       "      <td>34</td>\n",
       "      <td>1498</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent</th>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>145</td>\n",
       "      <td>103</td>\n",
       "      <td>66</td>\n",
       "      <td>145</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent_white</th>\n",
       "      <td>588</td>\n",
       "      <td>588</td>\n",
       "      <td>588</td>\n",
       "      <td>588</td>\n",
       "      <td>588</td>\n",
       "      <td>442</td>\n",
       "      <td>257</td>\n",
       "      <td>588</td>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Other</th>\n",
       "      <td>4302</td>\n",
       "      <td>4302</td>\n",
       "      <td>4302</td>\n",
       "      <td>4302</td>\n",
       "      <td>4302</td>\n",
       "      <td>55</td>\n",
       "      <td>32</td>\n",
       "      <td>4302</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unknown</th>\n",
       "      <td>2243</td>\n",
       "      <td>2243</td>\n",
       "      <td>2243</td>\n",
       "      <td>2243</td>\n",
       "      <td>1728</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>2243</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     f.eid  f.31.0.0  f.34.0.0  f.52.0.0  f.21000.0.0  \\\n",
       "ethnicity                                                               \n",
       "1                      526       526       526       526          526   \n",
       "1001                429231    429231    429231    429231       429226   \n",
       "1002                 12518     12518     12518     12518        12518   \n",
       "1003                 15543     15543     15543     15543        15542   \n",
       "2                       45        45        45        45           45   \n",
       "2001                   593       593       593       593          593   \n",
       "2002                   396       396       396       396          396   \n",
       "2003                   796       796       796       796          796   \n",
       "2004                   981       981       981       981          981   \n",
       "3                       42        42        42        42           42   \n",
       "3001                  5651      5651      5651      5651         5651   \n",
       "3002                  1739      1739      1739      1739         1739   \n",
       "3003                   221       221       221       221          221   \n",
       "3004                  1732      1732      1732      1732         1732   \n",
       "4                       25        25        25        25           25   \n",
       "4001                  4286      4286      4286      4286         4286   \n",
       "4002                  3199      3199      3199      3199         3199   \n",
       "4003                   116       116       116       116          116   \n",
       "Chinese               1498      1498      1498      1498         1498   \n",
       "Inconsistent           145       145       145       145          145   \n",
       "Inconsistent_white     588       588       588       588          588   \n",
       "Other                 4302      4302      4302      4302         4302   \n",
       "Unknown               2243      2243      2243      2243         1728   \n",
       "\n",
       "                    f.21000.1.0  f.21000.2.0  f.22001.0.0  f.22006.0.0  \n",
       "ethnicity                                                               \n",
       "1                             5            1          526            0  \n",
       "1001                      18462        10350       429231       407993  \n",
       "1002                        385          216        12518            0  \n",
       "1003                        374          192        15543            0  \n",
       "2                             0            0           45            0  \n",
       "2001                          9           18          593            0  \n",
       "2002                         15            6          396            0  \n",
       "2003                         18           10          796            0  \n",
       "2004                         13            7          981            0  \n",
       "3                             1            1           42            0  \n",
       "3001                         70           82         5651            0  \n",
       "3002                         39           26         1739            0  \n",
       "3003                          1            2          221            0  \n",
       "3004                         18           11         1732            0  \n",
       "4                             0            0           25            0  \n",
       "4001                         44           40         4286            0  \n",
       "4002                         38           20         3199            0  \n",
       "4003                          2            0          116            0  \n",
       "Chinese                      45           34         1498            0  \n",
       "Inconsistent                103           66          145            5  \n",
       "Inconsistent_white          442          257          588          146  \n",
       "Other                        55           32         4302            0  \n",
       "Unknown                      15            9         2243            0  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc.groupby(['ethnicity']).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "9a343932-5ea0-4ba4-902b-6c737cc28764",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc[\"ethnicity\"] = df_qc[ethnicity].apply(ancestry, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bdc46222-3ae7-4bb3-9477-bf8afa5f8b69",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Define white individuals \n",
    "\n",
    "We keep all of those that self-identified as white plus the inconsistent_white and unknown groups to do PCA analysis. We first define the non-white"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1cce29c8-1cb2-4f0e-ad70-fd64063f2cb9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def find_non_white(row):\n",
    "    return row[\"ethnicity\"] not in white and row[\"ethnicity\"] != \"Unknown\" and row[\"ethnicity\"] != \"Inconsistent_white\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "3a3585d8-332f-4765-97e9-73cc49337c8d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "ex_non_white = df_qc[[\"ethnicity\"]].apply(find_non_white, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "e5b3ddd7-6204-46e7-9c27-99142567f8c5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white = df_qc[~ex_non_white]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "e6817eed-64a4-4513-b39a-2851e4742d6e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1</td>\n",
       "      <td>1954</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1</td>\n",
       "      <td>1944</td>\n",
       "      <td>5</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502455</th>\n",
       "      <td>6025390</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502456</th>\n",
       "      <td>6025409</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502457</th>\n",
       "      <td>6025411</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502458</th>\n",
       "      <td>6025425</td>\n",
       "      <td>0</td>\n",
       "      <td>1963</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502459</th>\n",
       "      <td>6025438</td>\n",
       "      <td>1</td>\n",
       "      <td>1952</td>\n",
       "      <td>9</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>460649 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          f.eid f.31.0.0 f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0  \\\n",
       "0       1000019        0     1960       11        1001        <NA>   \n",
       "1       1000022        1     1954        8        1001        <NA>   \n",
       "2       1000035        1     1944        5        1001        <NA>   \n",
       "3       1000046        0     1946        3        1001        <NA>   \n",
       "4       1000054        0     1942        1        1001        <NA>   \n",
       "...         ...      ...      ...      ...         ...         ...   \n",
       "502455  6025390        0     1942        3        1001        <NA>   \n",
       "502456  6025409        0     1946       11        1001        1001   \n",
       "502457  6025411        0     1960       11        1001        <NA>   \n",
       "502458  6025425        0     1963        8        1001        <NA>   \n",
       "502459  6025438        1     1952        9        1001        <NA>   \n",
       "\n",
       "       f.21000.2.0 f.22001.0.0 f.22006.0.0 ethnicity  \n",
       "0             <NA>           0           1      1001  \n",
       "1             <NA>           1           1      1001  \n",
       "2             <NA>           1           1      1001  \n",
       "3             <NA>           0           1      1001  \n",
       "4             <NA>           0           1      1001  \n",
       "...            ...         ...         ...       ...  \n",
       "502455        <NA>           0        <NA>      1001  \n",
       "502456        <NA>           0           1      1001  \n",
       "502457        <NA>           0           1      1001  \n",
       "502458        <NA>           0           1      1001  \n",
       "502459        <NA>           1           1      1001  \n",
       "\n",
       "[460649 rows x 10 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_white"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "e7d9b14a-f1ce-4dd4-989a-203d63313af9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white = df_qc_white.rename(columns={'f.eid': 'IID', 'f.31.0.0': 'sex'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "e10a6a32-1e37-4a7d-b39c-db385a18863d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white['FID'] = df_qc_white['IID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "c6cb817e-68c2-4800-a767-748f4b37ff58",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>1000019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1</td>\n",
       "      <td>1954</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>1000022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1</td>\n",
       "      <td>1944</td>\n",
       "      <td>5</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>1000035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>1000046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>1000054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502455</th>\n",
       "      <td>6025390</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>3</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1001</td>\n",
       "      <td>6025390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502456</th>\n",
       "      <td>6025409</td>\n",
       "      <td>0</td>\n",
       "      <td>1946</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>6025409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502457</th>\n",
       "      <td>6025411</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>11</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>6025411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502458</th>\n",
       "      <td>6025425</td>\n",
       "      <td>0</td>\n",
       "      <td>1963</td>\n",
       "      <td>8</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>6025425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502459</th>\n",
       "      <td>6025438</td>\n",
       "      <td>1</td>\n",
       "      <td>1952</td>\n",
       "      <td>9</td>\n",
       "      <td>1001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001</td>\n",
       "      <td>6025438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>460649 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            IID sex f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0 f.21000.2.0  \\\n",
       "0       1000019   0     1960       11        1001        <NA>        <NA>   \n",
       "1       1000022   1     1954        8        1001        <NA>        <NA>   \n",
       "2       1000035   1     1944        5        1001        <NA>        <NA>   \n",
       "3       1000046   0     1946        3        1001        <NA>        <NA>   \n",
       "4       1000054   0     1942        1        1001        <NA>        <NA>   \n",
       "...         ...  ..      ...      ...         ...         ...         ...   \n",
       "502455  6025390   0     1942        3        1001        <NA>        <NA>   \n",
       "502456  6025409   0     1946       11        1001        1001        <NA>   \n",
       "502457  6025411   0     1960       11        1001        <NA>        <NA>   \n",
       "502458  6025425   0     1963        8        1001        <NA>        <NA>   \n",
       "502459  6025438   1     1952        9        1001        <NA>        <NA>   \n",
       "\n",
       "       f.22001.0.0 f.22006.0.0 ethnicity      FID  \n",
       "0                0           1      1001  1000019  \n",
       "1                1           1      1001  1000022  \n",
       "2                1           1      1001  1000035  \n",
       "3                0           1      1001  1000046  \n",
       "4                0           1      1001  1000054  \n",
       "...            ...         ...       ...      ...  \n",
       "502455           0        <NA>      1001  6025390  \n",
       "502456           0           1      1001  6025409  \n",
       "502457           0           1      1001  6025411  \n",
       "502458           0           1      1001  6025425  \n",
       "502459           1           1      1001  6025438  \n",
       "\n",
       "[460649 rows x 11 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_white"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "2a0af1b8-07e4-46fd-a3f1-830a145c02fe",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white[[\"FID\",\"IID\",\"ethnicity\"]].to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_460649.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "6262b304-320a-4d8f-8b5f-96746cc290a1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white=pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_460649.iid\", dtype=\"string\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c3492cfc-e9e3-4bc3-8111-cd861d2dd91b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "rel_white= pd.read_csv('~/UKBiobank/results/092821_PCA_results_500K/092821_king/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_king.related_id', dtype=\"string\", sep=\" \", names=[\"FID\", \"IID\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "68c695b4-e8af-4fb1-b635-69f457c96b1f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>1000019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1000035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000054</td>\n",
       "      <td>1000054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000224</td>\n",
       "      <td>1000224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000255</td>\n",
       "      <td>1000255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109214</th>\n",
       "      <td>6025176</td>\n",
       "      <td>6025176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109215</th>\n",
       "      <td>6025180</td>\n",
       "      <td>6025180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109216</th>\n",
       "      <td>6025322</td>\n",
       "      <td>6025322</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109217</th>\n",
       "      <td>6025425</td>\n",
       "      <td>6025425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109218</th>\n",
       "      <td>6025438</td>\n",
       "      <td>6025438</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109219 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            FID      IID\n",
       "0       1000019  1000019\n",
       "1       1000035  1000035\n",
       "2       1000054  1000054\n",
       "3       1000224  1000224\n",
       "4       1000255  1000255\n",
       "...         ...      ...\n",
       "109214  6025176  6025176\n",
       "109215  6025180  6025180\n",
       "109216  6025322  6025322\n",
       "109217  6025425  6025425\n",
       "109218  6025438  6025438\n",
       "\n",
       "[109219 rows x 2 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rel_white"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "db7df383-1e7f-425a-aa99-ce7e88860ff0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white_unrel=df_qc_white[~df_qc_white.FID.isin(rel_white.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "d93221a6-004a-4383-b309-666beaea06f9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1000022</td>\n",
       "      <td>1000022</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1000046</td>\n",
       "      <td>1000046</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1000063</td>\n",
       "      <td>1000063</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1000078</td>\n",
       "      <td>1000078</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1000081</td>\n",
       "      <td>1000081</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460642</th>\n",
       "      <td>6025363</td>\n",
       "      <td>6025363</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460643</th>\n",
       "      <td>6025378</td>\n",
       "      <td>6025378</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460644</th>\n",
       "      <td>6025390</td>\n",
       "      <td>6025390</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460645</th>\n",
       "      <td>6025409</td>\n",
       "      <td>6025409</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460646</th>\n",
       "      <td>6025411</td>\n",
       "      <td>6025411</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>351430 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            FID      IID ethnicity\n",
       "1       1000022  1000022      1001\n",
       "3       1000046  1000046      1001\n",
       "5       1000063  1000063      1001\n",
       "6       1000078  1000078      1001\n",
       "7       1000081  1000081      1001\n",
       "...         ...      ...       ...\n",
       "460642  6025363  6025363      1001\n",
       "460643  6025378  6025378      1001\n",
       "460644  6025390  6025390      1001\n",
       "460645  6025409  6025409      1001\n",
       "460646  6025411  6025411      1001\n",
       "\n",
       "[351430 rows x 3 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_white_unrel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "a045b073-1c39-4d1a-b856-3b0b52722bba",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white_unrel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_unrel_351430.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "7cf8c16e-5585-4476-bd64-0f0eb97e87c4",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white_rel=df_qc_white[df_qc_white.FID.isin(rel_white.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "2bf7af24-b7d9-4c53-a777-8d04abf5de5a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000019</td>\n",
       "      <td>1000019</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1000035</td>\n",
       "      <td>1000035</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1000054</td>\n",
       "      <td>1000054</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1000224</td>\n",
       "      <td>1000224</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1000255</td>\n",
       "      <td>1000255</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460626</th>\n",
       "      <td>6025176</td>\n",
       "      <td>6025176</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460627</th>\n",
       "      <td>6025180</td>\n",
       "      <td>6025180</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460638</th>\n",
       "      <td>6025322</td>\n",
       "      <td>6025322</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460647</th>\n",
       "      <td>6025425</td>\n",
       "      <td>6025425</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460648</th>\n",
       "      <td>6025438</td>\n",
       "      <td>6025438</td>\n",
       "      <td>1001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109219 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            FID      IID ethnicity\n",
       "0       1000019  1000019      1001\n",
       "2       1000035  1000035      1001\n",
       "4       1000054  1000054      1001\n",
       "20      1000224  1000224      1001\n",
       "23      1000255  1000255      1001\n",
       "...         ...      ...       ...\n",
       "460626  6025176  6025176      1001\n",
       "460627  6025180  6025180      1001\n",
       "460638  6025322  6025322      1001\n",
       "460647  6025425  6025425      1001\n",
       "460648  6025438  6025438      1001\n",
       "\n",
       "[109219 rows x 3 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_white_rel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "dfddd12e-3db4-46a6-ae0d-83c80f0d14bf",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_white_rel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_rel_109219.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "440c7d76-a7d9-45f9-bcc7-aecc7b36253d",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "#### Select individuals and remove variants HWE<1e-15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "13ef7e77-d816-4e69-821f-97a5f6c08c07",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/hg38_white_european_variantqc2_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wa31b293079fa4c8a) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To remove related samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_unrel_351430.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_white_european_variantqc2_$(date +\"%Y-%m-%d\").sbatch\n",
    "# common variants 1% MAF\n",
    "maf_filter=0.01\n",
    "#call rate 99%\n",
    "geno_filter=0.01\n",
    "#hwe\n",
    "hwe_filter=1e-15\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_white_eur_qc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4bfd3ca-84a1-4738-a10e-1097f62151c5",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Select individuals and remove variants HWE<1e-8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2195b32d-112a-45c8-af76-da45fc1997d7",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/hwe_1e-8/hg38_white_european_variantqc2_hwe1e-82023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wa795ccc4b346a471) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/hwe_1e-8\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To remove related samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_unrel_351430.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_white_european_variantqc2_hwe1e-8$(date +\"%Y-%m-%d\").sbatch\n",
    "# common variants 1% MAF\n",
    "maf_filter=0.01\n",
    "#call rate 99%\n",
    "geno_filter=0.01\n",
    "#hwe\n",
    "hwe_filter=1e-8\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_white_eur_qc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e313bfe-1f47-43ce-bafc-0da12a7d568a",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "source": [
    "#### Final file white europeans after individuals and variant qc_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "4ea9cebf-bbf7-4f8b-959a-32ee662da5d1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files/hg38_white_european_final_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wed2f26d4276f8614) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To keep related and unrelated samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_460649.iid\n",
    "# To keep variants after geno=0.01, maf=0.01 and hwe 1e-15\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.bim\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_white_european_final_$(date +\"%Y-%m-%d\").sbatch\n",
    "mem='30G'\n",
    "name='hg38_white_eur_qc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "geno=0.0\n",
    "mind=0.0\n",
    "hwe=0.0\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --geno_filter $geno\n",
    "    --mind_filter $mind\n",
    "    --hwe_filter $hwe\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f579fdd2-7b7d-44cf-b254-97cf34bcb9de",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 1. PCA for white europeans: keep unrelated and do LD prunning (get bed file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "36919e2d-0ce6-4fc6-8c66-bb62854b38f3",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/ldprun_unrelated/ldprun_unrelated_whiteEUR_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wb6ca060407205956) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/ldprun_unrelated\n",
    "## Use the qc version of the genotype array with the already filtered asian individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.bed\n",
    "#To keep the samples of asian and unrelated individuals only\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_unrel_351430.iid\n",
    "#GWAS QC variables: leave all the variables in 0 so there's no more filtering in the already filtered data\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#LD prunning variables\n",
    "window=50\n",
    "shift=10\n",
    "r2=0.1\n",
    "gwas_sbatch=$cwd/ldprun_unrelated_whiteEUR_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_args=\"\"\"qc\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --window $window\n",
    "    --shift $shift\n",
    "    --r2 $r2\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3d937bb-dbf4-440d-88bc-218e534fccfd",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 2 PCA: run for unrelated white European"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "d4982c4b-f1bc-4e1d-9d90-f1022518d4d4",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/pca_unrelated/flashpca_whiteEUR_unrelated_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w749ee929472f7952) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/pca_unrelated\n",
    "#This is the bfile originated after filtering unrelated individuals and pruning\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/ldprun_unrelated/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.filtered.prune.bed\n",
    "phenofile=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_unrel_351430.iid\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sos=~/project/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "pca_sbatch=$cwd/flashpca_whiteEUR_unrelated_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "maha_k=5\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "homogeneous=TRUE\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --pops $pops\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10cf31d4-4035-4dcc-a899-e52930d28082",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 3. Get the bed file for related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "f8c593e5-ba94-4983-b523-9bd4a9a3fbbe",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/related_whiteEUR/flashpca_whiteEUR_related_qc_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w569295ddb9de6d21) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/related_whiteEUR\n",
    "## Use qc'ed genotype array\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.bed\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_rel_109219.iid\n",
    "#Keep the same variants as above\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/ldprun_unrelated/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.filtered.prune.in\n",
    "#GWAS QC variables\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "gwas_sbatch=$cwd/flashpca_whiteEUR_related_qc_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b260e4b-74c9-40ed-87f2-0d8daeadf9a2",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 4. PCA project back related individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "ea58d7de-cccc-476a-925c-eda988bc90f8",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/flashpca_whiteEUR_related_genoarray_projected_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wc73d8ac3fde3ebca) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/related_whiteEUR/*.bed\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_460649.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/pca_unrelated/*.pca.rds\n",
    "pca_sbatch=$cwd/flashpca_whiteEUR_related_genoarray_projected_$(date +\"%Y-%m-%d\").sbatch\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "k=10\n",
    "maha_k=5\n",
    "prob=0.997\n",
    "pval=0.05\n",
    "## set the --homogeneous TRUE options to consider all the pops like one \n",
    "homogeneous=TRUE\n",
    "## For the plot you need to use the *.projected.rds and not the *.projected.mahalanobis.rds\n",
    "#plot_data=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.rds\n",
    "#outlier_file=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.outliers\n",
    "\n",
    "\n",
    "pca_args=\"\"\"project_samples\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --pca_model $pca_model\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --prob $prob\n",
    "    --pval $pval\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85edf68a-b69e-41a9-8a01-143520b8b6b1",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step. 5 PCA: plot and look for outliers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8e9ed233-18c7-4aa6-8ab7-53a7b27bc93f",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/plot/flashpca_whiteEUR_related_genoarray_plot_2023-02-02.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w43f05903bc6f2d1e) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "job_size=1\n",
    "numThreads=2\n",
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/plot\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/related_whiteEUR/*.bed\n",
    "## I had to modify the original file to add a super_pop and replace ethnicity for pop\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/white_IID/012323_ukb47922_white_expanded_qc_460649.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/pca_unrelated/012323_ukb47922_white_expanded_qc_460649.pca.rds\n",
    "plot_data=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/012323_ukb47922_white_expanded_qc_460649.pca.projected.rds\n",
    "outlier_file=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/012323_ukb47922_white_expanded_qc_460649.pca.outliers\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_whiteEUR_related_genoarray_plot_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "pca_sos=~/project/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "\n",
    "pca_args=\"\"\"plot_pca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --plot_data $plot_data\n",
    "    --outlier_file $outlier_file\n",
    "    --k $k\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg  \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86ce2a1a-dc80-4d86-a59b-931825d777f7",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Get individual ID and variants text files and final qc'ed bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "17201e18-1f70-4f4c-9e02-f0735e996ba8",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR.sh\" target=\"_blank\">/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save -f /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR.sh\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=36:00:00\n",
    "#$ -l h_vmem=30G\n",
    "#$ -N final_genoarray_whiteEUR_2023-01-31\n",
    "#$ -o /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR_2023-01-31.out\n",
    "#$ -e /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/final_genoarray_whiteEUR_2023-01-31.err\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load PLINK/2.0\n",
    "plink2 \\\n",
    "    --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted \\\n",
    "    --remove /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/PCA/project_related_whiteEUR/*.pca.projected.outliers \\\n",
    "    --write-snplist --write-samples --no-id-header \\\n",
    "    --threads 20 \\\n",
    "    --make-bed \\\n",
    "    --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_white_european_460649ind_hg38/final_files_no_outliers/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ddafa55-8491-4ba8-8c09-ea899981d386",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "\n",
    "### Define Asian individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "a15c81c3-b22e-4a26-a3e4-2d571a6a928e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# these should align with all possible options for ethnicity answers except for <NA>, Do not know, and Prefer not to answer\n",
    "white = ['1001', '1002', '1','1003']\n",
    "african = ['4001','2001', '4002', '2002', '4', '4003' ]\n",
    "asian = ['3001', '3002', '2003', '3004', '3003', '3']\n",
    "mixed = ['2', '2004']\n",
    "chinese = ['5']\n",
    "other = ['6']\n",
    "\n",
    "# figure out the ancestry of each individual\n",
    "def ancestry(row):\n",
    "    temp = [x for x in row[ethnicity] if not pd.isna(x) and x != \"-3\" and x != \"-1\"]\n",
    "    if len(temp) == 0:\n",
    "        return \"Unknown\"\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in asian: # if we have only one unique answer and the answer is in the asian variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len([x for x in temp if x in asian]) == len(temp):\n",
    "        return \"Inconsistent_asian\"\n",
    "    if len([x for x in temp if x in white]) == len(temp):\n",
    "        return \"White\"\n",
    "    if len([x for x in temp if x in african]) == len(temp):\n",
    "        return \"African\"\n",
    "    if len([x for x in temp if x in mixed]) == len(temp):\n",
    "        return \"Mixed\"\n",
    "    if len([x for x in temp if x in chinese]) == len(temp):\n",
    "        return \"Chinese\"\n",
    "    if len([x for x in temp if x in other]) == len(temp):\n",
    "        return \"Other\"\n",
    "    return \"Inconsistent\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "62d5a04b-1f81-41b2-8f0f-32250012e012",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian = filtered.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "2f29a77a-835a-4696-8e4f-b78124e1bfed",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian[\"ethnicity\"] = df_qc_asian[ethnicity].apply(ancestry, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "ac471370-d61e-464b-882d-b62773b63ac3",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def find_asian(row):\n",
    "    return row[\"ethnicity\"] in asian or row[\"ethnicity\"] == \"Inconsistent_asian\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "1bd9bfaa-5c60-49f1-9966-e1ea43b1c531",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "inc_asian = df_qc_asian[[\"ethnicity\"]].apply(find_asian, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "de50a773-9a0c-4a42-a3da-bce0e42d786d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10189 individuals considered asian\n"
     ]
    }
   ],
   "source": [
    "print(sum(inc_asian), \"individuals considered asian\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "ceb3d1dd-ad67-40e3-ace3-07b35cd10afe",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>796</td>\n",
       "      <td>18</td>\n",
       "      <td>10</td>\n",
       "      <td>796</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>42</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3001</th>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>5651</td>\n",
       "      <td>70</td>\n",
       "      <td>82</td>\n",
       "      <td>5651</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3002</th>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>1739</td>\n",
       "      <td>39</td>\n",
       "      <td>26</td>\n",
       "      <td>1739</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3003</th>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>221</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>221</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3004</th>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>1732</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>1732</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent_asian</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    f.eid  f.31.0.0  f.34.0.0  f.52.0.0  f.21000.0.0  \\\n",
       "ethnicity                                                              \n",
       "2003                  796       796       796       796          796   \n",
       "3                      42        42        42        42           42   \n",
       "3001                 5651      5651      5651      5651         5651   \n",
       "3002                 1739      1739      1739      1739         1739   \n",
       "3003                  221       221       221       221          221   \n",
       "3004                 1732      1732      1732      1732         1732   \n",
       "Inconsistent_asian      8         8         8         8            8   \n",
       "\n",
       "                    f.21000.1.0  f.21000.2.0  f.22001.0.0  f.22006.0.0  \n",
       "ethnicity                                                               \n",
       "2003                         18           10          796            0  \n",
       "3                             1            1           42            0  \n",
       "3001                         70           82         5651            0  \n",
       "3002                         39           26         1739            0  \n",
       "3003                          1            2          221            0  \n",
       "3004                         18           11         1732            0  \n",
       "Inconsistent_asian            3            6            8            0  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_asian=df_qc_asian[inc_asian]\n",
    "df_qc_asian.groupby(['ethnicity']).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "f334f735-c826-4f43-bdc4-243930cf1758",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian = df_qc_asian.rename(columns={'f.eid': 'IID', 'f.31.0.0': 'sex'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "3d057efb-3c84-4762-a1e2-1e41aa9d6309",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian['FID'] = df_qc_asian['IID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "4156a9bf-a3be-458c-9715-1d2316ea05b8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>1000906</td>\n",
       "      <td>1</td>\n",
       "      <td>1962</td>\n",
       "      <td>5</td>\n",
       "      <td>3003</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3003</td>\n",
       "      <td>1000906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>1001874</td>\n",
       "      <td>0</td>\n",
       "      <td>1947</td>\n",
       "      <td>8</td>\n",
       "      <td>3004</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3004</td>\n",
       "      <td>1001874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>1002712</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>12</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1002712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>1003025</td>\n",
       "      <td>1</td>\n",
       "      <td>1942</td>\n",
       "      <td>10</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1003025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>1003083</td>\n",
       "      <td>1</td>\n",
       "      <td>1961</td>\n",
       "      <td>10</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1003083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502347</th>\n",
       "      <td>6024313</td>\n",
       "      <td>0</td>\n",
       "      <td>1964</td>\n",
       "      <td>5</td>\n",
       "      <td>3004</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3004</td>\n",
       "      <td>6024313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502399</th>\n",
       "      <td>6024837</td>\n",
       "      <td>1</td>\n",
       "      <td>1965</td>\n",
       "      <td>5</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6024837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502405</th>\n",
       "      <td>6024898</td>\n",
       "      <td>0</td>\n",
       "      <td>1950</td>\n",
       "      <td>9</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6024898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502425</th>\n",
       "      <td>6025096</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>9</td>\n",
       "      <td>2003</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>2003</td>\n",
       "      <td>6025096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502427</th>\n",
       "      <td>6025113</td>\n",
       "      <td>1</td>\n",
       "      <td>1948</td>\n",
       "      <td>8</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6025113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10189 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            IID sex f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0 f.21000.2.0  \\\n",
       "89      1000906   1     1962        5        3003        <NA>        <NA>   \n",
       "186     1001874   0     1947        8        3004        <NA>        <NA>   \n",
       "270     1002712   0     1965       12        3001        <NA>        <NA>   \n",
       "301     1003025   1     1942       10        3001        <NA>        <NA>   \n",
       "307     1003083   1     1961       10        3001        <NA>        <NA>   \n",
       "...         ...  ..      ...      ...         ...         ...         ...   \n",
       "502347  6024313   0     1964        5        3004        <NA>        <NA>   \n",
       "502399  6024837   1     1965        5        3001        <NA>        <NA>   \n",
       "502405  6024898   0     1950        9        3001        <NA>        <NA>   \n",
       "502425  6025096   0     1965        9        2003        <NA>        <NA>   \n",
       "502427  6025113   1     1948        8        3001        <NA>        <NA>   \n",
       "\n",
       "       f.22001.0.0 f.22006.0.0 ethnicity      FID  \n",
       "89               1        <NA>      3003  1000906  \n",
       "186              0        <NA>      3004  1001874  \n",
       "270              0        <NA>      3001  1002712  \n",
       "301              1        <NA>      3001  1003025  \n",
       "307              1        <NA>      3001  1003083  \n",
       "...            ...         ...       ...      ...  \n",
       "502347           0        <NA>      3004  6024313  \n",
       "502399           1        <NA>      3001  6024837  \n",
       "502405           0        <NA>      3001  6024898  \n",
       "502425           0        <NA>      2003  6025096  \n",
       "502427           1        <NA>      3001  6025113  \n",
       "\n",
       "[10189 rows x 11 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_asian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "addfca20-866e-47d1-ace8-4b2595d8d8d4",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian[[\"FID\",\"IID\",\"ethnicity\"]].to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_10189.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7542351d-ef60-4070-bc89-cd361fae87cf",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian=pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_10189.iid\", sep=\"\\t\", dtype=\"string\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "19ccf22f-0f41-40ad-b8a8-095257ae3f80",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1011968</td>\n",
       "      <td>1011968</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID      IID\n",
       "0  1011968  1011968"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "call95_asian= pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/010622_asian_10189ind/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.010622_asian_10189ind.filtered.mindrem.id\", skiprows=[0], names=[\"FID\", \"IID\"], dtype=\"string\", sep='\\t')\n",
    "call95_asian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "c7ac20db-8a63-47c4-afe2-53a4544651c8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_call95=df_qc_asian[~df_qc_asian.FID.isin(call95_asian.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "89c88d2f-fa3d-44ce-86d0-2ea047a918ab",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_call95[[\"FID\",\"IID\", \"ethnicity\"]].to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_10188.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "258f4d16-bc44-4cd2-8593-c33e711b1f4f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4492584</td>\n",
       "      <td>4492584</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0        1\n",
       "0  4492584  4492584"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Relatedness per UKB variable f.22021.0.0>1\n",
    "rel_asian= pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/010622_asian_10189ind/010722_sampleQC_IID_asian_related.txt\", sep='\\t', dtype=\"string\", header=None)\n",
    "rel_asian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "0d1098ba-8de3-4ed5-b626-e7ac8c71d66f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1011060</td>\n",
       "      <td>1011060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1011143</td>\n",
       "      <td>1011143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1013331</td>\n",
       "      <td>1013331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1023766</td>\n",
       "      <td>1023766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1028494</td>\n",
       "      <td>1028494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106</th>\n",
       "      <td>6013816</td>\n",
       "      <td>6013816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1107</th>\n",
       "      <td>6015306</td>\n",
       "      <td>6015306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1108</th>\n",
       "      <td>6019749</td>\n",
       "      <td>6019749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1109</th>\n",
       "      <td>6021570</td>\n",
       "      <td>6021570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1110</th>\n",
       "      <td>6024313</td>\n",
       "      <td>6024313</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1111 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          FID      IID\n",
       "0     1011060  1011060\n",
       "1     1011143  1011143\n",
       "2     1013331  1013331\n",
       "3     1023766  1023766\n",
       "4     1028494  1028494\n",
       "...       ...      ...\n",
       "1106  6013816  6013816\n",
       "1107  6015306  6015306\n",
       "1108  6019749  6019749\n",
       "1109  6021570  6021570\n",
       "1110  6024313  6024313\n",
       "\n",
       "[1111 rows x 2 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Relatedness per king calculation\n",
    "rel_king_asian = pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/010622_asian_10189ind/010722_king_asian/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted.010722_king_asian.related_id\", names=[\"FID\", \"IID\"],sep=\" \", dtype=\"string\", header=None)\n",
    "rel_king_asian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "6c0ebe86-4c02-4e3e-8a97-343778cb1cbd",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#create relate and unrelated files\n",
    "##unrelated\n",
    "df_qc_asian_call95_unrel=df_qc_asian_call95[~df_qc_asian_call95.FID.isin(rel_king_asian.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "ee022022-c14a-4b9b-b4e0-e204a40afff5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_call95_unrel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_unrel_9077.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a20ed9fb-be35-4bdb-afdd-e3f67fe54238",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## related\n",
    "df_qc_asian_call95_rel=df_qc_asian_call95[df_qc_asian_call95.FID.isin(rel_king_asian.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "e69cf0c9-c255-4899-a489-fd829d9165c0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_call95_rel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_rel_1111.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "b54ca312-bcf6-4e3d-bbd6-f22ae78b8327",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_unrel=df_qc_asian[~df_qc_asian.FID.isin(call95_asian.FID) & ~df_qc_asian.FID.isin(rel_king_asian.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "ca47d4d7-aa1c-4d0b-b7c2-fa13d09d7179",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>1000906</td>\n",
       "      <td>1</td>\n",
       "      <td>1962</td>\n",
       "      <td>5</td>\n",
       "      <td>3003</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3003</td>\n",
       "      <td>1000906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>1001874</td>\n",
       "      <td>0</td>\n",
       "      <td>1947</td>\n",
       "      <td>8</td>\n",
       "      <td>3004</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3004</td>\n",
       "      <td>1001874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>1002712</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>12</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1002712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>1003025</td>\n",
       "      <td>1</td>\n",
       "      <td>1942</td>\n",
       "      <td>10</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1003025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>1003083</td>\n",
       "      <td>1</td>\n",
       "      <td>1961</td>\n",
       "      <td>10</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>1003083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502342</th>\n",
       "      <td>6024266</td>\n",
       "      <td>0</td>\n",
       "      <td>1943</td>\n",
       "      <td>2</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6024266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502399</th>\n",
       "      <td>6024837</td>\n",
       "      <td>1</td>\n",
       "      <td>1965</td>\n",
       "      <td>5</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6024837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502405</th>\n",
       "      <td>6024898</td>\n",
       "      <td>0</td>\n",
       "      <td>1950</td>\n",
       "      <td>9</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6024898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502425</th>\n",
       "      <td>6025096</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>9</td>\n",
       "      <td>2003</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>2003</td>\n",
       "      <td>6025096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502427</th>\n",
       "      <td>6025113</td>\n",
       "      <td>1</td>\n",
       "      <td>1948</td>\n",
       "      <td>8</td>\n",
       "      <td>3001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>3001</td>\n",
       "      <td>6025113</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9077 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            IID sex f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0 f.21000.2.0  \\\n",
       "89      1000906   1     1962        5        3003        <NA>        <NA>   \n",
       "186     1001874   0     1947        8        3004        <NA>        <NA>   \n",
       "270     1002712   0     1965       12        3001        <NA>        <NA>   \n",
       "301     1003025   1     1942       10        3001        <NA>        <NA>   \n",
       "307     1003083   1     1961       10        3001        <NA>        <NA>   \n",
       "...         ...  ..      ...      ...         ...         ...         ...   \n",
       "502342  6024266   0     1943        2        3001        <NA>        <NA>   \n",
       "502399  6024837   1     1965        5        3001        <NA>        <NA>   \n",
       "502405  6024898   0     1950        9        3001        <NA>        <NA>   \n",
       "502425  6025096   0     1965        9        2003        <NA>        <NA>   \n",
       "502427  6025113   1     1948        8        3001        <NA>        <NA>   \n",
       "\n",
       "       f.22001.0.0 f.22006.0.0 ethnicity      FID  \n",
       "89               1        <NA>      3003  1000906  \n",
       "186              0        <NA>      3004  1001874  \n",
       "270              0        <NA>      3001  1002712  \n",
       "301              1        <NA>      3001  1003025  \n",
       "307              1        <NA>      3001  1003083  \n",
       "...            ...         ...       ...      ...  \n",
       "502342           0        <NA>      3001  6024266  \n",
       "502399           1        <NA>      3001  6024837  \n",
       "502405           0        <NA>      3001  6024898  \n",
       "502425           0        <NA>      2003  6025096  \n",
       "502427           1        <NA>      3001  6025113  \n",
       "\n",
       "[9077 rows x 11 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_asian_unrel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "5bd287f8-981a-4936-85de-895efdff6ce6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_asian_unrel[[\"FID\",\"IID\"]].to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_unrel_9077.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f049c8f-ab5d-4da9-ac0b-b0e0ee3a4772",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "#### Select individuals and remove HWE < 1e-15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a49ef795-f241-4a01-b192-6189b1b5e279",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/hg38_asian_variantqc2_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wda6679e6762a601a) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To remove related samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_unrel_9077.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_asian_variantqc2_$(date +\"%Y-%m-%d\").sbatch\n",
    "# common variants 1% MAF\n",
    "maf_filter=0.01\n",
    "#call rate 99%\n",
    "geno_filter=0.01\n",
    "#hwe\n",
    "hwe_filter=1e-15\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_asianqc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "246433c8-8cf2-4e2d-9778-9d2067a85bc8",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Select individuals and remove HWE < 1e-8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6011e7e5-4cf8-4b2e-9d92-dc768ea4ec30",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/hwe_1e-8/hg38_asian_variantqc2_hwe1e-8_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wcbc523ef1f626070) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/hwe_1e-8\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To remove related samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_unrel_9077.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_asian_variantqc2_hwe1e-8_$(date +\"%Y-%m-%d\").sbatch\n",
    "# common variants 1% MAF\n",
    "maf_filter=0.01\n",
    "#call rate 99%\n",
    "geno_filter=0.01\n",
    "#hwe\n",
    "hwe_filter=1e-8\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_asianqc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d727f06f-a3cd-4a6f-bb0e-69e7ffd20d25",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Final file asians after variant qc_2 (contains related and unrelated individuals)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d0c59fc1-3feb-43a8-8f7f-033d723b7af4",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files/hg38_asian_ind_var_qc_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=waad2f31a456bc389) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files\n",
    "# bfile with variant_qc_1 N=486416 contains all samples and variants=674,489\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To keep related and unrelated asians\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_10188.iid\n",
    "# To keep variants using geno=0.01, maf=0.01 and hwe=1e-15\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.bim\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_asian_ind_var_qc_$(date +\"%Y-%m-%d\").sbatch\n",
    "mem='30G'\n",
    "name='hg38_asianqc'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "geno=0.0\n",
    "mind=0.0\n",
    "hwe=0.0\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --geno_filter $geno\n",
    "    --mind_filter $mind\n",
    "    --hwe_filter $hwe\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcc37dbd-442e-4a70-bc40-60dffadac99a",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 1. PCA for Asians: keep unrelated and do LD prunning (get bed file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "856f3e60-7058-4da8-8788-55f5ef5e1799",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/ldprun_unrelated/ldprun_unrelated_asian_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w5db99fd0b20fa2d9) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/ldprun_unrelated\n",
    "## Use the qc version of the genotype array with the already filtered asian individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.bed\n",
    "#To keep the samples of asian and unrelated individuals only\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_unrel_9077.iid\n",
    "#GWAS QC variables: leave all the variables in 0 so there's no more filtering in the already filtered data\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#LD prunning variables\n",
    "window=50\n",
    "shift=10\n",
    "r2=0.1\n",
    "gwas_sbatch=$cwd/ldprun_unrelated_asian_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_args=\"\"\"qc\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --window $window\n",
    "    --shift $shift\n",
    "    --r2 $r2\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcc8a692-c115-476d-9355-65b102ebb820",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 2 PCA: run for unrelated Asian"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "e5ff5ca5-0409-4d55-83d0-c27f276ef99e",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/pca_unrelated/flashpca_asian_unrelated_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wf6048da0d9ee6611) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/pca_unrelated\n",
    "#This is the bfile originated after filtering unrelated individuals and pruning\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/ldprun_unrelated/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.filtered.prune.bed\n",
    "## I had to modify the original file to add a super_pop and replace ethnicity for pop\n",
    "phenofile=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_unrel_9077.iid\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sos=~/project/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "pca_sbatch=$cwd/flashpca_asian_unrelated_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "maha_k=5\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "homogeneous=TRUE\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --pops $pops\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "15e52ba2-ef7f-46ed-ba7e-e435d4961eb0",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 3. Get the bed file for related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "33ccac2f-32a4-476c-85f4-32b431a9c644",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/related_asian/flashpca_asian_related_qc_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w41eac5ecc15c0f2e) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/related_asian\n",
    "## Use qc'ed genotype array\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.bed\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_rel_1111.iid\n",
    "#Keep the same variants as above\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/ldprun_unrelated/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.filtered.prune.in\n",
    "#GWAS QC variables\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "gwas_sbatch=$cwd/flashpca_asian_related_qc_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f654dd5-cb4b-4ad1-9b4b-64c50f8afa48",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 4. PCA project back related individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "320a8f34-a33a-415e-af50-70ec1ae1158d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/flashpca_asian_related_genoarray_projected_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wff275343d758f495) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/related_asian/*.bed\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_10188.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/pca_unrelated/*.pca.rds\n",
    "pca_sbatch=$cwd/flashpca_asian_related_genoarray_projected_$(date +\"%Y-%m-%d\").sbatch\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "k=10\n",
    "maha_k=5\n",
    "prob=0.997\n",
    "pval=0.05\n",
    "## set the --homogeneous TRUE options to consider all the pops like one \n",
    "homogeneous=TRUE\n",
    "## For the plot you need to use the *.projected.rds and not the *.projected.mahalanobis.rds\n",
    "#plot_data=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.rds\n",
    "#outlier_file=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.outliers\n",
    "\n",
    "\n",
    "pca_args=\"\"\"project_samples\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --pca_model $pca_model\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --prob $prob\n",
    "    --pval $pval\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd5f0dd7-427c-4bc3-96e4-8daa9f72e8a3",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step. 5 PCA: plot and look for outliers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "91a0d8a7-fef4-4045-ad91-0e897e180705",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/plot/flashpca_asian_related_genoarray_plot_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w9226415ae531a493) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/plot\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/related_asian/*.bed\n",
    "## I had to modify the original file to add a super_pop and replace ethnicity for pop\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/asian_IID/012323_ukb47922_asian_qc_call95_10188.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/pca_unrelated/*.pca.rds\n",
    "plot_data=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/*.pca.projected.rds\n",
    "outlier_file=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/*.pca.projected.outliers\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_asian_related_genoarray_plot_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "\n",
    "pca_args=\"\"\"plot_pca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --plot_data $plot_data\n",
    "    --outlier_file $outlier_file\n",
    "    --k $k\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg  \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aff11814-2abd-45d7-922b-8a82af7b4f60",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Get individual ID and variants text files and final qc'ed bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "40cc37c6-e225-4a70-9995-2984cc30223b",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/final_genoarray_asians.sh\" target=\"_blank\">/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/final_genoarray_asians.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/final_genoarray_asians.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save -f /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/final_genoarray_asians.sh\n",
    "# N=10,157 samples and 444,076 variants\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=36:00:00\n",
    "#$ -l h_vmem=30G\n",
    "#$ -N hg38_final_genoarray_2023-01-31\n",
    "#$ -o /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/hg38_final_genoarray_2023-01-31.out\n",
    "#$ -e /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/hg38_final_genoarray_2023-01-31.err\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load PLINK/2.0\n",
    "plink2 \\\n",
    "    --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted \\\n",
    "    --remove /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/PCA/project_related_asian/012323_ukb47922_asian_qc_call95_10188.pca.projected.outliers \\\n",
    "    --write-snplist --write-samples --no-id-header \\\n",
    "    --threads 20 \\\n",
    "    --make-bed \\\n",
    "    --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_asian_10189ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_10057ind_447567var_ASIAN"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c7968db-9be2-4249-890a-4090891bfb38",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Define African individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "9735b2ca-1fb8-4be8-af87-a78aa9547477",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# these should align with all possible options for ethnicity answers except for <NA>, Do not know, and Prefer not to answer\n",
    "white = ['1001', '1002', '1','1003']\n",
    "african = ['4001','2001', '4002', '2002', '4', '4003' ]\n",
    "asian = ['3001', '3002', '2003', '3004', '3003', '3']\n",
    "mixed = ['2', '2004']\n",
    "chinese = ['5']\n",
    "other = ['6']\n",
    "\n",
    "# figure out the ancestry of each individual\n",
    "def ancestry(row):\n",
    "    temp = [x for x in row[ethnicity] if not pd.isna(x) and x != \"-3\" and x != \"-1\"]\n",
    "    if len(temp) == 0:\n",
    "        return \"Unknown\"\n",
    "    \n",
    "    if len(set(temp)) == 1 and temp[0] in african: # if we have only one unique answer and the answer is in the asian variable\n",
    "        return \"_\".join(temp[0].split(\" \")) # return the unique answer\n",
    "    \n",
    "    if len([x for x in temp if x in african]) == len(temp):\n",
    "        return \"Inconsistent_african\"\n",
    "    if len([x for x in temp if x in white]) == len(temp):\n",
    "        return \"White\"\n",
    "    if len([x for x in temp if x in asian]) == len(temp):\n",
    "        return \"Asian\"\n",
    "    if len([x for x in temp if x in mixed]) == len(temp):\n",
    "        return \"Mixed\"\n",
    "    if len([x for x in temp if x in chinese]) == len(temp):\n",
    "        return \"Chinese\"\n",
    "    if len([x for x in temp if x in other]) == len(temp):\n",
    "        return \"Other\"\n",
    "    return \"Inconsistent\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "05147a76-bcdb-403e-be1e-5ad6abb53215",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr = filtered.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "cad94318-23a3-4261-9c35-1a0fbb429e7c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr[\"ethnicity\"] = df_qc_afr[ethnicity].apply(ancestry, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "e385c7d0-7a31-4db7-89d0-a2b6517cd8b7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def find_african(row):\n",
    "    return row[\"ethnicity\"] in african or row[\"ethnicity\"] == \"Inconsistent_african\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "13d74796-ee3e-4b4f-8d1b-f704a131deec",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "inc_african = df_qc_afr[[\"ethnicity\"]].apply(find_african, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "840af4a5-572c-4065-950c-951d2f015e7d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8621 individuals considered african\n"
     ]
    }
   ],
   "source": [
    "print(sum(inc_african), \"individuals considered african\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "a3375a83-394e-4a72-9a06-9a3230a1a42e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2001</th>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>593</td>\n",
       "      <td>9</td>\n",
       "      <td>18</td>\n",
       "      <td>593</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2002</th>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>396</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>396</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4001</th>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>4286</td>\n",
       "      <td>44</td>\n",
       "      <td>40</td>\n",
       "      <td>4286</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4002</th>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>3199</td>\n",
       "      <td>38</td>\n",
       "      <td>20</td>\n",
       "      <td>3199</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4003</th>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>116</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>116</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Inconsistent_african</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      f.eid  f.31.0.0  f.34.0.0  f.52.0.0  f.21000.0.0  \\\n",
       "ethnicity                                                                \n",
       "2001                    593       593       593       593          593   \n",
       "2002                    396       396       396       396          396   \n",
       "4                        25        25        25        25           25   \n",
       "4001                   4286      4286      4286      4286         4286   \n",
       "4002                   3199      3199      3199      3199         3199   \n",
       "4003                    116       116       116       116          116   \n",
       "Inconsistent_african      6         6         6         6            6   \n",
       "\n",
       "                      f.21000.1.0  f.21000.2.0  f.22001.0.0  f.22006.0.0  \n",
       "ethnicity                                                                 \n",
       "2001                            9           18          593            0  \n",
       "2002                           15            6          396            0  \n",
       "4                               0            0           25            0  \n",
       "4001                           44           40         4286            0  \n",
       "4002                           38           20         3199            0  \n",
       "4003                            2            0          116            0  \n",
       "Inconsistent_african            4            2            6            0  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter the asian individuals\n",
    "df_qc_afr = df_qc_afr[inc_african]\n",
    "df_qc_afr.groupby(['ethnicity']).count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "40febfc9-c4d4-4763-8de9-2633967f70c9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>f.eid</th>\n",
       "      <th>f.31.0.0</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>1000697</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>10</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>1001447</td>\n",
       "      <td>0</td>\n",
       "      <td>1961</td>\n",
       "      <td>6</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>1001465</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>6</td>\n",
       "      <td>2001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>1002004</td>\n",
       "      <td>1</td>\n",
       "      <td>1957</td>\n",
       "      <td>2</td>\n",
       "      <td>4002</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>1002354</td>\n",
       "      <td>0</td>\n",
       "      <td>1962</td>\n",
       "      <td>9</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502221</th>\n",
       "      <td>6023054</td>\n",
       "      <td>0</td>\n",
       "      <td>1941</td>\n",
       "      <td>9</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502390</th>\n",
       "      <td>6024740</td>\n",
       "      <td>0</td>\n",
       "      <td>1945</td>\n",
       "      <td>12</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502417</th>\n",
       "      <td>6025018</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>2</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502443</th>\n",
       "      <td>6025273</td>\n",
       "      <td>0</td>\n",
       "      <td>1947</td>\n",
       "      <td>7</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502454</th>\n",
       "      <td>6025381</td>\n",
       "      <td>1</td>\n",
       "      <td>1968</td>\n",
       "      <td>1</td>\n",
       "      <td>4002</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8621 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          f.eid f.31.0.0 f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0  \\\n",
       "68      1000697        0     1965       10        4001        <NA>   \n",
       "143     1001447        0     1961        6        4001        <NA>   \n",
       "145     1001465        0     1942        6        2001        <NA>   \n",
       "199     1002004        1     1957        2        4002        <NA>   \n",
       "234     1002354        0     1962        9        4001        <NA>   \n",
       "...         ...      ...      ...      ...         ...         ...   \n",
       "502221  6023054        0     1941        9        4001        <NA>   \n",
       "502390  6024740        0     1945       12        4001        <NA>   \n",
       "502417  6025018        0     1960        2        4001        <NA>   \n",
       "502443  6025273        0     1947        7        4001        <NA>   \n",
       "502454  6025381        1     1968        1        4002        <NA>   \n",
       "\n",
       "       f.21000.2.0 f.22001.0.0 f.22006.0.0 ethnicity  \n",
       "68            <NA>           0        <NA>      4001  \n",
       "143           <NA>           0        <NA>      4001  \n",
       "145           <NA>           0        <NA>      2001  \n",
       "199           <NA>           1        <NA>      4002  \n",
       "234           <NA>           0        <NA>      4001  \n",
       "...            ...         ...         ...       ...  \n",
       "502221        <NA>           0        <NA>      4001  \n",
       "502390        <NA>           0        <NA>      4001  \n",
       "502417        <NA>           0        <NA>      4001  \n",
       "502443        <NA>           0        <NA>      4001  \n",
       "502454        <NA>           1        <NA>      4002  \n",
       "\n",
       "[8621 rows x 10 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_afr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "a800d78d-b4d0-4a0b-8110-53ba74a2b5ef",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr = df_qc_afr.rename(columns={'f.eid': 'IID', 'f.31.0.0': 'sex'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "9f830821-a251-4c79-84f7-d6b443e7004b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr['FID'] = df_qc_afr['IID']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "b6387e09-b51f-4c5f-b989-e8c30b870a30",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>f.34.0.0</th>\n",
       "      <th>f.52.0.0</th>\n",
       "      <th>f.21000.0.0</th>\n",
       "      <th>f.21000.1.0</th>\n",
       "      <th>f.21000.2.0</th>\n",
       "      <th>f.22001.0.0</th>\n",
       "      <th>f.22006.0.0</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>FID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>1000697</td>\n",
       "      <td>0</td>\n",
       "      <td>1965</td>\n",
       "      <td>10</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>1000697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>1001447</td>\n",
       "      <td>0</td>\n",
       "      <td>1961</td>\n",
       "      <td>6</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>1001447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>1001465</td>\n",
       "      <td>0</td>\n",
       "      <td>1942</td>\n",
       "      <td>6</td>\n",
       "      <td>2001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>2001</td>\n",
       "      <td>1001465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>1002004</td>\n",
       "      <td>1</td>\n",
       "      <td>1957</td>\n",
       "      <td>2</td>\n",
       "      <td>4002</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4002</td>\n",
       "      <td>1002004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>1002354</td>\n",
       "      <td>0</td>\n",
       "      <td>1962</td>\n",
       "      <td>9</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>1002354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502221</th>\n",
       "      <td>6023054</td>\n",
       "      <td>0</td>\n",
       "      <td>1941</td>\n",
       "      <td>9</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>6023054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502390</th>\n",
       "      <td>6024740</td>\n",
       "      <td>0</td>\n",
       "      <td>1945</td>\n",
       "      <td>12</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>6024740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502417</th>\n",
       "      <td>6025018</td>\n",
       "      <td>0</td>\n",
       "      <td>1960</td>\n",
       "      <td>2</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>6025018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502443</th>\n",
       "      <td>6025273</td>\n",
       "      <td>0</td>\n",
       "      <td>1947</td>\n",
       "      <td>7</td>\n",
       "      <td>4001</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4001</td>\n",
       "      <td>6025273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502454</th>\n",
       "      <td>6025381</td>\n",
       "      <td>1</td>\n",
       "      <td>1968</td>\n",
       "      <td>1</td>\n",
       "      <td>4002</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>1</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>4002</td>\n",
       "      <td>6025381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8621 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            IID sex f.34.0.0 f.52.0.0 f.21000.0.0 f.21000.1.0 f.21000.2.0  \\\n",
       "68      1000697   0     1965       10        4001        <NA>        <NA>   \n",
       "143     1001447   0     1961        6        4001        <NA>        <NA>   \n",
       "145     1001465   0     1942        6        2001        <NA>        <NA>   \n",
       "199     1002004   1     1957        2        4002        <NA>        <NA>   \n",
       "234     1002354   0     1962        9        4001        <NA>        <NA>   \n",
       "...         ...  ..      ...      ...         ...         ...         ...   \n",
       "502221  6023054   0     1941        9        4001        <NA>        <NA>   \n",
       "502390  6024740   0     1945       12        4001        <NA>        <NA>   \n",
       "502417  6025018   0     1960        2        4001        <NA>        <NA>   \n",
       "502443  6025273   0     1947        7        4001        <NA>        <NA>   \n",
       "502454  6025381   1     1968        1        4002        <NA>        <NA>   \n",
       "\n",
       "       f.22001.0.0 f.22006.0.0 ethnicity      FID  \n",
       "68               0        <NA>      4001  1000697  \n",
       "143              0        <NA>      4001  1001447  \n",
       "145              0        <NA>      2001  1001465  \n",
       "199              1        <NA>      4002  1002004  \n",
       "234              0        <NA>      4001  1002354  \n",
       "...            ...         ...       ...      ...  \n",
       "502221           0        <NA>      4001  6023054  \n",
       "502390           0        <NA>      4001  6024740  \n",
       "502417           0        <NA>      4001  6025018  \n",
       "502443           0        <NA>      4001  6025273  \n",
       "502454           1        <NA>      4002  6025381  \n",
       "\n",
       "[8621 rows x 11 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_afr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "69209e49-662e-4587-9978-5e18dce71e7d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr[[\"FID\",\"IID\",\"ethnicity\"]].to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_8621.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9b64cdcc-20ab-40ff-951d-ef778e52f850",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr=pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_8621.iid\", sep=\"\\t\", dtype=\"string\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "e10f8d92-5348-463a-a8e5-2c7014ec194e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "call95_african= pd.read_csv('~/UKBiobank/genotype_files_processed/010622_african_9096ind/cache/UKB_genotypedatadownloaded083019.genotype_files_processed.filtered.extracted.010622_african_9096ind.filtered.mindrem.id', skiprows=[0], names=[\"FID\", \"IID\"], dtype=\"string\", sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "30a3cdc6-9276-4bd0-a3a0-ea7a72be13da",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3656538</td>\n",
       "      <td>3656538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3733695</td>\n",
       "      <td>3733695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4958925</td>\n",
       "      <td>4958925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5991763</td>\n",
       "      <td>5991763</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID      IID\n",
       "0  3656538  3656538\n",
       "1  3733695  3733695\n",
       "2  4958925  4958925\n",
       "3  5991763  5991763"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "call95_african"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4168f25a-5fef-47fe-bc3d-b2351ae91aec",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_call95=df_qc_afr[~df_qc_afr.FID.isin(call95_african.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f69c74ef-6af5-4194-8285-8af520599529",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_call95.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_call95_8617.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b811a1cc-d9b1-4658-b01f-dc0d05b09ea8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3304568</td>\n",
       "      <td>3304568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4094127</td>\n",
       "      <td>4094127</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0        1\n",
       "0  3304568  3304568\n",
       "1  4094127  4094127"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Relatedness per UKB variable f.22021.0.0>1\n",
    "rel_african= pd.read_csv(\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/010622_african_9096ind/010722_sampleQC_IID_african_related.txt\", sep='\\t', dtype=\"string\", header=None)\n",
    "rel_african"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "316d3f4d-7ef3-4618-afc4-af3ade0009d9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000697</td>\n",
       "      <td>1000697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001447</td>\n",
       "      <td>1001447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1006879</td>\n",
       "      <td>1006879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1009579</td>\n",
       "      <td>1009579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1011881</td>\n",
       "      <td>1011881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1336</th>\n",
       "      <td>6005674</td>\n",
       "      <td>6005674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1337</th>\n",
       "      <td>6013990</td>\n",
       "      <td>6013990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1338</th>\n",
       "      <td>6017509</td>\n",
       "      <td>6017509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1339</th>\n",
       "      <td>6018745</td>\n",
       "      <td>6018745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1340</th>\n",
       "      <td>6019021</td>\n",
       "      <td>6019021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1341 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          FID      IID\n",
       "0     1000697  1000697\n",
       "1     1001447  1001447\n",
       "2     1006879  1006879\n",
       "3     1009579  1009579\n",
       "4     1011881  1011881\n",
       "...       ...      ...\n",
       "1336  6005674  6005674\n",
       "1337  6013990  6013990\n",
       "1338  6017509  6017509\n",
       "1339  6018745  6018745\n",
       "1340  6019021  6019021\n",
       "\n",
       "[1341 rows x 2 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Relatedness per king calculation\n",
    "rel_king_african = pd.read_csv(\"~/UKBiobank/genotype_files_processed/010622_african_9096ind/010722_king_african/UKB_genotypedatadownloaded083019.010722_sample_var_final_qc.filtered.extracted.010722_king_african.related_id\", names=[\"FID\", \"IID\"],sep=\" \", dtype=\"string\", header=None)\n",
    "rel_king_african"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "96d1f0f2-0bed-4b7f-b2c4-8da41ae6cff0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_unrel=df_qc_afr[~df_qc_afr.FID.isin(call95_african.FID) & ~df_qc_afr.FID.isin(rel_king_african.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "35271b4d-6e6f-4413-a0cd-1c00e7a762ea",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1001465</td>\n",
       "      <td>1001465</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1002004</td>\n",
       "      <td>1002004</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1002354</td>\n",
       "      <td>1002354</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1002390</td>\n",
       "      <td>1002390</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1002608</td>\n",
       "      <td>1002608</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8616</th>\n",
       "      <td>6023054</td>\n",
       "      <td>6023054</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8617</th>\n",
       "      <td>6024740</td>\n",
       "      <td>6024740</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8618</th>\n",
       "      <td>6025018</td>\n",
       "      <td>6025018</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8619</th>\n",
       "      <td>6025273</td>\n",
       "      <td>6025273</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8620</th>\n",
       "      <td>6025381</td>\n",
       "      <td>6025381</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7276 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          FID      IID ethnicity\n",
       "2     1001465  1001465      2001\n",
       "3     1002004  1002004      4002\n",
       "4     1002354  1002354      4001\n",
       "5     1002390  1002390      4002\n",
       "6     1002608  1002608      4001\n",
       "...       ...      ...       ...\n",
       "8616  6023054  6023054      4001\n",
       "8617  6024740  6024740      4001\n",
       "8618  6025018  6025018      4001\n",
       "8619  6025273  6025273      4001\n",
       "8620  6025381  6025381      4002\n",
       "\n",
       "[7276 rows x 3 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_afr_unrel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d608aed0-770b-4736-a0a5-5c5cfa8861eb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_unrel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_unrel_7276.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "bca0b699-8e54-433d-b88f-886b616abfb1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_rel=df_qc_afr[~df_qc_afr.FID.isin(call95_african.FID) & df_qc_afr.FID.isin(rel_king_african.FID)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "7b296730-0e2e-41d9-9edf-bc208201414e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>ethnicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1000697</td>\n",
       "      <td>1000697</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1001447</td>\n",
       "      <td>1001447</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1006879</td>\n",
       "      <td>1006879</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1009579</td>\n",
       "      <td>1009579</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1011881</td>\n",
       "      <td>1011881</td>\n",
       "      <td>4001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8589</th>\n",
       "      <td>6005674</td>\n",
       "      <td>6005674</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8600</th>\n",
       "      <td>6013990</td>\n",
       "      <td>6013990</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8606</th>\n",
       "      <td>6017509</td>\n",
       "      <td>6017509</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8608</th>\n",
       "      <td>6018745</td>\n",
       "      <td>6018745</td>\n",
       "      <td>4002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8609</th>\n",
       "      <td>6019021</td>\n",
       "      <td>6019021</td>\n",
       "      <td>2001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1341 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          FID      IID ethnicity\n",
       "0     1000697  1000697      4001\n",
       "1     1001447  1001447      4001\n",
       "14    1006879  1006879      4001\n",
       "19    1009579  1009579      2001\n",
       "22    1011881  1011881      4001\n",
       "...       ...      ...       ...\n",
       "8589  6005674  6005674      2001\n",
       "8600  6013990  6013990      4002\n",
       "8606  6017509  6017509      2001\n",
       "8608  6018745  6018745      4002\n",
       "8609  6019021  6019021      2001\n",
       "\n",
       "[1341 rows x 3 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_qc_afr_rel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4a79b4ea-7e8b-48a3-a0de-ba4970355c51",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df_qc_afr_rel.to_csv(\"/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_rel_1341.iid\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8393a0b3-1313-4b54-a993-50b6d892990b",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "#### Select individuals and remove HWE <1e-15"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c8154d8f-33f0-4a77-a2b4-27874efdb42f",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/hg38_african_variantqc2_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w15f2a42f363f6ee3) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38\n",
    "# bfile with variant_qc_1 N=486416 and variants=674,489 hg38\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To keep unrelated samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_unrel_7276.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/hg38_african_variantqc2_$(date +\"%Y-%m-%d\").sbatch\n",
    "maf_filter=0.01\n",
    "geno_filter=0.01\n",
    "hwe_filter=1e-15\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_012323'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b3eebaa-c10f-4556-b00e-d0b7ab2cbd0a",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Select individuals and remove variants HWE < 1e-8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a87a4aef-c031-4969-a322-1928dbf4e905",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/hwe_1e-8/hg38_african_variantqc2_hwe1e-8_2023-01-30.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w2054e3258676deb8) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/hwe_1e-8\n",
    "# bfile with variant_qc_1 N=486416 and variants=674,489 hg38\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To keep unrelated samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_unrel_7276.iid\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_african_variantqc2_hwe1e-8_$(date +\"%Y-%m-%d\").sbatch\n",
    "maf_filter=0.01\n",
    "geno_filter=0.01\n",
    "hwe_filter=1e-8\n",
    "# Set mind filter to 0 not to filter out more individuals based on sample missingness\n",
    "mind_filter=0\n",
    "mem='30G'\n",
    "name='hg38_012323'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1cb1252-6d6f-4424-9a0f-f8e155b584c3",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Final file africans after individuals and variant qc_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "985f92d8-5b6c-4a54-91b9-e6e5e677ac16",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/hg38_african_variantqc2_hwe1e-8_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wadc7b85a19596c3f) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This is the path to the data transferred from Yale\n",
    "UKBB_yale=/mnt/mfs/statgen/archive/UKBiobank_Yale_transfer\n",
    "UKBB_PATH=/mnt/mfs/statgen/UKBiobank\n",
    "USER_PATH=$HOME/project\n",
    "OUT_PATH=$USER_PATH/UKBB_GWAS_dev/output\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "#Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files\n",
    "# bfile with variant_qc_1 N=486416 and variants=674,489 hg38\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.bed\n",
    "#To keep related and unrelated samples\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_call95_8617.iid\n",
    "# To keep variants after geno=0.01, maf=0.01 and hwe=1e-15\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.bim\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "gwasqc_sbatch=$cwd/hg38_african_variantqc2_hwe1e-8_$(date +\"%Y-%m-%d\").sbatch\n",
    "mem='30G'\n",
    "name='hg38_012323'\n",
    "job_size=1\n",
    "numThreads=2\n",
    "geno=0.0\n",
    "mind=0.0\n",
    "hwe=0.0\n",
    "\n",
    "gwasqc1_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --geno_filter $geno\n",
    "    --mind_filter $mind\n",
    "    --hwe_filter $hwe\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwasqc_sbatch \\\n",
    "    --args \"$gwasqc1_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6d965b1-879b-46cd-a01c-778a592d1bec",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 1. PCA for africans: keep unrelated and do LD prunning (get bed file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "74270f93-eddf-43c9-831a-fd7152f2efaa",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/ldprun_unrelated/ldprun_unrelated_asian_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w5c56d09b5cc2b696) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/ldprun_unrelated\n",
    "## Use the qc version of the genotype array with the already filtered asian individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/*.bed\n",
    "#To keep the samples of asian and unrelated individuals only\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_unrel_7276.iid\n",
    "#GWAS QC variables: leave all the variables in 0 so there's no more filtering in the already filtered data\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#LD prunning variables\n",
    "window=50\n",
    "shift=10\n",
    "r2=0.1\n",
    "gwas_sbatch=$cwd/ldprun_unrelated_african_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_args=\"\"\"qc\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --window $window\n",
    "    --shift $shift\n",
    "    --r2 $r2\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27433ce9-3d87-42cb-a5fb-56a260177ccd",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 2 PCA: run for unrelated African"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a8ae9790-f594-416a-ab60-f856c32584f6",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/pca_unrelated/flashpca_african_unrelated_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w2d2f5cd8191ffc9e) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/pca_unrelated\n",
    "#This is the bfile originated after filtering unrelated individuals and pruning\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/ldprun_unrelated/*.filtered.prune.bed\n",
    "#You need to input the correct file containing only the unrelated individuals, otherwise you'll get NA as a label\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_unrel_7276.iid\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sos=~/project/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "pca_sbatch=$cwd/flashpca_african_unrelated_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "maha_k=5\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "homogeneous=TRUE\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --pops $pops\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16d89b4b-9419-48b4-b252-b96b5294ec87",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 3. Get the bed file for related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "0e554e49-1e4c-49c9-a4ab-cabacc61ba38",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m (index=0) is \u001b[32mignored\u001b[0m due to saved signature\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/related_african/flashpca_african_related_qc_genoarray_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wdcf68ccfe350a5de) is ignored with 1 ignored step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/related_african\n",
    "## Use qc'ed genotype array\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/*.sorted.filtered.extracted.bed\n",
    "keep_samples=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_rel_1341.iid\n",
    "#Keep the same variants as above\n",
    "keep_variants=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/ldprun_unrelated/*.sorted.filtered.extracted.filtered.prune.in\n",
    "#GWAS QC variables\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "gwas_sbatch=$cwd/flashpca_african_related_qc_genoarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "numThreads=20\n",
    "mem='30G'\n",
    "\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d4633a0-c92e-4535-94be-d9a5610e622a",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step 4. PCA project back related individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "197f82f1-619c-4c02-be31-803374cde6c0",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/flashpca_african_related_genoarray_projected_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w2f210550b4dac3c1) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/related_african/*.bed\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_call95_8617.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/pca_unrelated/*.pca.rds\n",
    "pca_sbatch=$cwd/flashpca_african_related_genoarray_projected_$(date +\"%Y-%m-%d\").sbatch\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "k=10\n",
    "maha_k=5\n",
    "prob=0.997\n",
    "pval=0.05\n",
    "## set the --homogeneous TRUE options to consider all the pops like one \n",
    "homogeneous=TRUE\n",
    "## For the plot you need to use the *.projected.rds and not the *.projected.mahalanobis.rds\n",
    "#plot_data=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.rds\n",
    "#outlier_file=$UKBB_PATH/results/070921_pca_genotype_array/white_expanded_06_30_21_genoarray_projected/030821_ukb42495_exomed_white_189010ind.pheno.white_expanded_06_30_21_genoarray_projected.pca.projected.outliers\n",
    "\n",
    "\n",
    "pca_args=\"\"\"project_samples\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --pca_model $pca_model\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --prob $prob\n",
    "    --pval $pval\n",
    "    --homogeneous $homogeneous\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be9e2043-29f7-4530-b685-1fb0e592bb80",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### step. 5 PCA: plot and look for outliers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "041aeaa0-5c0a-479d-a375-63ae2bb03767",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/plot/flashpca_african_related_genoarray_plot_2023-01-31.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w106abc66920250ef) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Columbia's cluster\n",
    "cwd=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/plot\n",
    "#This is the bfile originated after filtering related individuals\n",
    "genoFile=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/related_african/*.bed\n",
    "## I had to modify the original file to add a super_pop and replace ethnicity for pop\n",
    "phenoFile=/mnt/mfs/statgen/UKBiobank/phenotype_files/african_IID/012323_ukb47922_african_qc_call95_8617.iid\n",
    "pca_model=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/pca_unrelated/*.pca.rds\n",
    "plot_data=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/*.pca.projected.rds\n",
    "outlier_file=/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/*.pca.projected.outliers\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_african_related_genoarray_plot_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=10\n",
    "\n",
    "pca_args=\"\"\"plot_pca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --plot_data $plot_data\n",
    "    --outlier_file $outlier_file\n",
    "    --k $k\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg  \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86e2a498-a0ba-45d6-88fe-e148b6c3b6ac",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Get individual ID and variants text files and final qc'ed bed files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "303f7de5-eb18-48f5-bbbd-be4fede94d16",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/final_genoarray_africans_hg38.sh\" target=\"_blank\">/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/final_genoarray_africans_hg38.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/final_genoarray_africans_hg38.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save -f /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/final_genoarray_africans_hg38.sh\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=36:00:00\n",
    "#$ -l h_vmem=30G\n",
    "#$ -N final_genoarray_africans_hg38_2023-01-31\n",
    "#$ -o /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg388/final_files_no_outliers/final_genoarray_africans_hg38_2023-01-31.out\n",
    "#$ -e /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/final_genoarray_africans_hg38_2023-01-31.err\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load PLINK/2.0\n",
    "plink2 \\\n",
    "    --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted \\\n",
    "    --remove /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/012323_ukb47922_african_qc_call95_8617.pca.projected.outliers \\\n",
    "    --write-snplist --write-samples --no-id-header \\\n",
    "    --threads 20 \\\n",
    "    --make-bed \\\n",
    "    --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "0fd048c0-fdcf-468a-a1df-7650d963a809",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a2.3LM 64-bit Intel (24 Jan 2020)   www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted\n",
      "  --make-bed\n",
      "  --no-id-header\n",
      "  --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN\n",
      "  --remove /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/012323_ukb47922_african_qc_call95_8617.pca.projected.outliers\n",
      "  --threads 20\n",
      "  --write-samples\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Tue Jan 31 15:12:41 2023\n",
      "257482 MiB RAM detected; reserving 128741 MiB for main workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 20 threads (change this with --threads).\n",
      "8617 samples (5001 females, 3616 males; 8617 founders) loaded from\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.fam.\n",
      "346727 variants loaded from\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted.bim.\n",
      "Note: No phenotype data present.\n",
      "--remove: 8591 samples remaining.\n",
      "8591 samples (4984 females, 3607 males; 8591 founders) remaining after main\n",
      "filters.\n",
      "--write-samples: Sample IDs written to\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.id\n",
      ".\n",
      "--write-snplist: Variant IDs written to\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.snplist\n",
      ".\n",
      "Writing\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.fam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.bim\n",
      "... done.\n",
      "Writing\n",
      "/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN.bed\n",
      "... 1837567594done.\n",
      "End time: Tue Jan 31 15:12:42 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load PLINK/2.0\n",
    "plink2 \\\n",
    "    --bfile /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.sorted.filtered.extracted \\\n",
    "    --remove /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/PCA/project_related_african/012323_ukb47922_african_qc_call95_8617.pca.projected.outliers \\\n",
    "    --write-snplist --write-samples --no-id-header \\\n",
    "    --threads 20 \\\n",
    "    --make-bed \\\n",
    "    --out /mnt/mfs/statgen/UKBiobank/genotype_files_processed/012323_african_9096ind_hg38/final_files_no_outliers/UKB_genotypedatadownloaded083019_hg38_8591ind_346727var_AFRICAN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "0d4b584f-ec17-4efd-a928-9fbeb4fcf7c8",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "file <- \"/mnt/mfs/statgen/UKBiobank/genotype_files_processed/012023_hg38_674489var_s486416ind/UKB_genotypedatadownloaded083019_hg38_674489variants_486416samples.bim\"\n",
    "\n",
    "df <- read.table(file, header= FALSE, stringsAsFactors = FALSE)\n",
    "colnames(df) <- c(\"chr\", \"var\",\"CM\",\"pos\", \"a1\", \"a2\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "f88f12d6-59c8-4477-ab31-f38f9378302b",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 20665 x 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chr</th><th scope=col>var</th><th scope=col>CM</th><th scope=col>pos</th><th scope=col>a1</th><th scope=col>a2</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>614991</th><td>19</td><td>chr19:260970:A:T</td><td>0</td><td>260970</td><td>A</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>614992</th><td>19</td><td>chr19:267039:C:T</td><td>0</td><td>267039</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>614993</th><td>19</td><td>chr19:267614:A:C</td><td>0</td><td>267614</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>614994</th><td>19</td><td>chr19:277715:C:G</td><td>0</td><td>277715</td><td>C</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>614995</th><td>19</td><td>chr19:280712:T:C</td><td>0</td><td>280712</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>614996</th><td>19</td><td>chr19:282181:T:C</td><td>0</td><td>282181</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>614997</th><td>19</td><td>chr19:282753:A:G</td><td>0</td><td>282753</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>614998</th><td>19</td><td>chr19:287703:A:G</td><td>0</td><td>287703</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>614999</th><td>19</td><td>chr19:288062:A:G</td><td>0</td><td>288062</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615000</th><td>19</td><td>chr19:288123:T:C</td><td>0</td><td>288123</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615001</th><td>19</td><td>chr19:288374:C:T</td><td>0</td><td>288374</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615002</th><td>19</td><td>chr19:288771:G:T</td><td>0</td><td>288771</td><td>G</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615003</th><td>19</td><td>chr19:289047:G:A</td><td>0</td><td>289047</td><td>G</td><td>A</td></tr>\n",
       "\t<tr><th scope=row>615004</th><td>19</td><td>chr19:289399:C:T</td><td>0</td><td>289399</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615005</th><td>19</td><td>chr19:292082:T:C</td><td>0</td><td>292082</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615006</th><td>19</td><td>chr19:292583:A:G</td><td>0</td><td>292583</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615007</th><td>19</td><td>chr19:293072:A:G</td><td>0</td><td>293072</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615008</th><td>19</td><td>chr19:294212:C:T</td><td>0</td><td>294212</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615009</th><td>19</td><td>chr19:294306:C:T</td><td>0</td><td>294306</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615010</th><td>19</td><td>chr19:294798:T:C</td><td>0</td><td>294798</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615011</th><td>19</td><td>chr19:295597:T:C</td><td>0</td><td>295597</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615012</th><td>19</td><td>chr19:295676:A:G</td><td>0</td><td>295676</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615013</th><td>19</td><td>chr19:296458:A:G</td><td>0</td><td>296458</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615014</th><td>19</td><td>chr19:296777:A:G</td><td>0</td><td>296777</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615015</th><td>19</td><td>chr19:302321:T:C</td><td>0</td><td>302321</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615016</th><td>19</td><td>chr19:307515:A:G</td><td>0</td><td>307515</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>615017</th><td>19</td><td>chr19:308045:C:T</td><td>0</td><td>308045</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>615018</th><td>19</td><td>chr19:308631:T:C</td><td>0</td><td>308631</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615019</th><td>19</td><td>chr19:314755:T:C</td><td>0</td><td>314755</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>615020</th><td>19</td><td>chr19:316586:A:G</td><td>0</td><td>316586</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>...</th><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td><td>...</td></tr>\n",
       "\t<tr><th scope=row>635627</th><td>19</td><td>chr19:58517318:A:G</td><td>0</td><td>58517318</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635628</th><td>19</td><td>chr19:58519213:T:G</td><td>0</td><td>58519213</td><td>T</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635629</th><td>19</td><td>chr19:58527517:T:G</td><td>0</td><td>58527517</td><td>T</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635630</th><td>19</td><td>chr19:58538989:T:G</td><td>0</td><td>58538989</td><td>T</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635631</th><td>19</td><td>chr19:58542921:A:G</td><td>0</td><td>58542921</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635632</th><td>19</td><td>chr19:58548152:C:A</td><td>0</td><td>58548152</td><td>C</td><td>A</td></tr>\n",
       "\t<tr><th scope=row>635633</th><td>19</td><td>chr19:58548362:A:G</td><td>0</td><td>58548362</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635634</th><td>19</td><td>chr19:58550025:T:C</td><td>0</td><td>58550025</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635635</th><td>19</td><td>chr19:58551790:A:G</td><td>0</td><td>58551790</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635636</th><td>19</td><td>chr19:58555024:G:C</td><td>0</td><td>58555024</td><td>G</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635637</th><td>19</td><td>chr19:58555551:A:C</td><td>0</td><td>58555551</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635638</th><td>19</td><td>chr19:58556256:T:C</td><td>0</td><td>58556256</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635639</th><td>19</td><td>chr19:58557118:A:C</td><td>0</td><td>58557118</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635640</th><td>19</td><td>chr19:58559211:T:G</td><td>0</td><td>58559211</td><td>T</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635641</th><td>19</td><td>chr19:58559549:C:T</td><td>0</td><td>58559549</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>635642</th><td>19</td><td>chr19:58562157:C:T</td><td>0</td><td>58562157</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>635643</th><td>19</td><td>chr19:58562434:A:G</td><td>0</td><td>58562434</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635644</th><td>19</td><td>chr19:58562968:A:C</td><td>0</td><td>58562968</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635645</th><td>19</td><td>chr19:58563505:A:C</td><td>0</td><td>58563505</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635646</th><td>19</td><td>chr19:58571001:T:C</td><td>0</td><td>58571001</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635647</th><td>19</td><td>chr19:58571158:A:G</td><td>0</td><td>58571158</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635648</th><td>19</td><td>chr19:58571238:T:C</td><td>0</td><td>58571238</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635649</th><td>19</td><td>chr19:58571901:A:G</td><td>0</td><td>58571901</td><td>A</td><td>G</td></tr>\n",
       "\t<tr><th scope=row>635650</th><td>19</td><td>chr19:58574094:C:T</td><td>0</td><td>58574094</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>635651</th><td>19</td><td>chr19:58576184:A:C</td><td>0</td><td>58576184</td><td>A</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635652</th><td>19</td><td>chr19:58578425:T:C</td><td>0</td><td>58578425</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635653</th><td>19</td><td>chr19:58581872:T:C</td><td>0</td><td>58581872</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635654</th><td>19</td><td>chr19:58582097:C:T</td><td>0</td><td>58582097</td><td>C</td><td>T</td></tr>\n",
       "\t<tr><th scope=row>635655</th><td>19</td><td>chr19:58583007:T:C</td><td>0</td><td>58583007</td><td>T</td><td>C</td></tr>\n",
       "\t<tr><th scope=row>635656</th><td>19</td><td>chr19:58586566:A:G</td><td>0</td><td>58586566</td><td>A</td><td>G</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 20665 x 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & chr & var & CM & pos & a1 & a2\\\\\n",
       "  & <int> & <chr> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t614991 & 19 & chr19:260970:A:T & 0 & 260970 & A & T\\\\\n",
       "\t614992 & 19 & chr19:267039:C:T & 0 & 267039 & C & T\\\\\n",
       "\t614993 & 19 & chr19:267614:A:C & 0 & 267614 & A & C\\\\\n",
       "\t614994 & 19 & chr19:277715:C:G & 0 & 277715 & C & G\\\\\n",
       "\t614995 & 19 & chr19:280712:T:C & 0 & 280712 & T & C\\\\\n",
       "\t614996 & 19 & chr19:282181:T:C & 0 & 282181 & T & C\\\\\n",
       "\t614997 & 19 & chr19:282753:A:G & 0 & 282753 & A & G\\\\\n",
       "\t614998 & 19 & chr19:287703:A:G & 0 & 287703 & A & G\\\\\n",
       "\t614999 & 19 & chr19:288062:A:G & 0 & 288062 & A & G\\\\\n",
       "\t615000 & 19 & chr19:288123:T:C & 0 & 288123 & T & C\\\\\n",
       "\t615001 & 19 & chr19:288374:C:T & 0 & 288374 & C & T\\\\\n",
       "\t615002 & 19 & chr19:288771:G:T & 0 & 288771 & G & T\\\\\n",
       "\t615003 & 19 & chr19:289047:G:A & 0 & 289047 & G & A\\\\\n",
       "\t615004 & 19 & chr19:289399:C:T & 0 & 289399 & C & T\\\\\n",
       "\t615005 & 19 & chr19:292082:T:C & 0 & 292082 & T & C\\\\\n",
       "\t615006 & 19 & chr19:292583:A:G & 0 & 292583 & A & G\\\\\n",
       "\t615007 & 19 & chr19:293072:A:G & 0 & 293072 & A & G\\\\\n",
       "\t615008 & 19 & chr19:294212:C:T & 0 & 294212 & C & T\\\\\n",
       "\t615009 & 19 & chr19:294306:C:T & 0 & 294306 & C & T\\\\\n",
       "\t615010 & 19 & chr19:294798:T:C & 0 & 294798 & T & C\\\\\n",
       "\t615011 & 19 & chr19:295597:T:C & 0 & 295597 & T & C\\\\\n",
       "\t615012 & 19 & chr19:295676:A:G & 0 & 295676 & A & G\\\\\n",
       "\t615013 & 19 & chr19:296458:A:G & 0 & 296458 & A & G\\\\\n",
       "\t615014 & 19 & chr19:296777:A:G & 0 & 296777 & A & G\\\\\n",
       "\t615015 & 19 & chr19:302321:T:C & 0 & 302321 & T & C\\\\\n",
       "\t615016 & 19 & chr19:307515:A:G & 0 & 307515 & A & G\\\\\n",
       "\t615017 & 19 & chr19:308045:C:T & 0 & 308045 & C & T\\\\\n",
       "\t615018 & 19 & chr19:308631:T:C & 0 & 308631 & T & C\\\\\n",
       "\t615019 & 19 & chr19:314755:T:C & 0 & 314755 & T & C\\\\\n",
       "\t615020 & 19 & chr19:316586:A:G & 0 & 316586 & A & G\\\\\n",
       "\t... & ... & ... & ... & ... & ... & ...\\\\\n",
       "\t635627 & 19 & chr19:58517318:A:G & 0 & 58517318 & A & G\\\\\n",
       "\t635628 & 19 & chr19:58519213:T:G & 0 & 58519213 & T & G\\\\\n",
       "\t635629 & 19 & chr19:58527517:T:G & 0 & 58527517 & T & G\\\\\n",
       "\t635630 & 19 & chr19:58538989:T:G & 0 & 58538989 & T & G\\\\\n",
       "\t635631 & 19 & chr19:58542921:A:G & 0 & 58542921 & A & G\\\\\n",
       "\t635632 & 19 & chr19:58548152:C:A & 0 & 58548152 & C & A\\\\\n",
       "\t635633 & 19 & chr19:58548362:A:G & 0 & 58548362 & A & G\\\\\n",
       "\t635634 & 19 & chr19:58550025:T:C & 0 & 58550025 & T & C\\\\\n",
       "\t635635 & 19 & chr19:58551790:A:G & 0 & 58551790 & A & G\\\\\n",
       "\t635636 & 19 & chr19:58555024:G:C & 0 & 58555024 & G & C\\\\\n",
       "\t635637 & 19 & chr19:58555551:A:C & 0 & 58555551 & A & C\\\\\n",
       "\t635638 & 19 & chr19:58556256:T:C & 0 & 58556256 & T & C\\\\\n",
       "\t635639 & 19 & chr19:58557118:A:C & 0 & 58557118 & A & C\\\\\n",
       "\t635640 & 19 & chr19:58559211:T:G & 0 & 58559211 & T & G\\\\\n",
       "\t635641 & 19 & chr19:58559549:C:T & 0 & 58559549 & C & T\\\\\n",
       "\t635642 & 19 & chr19:58562157:C:T & 0 & 58562157 & C & T\\\\\n",
       "\t635643 & 19 & chr19:58562434:A:G & 0 & 58562434 & A & G\\\\\n",
       "\t635644 & 19 & chr19:58562968:A:C & 0 & 58562968 & A & C\\\\\n",
       "\t635645 & 19 & chr19:58563505:A:C & 0 & 58563505 & A & C\\\\\n",
       "\t635646 & 19 & chr19:58571001:T:C & 0 & 58571001 & T & C\\\\\n",
       "\t635647 & 19 & chr19:58571158:A:G & 0 & 58571158 & A & G\\\\\n",
       "\t635648 & 19 & chr19:58571238:T:C & 0 & 58571238 & T & C\\\\\n",
       "\t635649 & 19 & chr19:58571901:A:G & 0 & 58571901 & A & G\\\\\n",
       "\t635650 & 19 & chr19:58574094:C:T & 0 & 58574094 & C & T\\\\\n",
       "\t635651 & 19 & chr19:58576184:A:C & 0 & 58576184 & A & C\\\\\n",
       "\t635652 & 19 & chr19:58578425:T:C & 0 & 58578425 & T & C\\\\\n",
       "\t635653 & 19 & chr19:58581872:T:C & 0 & 58581872 & T & C\\\\\n",
       "\t635654 & 19 & chr19:58582097:C:T & 0 & 58582097 & C & T\\\\\n",
       "\t635655 & 19 & chr19:58583007:T:C & 0 & 58583007 & T & C\\\\\n",
       "\t635656 & 19 & chr19:58586566:A:G & 0 & 58586566 & A & G\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 20665 x 6\n",
       "\n",
       "| <!--/--> | chr &lt;int&gt; | var &lt;chr&gt; | CM &lt;dbl&gt; | pos &lt;int&gt; | a1 &lt;chr&gt; | a2 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 614991 | 19 | chr19:260970:A:T | 0 | 260970 | A | T |\n",
       "| 614992 | 19 | chr19:267039:C:T | 0 | 267039 | C | T |\n",
       "| 614993 | 19 | chr19:267614:A:C | 0 | 267614 | A | C |\n",
       "| 614994 | 19 | chr19:277715:C:G | 0 | 277715 | C | G |\n",
       "| 614995 | 19 | chr19:280712:T:C | 0 | 280712 | T | C |\n",
       "| 614996 | 19 | chr19:282181:T:C | 0 | 282181 | T | C |\n",
       "| 614997 | 19 | chr19:282753:A:G | 0 | 282753 | A | G |\n",
       "| 614998 | 19 | chr19:287703:A:G | 0 | 287703 | A | G |\n",
       "| 614999 | 19 | chr19:288062:A:G | 0 | 288062 | A | G |\n",
       "| 615000 | 19 | chr19:288123:T:C | 0 | 288123 | T | C |\n",
       "| 615001 | 19 | chr19:288374:C:T | 0 | 288374 | C | T |\n",
       "| 615002 | 19 | chr19:288771:G:T | 0 | 288771 | G | T |\n",
       "| 615003 | 19 | chr19:289047:G:A | 0 | 289047 | G | A |\n",
       "| 615004 | 19 | chr19:289399:C:T | 0 | 289399 | C | T |\n",
       "| 615005 | 19 | chr19:292082:T:C | 0 | 292082 | T | C |\n",
       "| 615006 | 19 | chr19:292583:A:G | 0 | 292583 | A | G |\n",
       "| 615007 | 19 | chr19:293072:A:G | 0 | 293072 | A | G |\n",
       "| 615008 | 19 | chr19:294212:C:T | 0 | 294212 | C | T |\n",
       "| 615009 | 19 | chr19:294306:C:T | 0 | 294306 | C | T |\n",
       "| 615010 | 19 | chr19:294798:T:C | 0 | 294798 | T | C |\n",
       "| 615011 | 19 | chr19:295597:T:C | 0 | 295597 | T | C |\n",
       "| 615012 | 19 | chr19:295676:A:G | 0 | 295676 | A | G |\n",
       "| 615013 | 19 | chr19:296458:A:G | 0 | 296458 | A | G |\n",
       "| 615014 | 19 | chr19:296777:A:G | 0 | 296777 | A | G |\n",
       "| 615015 | 19 | chr19:302321:T:C | 0 | 302321 | T | C |\n",
       "| 615016 | 19 | chr19:307515:A:G | 0 | 307515 | A | G |\n",
       "| 615017 | 19 | chr19:308045:C:T | 0 | 308045 | C | T |\n",
       "| 615018 | 19 | chr19:308631:T:C | 0 | 308631 | T | C |\n",
       "| 615019 | 19 | chr19:314755:T:C | 0 | 314755 | T | C |\n",
       "| 615020 | 19 | chr19:316586:A:G | 0 | 316586 | A | G |\n",
       "| ... | ... | ... | ... | ... | ... | ... |\n",
       "| 635627 | 19 | chr19:58517318:A:G | 0 | 58517318 | A | G |\n",
       "| 635628 | 19 | chr19:58519213:T:G | 0 | 58519213 | T | G |\n",
       "| 635629 | 19 | chr19:58527517:T:G | 0 | 58527517 | T | G |\n",
       "| 635630 | 19 | chr19:58538989:T:G | 0 | 58538989 | T | G |\n",
       "| 635631 | 19 | chr19:58542921:A:G | 0 | 58542921 | A | G |\n",
       "| 635632 | 19 | chr19:58548152:C:A | 0 | 58548152 | C | A |\n",
       "| 635633 | 19 | chr19:58548362:A:G | 0 | 58548362 | A | G |\n",
       "| 635634 | 19 | chr19:58550025:T:C | 0 | 58550025 | T | C |\n",
       "| 635635 | 19 | chr19:58551790:A:G | 0 | 58551790 | A | G |\n",
       "| 635636 | 19 | chr19:58555024:G:C | 0 | 58555024 | G | C |\n",
       "| 635637 | 19 | chr19:58555551:A:C | 0 | 58555551 | A | C |\n",
       "| 635638 | 19 | chr19:58556256:T:C | 0 | 58556256 | T | C |\n",
       "| 635639 | 19 | chr19:58557118:A:C | 0 | 58557118 | A | C |\n",
       "| 635640 | 19 | chr19:58559211:T:G | 0 | 58559211 | T | G |\n",
       "| 635641 | 19 | chr19:58559549:C:T | 0 | 58559549 | C | T |\n",
       "| 635642 | 19 | chr19:58562157:C:T | 0 | 58562157 | C | T |\n",
       "| 635643 | 19 | chr19:58562434:A:G | 0 | 58562434 | A | G |\n",
       "| 635644 | 19 | chr19:58562968:A:C | 0 | 58562968 | A | C |\n",
       "| 635645 | 19 | chr19:58563505:A:C | 0 | 58563505 | A | C |\n",
       "| 635646 | 19 | chr19:58571001:T:C | 0 | 58571001 | T | C |\n",
       "| 635647 | 19 | chr19:58571158:A:G | 0 | 58571158 | A | G |\n",
       "| 635648 | 19 | chr19:58571238:T:C | 0 | 58571238 | T | C |\n",
       "| 635649 | 19 | chr19:58571901:A:G | 0 | 58571901 | A | G |\n",
       "| 635650 | 19 | chr19:58574094:C:T | 0 | 58574094 | C | T |\n",
       "| 635651 | 19 | chr19:58576184:A:C | 0 | 58576184 | A | C |\n",
       "| 635652 | 19 | chr19:58578425:T:C | 0 | 58578425 | T | C |\n",
       "| 635653 | 19 | chr19:58581872:T:C | 0 | 58581872 | T | C |\n",
       "| 635654 | 19 | chr19:58582097:C:T | 0 | 58582097 | C | T |\n",
       "| 635655 | 19 | chr19:58583007:T:C | 0 | 58583007 | T | C |\n",
       "| 635656 | 19 | chr19:58586566:A:G | 0 | 58586566 | A | G |\n",
       "\n"
      ],
      "text/plain": [
       "       chr var                CM  pos      a1  a2 \n",
       "614991 19  chr19:260970:A:T   0   260970   A   T  \n",
       "614992 19  chr19:267039:C:T   0   267039   C   T  \n",
       "614993 19  chr19:267614:A:C   0   267614   A   C  \n",
       "614994 19  chr19:277715:C:G   0   277715   C   G  \n",
       "614995 19  chr19:280712:T:C   0   280712   T   C  \n",
       "614996 19  chr19:282181:T:C   0   282181   T   C  \n",
       "614997 19  chr19:282753:A:G   0   282753   A   G  \n",
       "614998 19  chr19:287703:A:G   0   287703   A   G  \n",
       "614999 19  chr19:288062:A:G   0   288062   A   G  \n",
       "615000 19  chr19:288123:T:C   0   288123   T   C  \n",
       "615001 19  chr19:288374:C:T   0   288374   C   T  \n",
       "615002 19  chr19:288771:G:T   0   288771   G   T  \n",
       "615003 19  chr19:289047:G:A   0   289047   G   A  \n",
       "615004 19  chr19:289399:C:T   0   289399   C   T  \n",
       "615005 19  chr19:292082:T:C   0   292082   T   C  \n",
       "615006 19  chr19:292583:A:G   0   292583   A   G  \n",
       "615007 19  chr19:293072:A:G   0   293072   A   G  \n",
       "615008 19  chr19:294212:C:T   0   294212   C   T  \n",
       "615009 19  chr19:294306:C:T   0   294306   C   T  \n",
       "615010 19  chr19:294798:T:C   0   294798   T   C  \n",
       "615011 19  chr19:295597:T:C   0   295597   T   C  \n",
       "615012 19  chr19:295676:A:G   0   295676   A   G  \n",
       "615013 19  chr19:296458:A:G   0   296458   A   G  \n",
       "615014 19  chr19:296777:A:G   0   296777   A   G  \n",
       "615015 19  chr19:302321:T:C   0   302321   T   C  \n",
       "615016 19  chr19:307515:A:G   0   307515   A   G  \n",
       "615017 19  chr19:308045:C:T   0   308045   C   T  \n",
       "615018 19  chr19:308631:T:C   0   308631   T   C  \n",
       "615019 19  chr19:314755:T:C   0   314755   T   C  \n",
       "615020 19  chr19:316586:A:G   0   316586   A   G  \n",
       "...    ... ...                ... ...      ... ...\n",
       "635627 19  chr19:58517318:A:G 0   58517318 A   G  \n",
       "635628 19  chr19:58519213:T:G 0   58519213 T   G  \n",
       "635629 19  chr19:58527517:T:G 0   58527517 T   G  \n",
       "635630 19  chr19:58538989:T:G 0   58538989 T   G  \n",
       "635631 19  chr19:58542921:A:G 0   58542921 A   G  \n",
       "635632 19  chr19:58548152:C:A 0   58548152 C   A  \n",
       "635633 19  chr19:58548362:A:G 0   58548362 A   G  \n",
       "635634 19  chr19:58550025:T:C 0   58550025 T   C  \n",
       "635635 19  chr19:58551790:A:G 0   58551790 A   G  \n",
       "635636 19  chr19:58555024:G:C 0   58555024 G   C  \n",
       "635637 19  chr19:58555551:A:C 0   58555551 A   C  \n",
       "635638 19  chr19:58556256:T:C 0   58556256 T   C  \n",
       "635639 19  chr19:58557118:A:C 0   58557118 A   C  \n",
       "635640 19  chr19:58559211:T:G 0   58559211 T   G  \n",
       "635641 19  chr19:58559549:C:T 0   58559549 C   T  \n",
       "635642 19  chr19:58562157:C:T 0   58562157 C   T  \n",
       "635643 19  chr19:58562434:A:G 0   58562434 A   G  \n",
       "635644 19  chr19:58562968:A:C 0   58562968 A   C  \n",
       "635645 19  chr19:58563505:A:C 0   58563505 A   C  \n",
       "635646 19  chr19:58571001:T:C 0   58571001 T   C  \n",
       "635647 19  chr19:58571158:A:G 0   58571158 A   G  \n",
       "635648 19  chr19:58571238:T:C 0   58571238 T   C  \n",
       "635649 19  chr19:58571901:A:G 0   58571901 A   G  \n",
       "635650 19  chr19:58574094:C:T 0   58574094 C   T  \n",
       "635651 19  chr19:58576184:A:C 0   58576184 A   C  \n",
       "635652 19  chr19:58578425:T:C 0   58578425 T   C  \n",
       "635653 19  chr19:58581872:T:C 0   58581872 T   C  \n",
       "635654 19  chr19:58582097:C:T 0   58582097 C   T  \n",
       "635655 19  chr19:58583007:T:C 0   58583007 T   C  \n",
       "635656 19  chr19:58586566:A:G 0   58586566 A   G  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df[df$chr == 19 ,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6c01d619-9ddd-477a-a858-0011f164f81b",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": [
    "df[df$gender == 'M',]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     {
      "name": "ipython",
      "version": 3
     }
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ],
    [
     "SoS",
     "sos",
     "",
     "",
     "sos"
    ]
   ],
   "version": "0.22.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
