{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "29fef0c3-37f1-4e94-8ff4-63a02e4a67af",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# Regenie"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74af9921-6530-4e1c-a1d1-d3d5703e07d5",
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb4ecf1c-a458-43e0-997f-6193f68687ba",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "[global]\n",
    "# the output directory for generated files\n",
    "parameter: cwd = path\n",
    "# Path to sample file\n",
    "parameter: sampleFile = path('.')\n",
    "# Genotype files in plink binary this is used for computing the GRM\n",
    "parameter: bfile = path\n",
    "# Path to bgen or bed files \n",
    "parameter: genoFile = str\n",
    "# Phenotype file for quantitative trait (BMI)\n",
    "parameter: phenoFile = []\n",
    "# Phenotype to be analyzed (specify the column)\n",
    "parameter: phenoCol = list\n",
    "# Covariate file path. Will use phenoFile if empty\n",
    "parameter: covarFile = path('.')\n",
    "# Summary statisticss format file path used for unifying output column names. Will not unify names if empty\n",
    "parameter: formatFile = path('.')\n",
    "# Qualitative covariates to be used in the analysis\n",
    "parameter: covarCol = []\n",
    "# Quantitative covariates to be used in the analysis\n",
    "parameter: qCovarCol = []\n",
    "# Specific number of threads to use\n",
    "parameter: numThreads = 2\n",
    "# Minimum MAF to be used\n",
    "parameter: bgenMinMAF = 0.001\n",
    "# Mimimum info score to be used\n",
    "parameter: bgenMinINFO = 0.8\n",
    "# For cluster jobs, number commands to run per job\n",
    "parameter: job_size = 1\n",
    "# The container with the lmm software. Can be either a dockerhub image or a singularity `sif` file.\n",
    "# Default is set to using dockerhub image\n",
    "parameter: container_lmm = 'statisticalgenetics/lmm:2.4'\n",
    "parameter: container_marp = 'gaow/marp'\n",
    "if not covarFile.is_file():\n",
    "    covarFile = phenoFile\n",
    "cwd = path(f\"{cwd:a}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "399771c4-eff9-4a42-aa4c-38d074bba31d",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Run REGENIE step 1: fitting the null\n",
    "[fit]\n",
    "# Size of the genotype blocks to be used \n",
    "parameter: bsize = 400\n",
    "# Path to temporarily store block predictions\n",
    "parameter: lowmem_dir = cwd\n",
    "# Specify that traits are binary with 0=control,1=case,NA=missing (default is quantitative)\n",
    "parameter: trait = 'bt'\n",
    "# extract and prepare phenotype & covariate files\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "dat = pd.read_csv(phenoFile, header=0, delimiter='\\t', dtype=str)\n",
    "dat = dat.replace(to_replace =np.nan, value =\"NA\")\n",
    "if len(phenoCol) > 0:    \n",
    "    dat.to_csv(f\"{cwd}/{phenoFile:bn}.regenie_phenotype\", sep='\\t', index=False, columns = ['FID', 'IID'] + phenoCol)\n",
    "dat = pd.read_csv(covarFile, header=0, delimiter='\\t', dtype=str)\n",
    "if len(covarCol) > 0 or len(qCovarCol) > 0:\n",
    "    dat = dat.dropna(subset=covarCol)\n",
    "    dat = dat.dropna(subset=qCovarCol)\n",
    "    dat.replace(to_replace =np.nan, value =\"NA\")\n",
    "    dat1 = pd.DataFrame(dat, columns = ['FID','IID'] + covarCol)\n",
    "    #dat1 = dat1.astype(int)\n",
    "    dat2 = pd.DataFrame(dat, columns = ['IID'] + qCovarCol)\n",
    "    merged_left = pd.merge(left=dat1, right=dat2, how='left', left_on='IID', right_on='IID')\n",
    "    merged_left.to_csv(f\"{cwd}/{phenoFile:bn}.regenie_covar\", sep=' ', index=False)\n",
    "depends: f'{cwd}/cache/{bfile:bn}.qc_pass.snplist', f'{cwd}/cache/{bfile:bn}.qc_pass.id'\n",
    "input: geno = bfile, pheno = f\"{cwd}/{phenoFile:bn}.regenie_phenotype\", covar = f\"{cwd}/{phenoFile:bn}.regenie_covar\", qc = output_from(\"regenie_qc\")\n",
    "output: f'{cwd}/{phenoFile:bn}_' + \"_\".join([x for x in phenoCol]) + '.regenie_pred.list'\n",
    "task: trunk_workers = 1, trunk_size = job_size, walltime = '12h', mem = '15G', cores = numThreads, tags = f'{step_name}_{_output[0]:bn}'\n",
    "bash: container=container_lmm, expand = \"${ }\", stderr = f'{_output:n}.stderr', stdout = f'{_output:n}.stdout', volumes = [f\"{lowmem_dir:a}:{lowmem_dir:a}\"]\n",
    "    regenie \\\n",
    "      --step 1 \\\n",
    "      --bed ${_input[\"geno\"]:n} \\\n",
    "      --phenoFile ${_input[\"pheno\"]} \\\n",
    "      --covarFile ${_input[\"covar\"]} \\\n",
    "      --keep ${_input[\"qc\"][0]} \\\n",
    "      --extract ${_input[\"qc\"][1]} \\\n",
    "      ${('--' + trait) if trait in ['bt'] else ''} \\\n",
    "      --bsize ${bsize} \\\n",
    "      --lowmem --lowmem-prefix ${lowmem_dir:a}/${_output:bn} \\\n",
    "      --threads ${numThreads} \\\n",
    "      --out ${_output:nn}.regenie"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c056cfc-a74b-4720-b6f6-f9b80b0cfb92",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Run REGENIE step 2: association analysis\n",
    "[asso]\n",
    "# Size of the genotype blocks to be used \n",
    "parameter: bsize = 400\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "parameter: trait = 'bt'\n",
    "parameter: pred = f'{cwd}/*.regenie_pred.list'\n",
    "input: phenoFile, group_by = 1\n",
    "input_options = f\"--bgen {genoFile} --sample {sampleFile}\" if genoFile.endswith('.bgen') else f\"--bed \" + genoFile.split('.bed')[0]\n",
    "output: [f'{cwd}/cache/{_input:bn}_'+ str(phenoCol[i]) + '.regenie.gz' for i in range(len(phenoCol))]\n",
    "task: trunk_workers = 1, trunk_size = 10, walltime = '0.5h', mem = '1G', cores = numThreads, tags = f'{step_name}_{_output[0]:bn}'\n",
    "bash:container=container_lmm, expand = \"${ }\", stderr = f'{cwd}/cache/{_input:bn}.stderr', stdout = f'{cwd}/cache/{_input:bn}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    set -e\n",
    "    regenie \\\n",
    "     --step 2 \\\n",
    "     ${input_options} \\\n",
    "     --phenoFile ${_input:n}.regenie_phenotype \\\n",
    "     --covarFile ${_input:n}.regenie_covar \\\n",
    "     --phenoColList ${','.join(phenoCol)} \\\n",
    "     ${('--' + trait) if trait in ['bt'] else ''} \\\n",
    "     --firth 0.01 --approx \\\n",
    "     --pred ${pred} \\\n",
    "     --bsize ${bsize} \\\n",
    "     --minMAC ${minMAC} \\\n",
    "     --minINFO ${bgenMinINFO}\\\n",
    "     --threads ${numThreads} \\\n",
    "     --out ${cwd}/cache/${_input:bn} && \\\n",
    "     gzip -f --best ${_output:n}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56398fb5-b2f9-4469-9d0d-80c41b08d480",
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "# Run REGENIE step 2: association analysis with the permutation of phenotype and covarites date for each job\n",
    "[asso_permutation]\n",
    "# ID column file containing and only containing the sample FID and IID column (with header)\n",
    "parameter: id_column = path\n",
    "# Phenotype file containing and only containing the pheontypes in phenoCol (with header). The sample order should the same with [id_column] \n",
    "parameter: pheno_column = path\n",
    "# Covariate file prefix containing and only containing the covariates (wiht header). It should ends with \".{pheontype}\". The sample order should the same with [pheno_column]\n",
    "parameter: covar_column = str\n",
    "# Phenotype to be analyzed (specify the column)\n",
    "parameter: phenoCol = list\n",
    "parameter: phenolist = ' '.join([pheno for pheno in phenoCol])\n",
    "# The index of permutation to test\n",
    "parameter: idstart = int\n",
    "parameter: idend = int\n",
    "# Size of the genotype blocks to be used \n",
    "parameter: bsize = 400\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "parameter: trait = 'bt'\n",
    "# pred list file prefix from regenie step 1 result (ends with .regenie_pred.list), The prefix should be the same for all the phenotypes\n",
    "parameter: pred = str\n",
    "parameter: indexnum = list(range(idstart,idend+1))\n",
    "input: for_each = \"indexnum\"\n",
    "input_options = f\"--bgen {genoFile} --sample {sampleFile}\" if genoFile.endswith('.bgen') else f\"--bed \" + genoFile.split('.bed')[0]\n",
    "output: [f'{cwd}/cache/permutation.{_indexnum}_'+ str(phenoCol[i]) + '.regenie.gz' for i in range(len(phenoCol))]\n",
    "task: trunk_workers = 1, trunk_size = 10, walltime = '0.5h', mem = '1G', cores = numThreads, tags = f'{step_name}_{_output[0]:bn}'\n",
    "bash:container=container_lmm, expand = \"${ }\", stderr = f'{cwd}/cache/permutation.{_indexnum}.stderr', stdout = f'{cwd}/cache/permutation.{_indexnum}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    # ID shifting\n",
    "    id=`sed '1d' ${id_column} | awk '{print $1,$2}' OFS='\\t' | shuf`\n",
    "    echo -e \"FID\\tIID\\n$id\" > ${cwd}/cache/${_indexnum}.id\n",
    "    # Combine the ID column with pheno_column and covar_column\n",
    "    for pheno in ${phenolist}\n",
    "    do\n",
    "    paste ${cwd}/cache/${_indexnum}.id \\\n",
    "    ${pheno_column} \\\n",
    "    > ${cwd}/cache/$pheno.${_indexnum}.regenie_phenotype\n",
    "    paste ${cwd}/cache/${_indexnum}.id \\\n",
    "    ${covar_column}.$pheno \\\n",
    "    > ${cwd}/cache/$pheno.${_indexnum}.regenie_covar\n",
    "    done\n",
    "    # Regenie by pheontypes\n",
    "    for pheno in ${phenolist}\n",
    "    do\n",
    "    set -e\n",
    "    regenie \\\n",
    "     --step 2 \\\n",
    "     ${input_options} \\\n",
    "     --phenoFile ${cwd}/cache/$pheno.${_indexnum}.regenie_phenotype \\\n",
    "     --covarFile ${cwd}/cache/$pheno.${_indexnum}.regenie_covar \\\n",
    "     --phenoColList $pheno \\\n",
    "     ${('--' + trait) if trait in ['bt'] else ''} \\\n",
    "     --firth 0.01 --approx \\\n",
    "     --pred ${pred}_$pheno.regenie_pred.list \\\n",
    "     --bsize ${bsize} \\\n",
    "     --minMAC ${minMAC} \\\n",
    "     --minINFO ${bgenMinINFO}\\\n",
    "     --threads ${numThreads} \\\n",
    "     --out ${cwd}/cache/permutation.${_indexnum} && \\\n",
    "    gzip -f --best ${cwd}/cache/permutation.${_indexnum}_$pheno.regenie\n",
    "    done\n",
    "    # Delete the permuted phenotype files\n",
    "    #rm -rf ${cwd}/cache/${_indexnum}.id ${cwd}/cache/*.${_indexnum}.regenie_phenotype ${cwd}/cache/*.${_indexnum}.regenie_covar ${cwd}/cache/permutation.${_indexnum}.stderr ${cwd}/cache/permutation.${_indexnum}.stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88e74e52-3943-4f99-bbb3-c2183a8a477e",
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "# Run regenie burden test step 2\n",
    "[regenie_burden]\n",
    "# Size of the genotype blocks to be used \n",
    "parameter: bsize = 400\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "parameter: trait = 'bt'\n",
    "parameter: pred = f'{cwd}/*.regenie_pred.list'\n",
    "# Specify that traits are binary with 0=control,1=case,NA=missing (default is quantitative)\n",
    "parameter: trait = 'bt'\n",
    "# Annotation file format: variantID, gene and functional annotation (space/tab delimited)\n",
    "parameter: anno_file = path\n",
    "# This file lists variants within each set/gene to use when building masks. Format: set/gene name, chromosome, physical pos set/gene, then by a comma-separated list of variants included in the set/gene.\n",
    "parameter: set_list = path\n",
    "# Select specific genes/sets to test\n",
    "parameter: keep_gene = path(\".\")\n",
    "# Allele frequency file. format: variantId, alternative allele frequency\n",
    "parameter: aaf_file = path(\".\")\n",
    "# Select the annotations to be used in the mask file. format: mask# annotation type\n",
    "parameter: mask_file = path(\".\")\n",
    "# Select the upper MAF to generate masks\n",
    "parameter: aaf_bins =[0.01]\n",
    "# The way in which the alternative alleles are counted\n",
    "parameter: build_mask = 'max'\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "input: phenoFile, group_by = 1\n",
    "input_options = f\"--bgen {genoFile} --sample {sampleFile}\" if genoFile.endswith('.bgen') else f\"--bed \" + genoFile.split('.bed')[0]\n",
    "output: [f'{cwd}/cache/{_input:bn}_burden_'+ str(phenoCol[i]) + '.regenie.gz' for i in range(len(phenoCol))]\n",
    "task: trunk_workers = 1, trunk_size = 10, walltime = '0.5h', mem = '5G', cores = numThreads, tags = f'{step_name}_{_output[0]:bn}'\n",
    "python:container=container_lmm, expand = \"${ }\", stderr = f'{cwd}/cache/{_input:bn}.stderr', stdout = f'{cwd}/cache/{_input:bn}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    import pandas as pd\n",
    "    import numpy as np\n",
    "    dat = pd.read_csv(\"${_input:n}.regenie_phenotype\", header=0, delimiter='\\t', dtype=str)\n",
    "    dat = dat.replace(to_replace =np.nan, value =\"NA\")\n",
    "    dat = dat.dropna(subset=${phenoCol})\n",
    "    if len(${phenoCol}) > 0:    \n",
    "        dat.to_csv(f\"${cwd}/cache/${_input:nb}.regenie_phenotype\", sep='\\t', index=False, columns = ['FID', 'IID']+${phenoCol})\n",
    "    dat = pd.read_csv(\"${_input:n}.regenie_covar\", header=0, delimiter='\\t', dtype=str)\n",
    "    if len(${covarCol}) > 0 or len(${qCovarCol}) > 0:\n",
    "        dat = dat.dropna(subset=${covarCol})\n",
    "        dat = dat.dropna(subset=${qCovarCol})\n",
    "        dat.replace(to_replace =np.nan, value =\"NA\")\n",
    "        dat1 = pd.DataFrame(dat, columns = ['FID','IID'] + ${covarCol})\n",
    "        #dat1 = dat1.astype(int)\n",
    "        dat2 = pd.DataFrame(dat, columns = ['IID'] + ${qCovarCol})\n",
    "        merged_left = pd.merge(left=dat1, right=dat2, how='left', left_on='IID', right_on='IID')\n",
    "        merged_left.to_csv(f\"${cwd}/cache/${_input:nb}.regenie_covar\", sep=' ', index=False)\n",
    "bash:container=container_lmm, expand = \"${ }\", stderr = f'{cwd}/cache/{_input:bn}.stderr', stdout = f'{cwd}/cache/{_input:bn}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    set -e\n",
    "    regenie \\\n",
    "        --step 2 \\\n",
    "        ${input_options} \\\n",
    "        --phenoFile ${cwd}/cache/${_input:nb}.regenie_phenotype \\\n",
    "        --covarFile ${cwd}/cache/${_input:nb}.regenie_covar \\\n",
    "        --phenoColList ${','.join(phenoCol)} \\\n",
    "        ${('--' + trait) if trait in ['bt'] else ''} \\\n",
    "        ${(\"--extract-sets \" + str(keep_gene)) if keep_gene.is_file() else \"\"} \\\n",
    "        --firth --approx \\\n",
    "        --pred ${pred} \\\n",
    "        --set-list ${set_list} \\\n",
    "        --anno-file ${anno_file} \\\n",
    "        --mask-def ${mask_file} \\\n",
    "        --aaf-bins ${\",\".join([str(x) for x in aaf_bins])}\\\n",
    "        ${('--build-mask ' + build_mask) if build_mask in ['max','sum','comphet'] else ''} \\\n",
    "        ${('--aaf-file ' + str(aaf_file)) if aaf_file.is_file() else \"\"}\\\n",
    "        --singleton-carrier \\\n",
    "        --minMAC ${minMAC} \\\n",
    "        --bsize ${bsize} \\\n",
    "        --check-burden-files \\\n",
    "        --out  ${cwd}/cache/${_input:bn}_burden && \\\n",
    "        gzip -f --best ${_output:n}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1df0a12b-0282-4e96-b3eb-482805af5dd5",
   "metadata": {
    "kernel": "SoS"
   },
   "outputs": [],
   "source": [
    "# Run regenie burden test step 2 with the permutation of phenotype and covarites date for each job\n",
    "[regenie_burden_permutation]\n",
    "# ID column file containing and only containing the sample FID and IID column (with header)\n",
    "parameter: id_column = path\n",
    "# Phenotype file containing and only containing the pheontypes in phenoCol (wiht header). The sample order should the same with [id_column] \n",
    "parameter: pheno_column = path\n",
    "# Covariate file prefix containing and only containing the covariates (wiht header). It should ends with \".{pheontype}\". The sample order should the same with [pheno_column]\n",
    "parameter: covar_column = str\n",
    "# Phenotype to be analyzed (specify the column)\n",
    "parameter: phenoCol = list\n",
    "parameter: phenolist = ' '.join([pheno for pheno in phenoCol])\n",
    "# The index of permutation to test\n",
    "parameter: idstart = int\n",
    "parameter: idend = int\n",
    "parameter: indexnum = list(range(idstart,idend+1))\n",
    "# Size of the genotype blocks to be used \n",
    "parameter: bsize = 400\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "parameter: trait = 'bt'\n",
    "# Specify that traits are binary with 0=control,1=case,NA=missing (default is quantitative)\n",
    "parameter: trait = 'bt'\n",
    "# Annotation file format: variantID, gene and functional annotation (space/tab delimited)\n",
    "parameter: anno_file = path\n",
    "# This file lists variants within each set/gene to use when building masks. Format: set/gene name, chromosome, physical pos set/gene, then by a comma-separated list of variants included in the set/gene.\n",
    "parameter: set_list = path\n",
    "# Select specific genes/sets to test\n",
    "parameter: keep_gene = path(\".\")\n",
    "# Allele frequency file. format: variantId, alternative allele frequency\n",
    "parameter: aaf_file = path(\".\")\n",
    "# Select the annotations to be used in the mask file. format: mask# annotation type\n",
    "parameter: mask_file = path(\".\")\n",
    "# Select the upper MAF to generate masks\n",
    "parameter: aaf_bins =[0.01]\n",
    "# The way in which the alternative alleles are counted\n",
    "parameter: build_mask = 'max'\n",
    "# Mimimum allele count to be used\n",
    "parameter: minMAC = int\n",
    "input: for_each = \"indexnum\"\n",
    "input_options = f\"--bgen {genoFile} --sample {sampleFile}\" if genoFile.endswith('.bgen') else f\"--bed \" + genoFile.split('.bed')[0]\n",
    "output: [f'{cwd}/cache/permutation.{_indexnum}_burden_'+ str(phenoCol[i]) + '.regenie.gz' for i in range(len(phenoCol))]\n",
    "task: trunk_workers = 1, trunk_size = 2, walltime = '2h', mem = '5G', cores = 1, tags = f'{step_name}_{_output[0]:bn}'\n",
    "bash:container=container_lmm, expand = \"${ }\", stderr = f'{cwd}/cache/permutation.{_indexnum}.stderr', stdout = f'{cwd}/cache/permutation.{_indexnum}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    # ID shifting\n",
    "    id=`sed '1d' ${id_column} | awk '{print $1,$2}' OFS='\\t' | shuf`\n",
    "    echo -e \"FID\\tIID\\n$id\" > ${cwd}/cache/${_indexnum}.id\n",
    "    # Combine the ID column with pheno_column and covar_column\n",
    "    for pheno in ${phenolist}\n",
    "    do\n",
    "    paste ${cwd}/cache/${_indexnum}.id \\\n",
    "    ${pheno_column} \\\n",
    "    > ${cwd}/cache/$pheno.${_indexnum}.regenie_phenotype\n",
    "    paste ${cwd}/cache/${_indexnum}.id \\\n",
    "    ${covar_column}.$pheno \\\n",
    "    > ${cwd}/cache/$pheno.${_indexnum}.regenie_covar\n",
    "    done\n",
    "    # Regenie by pheontypes\n",
    "    for pheno in ${phenolist}\n",
    "    do\n",
    "    set -e\n",
    "    regenie \\\n",
    "        --step 2 \\\n",
    "        ${input_options} \\\n",
    "        --phenoFile ${cwd}/cache/$pheno.${_indexnum}.regenie_phenotype \\\n",
    "        --covarFile ${cwd}/cache/$pheno.${_indexnum}.regenie_covar \\\n",
    "        --phenoColList $pheno \\\n",
    "        ${('--' + trait) if trait in ['bt'] else ''} \\\n",
    "        ${(\"--extract-sets \" + str(keep_gene)) if keep_gene.is_file() else \"\"} \\\n",
    "        --firth --approx \\\n",
    "        --pred ${cwd}/$pheno.regenie_pred.list \\\n",
    "        --set-list ${set_list} \\\n",
    "        --anno-file ${anno_file} \\\n",
    "        --mask-def ${mask_file} \\\n",
    "        --aaf-bins ${\",\".join([str(x) for x in aaf_bins])}\\\n",
    "        ${('--build-mask ' + build_mask) if build_mask in ['max','sum','comphet'] else ''} \\\n",
    "        ${('--aaf-file ' + str(aaf_file)) if aaf_file.is_file() else \"\"}\\\n",
    "        --singleton-carrier \\\n",
    "        --minMAC ${minMAC} \\\n",
    "        --bsize ${bsize} \\\n",
    "        --check-burden-files \\\n",
    "        --out  ${cwd}/cache/permutation.${_indexnum}_burden && \\\n",
    "    gzip -f --best ${cwd}/cache/permutation.${_indexnum}_burden_$pheno.regenie\n",
    "    done\n",
    "    # Delete the permuted phenotype files\n",
    "    #rm -rf ${cwd}/cache/${_indexnum}.id ${cwd}/cache/*.${_indexnum}.regenie_phenotype ${cwd}/cache/*.${_indexnum}.regenie_covar ${cwd}/cache/permutation.${_indexnum}.stderr ${cwd}/cache/permutation.${_indexnum}.stdout"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "972cefc7-ac11-4d08-bbc1-7fa2fb02468c",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Merge results and log files\n",
    "[plot_1]\n",
    "parameter:reverse_log_p = False\n",
    "depends: formatFile\n",
    "input: [f'{cwd}/cache/{phenoFile:bn}_'+ str(phenoCol[i]) + '.regenie.gz' for i in range(len(phenoCol))], group_by=1\n",
    "output: f'{cwd}/{_input:bn}.{step_name.rsplit(\"_\",1)[0]}.snp_stats.gz',\n",
    "        f'{cwd}/{_input:bn}.{step_name.rsplit(\"_\",1)[0]}.snp_counts.txt'\n",
    "task: trunk_workers = 1, trunk_size = 1, walltime = '0.5h', mem = '16G', cores = 1, tags = f'{step_name}_{_output[0]:bn}'\n",
    "python: container=container_lmm, expand ='${ }', stderr = f'{cwd}/cache/{_input:bn}.stderr', stdout = f'{cwd}/cache/{_input:bn}.stdout', volumes = [f\"{cwd:a}:{cwd:a}\"]\n",
    "    import gzip\n",
    "    import pandas as pd\n",
    "    if ${formatFile.is_file()}:\n",
    "        output = '${_output[0]:n}' + '_original_columns' + '${_output[0]:x}'\n",
    "    else:\n",
    "        output = '${_output[0]}'\n",
    "   \n",
    "    data = pd.concat([pd.read_csv(f, compression='gzip', header=0, delim_whitespace=True, quotechar='\"', comment='#') for f in [${_input:r,}]], ignore_index=True)\n",
    "    data.to_csv(output, compression='gzip', sep='\\t', header = True, index = False)\n",
    "    # unify output format\n",
    "    if ${formatFile.is_file()} or ${reverse_log_p}:\n",
    "        sumstats = pd.read_csv(output, compression='gzip', header=0, delim_whitespace=True, quotechar='\"')  \n",
    "        if ${formatFile.is_file()}:\n",
    "            import yaml\n",
    "            config = yaml.safe_load(open(${formatFile:r}, 'r'))\n",
    "        try:\n",
    "            sumstats = sumstats.loc[:,list(config.values())]\n",
    "        except:\n",
    "            raise ValueError(f'According to ${formatFile}, input summary statistics should have the following columns: {list(config.values())}.')\n",
    "        sumstats.columns = list(config.keys())\n",
    "        if ${reverse_log_p}:\n",
    "            sumstats['P'] = sumstats['P'].apply(lambda row: 10**-row)\n",
    "        sumstats.to_csv(${_output[0]:r}, compression='gzip', sep='\\t', header = True, index = False)        \n",
    "\n",
    "bash: container=container_lmm, expand=\"$( )\"\n",
    "    # count result SNPs\n",
    "    for f in $(_input); do echo \"$f: `zcat $f | wc -l`\"; done > $(_output[1])\n",
    "    # merge stderr and stdout files\n",
    "    for f in $(_input); do \n",
    "        for ext in stderr stdout log; do\n",
    "            echo \"$f $ext:\"\n",
    "            cat ${f%.gz}.$ext 2>/dev/null || true\n",
    "            rm -f ${f%.gz}.$ext \n",
    "        done\n",
    "    done > $(_output[0]:n).log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70aefcf9-5ca2-462f-ab66-af0bcfd19586",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Manhattan and QQ plots using `qqman`\n",
    "[plot_2]\n",
    "# Column name for BP\n",
    "parameter: bp = 'POS'\n",
    "# Column name for p-value\n",
    "parameter: pval = 'P'\n",
    "# Column name for SNP\n",
    "parameter: snp = 'SNP'\n",
    "# Plot only on p-values smaller than this\n",
    "parameter: p_filter = '0.05'\n",
    "# Higlight SNPs with P-values lower than this\n",
    "parameter: sigp = 5e-08\n",
    "# ylim set to 0 to use maximum -log10(p) in data\n",
    "parameter: ylim = 0\n",
    "# Path to the annotation file if present\n",
    "parameter: anno_file = path()\n",
    "# The label to annotate in the manhattan plot options are (SNP, avsnp150, Gene)\n",
    "parameter: label_annotate = str\n",
    "# Option to add annotation data\n",
    "parameter: annotate = False\n",
    "# Decide wether to use top_snps list\n",
    "parameter: top_snps = False\n",
    "# Provide a list of SNPs to highlight\n",
    "parameter: snp_list = path()\n",
    "sep = '\\n\\n---\\n'\n",
    "if any(['fastGWA' in step_name]):\n",
    "    heritability = get_output(f'grep Heritability {_input[0]:n}.log | head -1').strip()\n",
    "else:\n",
    "    heritability = None\n",
    "depends: phenoFile\n",
    "input: group_by = 2, group_with = 'phenoCol'\n",
    "output: manhattan = f'{_input[0]:nn}.manhattan.png',\n",
    "        qq = f'{_input[0]:nn}.qq.png',\n",
    "        annotated_manhattan = f'{_input[0]:nn}.manhattan_annotated.png',\n",
    "        analysis_summary = f'{_input[0]:nn}.analysis_summary.md',\n",
    "        plot_data = f'{_input[0]:nn}.plot_data.rds',\n",
    "        manhattan_pdf = f'{_input[0]:nn}.manhattan_annotated.pdf'\n",
    "task: trunk_workers = 1, trunk_size = job_size, walltime = '3h', mem = '64G', tags = f'{step_name}_{_output[0]:bn}'    \n",
    "bash: container=container_lmm, expand = \"${ }\"\n",
    "    echo '''---\n",
    "    theme: base-theme\n",
    "    style: |\n",
    "      img {\n",
    "        height: 80%;\n",
    "        display: block;\n",
    "        margin-left: auto;\n",
    "        margin-right: auto;\n",
    "      }\n",
    "    ---    \n",
    "    ''' > ${_output[3]}\n",
    "    \n",
    "R: container=container_lmm, expand='${ }', stderr = f'{_output[0]:n}.stderr', stdout = f'{_output[0]:n}.stdout'\n",
    "    # some summary statistics for phenotype\n",
    "    pheno = read.table(${phenoFile:r}, header=T, sep = \"\\t\")$${_phenoCol}\n",
    "    if (length(unique(pheno))>2) {\n",
    "      out = capture.output(summary(pheno))\n",
    "    } else {\n",
    "      out = as.data.frame(table(pheno))\n",
    "      rownames(out) = c('n_ctrl', 'n_case')\n",
    "      out = out[,2,drop=F]\n",
    "    }\n",
    "    write('# ${_phenoCol} result summary\\n## Phenotype summary:\\n```', ${_output[3]:r}, append = T)\n",
    "    write.table(out, ${_output[3]:r}, append = T)\n",
    "    write('${(\" Heritability is %s\" % heritability) if heritability is not None else ''}', ${_output[3]:r}, append = T)\n",
    "    write(\"```\", ${_output[3]:r}, append = T)\n",
    "\n",
    "R: container=container_lmm, expand='${ }', stderr = f'{_output[0]:n}.stderr', stdout = f'{_output[0]:n}.stdout'\n",
    "    library('qqman')\n",
    "    data <- read.table(gzfile('${_input[0]}'), sep='\\t', header=T)\n",
    "    lambda <- median(qchisq(1-data$${pval},1), na.rm=TRUE)/qchisq(0.5,1)\n",
    "    ifelse((${ylim} == 0 && min(data$${pval}, na.rm=TRUE)!=0), ylim <- abs(floor(log10(min(data$${pval}, na.rm=TRUE)))), ylim <- abs(floor(log10(2.225074e-308))))\n",
    "    # Creating manhattan plot\n",
    "    png('${_output[0]}', width = 6, height = 4, unit='in', res=300)\n",
    "    manhattan_plot <- manhattan(data, chr='CHR', bp='${bp}', snp='${snp}', p='${pval}', main = 'Manhattan plot for ${_phenoCol} (${step_name.rsplit(\"_\",1)[0]})', ylim = c(0, ylim), cex = 0.6, \n",
    "    cex.axis = 0.9, col = c(\"blue4\", \"orange3\"), chrlabs = as.character(c(1:22)))\n",
    "    dev.off()\n",
    "    # Creating qqplot\n",
    "    png('${_output[1]}', width = 5, height = 5, unit='in', res=300)\n",
    "    qq_plot <- qq(data$${pval}, main = 'QQ Plot for ${_phenoCol} (${step_name.rsplit(\"_\",1)[0]})', xlim = c(0, 8), ylim = c(0, ylim), pch = 18, col = \"blue4\", cex = 1.5, las = 1)\n",
    "    dev.off()\n",
    "    write('## p-value summary:', ${_output[3]:r}, append=T)\n",
    "    write(paste(\"Genomic inflation factor is\", round(lambda,3), \"for\", nrow(data), \"variants analyzed.${sep}\"), ${_output[3]:r}, append=T)\n",
    "    \n",
    "  \n",
    "R: container=container_lmm, expand='${ }', stderr = f'{_output[2]:n}.stderr', stdout = f'{_output[2]:n}.stdout'\n",
    "    library('dplyr')\n",
    "    library('ggrepel')\n",
    "    library('tidyr')\n",
    "    library('ggplot2')\n",
    "    library('svglite')\n",
    "    #Load your data\n",
    "    gwas <- read.table(gzfile('${_input[0]}'),sep='\\t', header=T)\n",
    "    # Select SNPs of interest\n",
    "    if(${\"TRUE\" if top_snps else \"FALSE\"}){\n",
    "      snpsOfInterest <- read.table('${snp_list}', sep=\"\\t\", header=T)\n",
    "      } else {\n",
    "          snpsOfInterest <- gwas %>%\n",
    "          filter(${pval}<${sigp})\n",
    "          write.table(snpsOfInterest,'${_output[5]:nn}.top_snps.tsv', sep='\\t', row.names=FALSE, quote=FALSE)\n",
    "      }\n",
    "    \n",
    "    # Read in the annotation file if present\n",
    "    if (${\"TRUE\" if annotate else \"FALSE\"}) {\n",
    "         print(\"The annotation file has been provided\")\n",
    "         annot <- read.csv('${anno_file}')\n",
    "         annot <- annot %>%\n",
    "         select(\"alternate_id\", \"Gene.refGene\", \"avsnp150\") %>%\n",
    "         separate(\"Gene.refGene\", \"Gene\") %>%\n",
    "         rename(SNP = alternate_id)\n",
    "     # Merge snps of interest with the annotation info\n",
    "         snps_annot <- merge(snpsOfInterest,annot,by=\"SNP\")\n",
    "    } else {\n",
    "          print(\"The annotation file has not been provided\")\n",
    "    }\n",
    "  \n",
    "    # Prepare the dataset\n",
    "    dat <- gwas %>%  \n",
    "    # Compute chromosome size\n",
    "    group_by(CHR) %>% \n",
    "    summarise(chr_len=max(${bp})) %>%\n",
    "    # Calculate cumulative position of each chromosome\n",
    "    mutate(tot=cumsum(as.numeric(chr_len))- as.numeric(chr_len)) %>%\n",
    "    select(-chr_len) %>%      \n",
    "    # Add this info to the initial dataset\n",
    "    left_join(gwas, ., by=c(\"CHR\"=\"CHR\")) %>%\n",
    "    # Add a cumulative position of each SNP\n",
    "    arrange(CHR, ${bp}) %>%\n",
    "    mutate(BPcum=${bp}+tot) %>%  \n",
    "    # Add highlight and annotation information\n",
    "    mutate(is_highlight=ifelse(${snp} %in% snpsOfInterest$SNP, \"yes\", \"no\")) %>%\n",
    "    mutate(is_annotate=ifelse(${pval}<${sigp}, \"yes\", \"no\")) %>%\n",
    "    # Filter SNP to make the plot lighter\n",
    "    filter(-log10(P)>${p_filter})\n",
    "    \n",
    "    # Merge the gwas filtered data with the annotation\n",
    "    if (${\"TRUE\" if annotate else \"FALSE\"}) {\n",
    "          print(\"Merging the gwas results with the annotation file\")\n",
    "          dat_anno <- merge(x = dat, y = snps_annot[ , c(\"SNP\", \"Gene\", \"avsnp150\")], by = \"SNP\", all.x=TRUE)\n",
    "          gene_anno <- dat_anno %>% \n",
    "              filter(!is.na(Gene)) %>%\n",
    "              group_by(Gene) %>% \n",
    "              slice_min(order_by=P)\n",
    "          dat_anno <- dat_anno %>% mutate(is_top = ifelse(SNP %in% gene_anno$SNP, \"yes\", \"no\"))\n",
    "        } else {\n",
    "          print(\"The gwas results are kept as they are because no annotation file is provided\")\n",
    "          dat_anno <- dat\n",
    "        }\n",
    "    \n",
    "    # Create axis for the manhattan plot\n",
    "    axisdf = dat %>% group_by(CHR) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )\n",
    "  \n",
    "    # Make the plot\n",
    "    sig <- ${sigp}\n",
    "    if (${ylim} == 0) ylim <- abs(floor(log10(min(dat$${pval})))) + 2\n",
    "    y_limits <- c(8, NA)\n",
    "\n",
    "    manhplot <- ggplot(dat_anno, aes(x=BPcum, y=-log10(${pval}))) +\n",
    "        # Show all points\n",
    "        geom_point( aes(color=as.factor(CHR)), alpha=0.8, size=1.3) +\n",
    "        scale_color_manual(values = rep(c(\"black\", \"grey\"), 22 )) +\n",
    "        # custom X axis:\n",
    "        scale_x_continuous(label = axisdf$CHR, breaks= axisdf$center) +\n",
    "        scale_y_continuous(expand = c(0, 0), limits=c(NA, ylim)) +   # remove space between plot area and x axis\n",
    "        # Add highlighted points\n",
    "        geom_point(data=subset(dat_anno, is_highlight==\"yes\"), color=\"red\", size=2) +\n",
    "        # Add label using ggrepel to avoid overlapping, if annotate option is used label the top snps in each gene\n",
    "        geom_label_repel( data=subset(dat_anno, ${'is_top==\"yes\"' if annotate else 'is_annotate==\"yes\"'}), aes(label=${label_annotate}), size=2,  max.overlaps = Inf, ylim = y_limits, box.padding = unit(0.35, \"lines\"), point.padding = unit(0.5, \"lines\")) +\n",
    "        # Add significance level line\n",
    "        geom_hline(yintercept = -log10(sig), color = \"red1\", linetype = \"dashed\") +\n",
    "        #create X and Y axes labels\n",
    "        labs(x = \"Chromosome\",\n",
    "         y = \"-log10(p)\",\n",
    "         title ='Manhattan plot for ${_phenoCol} (${step_name.rsplit(\"_\",1)[0]})') +\n",
    "        # Custom the theme:\n",
    "        theme_classic() +\n",
    "        theme( \n",
    "          legend.position=\"none\",\n",
    "          panel.border = element_blank(),\n",
    "          panel.grid.major.x = element_blank(),\n",
    "          panel.grid.minor.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5))\n",
    "  \n",
    "    # To save a plot created with ggplot2 you have to use to print() function\n",
    "\n",
    "    png('${_output[2]}', width = 6, height = 4, unit='in', res=300)\n",
    "    print(manhplot)\n",
    "    dev.off()\n",
    "    ggsave(${_output[5]:r}, plot= manhplot, device = \"pdf\", width = 6, height = 4, dpi=300, unit='in')\n",
    "    # save significant data to a file for further evaluations\n",
    "    tmp = dat[,c('CHR', '${bp}', 'BPcum', '${snp}', '${pval}')]\n",
    "    colnames(tmp) = c('CHR', 'POS', 'POScum', 'SNP', 'pvalue')\n",
    "    saveRDS(list(data = tmp, \n",
    "                 ylim = abs(floor(log10(min(dat$${pval})))) + 2,\n",
    "                 axisdf = axisdf), ${_output[4]:r})\n",
    "\n",
    "bash: container=container_lmm, expand = True\n",
    "  set -e\n",
    "  echo -e \"# QQ plot for {_phenoCol}\\n\" >> {_output[3]}\n",
    "  echo -e \"![]({_output[1]:bn}.png){sep}\" >> {_output[3]}\n",
    "  echo -e \"# Manhattan plot for {_phenoCol}\\n\" >> {_output[3]}\n",
    "  echo -e \"![]({_output[0]:bn}.png){sep}\" >> {_output[3]}\n",
    "  echo -e \"# Manhattan plot for {_phenoCol}\\n\" >> {_output[3]}\n",
    "  echo -e \"![]({_output[2]:bn}.png){sep}\" >> {_output[3]}\n",
    "  echo -e \"# Manhattan plot for {_phenoCol}\\n\" >> {_output[3]}\n",
    "  echo -e \"![]({_output[2]:bn}.svg){sep}\" >> {_output[3]}\n",
    "  echo -e \"# Result files\\n\\`\\`\\`\" >> {_output[3]}\n",
    "  ls {_input[0]:nn}.* | grep -vP 'stderr|stdout'>> {_output[3]}\n",
    "  echo -e \"\\`\\`\\`\" >> {_output[3]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57cd31e6-256f-4013-b9a8-3adb33bc0b03",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Generate analysis report: HTML file, and optionally PPTX file\n",
    "[plot_3]\n",
    "input: group_by = 6, group_with='phenoCol'\n",
    "output: f\"{_input['analysis_summary']:n}.html\"\n",
    "sh: container=container_marp, expand = True, stderr = f'{_output:n}.stderr', stdout = f'{_output:n}.stdout'\n",
    "    node /opt/marp/.cli/marp-cli.js {_input['analysis_summary']} -o {_output:a} \\\n",
    "        --title '{_phenoCol} {step_name.rsplit(\"_\",1)[0]} analysis' \\\n",
    "        --allow-local-files\n",
    "    node /opt/marp/.cli/marp-cli.js {_input['analysis_summary']} -o {_output:an}.pptx \\\n",
    "        --title '{_phenoCol} {step_name.rsplit(\"_\",1)[0]} analysis' \\\n",
    "        --allow-local-files"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "SoS",
     "sos",
     "",
     "",
     "sos"
    ]
   ],
   "version": "0.23.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
