{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fd25e634-0e11-4a02-82ac-b9b36652da3a",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# Analysis of pleiotopy between Alzheimer's Disease and ARHI traits "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd628a18-daff-4fe4-ae5f-9dabb0f910b3",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "First of all a review of the available references is made regarding AD. Particularly for GWAS using AD and meta-analysis from different studies. \n",
    "\n",
    "1. Kunkle et al https://www.nature.com/articles/s41588-019-0358-2\n",
    "2. Jansen et al https://pubmed.ncbi.nlm.nih.gov/30617256/\n",
    "3. Wightman et al https://www.nature.com/articles/s41588-021-00921-z\n",
    "4. [family history based] https://pubmed.ncbi.nlm.nih.gov/29777097/\n",
    "5. Schwartzentruber et al https://pubmed.ncbi.nlm.nih.gov/33589840/ \n",
    "6. Bellenguez et al https://www.nature.com/articles/s41588-022-01024-z\n",
    "\n",
    "**Using AD proxy case definition**\n",
    "\n",
    "[Schwartzentruber et al 2021](https://www.nature.com/articles/s41588-020-00776-w)\n",
    "\n",
    "**Using clinically diagnosed LOAD**\n",
    "\n",
    "[Kunkle et al 2019](https://www.nature.com/articles/s41588-019-0358-2)\n",
    "\n",
    "**Using clinically diagnosed/proxy AD cases**\n",
    "\n",
    "[Belenguez et al 2022](https://www.nature.com/articles/s41588-022-01024-z)\n",
    "\n",
    "- two stage GWAS\n",
    "- 111,326 clinically diagnosed/‘proxy’ AD cases and 677,663 controls\n",
    "- The European Alzheimer & Dementia Biobank  + UK Biobank\n",
    "- GWAS meta-analysis results can be downloaded here https://www.ebi.ac.uk/gwas/studies/GCST90027158\n",
    "\n",
    "The results from this meta-analysis have been downloaded to our cluster and are located here:\n",
    "\n",
    "`/mnt/vast/hpc/csg/data_public/GWAS_sumstats/GCST90027158_buildGRCh38.tsv.gz`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "190dcf85-54d2-44bc-9dcd-e9dddedcc468",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## GWAS for ARHI used in the pleiotropy analysis\n",
    "\n",
    "Parameters used for REGENIE\n",
    "\n",
    "```\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "bgenMinINFO=0.8\n",
    "minMAC=4\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "genoFile=`echo ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "```\n",
    "\n",
    "The GWAS results are stored here (Note: these results correspond to white European individuals:\n",
    "\n",
    "- /mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/2021_10_07_f3393_500K\n",
    "- /mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/2021_10_07_f2247_500K\n",
    "- /mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/2021_10_07_f2257_500K\n",
    "- /mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/2021_10_07_combined_500K"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66aac625-3db4-46ce-8f2e-841e51642015",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# Step 1. Liftover of the meta-analysis sumtats of AD to match our GWAS results in hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbb7c26e-76d3-4201-b4ad-b43a48f6fa8c",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "module load Singularity\n",
    "USER_PATH=~/working\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "sos run $USER_PATH/bioworkflows/GWAS/liftover.ipynb \\\n",
    "--cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI \\\n",
    "--input_file /mnt/vast/hpc/csg/data_public/GWAS_sumstats/GCST90027158_buildGRCh38.tsv.gz \\\n",
    "--output_file GCST90027158_buildGRCh37.tsv \\\n",
    "--fr 'hg38' \\\n",
    "--to 'hg19' \\\n",
    "--yml_file $USER_PATH/UKBB_GWAS_dev/data/liftover.yml"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5756148d-f093-4c05-9663-a4a366526eac",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Step 2. Association analysis of the imputed data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9803db17-fccd-49a1-9fd4-aa71abd7f7d9",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Do the association analysis for the Hearing impairment traits with the white european individuals present in the 500K samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d79f4870-22d0-4163-9bd1-58389664e53c",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# Common variables Columbia's cluster\n",
    "UKBB_PATH=$HOME/UKBiobank\n",
    "UKBB_yale=$HOME/UKBiobank_Yale_transfer\n",
    "USER_PATH=$HOME/project\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "container_marp=$HOME/containers/marp.sif\n",
    "container_annovar=$HOME/containers/gatk4-annovar.sif\n",
    "hearing_pheno_path=$UKBB_PATH/phenotype_files/hearing_impairment\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "formatFile_fastgwa=$USER_PATH/UKBB_GWAS_dev/data/fastGWA_template.yml\n",
    "formatFile_bolt=$USER_PATH/UKBB_GWAS_dev/data/boltlmm_template.yml\n",
    "formatFile_saige=$USER_PATH/UKBB_GWAS_dev/data/saige_template.yml\n",
    "formatFile_regenie=$USER_PATH/UKBB_GWAS_dev/data/regenie_template.yml\n",
    "# Workflows\n",
    "lmm_sos=$USER_PATH/bioworkflows/GWAS/LMM.ipynb\n",
    "anno_sos=$USER_PATH/bioworkflows/variant-annotation/annovar.ipynb\n",
    "clumping_sos=$USER_PATH/bioworkflows/GWAS/LD_Clumping.ipynb\n",
    "extract_sos=$USER_PATH/bioworkflows/GWAS/Region_Extraction.ipynb\n",
    "snptogene_sos=$USER_PATH/UKBB_GWAS_dev/workflow/snptogene.ipynb\n",
    "\n",
    "# LMM directories for imputed data\n",
    "lmm_imp_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_imputed_data\n",
    "lmm_imp_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_imputed_data\n",
    "lmm_imp_dir_saige=$UKBB_PATH/results/SAIGE_results/results_imputed_data\n",
    "lmm_imp_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_imputed_data\n",
    "\n",
    "# LMM directories for exome data\n",
    "lmm_exome_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_exome_data\n",
    "lmm_exome_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_exome_data\n",
    "lmm_exome_dir_saige=$UKBB_PATH/results/SAIGE_results/results_exome_data\n",
    "lmm_exome_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_exome_data\n",
    "\n",
    "## LMM variables \n",
    "## Specific to Bolt_LMM\n",
    "LDscoresFile=$UKBB_PATH/LDSCORE.1000G_EUR.tab.gz\n",
    "geneticMapFile=$UKBB_PATH/genetic_map_hg19_withX.txt.gz\n",
    "covarMaxLevels=10\n",
    "numThreads=20\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "lmm_job_size=1\n",
    "ylim=0\n",
    "\n",
    "### Specific to FastGWA (depeding if you run from Yale or Columbia)\n",
    "####Yale's cluster\n",
    "grmFile=$UKBB_PATH/results/FastGWA_results/results_imputed_data/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.grm.sp\n",
    "####Columbia's cluster\n",
    "grmFile=$UKBB_yale/results/FastGWA_results/results_imputed_data/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.grm.sp\n",
    "\n",
    "### Specific to SAIGE\n",
    "bgenMinMAC=4\n",
    "trait_type=binary\n",
    "loco=TRUE\n",
    "sampleCol=IID\n",
    "\n",
    "### Specific to REGENIE\n",
    "bsize=1000\n",
    "lowmem=$HOME/scratch60/\n",
    "lowmem_dir=$HOME/scratch60/predictions\n",
    "trait=bt\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "reverse_log_p=True"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05a364ef-5f8e-4893-b67a-a03b3a1e6306",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## New PCA\n",
    "### Exclude the samples wiht code 1415 but without the sub-category code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d827c77-3df5-43e0-ae1c-3c5c70aa37a2",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "sample500k = pd.read_csv(\"/home/gl2776/UKBiobank/phenotype_files/HI_UKBB/ukb47922_white_460649ind.keep_id\",header=None,sep=\" \")\n",
    "sample500k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "711d32a1-b9e9-4350-b3e7-eb05f7a00f54",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "exclusion = pd.read_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/with_1415_without_subcat.500k.sample_id.txt\",header=0,sep=\"\\t\")\n",
    "exclusion = exclusion[\"IID\"].to_list()\n",
    "len(exclusion)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07540516-646e-4dca-b0b5-196cc6abd603",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "sample500k[~sample500k[0].isin(exclusion)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0d101cf-1118-4fc4-bbe7-dadf405721e0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "sample500k[~sample500k[0].isin(exclusion)].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/ukb47922_white_456561ind_exclu_1415.keep_id\",header=False,index=False,sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "caf0f2bd-dcec-4d00-957c-0175fb2b227a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## Keep_id for PCA\n",
    "# f3393\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[[\"FID\", \"IID\"]].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2.keep_id\", sep='\\t', index=False, header=False)\n",
    "# f2247\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[[\"FID\", \"IID\"]].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2.keep_id\", sep='\\t', index=False, header=False)\n",
    "# f2257\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[[\"FID\", \"IID\"]].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2.keep_id\", sep='\\t', index=False, header=False)\n",
    "# f2247_f2257\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[[\"FID\", \"IID\"]].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2.keep_id\", sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c65d208-3d5a-42d9-b370-08291037170e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# phenopca\n",
    "import pandas as pd\n",
    "# f3393\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\"\n",
    "list1415 = pd.read_csv(file,header=0,sep=\"\\t\")[\"IID\"].to_list()\n",
    "file = \"/home/gl2776/UKBiobank/results/092821_PCA_results_500K/100521_UKBB_Hearing_aid_f3393_expandedwhite_15601cases_237318ctrl_500k.phenopca\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[phe[\"IID\"].isin(list1415)].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k.phenopca\", sep='\\t', index=False)\n",
    "# f2247\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\"\n",
    "list1415 = pd.read_csv(file,header=0,sep=\"\\t\")[\"IID\"].to_list()\n",
    "file = \"/home/gl2776/UKBiobank/results/092821_PCA_results_500K/100521_UKBB_Hearing_difficulty_f2247_expandedwhite_110453cases_237318ctrl_500k.phenopca\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[phe[\"IID\"].isin(list1415)].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k.phenopca\", sep='\\t', index=False)\n",
    "# f2247\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\"\n",
    "list1415 = pd.read_csv(file,header=0,sep=\"\\t\")[\"IID\"].to_list()\n",
    "file = \"/home/gl2776/UKBiobank/results/092821_PCA_results_500K/100521_UKBB_Hearing_noise_f2257_expandedwhite_161443cases_237318ctrl_500k.phenopca\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[phe[\"IID\"].isin(list1415)].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k.phenopca\", sep='\\t', index=False)\n",
    "# f2247\n",
    "file = \"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\"\n",
    "list1415 = pd.read_csv(file,header=0,sep=\"\\t\")[\"IID\"].to_list()\n",
    "file = \"/home/gl2776/UKBiobank/results/092821_PCA_results_500K/100521_UKBB_Combined_f2247_f2257_expandedwhite_93258cases_237318ctrl_500k.phenopca\"\n",
    "phe = pd.read_csv(file,header=0,sep=\"\\t\")\n",
    "phe[phe[\"IID\"].isin(list1415)].to_csv(\"/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k.phenopca\", sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11de07be-7517-4b66-a479-b5903aea001d",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Calculate the first 2 PC's for each of these phenotypes to run the association analysis with the imputed data\n",
    "#### f.3393\n",
    "##### Step 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07933097-4acf-4181-87b5-9a9367bdfd28",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f3393\n",
    "#gwas_sbatch=$USER_PATH/UKBB_GWAS_dev/output/qc1_f3393_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "gwas_sbatch=$cwd/qc1_f3393_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "## Use qc'ed genotype array\n",
    "genoFile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "keep_samples=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2.keep_id\n",
    "#Keep variants after LD pruning\n",
    "keep_variants=$UKBB_PATH/results/092821_PCA_results_500K/092821_ldprun_unrelated/cache/*092821_ldprun_unrelated.filtered.prune.in\n",
    "\n",
    "#GWAS QC variables set all of this variables to 0 to avoid doing more filtering\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "numThreads=1\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7d9be72-15b9-4bc7-8dfc-92250ea48148",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "##### Step 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6744038d-695e-46f3-9b86-ca05c707c0c4",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f3393\n",
    "#This is the bfile obtained in step 1\n",
    "genoFile=$cwd/*.bed\n",
    "# Format FID, IID, pop\n",
    "phenoFile=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k.phenopca\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_f3393_pc_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=2\n",
    "maha_k=2\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "\n",
    "pca_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d437b20-fda8-47c5-852b-6d7562a57d31",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### f.2247\n",
    "\n",
    "##### Step 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa888c01-5401-4960-91f6-e79827d83175",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247\n",
    "#gwas_sbatch=$USER_PATH/UKBB_GWAS_dev/output/qc1_f2247_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "gwas_sbatch=$cwd/qc1_f2247_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "## Use qc'ed genotype array\n",
    "genoFile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "keep_samples=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2.keep_id\n",
    "#Keep variants after LD pruning\n",
    "keep_variants=$UKBB_PATH/results/092821_PCA_results_500K/092821_ldprun_unrelated/cache/*092821_ldprun_unrelated.filtered.prune.in\n",
    "\n",
    "#GWAS QC variables set all of this variables to 0 to avoid doing more filtering\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "numThreads=1\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "793d22a5-3ff2-4610-b971-f4d578a94c0f",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "##### Step 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4348318b-0030-4349-92b3-602eae6d3475",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247\n",
    "#This is the bfile obtained in step 1\n",
    "genoFile=$cwd/*.bed\n",
    "# Format FID, IID, pop\n",
    "phenoFile=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k.phenopca\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_f2247_pc_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=2\n",
    "maha_k=2\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "\n",
    "pca_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run  $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6e24976-1365-4fa3-8785-838395e38676",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### f.2257\n",
    "\n",
    "##### Step 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "baeaad1d-2c31-48e9-8cf8-9a9852a5bf98",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2257\n",
    "#gwas_sbatch=$USER_PATH/UKBB_GWAS_dev/output/qc1_f2257_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "gwas_sbatch=$cwd/qc1_f2257_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "## Use qc'ed genotype array\n",
    "genoFile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "keep_samples=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2.keep_id\n",
    "#Keep variants after LD pruning\n",
    "keep_variants=$UKBB_PATH/results/092821_PCA_results_500K/092821_ldprun_unrelated/cache/*092821_ldprun_unrelated.filtered.prune.in\n",
    "\n",
    "#GWAS QC variables set all of this variables to 0 to avoid doing more filtering\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "numThreads=1\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "99e71584-7a87-48d4-8fdc-39eca467320b",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "##### Step 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "107fca30-2cd9-463a-a629-f623fdf1dc91",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2257\n",
    "#This is the bfile obtained in step 1\n",
    "genoFile=$cwd/*.bed\n",
    "# Format FID, IID, pop\n",
    "phenoFile=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k.phenopca\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_f2257_pc_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=2\n",
    "maha_k=2\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "\n",
    "pca_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run  $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "295fa416-a810-4b58-9cb4-1afb15dec939",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Combined f.2247 & f.2257\n",
    "\n",
    "##### Step 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b54c12a2-6d85-46b7-856d-0460956987b1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247_f2257\n",
    "#gwas_sbatch=$USER_PATH/UKBB_GWAS_dev/output/qc1_f2247_f2257_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "gwas_sbatch=$cwd/qc1_f2247_f2257_qcarray_$(date +\"%Y-%m-%d\").sbatch\n",
    "## Use qc'ed genotype array\n",
    "genoFile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "keep_samples=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2.keep_id\n",
    "#Keep variants after LD pruning\n",
    "keep_variants=$UKBB_PATH/results/092821_PCA_results_500K/092821_ldprun_unrelated/cache/*092821_ldprun_unrelated.filtered.prune.in\n",
    "\n",
    "#GWAS QC variables set all of this variables to 0 to avoid doing more filtering\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "numThreads=1\n",
    "mem='30G'\n",
    "\n",
    "gwasqc_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/GWAS_QC.ipynb\n",
    "\n",
    "gwasqc_args=\"\"\"qc:1\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --keep_samples $keep_samples\n",
    "    --keep_variants $keep_variants\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "    --mem $mem\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $gwasqc_sos \\\n",
    "    --to-script $gwas_sbatch \\\n",
    "    --args \"$gwasqc_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92d4cdca-a740-49db-9a54-f13b077e1242",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "##### Step 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30fc887a-81ed-463b-bfc3-f48d510ff6af",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "## Columbia's cluster\n",
    "cwd=$UKBB_PATH/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247_f2257\n",
    "#This is the bfile obtained in step 1\n",
    "genoFile=$cwd/*.bed\n",
    "# Format FID, IID, pop\n",
    "phenoFile=$UKBB_PATH/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071122_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k.phenopca\n",
    "label_col=ethnicity\n",
    "pop_col=ethnicity\n",
    "pca_sbatch=$cwd/flashpca_f2247_f2257_pc_$(date +\"%Y-%m-%d\").sbatch\n",
    "k=2\n",
    "maha_k=2\n",
    "min_axis=\"\"\n",
    "max_axis=\"\"\n",
    "\n",
    "pca_sos=$USER_PATH/xqtl-pipeline/code/data_preprocessing/genotype/PCA.ipynb\n",
    "\n",
    "pca_args=\"\"\"flashpca\n",
    "    --cwd $cwd\n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile\n",
    "    --k $k\n",
    "    --maha_k $maha_k\n",
    "    --label_col $label_col\n",
    "    --pop_col $pop_col\n",
    "    --min_axis $min_axis\n",
    "    --max_axis $max_axis\n",
    "    --numThreads $numThreads \n",
    "    --job_size $job_size\n",
    "    --container $container_lmm\n",
    "\"\"\"\n",
    "\n",
    "sos run  $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $pca_sos \\\n",
    "    --to-script $pca_sbatch \\\n",
    "    --args \"$pca_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e6c6dff-57a1-4e90-8ba6-8d5c68a16891",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Combine the PCs in the phenotype files\n",
    "\n",
    "Phenotype files named \"070722_*_500k_PC1_PC2\" are with original PCs calculated by original sample size with code 1415. \"071222_*_500k_PC1_PC2\" will be the new phenotype files with the newly calculated PCs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ded105c7-b614-41c3-8d0d-51b7f7f9a435",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# f3393\n",
    "import pandas as pd\n",
    "phe = pd.read_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\",header=0,sep=\"\\t\")\n",
    "pc = pd.read_csv(\"~/UKBiobank/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f3393/071122_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k.pca.txt\",header=0,sep=\"\\t\")\n",
    "phe = phe[['FID', 'IID', 'sex', 'f3393', 'age']]\n",
    "phe = phe.merge(pc[[\"IID\",\"PC1\",\"PC2\"]],how=\"left\",left_on=\"IID\",right_on=\"IID\")\n",
    "phe.to_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\",index=False,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5daa9b61-f8b3-40b4-b4a8-0aaf3e96a3e2",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# f2247\n",
    "phe = pd.read_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\",header=0,sep=\"\\t\")\n",
    "pc = pd.read_csv(\"~/UKBiobank/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247/071122_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k.pca.txt\",header=0,sep=\"\\t\")\n",
    "phe = phe[['FID', 'IID', 'sex', 'f2247', 'age']]\n",
    "phe = phe.merge(pc[[\"IID\",\"PC1\",\"PC2\"]],how=\"left\",left_on=\"IID\",right_on=\"IID\")\n",
    "phe.to_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\",index=False,sep=\"\\t\")\n",
    "# f2257\n",
    "phe = pd.read_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\",header=0,sep=\"\\t\")\n",
    "pc = pd.read_csv(\"~/UKBiobank/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2257/071122_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k.pca.txt\",header=0,sep=\"\\t\")\n",
    "phe = phe[['FID', 'IID', 'sex', 'f2257', 'age']]\n",
    "phe = phe.merge(pc[[\"IID\",\"PC1\",\"PC2\"]],how=\"left\",left_on=\"IID\",right_on=\"IID\")\n",
    "phe.to_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\",index=False,sep=\"\\t\")\n",
    "# f2247_f2257\n",
    "phe = pd.read_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/070722_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\",header=0,sep=\"\\t\")\n",
    "pc = pd.read_csv(\"~/UKBiobank/results/092821_PCA_results_500K/071122_PCA_500k_exclu_1415/f2247_f2257/071122_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k.pca.txt\",header=0,sep=\"\\t\")\n",
    "phe = phe[['FID', 'IID', 'sex', 'f2247_f2257', 'age']]\n",
    "phe = phe.merge(pc[[\"IID\",\"PC1\",\"PC2\"]],how=\"left\",left_on=\"IID\",right_on=\"IID\")\n",
    "phe.to_csv(\"~/UKBiobank/phenotype_files/hearing_impairment/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\",index=False,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01f089a9-c9be-488a-802b-d051a930b056",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## Run association analysis with imputed data for each phenotype\n",
    "### 500k\n",
    "\n",
    "#### f.3393"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "de789a3e-3f48-42ba-85e0-11e69a1a757d",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/dmc2245/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f3393/f3393_500K_impdata_regenie_2022-07-26.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w2e19c78363e8a433) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cwd=~/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f3393\n",
    "lmm_dir_regenie=$cwd\n",
    "lmm_sbatch_regenie=$cwd/f3393_500K_impdata_regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\n",
    "covarFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2\n",
    "phenoCol=f3393\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "bgenMinINFO=0.8\n",
    "bgenMinMAF=0.001\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#Use the original bed files for the genotype array on regenie step1\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the imputed data hg19 to run the association analysis\n",
    "genoFile=`echo ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "label_annotate=SNP\n",
    "lowmem_dir=$cwd/predictions\n",
    "ref_first=True\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --label_annotate $label_annotate\n",
    "    --no-annotate\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --ref_first $ref_first\n",
    "    --numThreads $numThreads\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d22ed9cc-1243-4bd5-936d-b5e741fe78e1",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### f.2247"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "1eba0502-94be-4f1c-ab1d-a3bce1e7c381",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/dmc2245/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247/f2247_500K_impdata_regenie_2022-07-26.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w416eb35379405fde) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cwd=~/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247\n",
    "lmm_dir_regenie=$cwd\n",
    "lmm_sbatch_regenie=$cwd/f2247_500K_impdata_regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\n",
    "covarFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2\n",
    "phenoCol=f2247\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "bgenMinINFO=0.8\n",
    "bgenMinMAF=0.001\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#Use the original bed files for the genotype array on regenie step1\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the imputed data hg19 to run the association analysis\n",
    "genoFile=`echo ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "label_annotate=SNP\n",
    "lowmem_dir=$cwd/predictions\n",
    "ref_first=True\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --label_annotate $label_annotate\n",
    "    --no-annotate\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --ref_first $ref_first\n",
    "    --numThreads $numThreads\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "371bb66b-1903-4562-b4c3-81af39cfcf98",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### f.2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "17af5199-53e0-42ca-883e-dbf96d4c510f",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/dmc2245/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2257/f2257_500K_impdata_regenie_2022-07-26.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w3ba10e625e2c3b71) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cwd=~/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2257\n",
    "lmm_dir_regenie=$cwd\n",
    "lmm_sbatch_regenie=$cwd/f2257_500K_impdata_regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\n",
    "covarFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2\n",
    "phenoCol=f2257\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "bgenMinINFO=0.8\n",
    "bgenMinMAF=0.001\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#Use the original bed files for the genotype array on regenie step1\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the imputed data hg19 to run the association analysis\n",
    "genoFile=`echo ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "label_annotate=SNP\n",
    "lowmem_dir=$cwd/predictions\n",
    "ref_first=True\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --label_annotate $label_annotate\n",
    "    --no-annotate\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --ref_first $ref_first\n",
    "    --numThreads $numThreads\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51923fbe-4e1b-4e11-b875-0cf1915ce9fb",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "#### Combined f.2247 & f.2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "e7b32059-771e-496c-9561-88aa5124b319",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/dmc2245/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247_f2257/f2247_f2257_500K_impdata_regenie_2022-07-26.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w0f1374ff2f10b381) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cwd=~/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247_f2257\n",
    "lmm_dir_regenie=$cwd\n",
    "lmm_sbatch_regenie=$cwd/f2247_f2257_500K_impdata_regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\n",
    "covarFile=$hearing_pheno_path/fulldb_500K/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2\n",
    "phenoCol=f2247_f2257\n",
    "covarCol=sex\n",
    "qCovarCol=\"age PC1 PC2\"\n",
    "bgenMinINFO=0.8\n",
    "bgenMinMAF=0.001\n",
    "minMAC=4\n",
    "maf_filter=0\n",
    "geno_filter=0\n",
    "hwe_filter=0\n",
    "mind_filter=0\n",
    "#Use the original bed files for the genotype array on regenie step1\n",
    "bfile=$UKBB_PATH/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the imputed data hg19 to run the association analysis\n",
    "genoFile=`echo ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "label_annotate=SNP\n",
    "lowmem_dir=$cwd/predictions\n",
    "ref_first=True\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait \n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --ylim $ylim\n",
    "    --label_annotate $label_annotate\n",
    "    --no-annotate\n",
    "    --reverse_log_p $reverse_log_p\n",
    "    --ref_first $ref_first\n",
    "    --numThreads $numThreads\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run $USER_PATH/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $lmm_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "711452b4-3699-4d28-a959-f1729db41625",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Step 3. LD clumping between different traits"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8d2ac42-303a-4604-a85d-f6c2c206f90c",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Format file preparation so that all summary stats match "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5cea3345-21ea-4660-a5c5-c8baa0c6e6d4",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n",
      "/tmp/2658818.1.csg2.q/ipykernel_16933/2685489524.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working\n",
      "  from collections import Sized, KeysView, Sequence\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "data = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37.tsv.gz', compression='gzip', sep=\"\\t\", header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "586daad4-1900-4d05-8824-8047b45e2b6d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "639f9d2d-fbb9-4cd6-8ddf-fb4427e6e745",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "data.rename(columns={'A0':'REF', 'A1':'ALT', 'STAT':'BETA'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ba754fb5-b141-4f70-acd4-1bae7625c0c3",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>chr1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>chr1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT                SNP    BETA      SE       P\n",
       "0    1  529825   C    T    chr1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  chr1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    chr1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    chr1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    chr1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c5e99edb-053b-4397-861a-8e1ddb7172b7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "data['SNP'] = data['SNP'].str.replace('chr', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bea86c28-737d-4e22-9029-0a6146ed224b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2c489f3e-39b6-4458-ae47-2bd920f89ef7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "data.to_csv(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\", sep='\\t', index=False, header=True, compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37688145-eea5-47bc-b2f5-641d456298a0",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Re-format the SNP in the 500K GWAS summary stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d173039f-b1e3-4e17-b877-9cde338df064",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "\n",
    "f3393 = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f3393/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c15b6c26-e74c-4045-b9a2-cece1e82eda9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>0.272939253875768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.5226211725664186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>0.06150635633786312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>0.303832341109345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>-0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.7267499323234313</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS REF ALT          SNP       BETA        SE                    P\n",
       "0   1  13259   A   G  rs562993331  -0.444965  0.405872    0.272939253875768\n",
       "1   1  17569   A   C  rs535086049    1.19541   1.86985   0.5226211725664186\n",
       "2   1  17641   A   G  rs578081284   0.435933  0.233139  0.06150635633786312\n",
       "3   1  30741   A   C  rs558169846    1.28043   1.24525    0.303832341109345\n",
       "4   1  52144   A   T  rs190291950  -0.117852  0.337249   0.7267499323234313"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "21ac3f75-8afc-4f9b-a65f-27401b290c2c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f3393['var'] = f3393['CHR'] + ':' + f3393['POS'] + ':' + f3393['REF'] + ':' + f3393['ALT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "1c69f5b7-71a0-4037-ba00-3eddcb43dc94",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>0.272939253875768</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.5226211725664186</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>0.06150635633786312</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>0.303832341109345</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>-0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.7267499323234313</td>\n",
       "      <td>1:52144:A:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS REF ALT          SNP       BETA        SE                    P  \\\n",
       "0   1  13259   A   G  rs562993331  -0.444965  0.405872    0.272939253875768   \n",
       "1   1  17569   A   C  rs535086049    1.19541   1.86985   0.5226211725664186   \n",
       "2   1  17641   A   G  rs578081284   0.435933  0.233139  0.06150635633786312   \n",
       "3   1  30741   A   C  rs558169846    1.28043   1.24525    0.303832341109345   \n",
       "4   1  52144   A   T  rs190291950  -0.117852  0.337249   0.7267499323234313   \n",
       "\n",
       "           var  \n",
       "0  1:13259:A:G  \n",
       "1  1:17569:A:C  \n",
       "2  1:17641:A:G  \n",
       "3  1:30741:A:C  \n",
       "4  1:52144:A:T  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "64bb807b-d859-4fe1-a9a8-14313aa4ea5c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.gz\n",
      "  CHR    POS REF ALT        varid        BETA        SE                     P  \\\n",
      "0   1  13259   G   A  rs562993331    0.387653  0.169082  0.023122777490193567   \n",
      "1   1  17569   C   A  rs535086049   -0.331285  0.764723    0.6648627475668728   \n",
      "2   1  17641   G   A  rs578081284  -0.0451544  0.103088    0.6613738204754334   \n",
      "3   1  30741   C   A  rs558169846    -1.76303  0.991047  0.024469776350843825   \n",
      "4   1  57222   T   C  rs576081345   0.0501817  0.114946    0.6624254356036648   \n",
      "\n",
      "           SNP  \n",
      "0  1:13259:G:A  \n",
      "1  1:17569:C:A  \n",
      "2  1:17641:G:A  \n",
      "3  1:30741:C:A  \n",
      "4  1:57222:T:C  \n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "rootdir = '/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/'\n",
    "\n",
    "for subdir, dirs, files in os.walk(rootdir):\n",
    "    for filename in files:\n",
    "        if filename.endswith(\"071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.gz\"):\n",
    "            print(os.path.join(subdir, filename))\n",
    "            df = pd.read_csv(os.path.join(subdir, filename), dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "            df.rename(columns={'SNP':'varid'}, inplace=True)\n",
    "            df['SNP'] = df['CHR'] + ':' + df['POS'] + ':' + df['REF'] + ':' + df['ALT']\n",
    "            df.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/' + filename, sep='\\t', index=False, header=True, compression=\"gzip\")\n",
    "            print(df.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "628890b0-f9ee-4912-af8d-a00e12b2423a",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "# Variables for LD clumping "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2d42a0c3-2370-4925-a322-f1f5668113a9",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "clumping_sos=~/project/bioworkflows/GWAS/LD_Clumping.ipynb\n",
    "## GenoFile is not necessary to run LD clumping but it is needed by the global parameters. It won't be use in this module but you have to provide path\n",
    "genoFile=`echo $UKBB_yale/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "reference_genotype_prefix=~/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.nondup\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "container=~/containers/lmm.sif"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70aa9caf-4596-441d-a6f1-03e642bcf347",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Solution to empty clumped files\n",
    "\n",
    "The empty clumped files were showing up because the varid in the reference file is rsid whereas in the summary stats files it is based on chr:pos:ref:alt\n",
    "Therefore, I have decided to re-name all of the variants in the reference file with rsid for the more conventional notation CHR:POS:REF:ALT which can be acomplished using plink2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "592d693d-1d9a-41d8-b0e9-505b8c102397",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a2.3LM 64-bit Intel (24 Jan 2020)   www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno.log.\n",
      "Options in effect:\n",
      "  --bfile /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.2000.ref_geno\n",
      "  --make-bed\n",
      "  --new-id-max-allele-len 662\n",
      "  --out /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "\n",
      "Start time: Thu Jul 28 13:46:07 2022\n",
      "257481 MiB RAM detected; reserving 128740 MiB for main workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "2000 samples (1096 females, 904 males; 2000 founders) loaded from\n",
      "/home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.2000.ref_geno.fam.\n",
      "92457702 variants loaded from\n",
      "/home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.2000.ref_geno.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing\n",
      "/home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno.fam\n",
      "... done.\n",
      "Writing\n",
      "/home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno.bim\n",
      "... done.\n",
      "Writing\n",
      "/home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno.bed\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "End time: Thu Jul 28 13:47:19 2022\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load PLINK/2.0\n",
    "plink2 --bfile ~/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.2000.ref_geno \\\n",
    "       --make-bed \\\n",
    "       --out ~/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno\\\n",
    "       --new-id-max-allele-len 662 \\\n",
    "       --set-all-var-ids @:#:\\$r:\\$a\n",
    "       "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1eb93684-3de1-4583-95bb-0610f57b9cb8",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "There was still an error because some of the variants after renaming them were duplicated having the same ref/alt alleles. Therefore, I proceded to remove all the duplicated variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "6bed2a08-857b-4b56-bf4b-b63ceee2580f",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b4.9 64-bit (13 Oct 2017)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2017 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno\n",
      "  --list-duplicate-vars require-same-ref\n",
      "  --memory 60000\n",
      "\n",
      "Segmentation fault (core dumped)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load PLINK/2.0\n",
    "plink2 --bfile ~/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.2000.ref_geno \\\n",
    "        --rm-dup  'force-first' 'list' --make-bed \\\n",
    "        --out ~/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.nondup.2000.ref_geno"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d23cc3e-e22e-47c4-a41b-0ba82c246eb1",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD_meta and f3393"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0ce89594-6bdc-43fa-ab54-0a53eb5f819d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080722_f3393_ADmeta/f3393_ADmeta-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w08dba5e8d7034f9d) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f3393_ADmeta\n",
    "clumping_sbatch=$clumping_dir/f3393_ADmeta-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0eeee596-9728-402a-bdbd-067a1f11726a",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Check overlap between AD sumstats and f3393 sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dbe10d28-ddb2-46eb-b10e-34b82ce5a22d",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\"\n",
    "AD = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f49871a0-f80a-49d8-a6d4-6f04f6a5c7dc",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "5fafed0e-ffac-49af-96dc-fa3a854d783c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "AD_sig=AD[AD['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d6fd8563-a975-404f-9afe-826e6fd8e861",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.gz'\n",
    "f3393 = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a99f2c05-c93e-4d6b-8911-4840c29e6079",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>0.272939</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.195410</td>\n",
       "      <td>1.869850</td>\n",
       "      <td>0.522621</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>0.061506</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.280430</td>\n",
       "      <td>1.245250</td>\n",
       "      <td>0.303832</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>-0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.726750</td>\n",
       "      <td>1:52144:A:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   A   G  rs562993331 -0.444965  0.405872  0.272939  1:13259:A:G\n",
       "1    1  17569   A   C  rs535086049  1.195410  1.869850  0.522621  1:17569:A:C\n",
       "2    1  17641   A   G  rs578081284  0.435933  0.233139  0.061506  1:17641:A:G\n",
       "3    1  30741   A   C  rs558169846  1.280430  1.245250  0.303832  1:30741:A:C\n",
       "4    1  52144   A   T  rs190291950 -0.117852  0.337249  0.726750  1:52144:A:T"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c7f442ab-20d7-448c-a024-785951f7213e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f3393_sig=f3393[f3393['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "9a1a7c64-ab68-48a1-86f4-5782ac86b383",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8200715</th>\n",
       "      <td>6</td>\n",
       "      <td>32654502</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32654502:G:A</td>\n",
       "      <td>0.0798</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.004000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201118</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32665728:G:A</td>\n",
       "      <td>0.0766</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>2.682000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201135</th>\n",
       "      <td>6</td>\n",
       "      <td>32666173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666173:G:A</td>\n",
       "      <td>0.0795</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.218000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201140</th>\n",
       "      <td>6</td>\n",
       "      <td>32666295</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666295:G:A</td>\n",
       "      <td>0.0793</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.323000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201173</th>\n",
       "      <td>6</td>\n",
       "      <td>32666875</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666875:G:A</td>\n",
       "      <td>0.0760</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>3.967000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201174</th>\n",
       "      <td>6</td>\n",
       "      <td>32666899</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666899:G:A</td>\n",
       "      <td>0.0761</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>3.775000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201177</th>\n",
       "      <td>6</td>\n",
       "      <td>32666968</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:32666968:C:T</td>\n",
       "      <td>0.0760</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>4.034000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201179</th>\n",
       "      <td>6</td>\n",
       "      <td>32667006</td>\n",
       "      <td>C</td>\n",
       "      <td>CATG</td>\n",
       "      <td>6:32667006:C:CATG</td>\n",
       "      <td>0.0790</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.423000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201192</th>\n",
       "      <td>6</td>\n",
       "      <td>32667343</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32667343:T:A</td>\n",
       "      <td>0.0727</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>2.167000e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201204</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32667895:G:C</td>\n",
       "      <td>0.0778</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>2.088000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18707312</th>\n",
       "      <td>17</td>\n",
       "      <td>43946318</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43946318:C:T</td>\n",
       "      <td>0.0549</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.008000e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF   ALT                SNP    BETA      SE  \\\n",
       "8200715     6  32654502   G     A     6:32654502:G:A  0.0798  0.0123   \n",
       "8201118     6  32665728   G     A     6:32665728:G:A  0.0766  0.0121   \n",
       "8201135     6  32666173   G     A     6:32666173:G:A  0.0795  0.0123   \n",
       "8201140     6  32666295   G     A     6:32666295:G:A  0.0793  0.0123   \n",
       "8201173     6  32666875   G     A     6:32666875:G:A  0.0760  0.0122   \n",
       "8201174     6  32666899   G     A     6:32666899:G:A  0.0761  0.0122   \n",
       "8201177     6  32666968   C     T     6:32666968:C:T  0.0760  0.0122   \n",
       "8201179     6  32667006   C  CATG  6:32667006:C:CATG  0.0790  0.0123   \n",
       "8201192     6  32667343   T     A     6:32667343:T:A  0.0727  0.0121   \n",
       "8201204     6  32667895   G     C     6:32667895:G:C  0.0778  0.0122   \n",
       "18707312   17  43946318   C     T    17:43946318:C:T  0.0549  0.0098   \n",
       "\n",
       "                     P  \n",
       "8200715   1.004000e-10  \n",
       "8201118   2.682000e-10  \n",
       "8201135   1.218000e-10  \n",
       "8201140   1.323000e-10  \n",
       "8201173   3.967000e-10  \n",
       "8201174   3.775000e-10  \n",
       "8201177   4.034000e-10  \n",
       "8201179   1.423000e-10  \n",
       "8201192   2.167000e-09  \n",
       "8201204   2.088000e-10  \n",
       "18707312  2.008000e-08  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_sig[AD_sig['SNP'].isin(f3393_sig['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ac5d8c09-4b47-4316-8b95-54579a1695da",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8515913</th>\n",
       "      <td>6</td>\n",
       "      <td>32654502</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275183</td>\n",
       "      <td>-0.105956</td>\n",
       "      <td>0.018669</td>\n",
       "      <td>1.915226e-08</td>\n",
       "      <td>6:32654502:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516436</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275312</td>\n",
       "      <td>-0.102268</td>\n",
       "      <td>0.018389</td>\n",
       "      <td>3.606783e-08</td>\n",
       "      <td>6:32665728:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516462</th>\n",
       "      <td>6</td>\n",
       "      <td>32666173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275318</td>\n",
       "      <td>-0.106932</td>\n",
       "      <td>0.018668</td>\n",
       "      <td>1.418208e-08</td>\n",
       "      <td>6:32666173:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516467</th>\n",
       "      <td>6</td>\n",
       "      <td>32666295</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275319</td>\n",
       "      <td>-0.106943</td>\n",
       "      <td>0.018673</td>\n",
       "      <td>1.427414e-08</td>\n",
       "      <td>6:32666295:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516498</th>\n",
       "      <td>6</td>\n",
       "      <td>32666875</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275330</td>\n",
       "      <td>-0.104214</td>\n",
       "      <td>0.018446</td>\n",
       "      <td>2.200645e-08</td>\n",
       "      <td>6:32666875:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516499</th>\n",
       "      <td>6</td>\n",
       "      <td>32666899</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275331</td>\n",
       "      <td>-0.104226</td>\n",
       "      <td>0.018446</td>\n",
       "      <td>2.191341e-08</td>\n",
       "      <td>6:32666899:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516502</th>\n",
       "      <td>6</td>\n",
       "      <td>32666968</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275333</td>\n",
       "      <td>-0.104285</td>\n",
       "      <td>0.018445</td>\n",
       "      <td>2.151642e-08</td>\n",
       "      <td>6:32666968:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516504</th>\n",
       "      <td>6</td>\n",
       "      <td>32667006</td>\n",
       "      <td>C</td>\n",
       "      <td>CATG</td>\n",
       "      <td>rs760571019</td>\n",
       "      <td>-0.104240</td>\n",
       "      <td>0.018448</td>\n",
       "      <td>2.189626e-08</td>\n",
       "      <td>6:32667006:C:CATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516516</th>\n",
       "      <td>6</td>\n",
       "      <td>32667343</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275338</td>\n",
       "      <td>-0.108311</td>\n",
       "      <td>0.018419</td>\n",
       "      <td>5.866648e-09</td>\n",
       "      <td>6:32667343:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516550</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275358</td>\n",
       "      <td>-0.104878</td>\n",
       "      <td>0.018604</td>\n",
       "      <td>2.362490e-08</td>\n",
       "      <td>6:32667895:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19375028</th>\n",
       "      <td>17</td>\n",
       "      <td>43946318</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs753236</td>\n",
       "      <td>-0.082377</td>\n",
       "      <td>0.014885</td>\n",
       "      <td>3.742743e-08</td>\n",
       "      <td>17:43946318:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF   ALT        varid      BETA        SE  \\\n",
       "8515913     6  32654502   G     A    rs9275183 -0.105956  0.018669   \n",
       "8516436     6  32665728   G     A    rs9275312 -0.102268  0.018389   \n",
       "8516462     6  32666173   G     A    rs9275318 -0.106932  0.018668   \n",
       "8516467     6  32666295   G     A    rs9275319 -0.106943  0.018673   \n",
       "8516498     6  32666875   G     A    rs9275330 -0.104214  0.018446   \n",
       "8516499     6  32666899   G     A    rs9275331 -0.104226  0.018446   \n",
       "8516502     6  32666968   C     T    rs9275333 -0.104285  0.018445   \n",
       "8516504     6  32667006   C  CATG  rs760571019 -0.104240  0.018448   \n",
       "8516516     6  32667343   T     A    rs9275338 -0.108311  0.018419   \n",
       "8516550     6  32667895   G     C    rs9275358 -0.104878  0.018604   \n",
       "19375028   17  43946318   C     T     rs753236 -0.082377  0.014885   \n",
       "\n",
       "                     P                SNP  \n",
       "8515913   1.915226e-08     6:32654502:G:A  \n",
       "8516436   3.606783e-08     6:32665728:G:A  \n",
       "8516462   1.418208e-08     6:32666173:G:A  \n",
       "8516467   1.427414e-08     6:32666295:G:A  \n",
       "8516498   2.200645e-08     6:32666875:G:A  \n",
       "8516499   2.191341e-08     6:32666899:G:A  \n",
       "8516502   2.151642e-08     6:32666968:C:T  \n",
       "8516504   2.189626e-08  6:32667006:C:CATG  \n",
       "8516516   5.866648e-09     6:32667343:T:A  \n",
       "8516550   2.362490e-08     6:32667895:G:C  \n",
       "19375028  3.742743e-08    17:43946318:C:T  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393_sig[f3393_sig['SNP'].isin(AD_sig['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d5167e04-28bf-463e-8325-5a326acf1751",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1</td>\n",
       "      <td>693731</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:693731:G:A</td>\n",
       "      <td>-0.0243</td>\n",
       "      <td>0.0161</td>\n",
       "      <td>0.13180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>1</td>\n",
       "      <td>730087</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:730087:C:T</td>\n",
       "      <td>-0.0078</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>0.72810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>1</td>\n",
       "      <td>731718</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:731718:C:T</td>\n",
       "      <td>-0.0295</td>\n",
       "      <td>0.0155</td>\n",
       "      <td>0.05685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>1</td>\n",
       "      <td>732032</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1:732032:C:A</td>\n",
       "      <td>-0.0343</td>\n",
       "      <td>0.0162</td>\n",
       "      <td>0.03397</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>1</td>\n",
       "      <td>734349</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:734349:C:T</td>\n",
       "      <td>-0.0308</td>\n",
       "      <td>0.0155</td>\n",
       "      <td>0.04682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101031</th>\n",
       "      <td>22</td>\n",
       "      <td>51219387</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>22:51219387:C:T</td>\n",
       "      <td>0.0004</td>\n",
       "      <td>0.0168</td>\n",
       "      <td>0.97900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101049</th>\n",
       "      <td>22</td>\n",
       "      <td>51221731</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>22:51221731:C:T</td>\n",
       "      <td>0.0021</td>\n",
       "      <td>0.0168</td>\n",
       "      <td>0.90140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101081</th>\n",
       "      <td>22</td>\n",
       "      <td>51229805</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>22:51229805:C:T</td>\n",
       "      <td>-0.0001</td>\n",
       "      <td>0.0169</td>\n",
       "      <td>0.99430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101103</th>\n",
       "      <td>22</td>\n",
       "      <td>51236013</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>22:51236013:AT:A</td>\n",
       "      <td>0.0078</td>\n",
       "      <td>0.0110</td>\n",
       "      <td>0.47460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101107</th>\n",
       "      <td>22</td>\n",
       "      <td>51237063</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>22:51237063:C:T</td>\n",
       "      <td>-0.0044</td>\n",
       "      <td>0.0103</td>\n",
       "      <td>0.66660</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6488159 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF ALT               SNP    BETA      SE        P\n",
       "16          1    693731   G   A      1:693731:G:A -0.0243  0.0161  0.13180\n",
       "87          1    730087   C   T      1:730087:C:T -0.0078  0.0225  0.72810\n",
       "93          1    731718   C   T      1:731718:C:T -0.0295  0.0155  0.05685\n",
       "94          1    732032   C   A      1:732032:C:A -0.0343  0.0162  0.03397\n",
       "114         1    734349   C   T      1:734349:C:T -0.0308  0.0155  0.04682\n",
       "...       ...       ...  ..  ..               ...     ...     ...      ...\n",
       "21101031   22  51219387   C   T   22:51219387:C:T  0.0004  0.0168  0.97900\n",
       "21101049   22  51221731   C   T   22:51221731:C:T  0.0021  0.0168  0.90140\n",
       "21101081   22  51229805   C   T   22:51229805:C:T -0.0001  0.0169  0.99430\n",
       "21101103   22  51236013  AT   A  22:51236013:AT:A  0.0078  0.0110  0.47460\n",
       "21101107   22  51237063   C   T   22:51237063:C:T -0.0044  0.0103  0.66660\n",
       "\n",
       "[6488159 rows x 8 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD[AD['SNP'].isin(f3393['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ec201ed4-8743-46d5-aa75-a7bd8fb9e7e9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>1</td>\n",
       "      <td>693731</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs12238997</td>\n",
       "      <td>-0.017106</td>\n",
       "      <td>0.020979</td>\n",
       "      <td>0.414868</td>\n",
       "      <td>1:693731:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>1</td>\n",
       "      <td>730087</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs148120343</td>\n",
       "      <td>0.008566</td>\n",
       "      <td>0.029290</td>\n",
       "      <td>0.769938</td>\n",
       "      <td>1:730087:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1</td>\n",
       "      <td>731718</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs58276399</td>\n",
       "      <td>-0.015556</td>\n",
       "      <td>0.019901</td>\n",
       "      <td>0.434411</td>\n",
       "      <td>1:731718:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1</td>\n",
       "      <td>732032</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs61770163</td>\n",
       "      <td>-0.025332</td>\n",
       "      <td>0.021278</td>\n",
       "      <td>0.233832</td>\n",
       "      <td>1:732032:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>1</td>\n",
       "      <td>734349</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs141242758</td>\n",
       "      <td>-0.015371</td>\n",
       "      <td>0.019910</td>\n",
       "      <td>0.440108</td>\n",
       "      <td>1:734349:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735882</th>\n",
       "      <td>22</td>\n",
       "      <td>51219387</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9616832</td>\n",
       "      <td>-0.040643</td>\n",
       "      <td>0.024311</td>\n",
       "      <td>0.094563</td>\n",
       "      <td>22:51219387:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735893</th>\n",
       "      <td>22</td>\n",
       "      <td>51221731</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs115055839</td>\n",
       "      <td>-0.043280</td>\n",
       "      <td>0.024333</td>\n",
       "      <td>0.075297</td>\n",
       "      <td>22:51221731:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735917</th>\n",
       "      <td>22</td>\n",
       "      <td>51229805</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9616985</td>\n",
       "      <td>-0.043153</td>\n",
       "      <td>0.024427</td>\n",
       "      <td>0.077291</td>\n",
       "      <td>22:51229805:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735922</th>\n",
       "      <td>22</td>\n",
       "      <td>51236013</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>rs200507571</td>\n",
       "      <td>0.002709</td>\n",
       "      <td>0.016160</td>\n",
       "      <td>0.866861</td>\n",
       "      <td>22:51236013:AT:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735923</th>\n",
       "      <td>22</td>\n",
       "      <td>51237063</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs3896457</td>\n",
       "      <td>0.008595</td>\n",
       "      <td>0.014843</td>\n",
       "      <td>0.562525</td>\n",
       "      <td>22:51237063:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6487845 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF ALT        varid      BETA        SE         P  \\\n",
       "32          1    693731   G   A   rs12238997 -0.017106  0.020979  0.414868   \n",
       "53          1    730087   C   T  rs148120343  0.008566  0.029290  0.769938   \n",
       "54          1    731718   C   T   rs58276399 -0.015556  0.019901  0.434411   \n",
       "56          1    732032   C   A   rs61770163 -0.025332  0.021278  0.233832   \n",
       "57          1    734349   C   T  rs141242758 -0.015371  0.019910  0.440108   \n",
       "...       ...       ...  ..  ..          ...       ...       ...       ...   \n",
       "21735882   22  51219387   C   T    rs9616832 -0.040643  0.024311  0.094563   \n",
       "21735893   22  51221731   C   T  rs115055839 -0.043280  0.024333  0.075297   \n",
       "21735917   22  51229805   C   T    rs9616985 -0.043153  0.024427  0.077291   \n",
       "21735922   22  51236013  AT   A  rs200507571  0.002709  0.016160  0.866861   \n",
       "21735923   22  51237063   C   T    rs3896457  0.008595  0.014843  0.562525   \n",
       "\n",
       "                       SNP  \n",
       "32            1:693731:G:A  \n",
       "53            1:730087:C:T  \n",
       "54            1:731718:C:T  \n",
       "56            1:732032:C:A  \n",
       "57            1:734349:C:T  \n",
       "...                    ...  \n",
       "21735882   22:51219387:C:T  \n",
       "21735893   22:51221731:C:T  \n",
       "21735917   22:51229805:C:T  \n",
       "21735922  22:51236013:AT:A  \n",
       "21735923   22:51237063:C:T  \n",
       "\n",
       "[6487845 rows x 9 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393[f3393['SNP'].isin(AD['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bbf097da-0e8d-4686-af67-f1a53135ed64",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21735926"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(f3393['SNP'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f90acefb-b956-4417-914f-458893994cb5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21101114"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(AD['SNP'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4cc10cea-23d6-46fc-bdc3-78a58efd9bca",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## AD_meta and f2247"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "34e8c357-630c-4116-8df8-e66a576aa94b",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADmeta/f2247_ADmeta-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=we08732f2bd56b244) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADmeta\n",
    "clumping_sbatch=$clumping_dir/f2247_ADmeta-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd67ce21-1a70-4ff8-9a2b-c4f9e8fc2c79",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Check overalp between AD sumstats and f2247 sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a56d8f78-fb40-4974-b91e-036995f8e921",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\"\n",
    "AD = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "443ffd59-ba0f-46a2-8cd0-acf75d89d15b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7d5649d4-64d5-459f-b328-884f65b24c53",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "AD_sig=AD[AD['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b5a780bc-c679-4989-983f-6731b45f948a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.gz'\n",
    "f2247 = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "6b4ce5d8-f3cc-4c6a-b46a-2beb2ec18e58",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.387653</td>\n",
       "      <td>0.169082</td>\n",
       "      <td>0.023123</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>0.331285</td>\n",
       "      <td>0.764723</td>\n",
       "      <td>0.664863</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.045154</td>\n",
       "      <td>0.103088</td>\n",
       "      <td>0.661374</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.763030</td>\n",
       "      <td>0.991047</td>\n",
       "      <td>0.024470</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>-0.050182</td>\n",
       "      <td>0.114946</td>\n",
       "      <td>0.662425</td>\n",
       "      <td>1:57222:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   A   G  rs562993331 -0.387653  0.169082  0.023123  1:13259:A:G\n",
       "1    1  17569   A   C  rs535086049  0.331285  0.764723  0.664863  1:17569:A:C\n",
       "2    1  17641   A   G  rs578081284  0.045154  0.103088  0.661374  1:17641:A:G\n",
       "3    1  30741   A   C  rs558169846  1.763030  0.991047  0.024470  1:30741:A:C\n",
       "4    1  57222   C   T  rs576081345 -0.050182  0.114946  0.662425  1:57222:C:T"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f2247.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e4122e4f-295e-4a1b-878b-ab26a9910519",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f2247_sig=f2247[f2247['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "57256793-ddbf-477a-95f5-935f892c4fa0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8199044</th>\n",
       "      <td>6</td>\n",
       "      <td>32577380</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32577380:G:A</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>0.0103</td>\n",
       "      <td>1.829000e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8199110</th>\n",
       "      <td>6</td>\n",
       "      <td>32578590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32578590:G:A</td>\n",
       "      <td>0.0736</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>1.362000e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8199369</th>\n",
       "      <td>6</td>\n",
       "      <td>32584693</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32584693:G:C</td>\n",
       "      <td>0.0747</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>9.003000e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8200715</th>\n",
       "      <td>6</td>\n",
       "      <td>32654502</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32654502:G:A</td>\n",
       "      <td>0.0798</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.004000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201118</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32665728:G:A</td>\n",
       "      <td>0.0766</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>2.682000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201135</th>\n",
       "      <td>6</td>\n",
       "      <td>32666173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666173:G:A</td>\n",
       "      <td>0.0795</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.218000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201140</th>\n",
       "      <td>6</td>\n",
       "      <td>32666295</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666295:G:A</td>\n",
       "      <td>0.0793</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.323000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201173</th>\n",
       "      <td>6</td>\n",
       "      <td>32666875</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666875:G:A</td>\n",
       "      <td>0.0760</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>3.967000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201174</th>\n",
       "      <td>6</td>\n",
       "      <td>32666899</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32666899:G:A</td>\n",
       "      <td>0.0761</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>3.775000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201177</th>\n",
       "      <td>6</td>\n",
       "      <td>32666968</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:32666968:C:T</td>\n",
       "      <td>0.0760</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>4.034000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201179</th>\n",
       "      <td>6</td>\n",
       "      <td>32667006</td>\n",
       "      <td>C</td>\n",
       "      <td>CATG</td>\n",
       "      <td>6:32667006:C:CATG</td>\n",
       "      <td>0.0790</td>\n",
       "      <td>0.0123</td>\n",
       "      <td>1.423000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201192</th>\n",
       "      <td>6</td>\n",
       "      <td>32667343</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32667343:T:A</td>\n",
       "      <td>0.0727</td>\n",
       "      <td>0.0121</td>\n",
       "      <td>2.167000e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201204</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32667895:G:C</td>\n",
       "      <td>0.0778</td>\n",
       "      <td>0.0122</td>\n",
       "      <td>2.088000e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8201206</th>\n",
       "      <td>6</td>\n",
       "      <td>32667937</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32667937:T:A</td>\n",
       "      <td>0.0793</td>\n",
       "      <td>0.0124</td>\n",
       "      <td>1.512000e-10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS REF   ALT                SNP    BETA      SE  \\\n",
       "8199044    6  32577380   G     A     6:32577380:G:A  0.0729  0.0103   \n",
       "8199110    6  32578590   G     A     6:32578590:G:A  0.0736  0.0104   \n",
       "8199369    6  32584693   G     C     6:32584693:G:C  0.0747  0.0105   \n",
       "8200715    6  32654502   G     A     6:32654502:G:A  0.0798  0.0123   \n",
       "8201118    6  32665728   G     A     6:32665728:G:A  0.0766  0.0121   \n",
       "8201135    6  32666173   G     A     6:32666173:G:A  0.0795  0.0123   \n",
       "8201140    6  32666295   G     A     6:32666295:G:A  0.0793  0.0123   \n",
       "8201173    6  32666875   G     A     6:32666875:G:A  0.0760  0.0122   \n",
       "8201174    6  32666899   G     A     6:32666899:G:A  0.0761  0.0122   \n",
       "8201177    6  32666968   C     T     6:32666968:C:T  0.0760  0.0122   \n",
       "8201179    6  32667006   C  CATG  6:32667006:C:CATG  0.0790  0.0123   \n",
       "8201192    6  32667343   T     A     6:32667343:T:A  0.0727  0.0121   \n",
       "8201204    6  32667895   G     C     6:32667895:G:C  0.0778  0.0122   \n",
       "8201206    6  32667937   T     A     6:32667937:T:A  0.0793  0.0124   \n",
       "\n",
       "                    P  \n",
       "8199044  1.829000e-12  \n",
       "8199110  1.362000e-12  \n",
       "8199369  9.003000e-13  \n",
       "8200715  1.004000e-10  \n",
       "8201118  2.682000e-10  \n",
       "8201135  1.218000e-10  \n",
       "8201140  1.323000e-10  \n",
       "8201173  3.967000e-10  \n",
       "8201174  3.775000e-10  \n",
       "8201177  4.034000e-10  \n",
       "8201179  1.423000e-10  \n",
       "8201192  2.167000e-09  \n",
       "8201204  2.088000e-10  \n",
       "8201206  1.512000e-10  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_sig[AD_sig['SNP'].isin(f2247_sig['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a5836a42-95da-42e1-be38-6331e132406a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8555319</th>\n",
       "      <td>6</td>\n",
       "      <td>32577380</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs660895</td>\n",
       "      <td>-0.038348</td>\n",
       "      <td>0.006534</td>\n",
       "      <td>4.579204e-09</td>\n",
       "      <td>6:32577380:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8555404</th>\n",
       "      <td>6</td>\n",
       "      <td>32578590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs3997868</td>\n",
       "      <td>-0.037369</td>\n",
       "      <td>0.006557</td>\n",
       "      <td>1.251065e-08</td>\n",
       "      <td>6:32578590:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8555800</th>\n",
       "      <td>6</td>\n",
       "      <td>32584693</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>rs510205</td>\n",
       "      <td>-0.038854</td>\n",
       "      <td>0.006549</td>\n",
       "      <td>3.103130e-09</td>\n",
       "      <td>6:32584693:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561048</th>\n",
       "      <td>6</td>\n",
       "      <td>32654502</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275183</td>\n",
       "      <td>-0.046044</td>\n",
       "      <td>0.008126</td>\n",
       "      <td>1.546002e-08</td>\n",
       "      <td>6:32654502:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561570</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275312</td>\n",
       "      <td>-0.044248</td>\n",
       "      <td>0.007994</td>\n",
       "      <td>3.286700e-08</td>\n",
       "      <td>6:32665728:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561596</th>\n",
       "      <td>6</td>\n",
       "      <td>32666173</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275318</td>\n",
       "      <td>-0.046374</td>\n",
       "      <td>0.008125</td>\n",
       "      <td>1.216942e-08</td>\n",
       "      <td>6:32666173:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561601</th>\n",
       "      <td>6</td>\n",
       "      <td>32666295</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275319</td>\n",
       "      <td>-0.046472</td>\n",
       "      <td>0.008127</td>\n",
       "      <td>1.144748e-08</td>\n",
       "      <td>6:32666295:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561632</th>\n",
       "      <td>6</td>\n",
       "      <td>32666875</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275330</td>\n",
       "      <td>-0.045466</td>\n",
       "      <td>0.008023</td>\n",
       "      <td>1.541452e-08</td>\n",
       "      <td>6:32666875:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561633</th>\n",
       "      <td>6</td>\n",
       "      <td>32666899</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275331</td>\n",
       "      <td>-0.045530</td>\n",
       "      <td>0.008023</td>\n",
       "      <td>1.469332e-08</td>\n",
       "      <td>6:32666899:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561636</th>\n",
       "      <td>6</td>\n",
       "      <td>32666968</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275333</td>\n",
       "      <td>-0.045572</td>\n",
       "      <td>0.008023</td>\n",
       "      <td>1.425706e-08</td>\n",
       "      <td>6:32666968:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561638</th>\n",
       "      <td>6</td>\n",
       "      <td>32667006</td>\n",
       "      <td>C</td>\n",
       "      <td>CATG</td>\n",
       "      <td>rs760571019</td>\n",
       "      <td>-0.044919</td>\n",
       "      <td>0.008025</td>\n",
       "      <td>2.305420e-08</td>\n",
       "      <td>6:32667006:C:CATG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561650</th>\n",
       "      <td>6</td>\n",
       "      <td>32667343</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275338</td>\n",
       "      <td>-0.045788</td>\n",
       "      <td>0.008027</td>\n",
       "      <td>1.243083e-08</td>\n",
       "      <td>6:32667343:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561684</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275358</td>\n",
       "      <td>-0.045173</td>\n",
       "      <td>0.008094</td>\n",
       "      <td>2.528192e-08</td>\n",
       "      <td>6:32667895:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8561687</th>\n",
       "      <td>6</td>\n",
       "      <td>32667937</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275360</td>\n",
       "      <td>-0.045895</td>\n",
       "      <td>0.008346</td>\n",
       "      <td>4.038499e-08</td>\n",
       "      <td>6:32667937:T:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS REF   ALT        varid      BETA        SE  \\\n",
       "8555319    6  32577380   G     A     rs660895 -0.038348  0.006534   \n",
       "8555404    6  32578590   G     A    rs3997868 -0.037369  0.006557   \n",
       "8555800    6  32584693   G     C     rs510205 -0.038854  0.006549   \n",
       "8561048    6  32654502   G     A    rs9275183 -0.046044  0.008126   \n",
       "8561570    6  32665728   G     A    rs9275312 -0.044248  0.007994   \n",
       "8561596    6  32666173   G     A    rs9275318 -0.046374  0.008125   \n",
       "8561601    6  32666295   G     A    rs9275319 -0.046472  0.008127   \n",
       "8561632    6  32666875   G     A    rs9275330 -0.045466  0.008023   \n",
       "8561633    6  32666899   G     A    rs9275331 -0.045530  0.008023   \n",
       "8561636    6  32666968   C     T    rs9275333 -0.045572  0.008023   \n",
       "8561638    6  32667006   C  CATG  rs760571019 -0.044919  0.008025   \n",
       "8561650    6  32667343   T     A    rs9275338 -0.045788  0.008027   \n",
       "8561684    6  32667895   G     C    rs9275358 -0.045173  0.008094   \n",
       "8561687    6  32667937   T     A    rs9275360 -0.045895  0.008346   \n",
       "\n",
       "                    P                SNP  \n",
       "8555319  4.579204e-09     6:32577380:G:A  \n",
       "8555404  1.251065e-08     6:32578590:G:A  \n",
       "8555800  3.103130e-09     6:32584693:G:C  \n",
       "8561048  1.546002e-08     6:32654502:G:A  \n",
       "8561570  3.286700e-08     6:32665728:G:A  \n",
       "8561596  1.216942e-08     6:32666173:G:A  \n",
       "8561601  1.144748e-08     6:32666295:G:A  \n",
       "8561632  1.541452e-08     6:32666875:G:A  \n",
       "8561633  1.469332e-08     6:32666899:G:A  \n",
       "8561636  1.425706e-08     6:32666968:C:T  \n",
       "8561638  2.305420e-08  6:32667006:C:CATG  \n",
       "8561650  1.243083e-08     6:32667343:T:A  \n",
       "8561684  2.528192e-08     6:32667895:G:C  \n",
       "8561687  4.038499e-08     6:32667937:T:A  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f2247_sig[f2247_sig['SNP'].isin(AD_sig['SNP'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90e4679b-ff1f-4b15-a0d2-9b3193a2a830",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## AD_meta and f2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "80521c24-7236-42a6-a476-852d63a94eed",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADmeta/f2257_ADmeta-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w1932b2f70073779a) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADmeta\n",
    "clumping_sbatch=$clumping_dir/f2257_ADmeta-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d2e715a-e56b-46f8-a901-10a7b61c90a9",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Check overalp between AD sumstats and f2257 sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0a1e8680-12b2-4a10-9577-e0ee2810994a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\"\n",
    "AD = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7eb9f4ed-3b72-4aaa-a5a8-5cb4bbf8001c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a35cafd5-7ee0-42c2-9346-3c8a2caab87d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "AD_sig=AD[AD['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "1816fbb3-825d-4434-9242-756b0e286346",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.gz'\n",
    "f2257 = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "4fec4a62-3aae-4a5e-8330-a803cd324efa",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.291280</td>\n",
       "      <td>0.157364</td>\n",
       "      <td>0.064170</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>0.301553</td>\n",
       "      <td>0.688140</td>\n",
       "      <td>0.661231</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.010873</td>\n",
       "      <td>0.090079</td>\n",
       "      <td>0.903924</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>0.665389</td>\n",
       "      <td>0.529693</td>\n",
       "      <td>0.209051</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>0.052396</td>\n",
       "      <td>0.102875</td>\n",
       "      <td>0.610529</td>\n",
       "      <td>1:57222:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   A   G  rs562993331 -0.291280  0.157364  0.064170  1:13259:A:G\n",
       "1    1  17569   A   C  rs535086049  0.301553  0.688140  0.661231  1:17569:A:C\n",
       "2    1  17641   A   G  rs578081284  0.010873  0.090079  0.903924  1:17641:A:G\n",
       "3    1  30741   A   C  rs558169846  0.665389  0.529693  0.209051  1:30741:A:C\n",
       "4    1  57222   C   T  rs576081345  0.052396  0.102875  0.610529  1:57222:C:T"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f2257.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "7da59229-df3f-4d07-8cf4-612798abd73f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f2257_sig=f2257[f2257['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0963f670-e306-4edf-9825-e944b73c3238",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "AD_res=AD_sig[AD_sig['SNP'].isin(f2257_sig['SNP'])]\n",
    "AD_res.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_gwashits_in_f2257imp.tsv' , sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ebc30b61-9ec0-4da0-a4c9-e0754f7b8343",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f2257_res=f2257_sig[f2257_sig['SNP'].isin(AD_sig['SNP'])]\n",
    "f2257_res.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/f2257imp_gwashits_in_AD.tsv' , sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70be212c-f05f-4030-b7ad-0f8664db14ae",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## AD_meta and f2247_f2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "da3801c0-e4f1-46dc-ab00-01ae9d1871bd",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADmeta/combined_ADmeta-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w807d8b080b4cc717) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# # Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADmeta\n",
    "clumping_sbatch=$clumping_dir/combined_ADmeta-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d84578c-92de-48eb-b397-0d3744fb3dab",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Check overlap between ADmeta sumstats and combined sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fc94222a-5a0d-4156-a694-318a9839d5cc",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\"\n",
    "AD = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bb9d82e2-3783-4b34-b2c1-a5bbbb15d1fd",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR     POS REF  ALT             SNP    BETA      SE       P\n",
       "0    1  529825   C    T    1:529825:C:T -0.0152  0.3236  0.9626\n",
       "1    1  531142   C  CTG  1:531142:C:CTG -0.2494  0.2247  0.2670\n",
       "2    1  566327   G    A    1:566327:G:A  0.0975  0.4320  0.8214\n",
       "3    1  581537   G    A    1:581537:G:A -0.3028  0.3180  0.3411\n",
       "4    1  661906   G    C    1:661906:G:C  0.2938  0.3171  0.3542"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b263f891-269a-4dea-bfc2-6f2a128da060",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "AD_sig=AD[AD['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "f41c4f94-a4b6-42e1-95db-5f6212c67de0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.gz'\n",
    "comb = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "c80982b2-5b89-423c-abc9-ee329955b6d6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.439782</td>\n",
       "      <td>0.176722</td>\n",
       "      <td>0.014219</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>0.082381</td>\n",
       "      <td>0.804463</td>\n",
       "      <td>0.918436</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.073457</td>\n",
       "      <td>0.109471</td>\n",
       "      <td>0.502211</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.587040</td>\n",
       "      <td>0.987036</td>\n",
       "      <td>0.045833</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>-0.030952</td>\n",
       "      <td>0.122759</td>\n",
       "      <td>0.800936</td>\n",
       "      <td>1:57222:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   A   G  rs562993331 -0.439782  0.176722  0.014219  1:13259:A:G\n",
       "1    1  17569   A   C  rs535086049  0.082381  0.804463  0.918436  1:17569:A:C\n",
       "2    1  17641   A   G  rs578081284  0.073457  0.109471  0.502211  1:17641:A:G\n",
       "3    1  30741   A   C  rs558169846  1.587040  0.987036  0.045833  1:30741:A:C\n",
       "4    1  57222   C   T  rs576081345 -0.030952  0.122759  0.800936  1:57222:C:T"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "8509da5f-3252-485e-98cc-9ca0fba193c0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "comb_sig=comb[comb['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5dd92fb6-4d67-4e46-8342-cd4e2be84b0c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8199044</th>\n",
       "      <td>6</td>\n",
       "      <td>32577380</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32577380:G:A</td>\n",
       "      <td>0.0729</td>\n",
       "      <td>0.0103</td>\n",
       "      <td>1.829000e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8199369</th>\n",
       "      <td>6</td>\n",
       "      <td>32584693</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32584693:G:C</td>\n",
       "      <td>0.0747</td>\n",
       "      <td>0.0105</td>\n",
       "      <td>9.003000e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705003</th>\n",
       "      <td>17</td>\n",
       "      <td>43691377</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43691377:C:T</td>\n",
       "      <td>0.0591</td>\n",
       "      <td>0.0101</td>\n",
       "      <td>4.438000e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705640</th>\n",
       "      <td>17</td>\n",
       "      <td>43781105</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:43781105:G:A</td>\n",
       "      <td>0.0534</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.805000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705643</th>\n",
       "      <td>17</td>\n",
       "      <td>43781426</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43781426:C:T</td>\n",
       "      <td>0.0534</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.623000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705734</th>\n",
       "      <td>17</td>\n",
       "      <td>43789640</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>17:43789640:C:A</td>\n",
       "      <td>0.0534</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.714000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705774</th>\n",
       "      <td>17</td>\n",
       "      <td>43792358</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:43792358:G:A</td>\n",
       "      <td>0.0534</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.648000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705778</th>\n",
       "      <td>17</td>\n",
       "      <td>43792586</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>17:43792586:C:A</td>\n",
       "      <td>0.0533</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.792000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705781</th>\n",
       "      <td>17</td>\n",
       "      <td>43792895</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43792895:G:T</td>\n",
       "      <td>0.0533</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.829000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705782</th>\n",
       "      <td>17</td>\n",
       "      <td>43792896</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43792896:C:T</td>\n",
       "      <td>0.0533</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.749000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18706345</th>\n",
       "      <td>17</td>\n",
       "      <td>43848750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43848750:C:T</td>\n",
       "      <td>0.0535</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.565000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18706391</th>\n",
       "      <td>17</td>\n",
       "      <td>43853133</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43853133:C:T</td>\n",
       "      <td>0.0532</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.871000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18707029</th>\n",
       "      <td>17</td>\n",
       "      <td>43922942</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43922942:C:T</td>\n",
       "      <td>0.0533</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>4.580000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18707047</th>\n",
       "      <td>17</td>\n",
       "      <td>43924219</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:43924219:C:T</td>\n",
       "      <td>0.0542</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.988000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708109</th>\n",
       "      <td>17</td>\n",
       "      <td>44027366</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44027366:G:T</td>\n",
       "      <td>0.0546</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.568000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708134</th>\n",
       "      <td>17</td>\n",
       "      <td>44030311</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44030311:G:T</td>\n",
       "      <td>0.0544</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.727000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708136</th>\n",
       "      <td>17</td>\n",
       "      <td>44030773</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44030773:G:A</td>\n",
       "      <td>0.0545</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.534000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708163</th>\n",
       "      <td>17</td>\n",
       "      <td>44033002</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44033002:C:T</td>\n",
       "      <td>0.0542</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>3.119000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708303</th>\n",
       "      <td>17</td>\n",
       "      <td>44048336</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44048336:C:T</td>\n",
       "      <td>0.0539</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>3.972000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708304</th>\n",
       "      <td>17</td>\n",
       "      <td>44048350</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44048350:C:T</td>\n",
       "      <td>0.0539</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>3.960000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708469</th>\n",
       "      <td>17</td>\n",
       "      <td>44067400</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44067400:C:T</td>\n",
       "      <td>0.0544</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.780000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708742</th>\n",
       "      <td>17</td>\n",
       "      <td>44095467</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44095467:C:T</td>\n",
       "      <td>0.0575</td>\n",
       "      <td>0.0097</td>\n",
       "      <td>2.991000e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708812</th>\n",
       "      <td>17</td>\n",
       "      <td>44103825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44103825:C:T</td>\n",
       "      <td>0.0555</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>1.779000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708836</th>\n",
       "      <td>17</td>\n",
       "      <td>44106220</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44106220:G:A</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.471000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708858</th>\n",
       "      <td>17</td>\n",
       "      <td>44108906</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44108906:G:A</td>\n",
       "      <td>0.0546</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.558000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18708907</th>\n",
       "      <td>17</td>\n",
       "      <td>44114362</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44114362:T:A</td>\n",
       "      <td>0.0546</td>\n",
       "      <td>0.0100</td>\n",
       "      <td>4.415000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18709006</th>\n",
       "      <td>17</td>\n",
       "      <td>44124535</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44124535:C:T</td>\n",
       "      <td>0.0553</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>1.748000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18709346</th>\n",
       "      <td>17</td>\n",
       "      <td>44161470</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44161470:C:A</td>\n",
       "      <td>0.0555</td>\n",
       "      <td>0.0100</td>\n",
       "      <td>3.175000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18709387</th>\n",
       "      <td>17</td>\n",
       "      <td>44165049</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44165049:G:A</td>\n",
       "      <td>0.0547</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.546000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18709388</th>\n",
       "      <td>17</td>\n",
       "      <td>44165069</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44165069:G:A</td>\n",
       "      <td>0.0546</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>2.593000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18710138</th>\n",
       "      <td>17</td>\n",
       "      <td>44305199</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>17:44305199:C:T</td>\n",
       "      <td>0.0545</td>\n",
       "      <td>0.0100</td>\n",
       "      <td>4.775000e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18710168</th>\n",
       "      <td>17</td>\n",
       "      <td>44315803</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>17:44315803:C:A</td>\n",
       "      <td>0.0559</td>\n",
       "      <td>0.0102</td>\n",
       "      <td>4.389000e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF ALT              SNP    BETA      SE             P\n",
       "8199044     6  32577380   G   A   6:32577380:G:A  0.0729  0.0103  1.829000e-12\n",
       "8199369     6  32584693   G   C   6:32584693:G:C  0.0747  0.0105  9.003000e-13\n",
       "18705003   17  43691377   C   T  17:43691377:C:T  0.0591  0.0101  4.438000e-09\n",
       "18705640   17  43781105   G   A  17:43781105:G:A  0.0534  0.0098  4.805000e-08\n",
       "18705643   17  43781426   C   T  17:43781426:C:T  0.0534  0.0098  4.623000e-08\n",
       "18705734   17  43789640   C   A  17:43789640:C:A  0.0534  0.0098  4.714000e-08\n",
       "18705774   17  43792358   G   A  17:43792358:G:A  0.0534  0.0098  4.648000e-08\n",
       "18705778   17  43792586   C   A  17:43792586:C:A  0.0533  0.0098  4.792000e-08\n",
       "18705781   17  43792895   G   T  17:43792895:G:T  0.0533  0.0098  4.829000e-08\n",
       "18705782   17  43792896   C   T  17:43792896:C:T  0.0533  0.0098  4.749000e-08\n",
       "18706345   17  43848750   C   T  17:43848750:C:T  0.0535  0.0098  4.565000e-08\n",
       "18706391   17  43853133   C   T  17:43853133:C:T  0.0532  0.0098  4.871000e-08\n",
       "18707029   17  43922942   C   T  17:43922942:C:T  0.0533  0.0098  4.580000e-08\n",
       "18707047   17  43924219   C   T  17:43924219:C:T  0.0542  0.0098  2.988000e-08\n",
       "18708109   17  44027366   G   T  17:44027366:G:T  0.0546  0.0098  2.568000e-08\n",
       "18708134   17  44030311   G   T  17:44030311:G:T  0.0544  0.0098  2.727000e-08\n",
       "18708136   17  44030773   G   A  17:44030773:G:A  0.0545  0.0098  2.534000e-08\n",
       "18708163   17  44033002   C   T  17:44033002:C:T  0.0542  0.0098  3.119000e-08\n",
       "18708303   17  44048336   C   T  17:44048336:C:T  0.0539  0.0098  3.972000e-08\n",
       "18708304   17  44048350   C   T  17:44048350:C:T  0.0539  0.0098  3.960000e-08\n",
       "18708469   17  44067400   C   T  17:44067400:C:T  0.0544  0.0098  2.780000e-08\n",
       "18708742   17  44095467   C   T  17:44095467:C:T  0.0575  0.0097  2.991000e-09\n",
       "18708812   17  44103825   C   T  17:44103825:C:T  0.0555  0.0098  1.779000e-08\n",
       "18708836   17  44106220   G   A  17:44106220:G:A  0.0547  0.0098  2.471000e-08\n",
       "18708858   17  44108906   G   A  17:44108906:G:A  0.0546  0.0098  2.558000e-08\n",
       "18708907   17  44114362   T   A  17:44114362:T:A  0.0546  0.0100  4.415000e-08\n",
       "18709006   17  44124535   C   T  17:44124535:C:T  0.0553  0.0098  1.748000e-08\n",
       "18709346   17  44161470   C   A  17:44161470:C:A  0.0555  0.0100  3.175000e-08\n",
       "18709387   17  44165049   G   A  17:44165049:G:A  0.0547  0.0098  2.546000e-08\n",
       "18709388   17  44165069   G   A  17:44165069:G:A  0.0546  0.0098  2.593000e-08\n",
       "18710138   17  44305199   C   T  17:44305199:C:T  0.0545  0.0100  4.775000e-08\n",
       "18710168   17  44315803   C   A  17:44315803:C:A  0.0559  0.0102  4.389000e-08"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_sig[AD_sig['SNP'].isin(comb_sig['SNP'])]\n",
    "#AD_res.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_gwashits_in_f2257imp.tsv' , sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "617a236c-fe08-40dd-9a9f-7645372c0adb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8543520</th>\n",
       "      <td>6</td>\n",
       "      <td>32577380</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs660895</td>\n",
       "      <td>-0.038974</td>\n",
       "      <td>0.006932</td>\n",
       "      <td>1.974104e-08</td>\n",
       "      <td>6:32577380:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8544002</th>\n",
       "      <td>6</td>\n",
       "      <td>32584693</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>rs510205</td>\n",
       "      <td>-0.039528</td>\n",
       "      <td>0.006947</td>\n",
       "      <td>1.335611e-08</td>\n",
       "      <td>6:32584693:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19452908</th>\n",
       "      <td>17</td>\n",
       "      <td>43691377</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs382362</td>\n",
       "      <td>-0.040797</td>\n",
       "      <td>0.007445</td>\n",
       "      <td>4.452665e-08</td>\n",
       "      <td>17:43691377:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453605</th>\n",
       "      <td>17</td>\n",
       "      <td>43781105</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs968028</td>\n",
       "      <td>-0.037328</td>\n",
       "      <td>0.006795</td>\n",
       "      <td>4.097698e-08</td>\n",
       "      <td>17:43781105:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453608</th>\n",
       "      <td>17</td>\n",
       "      <td>43781426</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62056879</td>\n",
       "      <td>-0.037083</td>\n",
       "      <td>0.006793</td>\n",
       "      <td>4.988845e-08</td>\n",
       "      <td>17:43781426:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453696</th>\n",
       "      <td>17</td>\n",
       "      <td>43789640</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs62056930</td>\n",
       "      <td>-0.037202</td>\n",
       "      <td>0.006790</td>\n",
       "      <td>4.442833e-08</td>\n",
       "      <td>17:43789640:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453736</th>\n",
       "      <td>17</td>\n",
       "      <td>43792358</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs56160448</td>\n",
       "      <td>-0.037528</td>\n",
       "      <td>0.006793</td>\n",
       "      <td>3.439537e-08</td>\n",
       "      <td>17:43792358:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453741</th>\n",
       "      <td>17</td>\n",
       "      <td>43792586</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs1880749</td>\n",
       "      <td>-0.037438</td>\n",
       "      <td>0.006791</td>\n",
       "      <td>3.682392e-08</td>\n",
       "      <td>17:43792586:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453742</th>\n",
       "      <td>17</td>\n",
       "      <td>43792895</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs1568951</td>\n",
       "      <td>-0.037377</td>\n",
       "      <td>0.006791</td>\n",
       "      <td>3.877484e-08</td>\n",
       "      <td>17:43792895:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19453743</th>\n",
       "      <td>17</td>\n",
       "      <td>43792896</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs1568950</td>\n",
       "      <td>-0.037377</td>\n",
       "      <td>0.006791</td>\n",
       "      <td>3.877484e-08</td>\n",
       "      <td>17:43792896:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19454293</th>\n",
       "      <td>17</td>\n",
       "      <td>43848750</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62055935</td>\n",
       "      <td>-0.037581</td>\n",
       "      <td>0.006837</td>\n",
       "      <td>4.039801e-08</td>\n",
       "      <td>17:43848750:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19454339</th>\n",
       "      <td>17</td>\n",
       "      <td>43853133</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62055950</td>\n",
       "      <td>-0.037080</td>\n",
       "      <td>0.006793</td>\n",
       "      <td>4.988041e-08</td>\n",
       "      <td>17:43853133:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19454902</th>\n",
       "      <td>17</td>\n",
       "      <td>43922942</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62621252</td>\n",
       "      <td>-0.037133</td>\n",
       "      <td>0.006788</td>\n",
       "      <td>4.672077e-08</td>\n",
       "      <td>17:43922942:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19454918</th>\n",
       "      <td>17</td>\n",
       "      <td>43924219</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs12373124</td>\n",
       "      <td>-0.037259</td>\n",
       "      <td>0.006825</td>\n",
       "      <td>4.975996e-08</td>\n",
       "      <td>17:43924219:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19455918</th>\n",
       "      <td>17</td>\n",
       "      <td>44027366</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62062797</td>\n",
       "      <td>-0.037304</td>\n",
       "      <td>0.006827</td>\n",
       "      <td>4.838269e-08</td>\n",
       "      <td>17:44027366:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19455942</th>\n",
       "      <td>17</td>\n",
       "      <td>44030311</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs17650651</td>\n",
       "      <td>-0.037078</td>\n",
       "      <td>0.006792</td>\n",
       "      <td>4.970156e-08</td>\n",
       "      <td>17:44030311:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19455945</th>\n",
       "      <td>17</td>\n",
       "      <td>44030773</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs62062805</td>\n",
       "      <td>-0.037128</td>\n",
       "      <td>0.006794</td>\n",
       "      <td>4.824919e-08</td>\n",
       "      <td>17:44030773:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19455961</th>\n",
       "      <td>17</td>\n",
       "      <td>44033002</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62063262</td>\n",
       "      <td>-0.037085</td>\n",
       "      <td>0.006793</td>\n",
       "      <td>4.981613e-08</td>\n",
       "      <td>17:44033002:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456092</th>\n",
       "      <td>17</td>\n",
       "      <td>44048336</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62063299</td>\n",
       "      <td>-0.037758</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>3.469765e-08</td>\n",
       "      <td>17:44048336:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456093</th>\n",
       "      <td>17</td>\n",
       "      <td>44048350</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs62063300</td>\n",
       "      <td>-0.037736</td>\n",
       "      <td>0.006836</td>\n",
       "      <td>3.520303e-08</td>\n",
       "      <td>17:44048350:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456260</th>\n",
       "      <td>17</td>\n",
       "      <td>44067400</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs10445337</td>\n",
       "      <td>-0.037398</td>\n",
       "      <td>0.006789</td>\n",
       "      <td>3.759932e-08</td>\n",
       "      <td>17:44067400:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456516</th>\n",
       "      <td>17</td>\n",
       "      <td>44095467</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs58879558</td>\n",
       "      <td>-0.036907</td>\n",
       "      <td>0.006737</td>\n",
       "      <td>4.459231e-08</td>\n",
       "      <td>17:44095467:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456579</th>\n",
       "      <td>17</td>\n",
       "      <td>44103825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs75010486</td>\n",
       "      <td>-0.038018</td>\n",
       "      <td>0.006858</td>\n",
       "      <td>3.083472e-08</td>\n",
       "      <td>17:44103825:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456603</th>\n",
       "      <td>17</td>\n",
       "      <td>44106220</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs79772576</td>\n",
       "      <td>-0.037374</td>\n",
       "      <td>0.006786</td>\n",
       "      <td>3.789658e-08</td>\n",
       "      <td>17:44106220:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456622</th>\n",
       "      <td>17</td>\n",
       "      <td>44108906</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs34579536</td>\n",
       "      <td>-0.037543</td>\n",
       "      <td>0.006784</td>\n",
       "      <td>3.262571e-08</td>\n",
       "      <td>17:44108906:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456667</th>\n",
       "      <td>17</td>\n",
       "      <td>44114362</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>rs62062324</td>\n",
       "      <td>-0.038000</td>\n",
       "      <td>0.006870</td>\n",
       "      <td>3.307425e-08</td>\n",
       "      <td>17:44114362:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19456758</th>\n",
       "      <td>17</td>\n",
       "      <td>44124535</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs79065019</td>\n",
       "      <td>-0.037293</td>\n",
       "      <td>0.006787</td>\n",
       "      <td>4.078965e-08</td>\n",
       "      <td>17:44124535:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19457074</th>\n",
       "      <td>17</td>\n",
       "      <td>44161470</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs80028338</td>\n",
       "      <td>-0.039447</td>\n",
       "      <td>0.007223</td>\n",
       "      <td>4.925496e-08</td>\n",
       "      <td>17:44161470:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19457121</th>\n",
       "      <td>17</td>\n",
       "      <td>44165049</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs3912063</td>\n",
       "      <td>-0.037290</td>\n",
       "      <td>0.006803</td>\n",
       "      <td>4.384398e-08</td>\n",
       "      <td>17:44165049:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19457122</th>\n",
       "      <td>17</td>\n",
       "      <td>44165069</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs3912062</td>\n",
       "      <td>-0.037306</td>\n",
       "      <td>0.006802</td>\n",
       "      <td>4.320062e-08</td>\n",
       "      <td>17:44165069:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19458378</th>\n",
       "      <td>17</td>\n",
       "      <td>44305199</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs2260227</td>\n",
       "      <td>-0.037357</td>\n",
       "      <td>0.006799</td>\n",
       "      <td>4.081502e-08</td>\n",
       "      <td>17:44305199:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19458438</th>\n",
       "      <td>17</td>\n",
       "      <td>44315803</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs2696673</td>\n",
       "      <td>-0.040794</td>\n",
       "      <td>0.007113</td>\n",
       "      <td>1.022869e-08</td>\n",
       "      <td>17:44315803:C:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF ALT       varid      BETA        SE             P  \\\n",
       "8543520     6  32577380   G   A    rs660895 -0.038974  0.006932  1.974104e-08   \n",
       "8544002     6  32584693   G   C    rs510205 -0.039528  0.006947  1.335611e-08   \n",
       "19452908   17  43691377   C   T    rs382362 -0.040797  0.007445  4.452665e-08   \n",
       "19453605   17  43781105   G   A    rs968028 -0.037328  0.006795  4.097698e-08   \n",
       "19453608   17  43781426   C   T  rs62056879 -0.037083  0.006793  4.988845e-08   \n",
       "19453696   17  43789640   C   A  rs62056930 -0.037202  0.006790  4.442833e-08   \n",
       "19453736   17  43792358   G   A  rs56160448 -0.037528  0.006793  3.439537e-08   \n",
       "19453741   17  43792586   C   A   rs1880749 -0.037438  0.006791  3.682392e-08   \n",
       "19453742   17  43792895   G   T   rs1568951 -0.037377  0.006791  3.877484e-08   \n",
       "19453743   17  43792896   C   T   rs1568950 -0.037377  0.006791  3.877484e-08   \n",
       "19454293   17  43848750   C   T  rs62055935 -0.037581  0.006837  4.039801e-08   \n",
       "19454339   17  43853133   C   T  rs62055950 -0.037080  0.006793  4.988041e-08   \n",
       "19454902   17  43922942   C   T  rs62621252 -0.037133  0.006788  4.672077e-08   \n",
       "19454918   17  43924219   C   T  rs12373124 -0.037259  0.006825  4.975996e-08   \n",
       "19455918   17  44027366   G   T  rs62062797 -0.037304  0.006827  4.838269e-08   \n",
       "19455942   17  44030311   G   T  rs17650651 -0.037078  0.006792  4.970156e-08   \n",
       "19455945   17  44030773   G   A  rs62062805 -0.037128  0.006794  4.824919e-08   \n",
       "19455961   17  44033002   C   T  rs62063262 -0.037085  0.006793  4.981613e-08   \n",
       "19456092   17  44048336   C   T  rs62063299 -0.037758  0.006836  3.469765e-08   \n",
       "19456093   17  44048350   C   T  rs62063300 -0.037736  0.006836  3.520303e-08   \n",
       "19456260   17  44067400   C   T  rs10445337 -0.037398  0.006789  3.759932e-08   \n",
       "19456516   17  44095467   C   T  rs58879558 -0.036907  0.006737  4.459231e-08   \n",
       "19456579   17  44103825   C   T  rs75010486 -0.038018  0.006858  3.083472e-08   \n",
       "19456603   17  44106220   G   A  rs79772576 -0.037374  0.006786  3.789658e-08   \n",
       "19456622   17  44108906   G   A  rs34579536 -0.037543  0.006784  3.262571e-08   \n",
       "19456667   17  44114362   T   A  rs62062324 -0.038000  0.006870  3.307425e-08   \n",
       "19456758   17  44124535   C   T  rs79065019 -0.037293  0.006787  4.078965e-08   \n",
       "19457074   17  44161470   C   A  rs80028338 -0.039447  0.007223  4.925496e-08   \n",
       "19457121   17  44165049   G   A   rs3912063 -0.037290  0.006803  4.384398e-08   \n",
       "19457122   17  44165069   G   A   rs3912062 -0.037306  0.006802  4.320062e-08   \n",
       "19458378   17  44305199   C   T   rs2260227 -0.037357  0.006799  4.081502e-08   \n",
       "19458438   17  44315803   C   A   rs2696673 -0.040794  0.007113  1.022869e-08   \n",
       "\n",
       "                      SNP  \n",
       "8543520    6:32577380:G:A  \n",
       "8544002    6:32584693:G:C  \n",
       "19452908  17:43691377:C:T  \n",
       "19453605  17:43781105:G:A  \n",
       "19453608  17:43781426:C:T  \n",
       "19453696  17:43789640:C:A  \n",
       "19453736  17:43792358:G:A  \n",
       "19453741  17:43792586:C:A  \n",
       "19453742  17:43792895:G:T  \n",
       "19453743  17:43792896:C:T  \n",
       "19454293  17:43848750:C:T  \n",
       "19454339  17:43853133:C:T  \n",
       "19454902  17:43922942:C:T  \n",
       "19454918  17:43924219:C:T  \n",
       "19455918  17:44027366:G:T  \n",
       "19455942  17:44030311:G:T  \n",
       "19455945  17:44030773:G:A  \n",
       "19455961  17:44033002:C:T  \n",
       "19456092  17:44048336:C:T  \n",
       "19456093  17:44048350:C:T  \n",
       "19456260  17:44067400:C:T  \n",
       "19456516  17:44095467:C:T  \n",
       "19456579  17:44103825:C:T  \n",
       "19456603  17:44106220:G:A  \n",
       "19456622  17:44108906:G:A  \n",
       "19456667  17:44114362:T:A  \n",
       "19456758  17:44124535:C:T  \n",
       "19457074  17:44161470:C:A  \n",
       "19457121  17:44165049:G:A  \n",
       "19457122  17:44165069:G:A  \n",
       "19458378  17:44305199:C:T  \n",
       "19458438  17:44315803:C:A  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "comb_sig[comb_sig['SNP'].isin(AD_sig['SNP'])]\n",
    "#f2257_res.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/f2257imp_gwashits_in_AD.tsv' , sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e04ce8f-ee7f-46a5-a334-7b21d4bc801e",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "# Bivariate LD_clumping analysis\n",
    "\n",
    "**Important Note:** please take into account that the number of clumps in the log file can (and possibly will) be different (higher) that the number of regions in the `*.clumped_regions` file because there will be some variants that are not clumped with any other and thus do not create a region. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66c26704-18bd-4049-9604-15f9acc611c2",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## AD_proxy data create format compatible with LD_clumping pipeline\n",
    "\n",
    "The file was provided by Andy DeWan from Yale cluster `/gpfs/gibbs/pi/dewan/data/UKBiobank/results/REGENIE_results/results_imputed_data/AD` "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d2b7adcf-f4ce-4570-81f0-87c4febdcac0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file = pd.read_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "97d3de77-1ba4-44f8-9e85-5f31ed00a5ee",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "formatFile= Path('~/project/UKBB_GWAS_dev/data/regenie_template.yml')\n",
    "import gzip\n",
    "import pandas as pd\n",
    "file = pd.read_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt', header=0,delim_whitespace=True, quotechar='\"' )\n",
    "if formatFile.is_file():\n",
    "    output = '~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all' + '_original_columns.snp_stats.gz'\n",
    "else:\n",
    "    output = '~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt' + '.snp_stats.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e649ac47-84b0-4f38-972c-b153a0ed3656",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.845414</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.602991</td>\n",
       "      <td>0.359083</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.437438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.843954</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>1.407020</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>3.579100</td>\n",
       "      <td>1.232760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.058511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000799</td>\n",
       "      <td>0.853630</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.078532</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.433839</td>\n",
       "      <td>0.292335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.510111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.903759</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.314490</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>2.901010</td>\n",
       "      <td>1.052940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.088524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000622</td>\n",
       "      <td>0.837846</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>5.008980</td>\n",
       "      <td>1.598320</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.025216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM  GENPOS           ID ALLELE0 ALLELE1    A1FREQ      INFO       N  \\\n",
       "0      1   13259  rs562993331       G       A  0.000268  0.845414  460649   \n",
       "1      1   17569  rs535086049       C       A  0.000012  0.843954  460649   \n",
       "2      1   17641  rs578081284       G       A  0.000799  0.853630  460649   \n",
       "3      1   30741  rs558169846       C       A  0.000022  0.903759  460649   \n",
       "4      1   57222  rs576081345       T       C  0.000622  0.837846  460649   \n",
       "\n",
       "  TEST      BETA        SE     CHISQ    LOG10P  EXTRA         P  \n",
       "0  ADD  0.158113  0.203616  0.602991  0.359083    NaN  0.437438  \n",
       "1  ADD  1.407020  0.719964  3.579100  1.232760    NaN  0.058511  \n",
       "2  ADD  0.078532  0.119229  0.433839  0.292335    NaN  0.510111  \n",
       "3  ADD -1.314490  0.771762  2.901010  1.052940    NaN  0.088524  \n",
       "4  ADD  0.291624  0.126625  5.008980  1.598320    NaN  0.025216  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7540f047-e44a-46f2-bf7c-47da64918e73",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file.to_csv(output, compression='gzip', sep='\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "dcad14b3-f6e8-4368-968f-0c8a227977ca",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    " # unify output format\n",
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "formatFile = Path('/home/dmc2245/project/UKBB_GWAS_dev/data/regenie_template.yml')\n",
    "reverse_log_p = True\n",
    "if formatFile.is_file() or reverse_log_p:\n",
    "    sumstats = pd.read_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt', header=0,delim_whitespace=True, quotechar='\"') \n",
    "    if formatFile.is_file():\n",
    "        import yaml\n",
    "        config = yaml.safe_load(open(formatFile, 'r'))\n",
    "    try:\n",
    "        sumstats = sumstats.loc[:,list(config.values())]\n",
    "    except:\n",
    "        raise ValueError(f'According to formatFile, input summary statistics should have the following columns: {list(config.values())}.')\n",
    "    sumstats.columns = list(config.keys())\n",
    "    if reverse_log_p:\n",
    "            sumstats['P'] = sumstats['P'].apply(lambda row: 10**-row)\n",
    "    sumstats.to_csv(\"~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.reheader.snp_stats.gz\", compression='gzip', sep='\\t', header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "805d25cc-846c-494b-b6d6-fb462e3b0fdb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.437438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.407020</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>0.058511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.078532</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.510111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>-1.314490</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>0.088524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>0.025216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT          SNP      BETA        SE         P\n",
       "0    1  13259   G   A  rs562993331  0.158113  0.203616  0.437438\n",
       "1    1  17569   C   A  rs535086049  1.407020  0.719964  0.058511\n",
       "2    1  17641   G   A  rs578081284  0.078532  0.119229  0.510111\n",
       "3    1  30741   C   A  rs558169846 -1.314490  0.771762  0.088524\n",
       "4    1  57222   T   C  rs576081345  0.291624  0.126625  0.025216"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sumstats.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2227e583-b8d5-478a-b7d9-def6224386b5",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### AD_proxy and f3393"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "93d2076e-5cb9-4c2d-9af2-130a7906401f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "data = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.reheader.snp_stats.gz', compression='gzip', sep=\"\\t\", header=0, dtype=str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "cd08715c-a94c-4486-9a35-442dc5feab35",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.43743849630213455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.40702</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>0.05851133403985255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.0785319</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.5101113656100561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>-1.31449</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>0.0885237901518778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>0.025216220879824072</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS REF ALT          SNP       BETA        SE                     P\n",
       "0   1  13259   G   A  rs562993331   0.158113  0.203616   0.43743849630213455\n",
       "1   1  17569   C   A  rs535086049    1.40702  0.719964   0.05851133403985255\n",
       "2   1  17641   G   A  rs578081284  0.0785319  0.119229    0.5101113656100561\n",
       "3   1  30741   C   A  rs558169846   -1.31449  0.771762    0.0885237901518778\n",
       "4   1  57222   T   C  rs576081345   0.291624  0.126625  0.025216220879824072"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "5f296de6-1e81-47b7-bb6f-c7c9290e339f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "data.rename(columns={'SNP':'varid'}, inplace=True)\n",
    "data['SNP'] = data['CHR'] + ':' + data['POS'] + ':' + data['REF'] + ':' + data['ALT']\n",
    "data.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.final.snp_stats.gz', sep='\\t', index=False, header=True, compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5087934d-3538-4dcd-9a67-983f0c7f3b1e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.43743849630213455</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.40702</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>0.05851133403985255</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.0785319</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.5101113656100561</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>-1.31449</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>0.0885237901518778</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>0.025216220879824072</td>\n",
       "      <td>1:57222:T:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS REF ALT        varid       BETA        SE                     P  \\\n",
       "0   1  13259   G   A  rs562993331   0.158113  0.203616   0.43743849630213455   \n",
       "1   1  17569   C   A  rs535086049    1.40702  0.719964   0.05851133403985255   \n",
       "2   1  17641   G   A  rs578081284  0.0785319  0.119229    0.5101113656100561   \n",
       "3   1  30741   C   A  rs558169846   -1.31449  0.771762    0.0885237901518778   \n",
       "4   1  57222   T   C  rs576081345   0.291624  0.126625  0.025216220879824072   \n",
       "\n",
       "           SNP  \n",
       "0  1:13259:G:A  \n",
       "1  1:17569:C:A  \n",
       "2  1:17641:G:A  \n",
       "3  1:30741:C:A  \n",
       "4  1:57222:T:C  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ada261f-b3ba-41ce-bba2-00e2a3f83b6a",
   "metadata": {
    "kernel": "Markdown",
    "tags": []
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b999aafb-edb6-4a3a-8b83-bc3ffa8ddd49",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f3393_ADproxy/f3393_ADproxy-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wf767271d8665d92f) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f3393_ADproxy\n",
    "clumping_sbatch=$clumping_dir/f3393_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "054eb4ff-cab3-4e96-b9ae-405adeac1143",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Check overlap between AD proxy sumstats and f3393 sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c22799a0-7ab5-4e6c-a0bc-cb03f7af725f",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.final.snp_stats.gz\"\n",
    "ADproxy = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "a11a5750-bbf1-4c52-b2af-a943a423cfe1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.437438</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.407020</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>0.058511</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.078532</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.510111</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>-1.314490</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>0.088524</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>0.025216</td>\n",
       "      <td>1:57222:T:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   G   A  rs562993331  0.158113  0.203616  0.437438  1:13259:G:A\n",
       "1    1  17569   C   A  rs535086049  1.407020  0.719964  0.058511  1:17569:C:A\n",
       "2    1  17641   G   A  rs578081284  0.078532  0.119229  0.510111  1:17641:G:A\n",
       "3    1  30741   C   A  rs558169846 -1.314490  0.771762  0.088524  1:30741:C:A\n",
       "4    1  57222   T   C  rs576081345  0.291624  0.126625  0.025216  1:57222:T:C"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ADproxy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d4b31a7c-3186-46c7-adbf-44d115dc4a8b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1458344</th>\n",
       "      <td>1</td>\n",
       "      <td>207684192</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>rs12037841</td>\n",
       "      <td>-0.046865</td>\n",
       "      <td>0.008055</td>\n",
       "      <td>6.716299e-09</td>\n",
       "      <td>1:207684192:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458360</th>\n",
       "      <td>1</td>\n",
       "      <td>207685965</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs4562624</td>\n",
       "      <td>-0.047314</td>\n",
       "      <td>0.008047</td>\n",
       "      <td>4.653932e-09</td>\n",
       "      <td>1:207685965:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458405</th>\n",
       "      <td>1</td>\n",
       "      <td>207692049</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs6656401</td>\n",
       "      <td>-0.048239</td>\n",
       "      <td>0.008065</td>\n",
       "      <td>2.519997e-09</td>\n",
       "      <td>1:207692049:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458559</th>\n",
       "      <td>1</td>\n",
       "      <td>207747296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs1752684</td>\n",
       "      <td>-0.045510</td>\n",
       "      <td>0.008029</td>\n",
       "      <td>1.613095e-08</td>\n",
       "      <td>1:207747296:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458600</th>\n",
       "      <td>1</td>\n",
       "      <td>207750568</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs679515</td>\n",
       "      <td>-0.048780</td>\n",
       "      <td>0.008080</td>\n",
       "      <td>1.791884e-09</td>\n",
       "      <td>1:207750568:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104267</th>\n",
       "      <td>8</td>\n",
       "      <td>27466912</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>rs11449170</td>\n",
       "      <td>0.044635</td>\n",
       "      <td>0.006392</td>\n",
       "      <td>2.747894e-12</td>\n",
       "      <td>8:27466912:G:GA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104276</th>\n",
       "      <td>8</td>\n",
       "      <td>27467686</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9331896</td>\n",
       "      <td>0.044401</td>\n",
       "      <td>0.006381</td>\n",
       "      <td>3.305978e-12</td>\n",
       "      <td>8:27467686:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104279</th>\n",
       "      <td>8</td>\n",
       "      <td>27467821</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>rs2070926</td>\n",
       "      <td>0.045557</td>\n",
       "      <td>0.006390</td>\n",
       "      <td>9.600636e-13</td>\n",
       "      <td>8:27467821:C:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104286</th>\n",
       "      <td>8</td>\n",
       "      <td>27468503</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs867230</td>\n",
       "      <td>0.045308</td>\n",
       "      <td>0.006406</td>\n",
       "      <td>1.451443e-12</td>\n",
       "      <td>8:27468503:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032329</th>\n",
       "      <td>8</td>\n",
       "      <td>139393909</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs1519369</td>\n",
       "      <td>0.036701</td>\n",
       "      <td>0.006671</td>\n",
       "      <td>3.941486e-08</td>\n",
       "      <td>8:139393909:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1742 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR        POS REF ALT       varid      BETA        SE  \\\n",
       "1458344     1  207684192   T   G  rs12037841 -0.046865  0.008055   \n",
       "1458360     1  207685965   A   C   rs4562624 -0.047314  0.008047   \n",
       "1458405     1  207692049   A   G   rs6656401 -0.048239  0.008065   \n",
       "1458559     1  207747296   A   G   rs1752684 -0.045510  0.008029   \n",
       "1458600     1  207750568   T   C    rs679515 -0.048780  0.008080   \n",
       "...       ...        ...  ..  ..         ...       ...       ...   \n",
       "21104267    8   27466912   G  GA  rs11449170  0.044635  0.006392   \n",
       "21104276    8   27467686   C   T   rs9331896  0.044401  0.006381   \n",
       "21104279    8   27467821   C   G   rs2070926  0.045557  0.006390   \n",
       "21104286    8   27468503   C   A    rs867230  0.045308  0.006406   \n",
       "22032329    8  139393909   C   T   rs1519369  0.036701  0.006671   \n",
       "\n",
       "                     P              SNP  \n",
       "1458344   6.716299e-09  1:207684192:T:G  \n",
       "1458360   4.653932e-09  1:207685965:A:C  \n",
       "1458405   2.519997e-09  1:207692049:A:G  \n",
       "1458559   1.613095e-08  1:207747296:A:G  \n",
       "1458600   1.791884e-09  1:207750568:T:C  \n",
       "...                ...              ...  \n",
       "21104267  2.747894e-12  8:27466912:G:GA  \n",
       "21104276  3.305978e-12   8:27467686:C:T  \n",
       "21104279  9.600636e-13   8:27467821:C:G  \n",
       "21104286  1.451443e-12   8:27468503:C:A  \n",
       "22032329  3.941486e-08  8:139393909:C:T  \n",
       "\n",
       "[1742 rows x 9 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "ADproxy[ADproxy['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "1855d602-dd49-4754-acaf-18c611e67c1c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.gz'\n",
    "f3393 = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "fd767807-380c-4a46-adf1-19e197e0ed78",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>-0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>0.272939</td>\n",
       "      <td>1:13259:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1.195410</td>\n",
       "      <td>1.869850</td>\n",
       "      <td>0.522621</td>\n",
       "      <td>1:17569:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>0.061506</td>\n",
       "      <td>1:17641:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1.280430</td>\n",
       "      <td>1.245250</td>\n",
       "      <td>0.303832</td>\n",
       "      <td>1:30741:A:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>-0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.726750</td>\n",
       "      <td>1:52144:A:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHR    POS REF ALT        varid      BETA        SE         P          SNP\n",
       "0    1  13259   A   G  rs562993331 -0.444965  0.405872  0.272939  1:13259:A:G\n",
       "1    1  17569   A   C  rs535086049  1.195410  1.869850  0.522621  1:17569:A:C\n",
       "2    1  17641   A   G  rs578081284  0.435933  0.233139  0.061506  1:17641:A:G\n",
       "3    1  30741   A   C  rs558169846  1.280430  1.245250  0.303832  1:30741:A:C\n",
       "4    1  52144   A   T  rs190291950 -0.117852  0.337249  0.726750  1:52144:A:T"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "4db35a6b-6e3f-4774-bba0-09ce28adc996",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2181439</th>\n",
       "      <td>2</td>\n",
       "      <td>54687254</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs13026575</td>\n",
       "      <td>-0.091392</td>\n",
       "      <td>0.016453</td>\n",
       "      <td>3.519978e-08</td>\n",
       "      <td>2:54687254:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181496</th>\n",
       "      <td>2</td>\n",
       "      <td>54693754</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>rs10169422</td>\n",
       "      <td>-0.089591</td>\n",
       "      <td>0.015662</td>\n",
       "      <td>1.338474e-08</td>\n",
       "      <td>2:54693754:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181535</th>\n",
       "      <td>2</td>\n",
       "      <td>54697387</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>2:54697387_AT_A</td>\n",
       "      <td>-0.081630</td>\n",
       "      <td>0.014796</td>\n",
       "      <td>4.111308e-08</td>\n",
       "      <td>2:54697387:A:AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181560</th>\n",
       "      <td>2</td>\n",
       "      <td>54699021</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9677271</td>\n",
       "      <td>-0.093448</td>\n",
       "      <td>0.016656</td>\n",
       "      <td>2.598723e-08</td>\n",
       "      <td>2:54699021:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181562</th>\n",
       "      <td>2</td>\n",
       "      <td>54699067</td>\n",
       "      <td>A</td>\n",
       "      <td>AACTTC</td>\n",
       "      <td>rs746705223</td>\n",
       "      <td>-0.093255</td>\n",
       "      <td>0.016662</td>\n",
       "      <td>2.809959e-08</td>\n",
       "      <td>2:54699067:A:AACTTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21733696</th>\n",
       "      <td>22</td>\n",
       "      <td>50956117</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs200126237</td>\n",
       "      <td>-0.909770</td>\n",
       "      <td>0.141901</td>\n",
       "      <td>3.488751e-09</td>\n",
       "      <td>22:50956117:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21734140</th>\n",
       "      <td>22</td>\n",
       "      <td>50993633</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22:50993633_C_T</td>\n",
       "      <td>-1.425390</td>\n",
       "      <td>0.175632</td>\n",
       "      <td>2.767579e-13</td>\n",
       "      <td>22:50993633:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21734565</th>\n",
       "      <td>22</td>\n",
       "      <td>51032411</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs776344563</td>\n",
       "      <td>-1.469330</td>\n",
       "      <td>0.183468</td>\n",
       "      <td>5.858683e-13</td>\n",
       "      <td>22:51032411:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735509</th>\n",
       "      <td>22</td>\n",
       "      <td>51146132</td>\n",
       "      <td>C</td>\n",
       "      <td>CAAATA</td>\n",
       "      <td>22:51146132_CAAATA_C</td>\n",
       "      <td>-1.262120</td>\n",
       "      <td>0.186638</td>\n",
       "      <td>8.077370e-10</td>\n",
       "      <td>22:51146132:C:CAAATA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735924</th>\n",
       "      <td>22</td>\n",
       "      <td>51237215</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs536109858</td>\n",
       "      <td>-1.345640</td>\n",
       "      <td>0.209808</td>\n",
       "      <td>6.360189e-09</td>\n",
       "      <td>22:51237215:T:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1337 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF     ALT                 varid      BETA        SE  \\\n",
       "2181439     2  54687254   T       C            rs13026575 -0.091392  0.016453   \n",
       "2181496     2  54693754   G       T            rs10169422 -0.089591  0.015662   \n",
       "2181535     2  54697387   A      AT       2:54697387_AT_A -0.081630  0.014796   \n",
       "2181560     2  54699021   C       T             rs9677271 -0.093448  0.016656   \n",
       "2181562     2  54699067   A  AACTTC           rs746705223 -0.093255  0.016662   \n",
       "...       ...       ...  ..     ...                   ...       ...       ...   \n",
       "21733696   22  50956117   A       G           rs200126237 -0.909770  0.141901   \n",
       "21734140   22  50993633   T       C       22:50993633_C_T -1.425390  0.175632   \n",
       "21734565   22  51032411   G       A           rs776344563 -1.469330  0.183468   \n",
       "21735509   22  51146132   C  CAAATA  22:51146132_CAAATA_C -1.262120  0.186638   \n",
       "21735924   22  51237215   T       C           rs536109858 -1.345640  0.209808   \n",
       "\n",
       "                     P                   SNP  \n",
       "2181439   3.519978e-08        2:54687254:T:C  \n",
       "2181496   1.338474e-08        2:54693754:G:T  \n",
       "2181535   4.111308e-08       2:54697387:A:AT  \n",
       "2181560   2.598723e-08        2:54699021:C:T  \n",
       "2181562   2.809959e-08   2:54699067:A:AACTTC  \n",
       "...                ...                   ...  \n",
       "21733696  3.488751e-09       22:50956117:A:G  \n",
       "21734140  2.767579e-13       22:50993633:T:C  \n",
       "21734565  5.858683e-13       22:51032411:G:A  \n",
       "21735509  8.077370e-10  22:51146132:C:CAAATA  \n",
       "21735924  6.360189e-09       22:51237215:T:C  \n",
       "\n",
       "[1337 rows x 9 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393[f3393['P'] < 5e-08]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "d4a8ffb9-fe69-4f53-b3d3-25570a3e0e77",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18189710</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs9275312</td>\n",
       "      <td>-0.053307</td>\n",
       "      <td>0.009735</td>\n",
       "      <td>3.834864e-08</td>\n",
       "      <td>6:32665728:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189768</th>\n",
       "      <td>6</td>\n",
       "      <td>32666802</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275327</td>\n",
       "      <td>-0.053083</td>\n",
       "      <td>0.009776</td>\n",
       "      <td>4.980696e-08</td>\n",
       "      <td>6:32666802:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189770</th>\n",
       "      <td>6</td>\n",
       "      <td>32666822</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275328</td>\n",
       "      <td>-0.053097</td>\n",
       "      <td>0.009776</td>\n",
       "      <td>4.938784e-08</td>\n",
       "      <td>6:32666822:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189814</th>\n",
       "      <td>6</td>\n",
       "      <td>32667638</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275348</td>\n",
       "      <td>-0.053767</td>\n",
       "      <td>0.009840</td>\n",
       "      <td>4.097227e-08</td>\n",
       "      <td>6:32667638:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189825</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>rs9275358</td>\n",
       "      <td>-0.054177</td>\n",
       "      <td>0.009860</td>\n",
       "      <td>3.447625e-08</td>\n",
       "      <td>6:32667895:C:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189831</th>\n",
       "      <td>6</td>\n",
       "      <td>32667957</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275362</td>\n",
       "      <td>-0.054326</td>\n",
       "      <td>0.009988</td>\n",
       "      <td>4.726830e-08</td>\n",
       "      <td>6:32667957:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18189841</th>\n",
       "      <td>6</td>\n",
       "      <td>32668125</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275365</td>\n",
       "      <td>-0.053562</td>\n",
       "      <td>0.009758</td>\n",
       "      <td>3.556805e-08</td>\n",
       "      <td>6:32668125:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF ALT      varid      BETA        SE             P  \\\n",
       "18189710    6  32665728   A   G  rs9275312 -0.053307  0.009735  3.834864e-08   \n",
       "18189768    6  32666802   C   T  rs9275327 -0.053083  0.009776  4.980696e-08   \n",
       "18189770    6  32666822   C   T  rs9275328 -0.053097  0.009776  4.938784e-08   \n",
       "18189814    6  32667638   G   A  rs9275348 -0.053767  0.009840  4.097227e-08   \n",
       "18189825    6  32667895   C   G  rs9275358 -0.054177  0.009860  3.447625e-08   \n",
       "18189831    6  32667957   C   T  rs9275362 -0.054326  0.009988  4.726830e-08   \n",
       "18189841    6  32668125   G   A  rs9275365 -0.053562  0.009758  3.556805e-08   \n",
       "\n",
       "                     SNP  \n",
       "18189710  6:32665728:A:G  \n",
       "18189768  6:32666802:C:T  \n",
       "18189770  6:32666822:C:T  \n",
       "18189814  6:32667638:G:A  \n",
       "18189825  6:32667895:C:G  \n",
       "18189831  6:32667957:C:T  \n",
       "18189841  6:32668125:G:A  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ADproxy_sig[ADproxy_sig['varid'].isin(f3393_sig['varid'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "3668e285-7e61-4738-96db-21f1ef727586",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>varid</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8516436</th>\n",
       "      <td>6</td>\n",
       "      <td>32665728</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs9275312</td>\n",
       "      <td>-0.102268</td>\n",
       "      <td>0.018389</td>\n",
       "      <td>3.606783e-08</td>\n",
       "      <td>6:32665728:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516493</th>\n",
       "      <td>6</td>\n",
       "      <td>32666802</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275327</td>\n",
       "      <td>-0.104226</td>\n",
       "      <td>0.018451</td>\n",
       "      <td>2.210192e-08</td>\n",
       "      <td>6:32666802:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516495</th>\n",
       "      <td>6</td>\n",
       "      <td>32666822</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275328</td>\n",
       "      <td>-0.104121</td>\n",
       "      <td>0.018451</td>\n",
       "      <td>2.281970e-08</td>\n",
       "      <td>6:32666822:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516539</th>\n",
       "      <td>6</td>\n",
       "      <td>32667638</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs9275348</td>\n",
       "      <td>-0.102989</td>\n",
       "      <td>0.018589</td>\n",
       "      <td>4.056113e-08</td>\n",
       "      <td>6:32667638:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516550</th>\n",
       "      <td>6</td>\n",
       "      <td>32667895</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275358</td>\n",
       "      <td>-0.104878</td>\n",
       "      <td>0.018604</td>\n",
       "      <td>2.362490e-08</td>\n",
       "      <td>6:32667895:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516555</th>\n",
       "      <td>6</td>\n",
       "      <td>32667957</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs9275362</td>\n",
       "      <td>-0.103782</td>\n",
       "      <td>0.018842</td>\n",
       "      <td>4.881463e-08</td>\n",
       "      <td>6:32667957:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8516565</th>\n",
       "      <td>6</td>\n",
       "      <td>32668125</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs9275365</td>\n",
       "      <td>-0.103314</td>\n",
       "      <td>0.018425</td>\n",
       "      <td>2.791130e-08</td>\n",
       "      <td>6:32668125:A:G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS REF ALT      varid      BETA        SE             P  \\\n",
       "8516436    6  32665728   G   A  rs9275312 -0.102268  0.018389  3.606783e-08   \n",
       "8516493    6  32666802   T   C  rs9275327 -0.104226  0.018451  2.210192e-08   \n",
       "8516495    6  32666822   T   C  rs9275328 -0.104121  0.018451  2.281970e-08   \n",
       "8516539    6  32667638   A   G  rs9275348 -0.102989  0.018589  4.056113e-08   \n",
       "8516550    6  32667895   G   C  rs9275358 -0.104878  0.018604  2.362490e-08   \n",
       "8516555    6  32667957   T   C  rs9275362 -0.103782  0.018842  4.881463e-08   \n",
       "8516565    6  32668125   A   G  rs9275365 -0.103314  0.018425  2.791130e-08   \n",
       "\n",
       "                    SNP  \n",
       "8516436  6:32665728:G:A  \n",
       "8516493  6:32666802:T:C  \n",
       "8516495  6:32666822:T:C  \n",
       "8516539  6:32667638:A:G  \n",
       "8516550  6:32667895:G:C  \n",
       "8516555  6:32667957:T:C  \n",
       "8516565  6:32668125:A:G  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Issues with REF/ALT alleles. GWAS was run again with --ref_first to account for these differences. \n",
    "f3393_sig[f3393_sig['varid'].isin(ADproxy_sig['varid'])]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01b6e4c7-ca46-416b-bf7c-0091174fdd16",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## AD_proxy and f2247"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95321466-a9a9-420c-a747-23ab4281bc5c",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADproxy/f2247_ADproxy-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wb761843e0224ccac) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADproxy\n",
    "clumping_sbatch=$clumping_dir/f2247_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cd2c7d3-685c-431f-93d9-8a6aa770a92c",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD proxy and f2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "17bb0719-0ca7-468d-8adb-f82e6bf1cdee",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADproxy/f2257_ADproxy-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w5fad6235b33f0392) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADproxy\n",
    "clumping_sbatch=$clumping_dir/f2257_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d16e0826-6d4d-44dd-921b-597f1b9e8216",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD proxy and f2247_f2257"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4c7e7741-d15f-4e66-a377-a88deb427756",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADproxy/combined_ADproxy-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w79db46045a15b733) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# # Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADproxy\n",
    "clumping_sbatch=$clumping_dir/combined_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "249cc4d2-0ef3-4146-bfa8-3a95a6596a04",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD proxy and ADmeta "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "7f1a4c42-1e88-45a0-83ca-9beedb118a9e",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_ADmeta_ADproxy/ADmeta_ADproxy-ldclumping_2022-08-10.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wccf54b3709973dd2) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# # Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_ADmeta_ADproxy\n",
    "clumping_sbatch=$clumping_dir/ADmeta_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz`\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09e17de2-a47c-4495-942d-af9c12057b87",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Use the replicate option\n",
    "\n",
    "This means that only clumps containing clumped SNPs with p2-significant results in more than one result file are shown."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ae823609-7fca-4763-a120-dda9d86d8f73",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/081822_ADmeta_ADproxy_replicate/ADmeta_ADproxy-ldclumping_2022-08-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wc52702c093137b46) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/081822_ADmeta_ADproxy_replicate\n",
    "clumping_sbatch=$clumping_dir/ADmeta_ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=`echo /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz`\n",
    "replicate=True\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=5e-06\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BETA\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate\n",
    "    --replicate $replicate\n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f048fdb5-e758-4898-8901-02dc78d0db73",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Find overlap between AD_meta and AD_proxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8f37e9b1-b3c4-4882-9b00-ef0a7f5a114c",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "file = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz\"\n",
    "AD_meta = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3779f61c-9e47-4bff-ab27-12450c9dcf4e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>529825</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0152</td>\n",
       "      <td>0.3236</td>\n",
       "      <td>0.9626</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "      <td>1:529825:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>531142</td>\n",
       "      <td>C</td>\n",
       "      <td>CTG</td>\n",
       "      <td>-0.2494</td>\n",
       "      <td>0.2247</td>\n",
       "      <td>0.2670</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "      <td>1:531142:C:CTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>566327</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0975</td>\n",
       "      <td>0.4320</td>\n",
       "      <td>0.8214</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "      <td>1:566327:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>581537</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.3028</td>\n",
       "      <td>0.3180</td>\n",
       "      <td>0.3411</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "      <td>1:581537:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>661906</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2938</td>\n",
       "      <td>0.3171</td>\n",
       "      <td>0.3542</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "      <td>1:661906:G:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101109</th>\n",
       "      <td>22</td>\n",
       "      <td>51237496</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.1502</td>\n",
       "      <td>0.2931</td>\n",
       "      <td>0.6083</td>\n",
       "      <td>22:51237496:C:T</td>\n",
       "      <td>22:51237496:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101110</th>\n",
       "      <td>22</td>\n",
       "      <td>51237535</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0655</td>\n",
       "      <td>0.2898</td>\n",
       "      <td>0.8213</td>\n",
       "      <td>22:51237535:C:A</td>\n",
       "      <td>22:51237535:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101111</th>\n",
       "      <td>22</td>\n",
       "      <td>51238249</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0098</td>\n",
       "      <td>0.0228</td>\n",
       "      <td>0.6673</td>\n",
       "      <td>22:51238249:C:A</td>\n",
       "      <td>22:51238249:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101112</th>\n",
       "      <td>22</td>\n",
       "      <td>51238318</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0476</td>\n",
       "      <td>0.0813</td>\n",
       "      <td>0.5584</td>\n",
       "      <td>22:51238318:T:A</td>\n",
       "      <td>22:51238318:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21101113</th>\n",
       "      <td>22</td>\n",
       "      <td>51238328</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0312</td>\n",
       "      <td>0.1082</td>\n",
       "      <td>0.7731</td>\n",
       "      <td>22:51238328:C:A</td>\n",
       "      <td>22:51238328:C:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21101114 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR       POS REF  ALT    BETA      SE       P     SNP_original  \\\n",
       "0           1    529825   C    T -0.0152  0.3236  0.9626     1:529825:C:T   \n",
       "1           1    531142   C  CTG -0.2494  0.2247  0.2670   1:531142:C:CTG   \n",
       "2           1    566327   G    A  0.0975  0.4320  0.8214     1:566327:G:A   \n",
       "3           1    581537   G    A -0.3028  0.3180  0.3411     1:581537:G:A   \n",
       "4           1    661906   G    C  0.2938  0.3171  0.3542     1:661906:G:C   \n",
       "...       ...       ...  ..  ...     ...     ...     ...              ...   \n",
       "21101109   22  51237496   C    T  0.1502  0.2931  0.6083  22:51237496:C:T   \n",
       "21101110   22  51237535   C    A -0.0655  0.2898  0.8213  22:51237535:C:A   \n",
       "21101111   22  51238249   C    A  0.0098  0.0228  0.6673  22:51238249:C:A   \n",
       "21101112   22  51238318   T    A  0.0476  0.0813  0.5584  22:51238318:T:A   \n",
       "21101113   22  51238328   C    A  0.0312  0.1082  0.7731  22:51238328:C:A   \n",
       "\n",
       "                      SNP  \n",
       "0            1:529825:C:T  \n",
       "1          1:531142:C:CTG  \n",
       "2            1:566327:G:A  \n",
       "3            1:581537:G:A  \n",
       "4            1:661906:G:C  \n",
       "...                   ...  \n",
       "21101109  22:51237496:C:T  \n",
       "21101110  22:51237535:C:A  \n",
       "21101111  22:51238249:C:A  \n",
       "21101112  22:51238318:T:A  \n",
       "21101113  22:51238328:C:A  \n",
       "\n",
       "[21101114 rows x 9 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "49e12478-52bb-42fa-b3eb-103ed8f6301f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "#Number of hits that are genome-wide significant\n",
    "AD_meta_sig=AD_meta[AD_meta['P'] < 5e-06]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c9898a24-df58-4c8c-aa34-eac6daa7bddb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "file='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz'\n",
    "AD_proxy = pd.read_csv(file,compression=\"gzip\",header=0,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "916b4cef-8b8e-49e8-b0ee-cff874c685c6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>varid</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.845414</td>\n",
       "      <td>0.158113</td>\n",
       "      <td>0.203616</td>\n",
       "      <td>0.437438</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.843954</td>\n",
       "      <td>1.407020</td>\n",
       "      <td>0.719964</td>\n",
       "      <td>0.058511</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000799</td>\n",
       "      <td>0.853630</td>\n",
       "      <td>0.078532</td>\n",
       "      <td>0.119229</td>\n",
       "      <td>0.510111</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.903759</td>\n",
       "      <td>-1.314490</td>\n",
       "      <td>0.771762</td>\n",
       "      <td>0.088524</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>57222</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000622</td>\n",
       "      <td>0.837846</td>\n",
       "      <td>0.291624</td>\n",
       "      <td>0.126625</td>\n",
       "      <td>0.025216</td>\n",
       "      <td>rs576081345</td>\n",
       "      <td>1:57222:T:C</td>\n",
       "      <td>1:57222:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23095994</th>\n",
       "      <td>9</td>\n",
       "      <td>141101939</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.022299</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.008858</td>\n",
       "      <td>0.021224</td>\n",
       "      <td>0.676424</td>\n",
       "      <td>9:141101939_C_T</td>\n",
       "      <td>9:141101939:C:T</td>\n",
       "      <td>9:141101939:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23095995</th>\n",
       "      <td>9</td>\n",
       "      <td>141102535</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000484</td>\n",
       "      <td>0.816239</td>\n",
       "      <td>0.266958</td>\n",
       "      <td>0.153428</td>\n",
       "      <td>0.081865</td>\n",
       "      <td>9:141102535_CT_C</td>\n",
       "      <td>9:141102535:CT:C</td>\n",
       "      <td>9:141102535:CT:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23095996</th>\n",
       "      <td>9</td>\n",
       "      <td>141102859</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.011236</td>\n",
       "      <td>0.809996</td>\n",
       "      <td>-0.013092</td>\n",
       "      <td>0.032992</td>\n",
       "      <td>0.691506</td>\n",
       "      <td>9:141102859_C_T</td>\n",
       "      <td>9:141102859:C:T</td>\n",
       "      <td>9:141102859:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23095997</th>\n",
       "      <td>9</td>\n",
       "      <td>141104957</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.010978</td>\n",
       "      <td>0.846640</td>\n",
       "      <td>0.023038</td>\n",
       "      <td>0.032529</td>\n",
       "      <td>0.478803</td>\n",
       "      <td>9:141104957_G_A</td>\n",
       "      <td>9:141104957:G:A</td>\n",
       "      <td>9:141104957:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23095998</th>\n",
       "      <td>9</td>\n",
       "      <td>141109216</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000211</td>\n",
       "      <td>0.925246</td>\n",
       "      <td>-0.394753</td>\n",
       "      <td>0.218611</td>\n",
       "      <td>0.070959</td>\n",
       "      <td>9:141109216_A_G</td>\n",
       "      <td>9:141109216:A:G</td>\n",
       "      <td>9:141109216:A:G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23095999 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR        POS REF ALT       MAF      INFO      BETA        SE  \\\n",
       "0           1      13259   G   A  0.000268  0.845414  0.158113  0.203616   \n",
       "1           1      17569   C   A  0.000012  0.843954  1.407020  0.719964   \n",
       "2           1      17641   G   A  0.000799  0.853630  0.078532  0.119229   \n",
       "3           1      30741   C   A  0.000022  0.903759 -1.314490  0.771762   \n",
       "4           1      57222   T   C  0.000622  0.837846  0.291624  0.126625   \n",
       "...       ...        ...  ..  ..       ...       ...       ...       ...   \n",
       "23095994    9  141101939   C   T  0.022299  1.000000 -0.008858  0.021224   \n",
       "23095995    9  141102535  CT   C  0.000484  0.816239  0.266958  0.153428   \n",
       "23095996    9  141102859   C   T  0.011236  0.809996 -0.013092  0.032992   \n",
       "23095997    9  141104957   G   A  0.010978  0.846640  0.023038  0.032529   \n",
       "23095998    9  141109216   A   G  0.000211  0.925246 -0.394753  0.218611   \n",
       "\n",
       "                 P             varid      SNP_original               SNP  \n",
       "0         0.437438       rs562993331       1:13259:G:A       1:13259:G:A  \n",
       "1         0.058511       rs535086049       1:17569:C:A       1:17569:C:A  \n",
       "2         0.510111       rs578081284       1:17641:G:A       1:17641:G:A  \n",
       "3         0.088524       rs558169846       1:30741:C:A       1:30741:C:A  \n",
       "4         0.025216       rs576081345       1:57222:T:C       1:57222:T:C  \n",
       "...            ...               ...               ...               ...  \n",
       "23095994  0.676424   9:141101939_C_T   9:141101939:C:T   9:141101939:C:T  \n",
       "23095995  0.081865  9:141102535_CT_C  9:141102535:CT:C  9:141102535:CT:C  \n",
       "23095996  0.691506   9:141102859_C_T   9:141102859:C:T   9:141102859:C:T  \n",
       "23095997  0.478803   9:141104957_G_A   9:141104957:G:A   9:141104957:G:A  \n",
       "23095998  0.070959   9:141109216_A_G   9:141109216:A:G   9:141109216:A:G  \n",
       "\n",
       "[23095999 rows x 12 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_proxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2a4d9956-bb4f-4feb-b869-d4c680da6711",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "AD_proxy_sig=AD_proxy[AD_proxy['P'] < 5e-06]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c276f212-0d13-415c-9b5f-dd94ad1818a1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>991475</th>\n",
       "      <td>1</td>\n",
       "      <td>161097241</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0451</td>\n",
       "      <td>0.0090</td>\n",
       "      <td>5.831000e-07</td>\n",
       "      <td>1:161097241:T:A</td>\n",
       "      <td>1:161097241:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991490</th>\n",
       "      <td>1</td>\n",
       "      <td>161100336</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0446</td>\n",
       "      <td>0.0090</td>\n",
       "      <td>7.636000e-07</td>\n",
       "      <td>1:161100336:T:A</td>\n",
       "      <td>1:161100336:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991505</th>\n",
       "      <td>1</td>\n",
       "      <td>161103445</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0491</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>2.677000e-08</td>\n",
       "      <td>1:161103445:G:T</td>\n",
       "      <td>1:161103445:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991511</th>\n",
       "      <td>1</td>\n",
       "      <td>161104124</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0447</td>\n",
       "      <td>0.0090</td>\n",
       "      <td>7.208000e-07</td>\n",
       "      <td>1:161104124:G:A</td>\n",
       "      <td>1:161104124:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991522</th>\n",
       "      <td>1</td>\n",
       "      <td>161106354</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>-0.0494</td>\n",
       "      <td>0.0088</td>\n",
       "      <td>2.242000e-08</td>\n",
       "      <td>1:161106354:G:T</td>\n",
       "      <td>1:161106354:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19936176</th>\n",
       "      <td>19</td>\n",
       "      <td>45806553</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.3074</td>\n",
       "      <td>0.0437</td>\n",
       "      <td>2.034000e-12</td>\n",
       "      <td>19:45806553:C:T</td>\n",
       "      <td>19:45806553:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19936327</th>\n",
       "      <td>19</td>\n",
       "      <td>45825418</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.3338</td>\n",
       "      <td>0.0521</td>\n",
       "      <td>1.538000e-10</td>\n",
       "      <td>19:45825418:C:T</td>\n",
       "      <td>19:45825418:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19936371</th>\n",
       "      <td>19</td>\n",
       "      <td>45830947</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.0451</td>\n",
       "      <td>0.0089</td>\n",
       "      <td>4.086000e-07</td>\n",
       "      <td>19:45830947:G:A</td>\n",
       "      <td>19:45830947:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19936431</th>\n",
       "      <td>19</td>\n",
       "      <td>45838141</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.6209</td>\n",
       "      <td>0.0764</td>\n",
       "      <td>4.215000e-16</td>\n",
       "      <td>19:45838141:C:T</td>\n",
       "      <td>19:45838141:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19936614</th>\n",
       "      <td>19</td>\n",
       "      <td>45855406</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.5474</td>\n",
       "      <td>0.0649</td>\n",
       "      <td>3.159000e-17</td>\n",
       "      <td>19:45855406:G:C</td>\n",
       "      <td>19:45855406:G:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2097 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR        POS REF ALT    BETA      SE             P  \\\n",
       "991475      1  161097241   T   A  0.0451  0.0090  5.831000e-07   \n",
       "991490      1  161100336   T   A  0.0446  0.0090  7.636000e-07   \n",
       "991505      1  161103445   G   T -0.0491  0.0088  2.677000e-08   \n",
       "991511      1  161104124   G   A  0.0447  0.0090  7.208000e-07   \n",
       "991522      1  161106354   G   T -0.0494  0.0088  2.242000e-08   \n",
       "...       ...        ...  ..  ..     ...     ...           ...   \n",
       "19936176   19   45806553   C   T  0.3074  0.0437  2.034000e-12   \n",
       "19936327   19   45825418   C   T  0.3338  0.0521  1.538000e-10   \n",
       "19936371   19   45830947   G   A -0.0451  0.0089  4.086000e-07   \n",
       "19936431   19   45838141   C   T  0.6209  0.0764  4.215000e-16   \n",
       "19936614   19   45855406   G   C  0.5474  0.0649  3.159000e-17   \n",
       "\n",
       "             SNP_original              SNP  \n",
       "991475    1:161097241:T:A  1:161097241:T:A  \n",
       "991490    1:161100336:T:A  1:161100336:T:A  \n",
       "991505    1:161103445:G:T  1:161103445:T:G  \n",
       "991511    1:161104124:G:A  1:161104124:G:A  \n",
       "991522    1:161106354:G:T  1:161106354:T:G  \n",
       "...                   ...              ...  \n",
       "19936176  19:45806553:C:T  19:45806553:C:T  \n",
       "19936327  19:45825418:C:T  19:45825418:C:T  \n",
       "19936371  19:45830947:G:A  19:45830947:G:A  \n",
       "19936431  19:45838141:C:T  19:45838141:C:T  \n",
       "19936614  19:45855406:G:C  19:45855406:G:C  \n",
       "\n",
       "[2097 rows x 9 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_meta_sig[AD_meta_sig['SNP'].isin(AD_proxy_sig['SNP'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0ed985b5-b5f2-4662-a362-cd9b22aeddd4",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>varid</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1065054</th>\n",
       "      <td>1</td>\n",
       "      <td>161097241</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.275104</td>\n",
       "      <td>0.999587</td>\n",
       "      <td>0.032051</td>\n",
       "      <td>0.006947</td>\n",
       "      <td>4.102135e-06</td>\n",
       "      <td>rs12727614</td>\n",
       "      <td>1:161097241:T:A</td>\n",
       "      <td>1:161097241:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065073</th>\n",
       "      <td>1</td>\n",
       "      <td>161100336</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.275114</td>\n",
       "      <td>0.999638</td>\n",
       "      <td>0.032019</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>4.190058e-06</td>\n",
       "      <td>rs11583045</td>\n",
       "      <td>1:161100336:T:A</td>\n",
       "      <td>1:161100336:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065090</th>\n",
       "      <td>1</td>\n",
       "      <td>161103445</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.295974</td>\n",
       "      <td>0.999809</td>\n",
       "      <td>0.032169</td>\n",
       "      <td>0.006802</td>\n",
       "      <td>2.336307e-06</td>\n",
       "      <td>rs10797093</td>\n",
       "      <td>1:161103445:T:G</td>\n",
       "      <td>1:161103445:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065094</th>\n",
       "      <td>1</td>\n",
       "      <td>161104124</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.275063</td>\n",
       "      <td>0.999811</td>\n",
       "      <td>0.031940</td>\n",
       "      <td>0.006946</td>\n",
       "      <td>4.421911e-06</td>\n",
       "      <td>rs4540676</td>\n",
       "      <td>1:161104124:G:A</td>\n",
       "      <td>1:161104124:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1065110</th>\n",
       "      <td>1</td>\n",
       "      <td>161106354</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.295966</td>\n",
       "      <td>0.999906</td>\n",
       "      <td>0.032252</td>\n",
       "      <td>0.006802</td>\n",
       "      <td>2.198214e-06</td>\n",
       "      <td>rs11265557</td>\n",
       "      <td>1:161106354:T:G</td>\n",
       "      <td>1:161106354:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104385</th>\n",
       "      <td>8</td>\n",
       "      <td>27481984</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.689115</td>\n",
       "      <td>0.987385</td>\n",
       "      <td>0.034706</td>\n",
       "      <td>0.006813</td>\n",
       "      <td>3.361940e-07</td>\n",
       "      <td>rs507341</td>\n",
       "      <td>8:27481984:C:T</td>\n",
       "      <td>8:27481984:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104388</th>\n",
       "      <td>8</td>\n",
       "      <td>27482354</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.310179</td>\n",
       "      <td>0.987466</td>\n",
       "      <td>-0.034708</td>\n",
       "      <td>0.006816</td>\n",
       "      <td>3.401496e-07</td>\n",
       "      <td>rs495150</td>\n",
       "      <td>8:27482354:T:G</td>\n",
       "      <td>8:27482354:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104425</th>\n",
       "      <td>8</td>\n",
       "      <td>27486916</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.310241</td>\n",
       "      <td>0.987670</td>\n",
       "      <td>-0.034207</td>\n",
       "      <td>0.006814</td>\n",
       "      <td>4.962837e-07</td>\n",
       "      <td>rs576748</td>\n",
       "      <td>8:27486916:G:A</td>\n",
       "      <td>8:27486916:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21104471</th>\n",
       "      <td>8</td>\n",
       "      <td>27494366</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.041300</td>\n",
       "      <td>0.983765</td>\n",
       "      <td>-0.077456</td>\n",
       "      <td>0.016219</td>\n",
       "      <td>1.470415e-06</td>\n",
       "      <td>rs138529507</td>\n",
       "      <td>8:27494366:T:C</td>\n",
       "      <td>8:27494366:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21105174</th>\n",
       "      <td>8</td>\n",
       "      <td>27574166</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.039412</td>\n",
       "      <td>0.978535</td>\n",
       "      <td>-0.080823</td>\n",
       "      <td>0.016641</td>\n",
       "      <td>9.640732e-07</td>\n",
       "      <td>rs117606656</td>\n",
       "      <td>8:27574166:T:G</td>\n",
       "      <td>8:27574166:T:G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2097 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          CHR        POS REF ALT       MAF      INFO      BETA        SE  \\\n",
       "1065054     1  161097241   T   A  0.275104  0.999587  0.032051  0.006947   \n",
       "1065073     1  161100336   T   A  0.275114  0.999638  0.032019  0.006946   \n",
       "1065090     1  161103445   T   G  0.295974  0.999809  0.032169  0.006802   \n",
       "1065094     1  161104124   G   A  0.275063  0.999811  0.031940  0.006946   \n",
       "1065110     1  161106354   T   G  0.295966  0.999906  0.032252  0.006802   \n",
       "...       ...        ...  ..  ..       ...       ...       ...       ...   \n",
       "21104385    8   27481984   C   T  0.689115  0.987385  0.034706  0.006813   \n",
       "21104388    8   27482354   T   G  0.310179  0.987466 -0.034708  0.006816   \n",
       "21104425    8   27486916   G   A  0.310241  0.987670 -0.034207  0.006814   \n",
       "21104471    8   27494366   T   C  0.041300  0.983765 -0.077456  0.016219   \n",
       "21105174    8   27574166   T   G  0.039412  0.978535 -0.080823  0.016641   \n",
       "\n",
       "                     P        varid     SNP_original              SNP  \n",
       "1065054   4.102135e-06   rs12727614  1:161097241:T:A  1:161097241:T:A  \n",
       "1065073   4.190058e-06   rs11583045  1:161100336:T:A  1:161100336:T:A  \n",
       "1065090   2.336307e-06   rs10797093  1:161103445:T:G  1:161103445:T:G  \n",
       "1065094   4.421911e-06    rs4540676  1:161104124:G:A  1:161104124:G:A  \n",
       "1065110   2.198214e-06   rs11265557  1:161106354:T:G  1:161106354:T:G  \n",
       "...                ...          ...              ...              ...  \n",
       "21104385  3.361940e-07     rs507341   8:27481984:C:T   8:27481984:T:C  \n",
       "21104388  3.401496e-07     rs495150   8:27482354:T:G   8:27482354:T:G  \n",
       "21104425  4.962837e-07     rs576748   8:27486916:G:A   8:27486916:G:A  \n",
       "21104471  1.470415e-06  rs138529507   8:27494366:T:C   8:27494366:T:C  \n",
       "21105174  9.640732e-07  rs117606656   8:27574166:T:G   8:27574166:T:G  \n",
       "\n",
       "[2097 rows x 12 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_proxy_sig[AD_proxy_sig['SNP'].isin(AD_meta_sig['SNP'])]\n",
    "#f2257_res.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/f2257imp_gwashits_in_AD.tsv' , sep='\\t', index=False, header=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "60caac20-f13a-435a-8b83-476c8af8e131",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Try univariate clumping to see which regions appear and lower p-value to 5e-06"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df161af9-8f6a-499d-8f88-ed1c61848a3d",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD_proxy clump\n",
    "\n",
    "Run first round of univariate clump to identify variants with flipped alleles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0ce4922d-ac02-47d1-8a53-3f6e46d4fb26",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/ADproxy/ADproxy-ldclumping_2022-08-04.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w117c1f46c49790a7) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy\n",
    "clumping_sbatch=$clumping_dir/ADproxy-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.final.snp_stats.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cd4c109-690d-4699-afb9-eed0135199fc",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Create Allele flip file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6b3ed952-3111-4660-be45-6a3c85ade65c",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# To get the variant ID of the alleles that need to be flipped\n",
    "cd ~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy\n",
    "cat AD_step2_BT_all.final.snp_stats.clumped | grep \"not found in dataset\" | awk '{print $1}' > ADproxy_allele_flip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "23b63aba-ba32-4aa6-a58f-846ebab7ab17",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "stat.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','P':'P'}, inplace=True)\n",
    "#stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "#stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy/ADproxy_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "16eed810-a4cf-457b-b69a-eedf870e0eb7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b2a6940d-b156-4a8e-915c-349d16145448",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "173447ae-89f3-4f1e-8ce5-7c286e49f446",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>CHR_POS</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1458289</th>\n",
       "      <td>1</td>\n",
       "      <td>207679307</td>\n",
       "      <td>rs4844600</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.811141</td>\n",
       "      <td>0.993394</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0389473</td>\n",
       "      <td>0.00792551</td>\n",
       "      <td>24.0149</td>\n",
       "      <td>6.01957</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.55938603503361e-07</td>\n",
       "      <td>1:207679307:A:G</td>\n",
       "      <td>1:207679307</td>\n",
       "      <td>1:207679307:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458344</th>\n",
       "      <td>1</td>\n",
       "      <td>207684192</td>\n",
       "      <td>rs12037841</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.82115</td>\n",
       "      <td>0.998536</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0468646</td>\n",
       "      <td>0.00805513</td>\n",
       "      <td>33.6153</td>\n",
       "      <td>8.17287</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.7162986587847305e-09</td>\n",
       "      <td>1:207684192:T:G</td>\n",
       "      <td>1:207684192</td>\n",
       "      <td>1:207684192:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458359</th>\n",
       "      <td>1</td>\n",
       "      <td>207685786</td>\n",
       "      <td>rs4266886</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.811316</td>\n",
       "      <td>0.99526</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0400964</td>\n",
       "      <td>0.007919</td>\n",
       "      <td>25.4903</td>\n",
       "      <td>6.35201</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.44621029573694e-07</td>\n",
       "      <td>1:207685786:T:C</td>\n",
       "      <td>1:207685786</td>\n",
       "      <td>1:207685786:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458360</th>\n",
       "      <td>1</td>\n",
       "      <td>207685965</td>\n",
       "      <td>rs4562624</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.821226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0473144</td>\n",
       "      <td>0.00804713</td>\n",
       "      <td>34.329</td>\n",
       "      <td>8.33218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.6539316421925305e-09</td>\n",
       "      <td>1:207685965:A:C</td>\n",
       "      <td>1:207685965</td>\n",
       "      <td>1:207685965:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1458405</th>\n",
       "      <td>1</td>\n",
       "      <td>207692049</td>\n",
       "      <td>rs6656401</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.82207</td>\n",
       "      <td>0.999179</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0482389</td>\n",
       "      <td>0.00806466</td>\n",
       "      <td>35.5235</td>\n",
       "      <td>8.5986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.5199968621092403e-09</td>\n",
       "      <td>1:207692049:A:G</td>\n",
       "      <td>1:207692049</td>\n",
       "      <td>1:207692049:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032102</th>\n",
       "      <td>8</td>\n",
       "      <td>139366534</td>\n",
       "      <td>rs11166817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.644352</td>\n",
       "      <td>0.998075</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0305679</td>\n",
       "      <td>0.00650746</td>\n",
       "      <td>22.0191</td>\n",
       "      <td>5.56872</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.69947929013705e-06</td>\n",
       "      <td>8:139366534:C:T</td>\n",
       "      <td>8:139366534</td>\n",
       "      <td>8:139366534:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032107</th>\n",
       "      <td>8</td>\n",
       "      <td>139367027</td>\n",
       "      <td>rs17722613</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.644394</td>\n",
       "      <td>0.998056</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0305161</td>\n",
       "      <td>0.00650787</td>\n",
       "      <td>21.9418</td>\n",
       "      <td>5.55123</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.81041205724176e-06</td>\n",
       "      <td>8:139367027:G:A</td>\n",
       "      <td>8:139367027</td>\n",
       "      <td>8:139367027:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032108</th>\n",
       "      <td>8</td>\n",
       "      <td>139367262</td>\n",
       "      <td>rs11782012</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.643547</td>\n",
       "      <td>0.998113</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0301768</td>\n",
       "      <td>0.00650446</td>\n",
       "      <td>21.4799</td>\n",
       "      <td>5.44666</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.57552649044493e-06</td>\n",
       "      <td>8:139367262:C:T</td>\n",
       "      <td>8:139367262</td>\n",
       "      <td>8:139367262:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032207</th>\n",
       "      <td>8</td>\n",
       "      <td>139381546</td>\n",
       "      <td>rs340708</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.644328</td>\n",
       "      <td>0.999248</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0303928</td>\n",
       "      <td>0.00650423</td>\n",
       "      <td>21.7895</td>\n",
       "      <td>5.51675</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.04263600373233e-06</td>\n",
       "      <td>8:139381546:G:T</td>\n",
       "      <td>8:139381546</td>\n",
       "      <td>8:139381546:T:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22032212</th>\n",
       "      <td>8</td>\n",
       "      <td>139382205</td>\n",
       "      <td>rs340710</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.64429</td>\n",
       "      <td>1.0</td>\n",
       "      <td>460649</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0310882</td>\n",
       "      <td>0.00650286</td>\n",
       "      <td>22.8065</td>\n",
       "      <td>5.74677</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.7915544003254e-06</td>\n",
       "      <td>8:139382205:C:A</td>\n",
       "      <td>8:139382205</td>\n",
       "      <td>8:139382205:A:C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>740 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR        POS       varid REF ALT       MAF      INFO       N TEST  \\\n",
       "1458289    1  207679307   rs4844600   A   G  0.811141  0.993394  460649  ADD   \n",
       "1458344    1  207684192  rs12037841   T   G   0.82115  0.998536  460649  ADD   \n",
       "1458359    1  207685786   rs4266886   T   C  0.811316   0.99526  460649  ADD   \n",
       "1458360    1  207685965   rs4562624   A   C  0.821226       1.0  460649  ADD   \n",
       "1458405    1  207692049   rs6656401   A   G   0.82207  0.999179  460649  ADD   \n",
       "...       ..        ...         ...  ..  ..       ...       ...     ...  ...   \n",
       "22032102   8  139366534  rs11166817   C   T  0.644352  0.998075  460649  ADD   \n",
       "22032107   8  139367027  rs17722613   G   A  0.644394  0.998056  460649  ADD   \n",
       "22032108   8  139367262  rs11782012   C   T  0.643547  0.998113  460649  ADD   \n",
       "22032207   8  139381546    rs340708   G   T  0.644328  0.999248  460649  ADD   \n",
       "22032212   8  139382205    rs340710   C   A   0.64429       1.0  460649  ADD   \n",
       "\n",
       "                BETA          SE    CHISQ   LOG10P EXTRA  \\\n",
       "1458289   -0.0389473  0.00792551  24.0149  6.01957   NaN   \n",
       "1458344   -0.0468646  0.00805513  33.6153  8.17287   NaN   \n",
       "1458359   -0.0400964    0.007919  25.4903  6.35201   NaN   \n",
       "1458360   -0.0473144  0.00804713   34.329  8.33218   NaN   \n",
       "1458405   -0.0482389  0.00806466  35.5235   8.5986   NaN   \n",
       "...              ...         ...      ...      ...   ...   \n",
       "22032102  -0.0305679  0.00650746  22.0191  5.56872   NaN   \n",
       "22032107  -0.0305161  0.00650787  21.9418  5.55123   NaN   \n",
       "22032108  -0.0301768  0.00650446  21.4799  5.44666   NaN   \n",
       "22032207  -0.0303928  0.00650423  21.7895  5.51675   NaN   \n",
       "22032212  -0.0310882  0.00650286  22.8065  5.74677   NaN   \n",
       "\n",
       "                               P     SNP_original      CHR_POS  \\\n",
       "1458289     9.55938603503361e-07  1:207679307:A:G  1:207679307   \n",
       "1458344   6.7162986587847305e-09  1:207684192:T:G  1:207684192   \n",
       "1458359     4.44621029573694e-07  1:207685786:T:C  1:207685786   \n",
       "1458360   4.6539316421925305e-09  1:207685965:A:C  1:207685965   \n",
       "1458405   2.5199968621092403e-09  1:207692049:A:G  1:207692049   \n",
       "...                          ...              ...          ...   \n",
       "22032102    2.69947929013705e-06  8:139366534:C:T  8:139366534   \n",
       "22032107    2.81041205724176e-06  8:139367027:G:A  8:139367027   \n",
       "22032108    3.57552649044493e-06  8:139367262:C:T  8:139367262   \n",
       "22032207    3.04263600373233e-06  8:139381546:G:T  8:139381546   \n",
       "22032212     1.7915544003254e-06  8:139382205:C:A  8:139382205   \n",
       "\n",
       "                      SNP  \n",
       "1458289   1:207679307:G:A  \n",
       "1458344   1:207684192:G:T  \n",
       "1458359   1:207685786:C:T  \n",
       "1458360   1:207685965:C:A  \n",
       "1458405   1:207692049:G:A  \n",
       "...                   ...  \n",
       "22032102  8:139366534:T:C  \n",
       "22032107  8:139367027:A:G  \n",
       "22032108  8:139367262:T:C  \n",
       "22032207  8:139381546:T:G  \n",
       "22032212  8:139382205:A:C  \n",
       "\n",
       "[740 rows x 18 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat[stat['SNP_original'].isin(changelist)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "8a861af8-58f7-49c4-ab98-e5eacb7c0ff0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"MAF\", \"INFO\", \"BETA\", \"SE\", \"P\", \"varid\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "83b8e5a3-08a1-418e-b566-0763f60870e7",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy/ADproxy-ldclumping_flip_2022-08-04.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wdd06a4f2948a0bc9) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy\n",
    "clumping_sbatch=$clumping_dir/ADproxy-ldclumping_flip_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/AD_step2_BT_all.txt.snp_stats.ldclump.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee4b3c30-5989-492a-9461-3f80e905dd26",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### After applying the allele_flip code there are still 15 top variants not present in the reference_file\n",
    "\n",
    "These variants can be found at the tail of this file \n",
    "`~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_ADproxy/*.regenie.snp_stats.ldclump.clumped`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83638447-cfae-40fb-9856-717a6262fae2",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## f3393 clump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "96e8e6fd-ad53-487b-b23e-c1e2b86ad7b2",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393/f3393-ldclumping2022-08-05.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w23848caebd857ce7) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "# memory 80GB\n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393\n",
    "clumping_sbatch=$clumping_dir/f3393-ldclumping$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.gz\n",
    "genoFile=`echo $UKBB_yale/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=1\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2d770863-4b20-47f2-af9c-55e0d6e16cfc",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "After replacing the reference file for the new one where all the variants are names by CHR:POS:REF:ALT and the duplicated ones are removed we get the following results\n",
    "\n",
    "\n",
    "This new problem arises where some of the alleles are still flipped in the summary stats/reference sequence\n",
    "\n",
    "```\n",
    "941 top variant IDs missing; see the end of the .clumped file.\n",
    "--clump: 135 clumps formed from 6544 top variants\n",
    "```\n",
    "\n",
    "In the reference.bim file `ukb39554_c1_22_v3.imputed.renamedvar.nondup.2000.ref_geno.bim`: \n",
    "\n",
    "Example:\n",
    "```\n",
    "6:32585755:A(major):T(minor)\n",
    "6\t6:32585755:A:T\t0\t32585755\tT\tA\n",
    "```\n",
    "\n",
    "In the summary stats file\n",
    "```\n",
    "6:32585755:T(ref):A(alt)\n",
    "6\t32585755\tT\tA\trs3129762\t0.0818472\t0.0180549\t4.992637104333426e-06\t6:32585755:T:A\n",
    "```\n",
    "\n",
    "In dbsnp\n",
    "\n",
    "```\n",
    "Europeans T=0.14 (reference) and A=0.86 (alternative)\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9ec540f-f7ce-48ab-862a-49db715c7f8a",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "I can think of one solution to solve this temporarly \n",
    "\n",
    "1. Create an extra column in the sumstats of the f3393 and flip all the variants having this issue to the format found in the bim file (major:minor) keeping the original SNP column renamed to SPN_ref_alt and the other one would be SNP (major/minor)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e3d6a1b6-3860-45d6-b578-ee16e12f2d55",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "f3393 = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f3393/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats_original_columns.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "76eb043f-d408-41a8-a200-acd15fb81817",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000240613</td>\n",
       "      <td>0.844357</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>1.20192</td>\n",
       "      <td>0.563934</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.34679e-05</td>\n",
       "      <td>0.894182</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.408719</td>\n",
       "      <td>0.281813</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000794684</td>\n",
       "      <td>0.852345</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>3.4963</td>\n",
       "      <td>1.21108</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.55789e-05</td>\n",
       "      <td>0.910708</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>1.0573</td>\n",
       "      <td>0.517366</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000485679</td>\n",
       "      <td>0.802923</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.122117</td>\n",
       "      <td>0.138615</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM GENPOS           ID ALLELE0 ALLELE1       A1FREQ      INFO       N  \\\n",
       "0     1  13259  rs562993331       G       A  0.000240613  0.844357  251433   \n",
       "1     1  17569  rs535086049       C       A  1.34679e-05  0.894182  251433   \n",
       "2     1  17641  rs578081284       G       A  0.000794684  0.852345  251433   \n",
       "3     1  30741  rs558169846       C       A  2.55789e-05  0.910708  251433   \n",
       "4     1  52144  rs190291950       T       A  0.000485679  0.802923  251433   \n",
       "\n",
       "  TEST       BETA        SE     CHISQ    LOG10P EXTRA  \n",
       "0  ADD   0.444965  0.405872   1.20192  0.563934   NaN  \n",
       "1  ADD   -1.19541   1.86985  0.408719  0.281813   NaN  \n",
       "2  ADD  -0.435933  0.233139    3.4963   1.21108   NaN  \n",
       "3  ADD   -1.28043   1.24525    1.0573  0.517366   NaN  \n",
       "4  ADD   0.117852  0.337249  0.122117  0.138615   NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d74c5bdf-e9fa-4906-9a63-8314b05cd896",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f3393.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','LOG10P':'P'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "e1651e33-fd34-4261-aac0-17ad85a5a710",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000240613</td>\n",
       "      <td>0.844357</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>1.20192</td>\n",
       "      <td>0.272939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.34679e-05</td>\n",
       "      <td>0.894182</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.408719</td>\n",
       "      <td>0.522621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000794684</td>\n",
       "      <td>0.852345</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>3.4963</td>\n",
       "      <td>0.061506</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.55789e-05</td>\n",
       "      <td>0.910708</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>1.0573</td>\n",
       "      <td>0.303832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000485679</td>\n",
       "      <td>0.802923</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.122117</td>\n",
       "      <td>0.726750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS        varid REF ALT          MAF      INFO       N TEST  \\\n",
       "0   1  13259  rs562993331   G   A  0.000240613  0.844357  251433  ADD   \n",
       "1   1  17569  rs535086049   C   A  1.34679e-05  0.894182  251433  ADD   \n",
       "2   1  17641  rs578081284   G   A  0.000794684  0.852345  251433  ADD   \n",
       "3   1  30741  rs558169846   C   A  2.55789e-05  0.910708  251433  ADD   \n",
       "4   1  52144  rs190291950   T   A  0.000485679  0.802923  251433  ADD   \n",
       "\n",
       "        BETA        SE     CHISQ         P EXTRA SNP_original  \n",
       "0   0.444965  0.405872   1.20192  0.272939   NaN  1:13259:G:A  \n",
       "1   -1.19541   1.86985  0.408719  0.522621   NaN  1:17569:C:A  \n",
       "2  -0.435933  0.233139    3.4963  0.061506   NaN  1:17641:G:A  \n",
       "3   -1.28043   1.24525    1.0573  0.303832   NaN  1:30741:C:A  \n",
       "4   0.117852  0.337249  0.122117  0.726750   NaN  1:52144:T:A  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b208bb6c-a485-4894-af76-3f9e1897daca",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f3393[['P']] = f3393[['P']].apply(pd.to_numeric)\n",
    "f3393['P'] = f3393['P'].apply(lambda row: 10**-row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9f0d738f-417f-439d-b679-7497d8b9187d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "f3393['SNP_original'] = f3393['CHR'] + ':' + f3393['POS'] + ':' + f3393['REF'] + ':' + f3393['ALT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "feb9cf69-0e2f-45e8-84e9-35ec99b11b9d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000240613</td>\n",
       "      <td>0.844357</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>1.20192</td>\n",
       "      <td>0.272939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.34679e-05</td>\n",
       "      <td>0.894182</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.408719</td>\n",
       "      <td>0.522621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000794684</td>\n",
       "      <td>0.852345</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>3.4963</td>\n",
       "      <td>0.061506</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.55789e-05</td>\n",
       "      <td>0.910708</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>1.0573</td>\n",
       "      <td>0.303832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000485679</td>\n",
       "      <td>0.802923</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.122117</td>\n",
       "      <td>0.726750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS        varid REF ALT       A1FREQ      INFO       N TEST  \\\n",
       "0   1  13259  rs562993331   G   A  0.000240613  0.844357  251433  ADD   \n",
       "1   1  17569  rs535086049   C   A  1.34679e-05  0.894182  251433  ADD   \n",
       "2   1  17641  rs578081284   G   A  0.000794684  0.852345  251433  ADD   \n",
       "3   1  30741  rs558169846   C   A  2.55789e-05  0.910708  251433  ADD   \n",
       "4   1  52144  rs190291950   T   A  0.000485679  0.802923  251433  ADD   \n",
       "\n",
       "        BETA        SE     CHISQ         P EXTRA SNP_original  \n",
       "0   0.444965  0.405872   1.20192  0.272939   NaN  1:13259:G:A  \n",
       "1   -1.19541   1.86985  0.408719  0.522621   NaN  1:17569:C:A  \n",
       "2  -0.435933  0.233139    3.4963  0.061506   NaN  1:17641:G:A  \n",
       "3   -1.28043   1.24525    1.0573  0.303832   NaN  1:30741:C:A  \n",
       "4   0.117852  0.337249  0.122117  0.726750   NaN  1:52144:T:A  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "cd8bc5f0-ac3d-48c8-aedc-44c32f756379",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/f3393/f3393_allele_flip_list', dtype=str , header=0 , names=['SNP'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4e33a81c-63e2-4fcb-8f41-0f213e178671",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5:73090165:A:AGTT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5:73090533:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5:73090261:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5:73073615:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5:73090589:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 SNP\n",
       "0  5:73090165:A:AGTT\n",
       "1     5:73090533:A:G\n",
       "2     5:73090261:T:C\n",
       "3     5:73073615:T:C\n",
       "4     5:73090589:G:A"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "flip_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "695a380f-0ffe-47cf-b39c-d5b880ae4939",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "flip_var = flip_list['SNP'].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "71117fa2-9d8f-4480-8ff9-07f213ea8fb8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000240613</td>\n",
       "      <td>0.844357</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>1.20192</td>\n",
       "      <td>0.272939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.34679e-05</td>\n",
       "      <td>0.894182</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.408719</td>\n",
       "      <td>0.522621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000794684</td>\n",
       "      <td>0.852345</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>3.4963</td>\n",
       "      <td>0.061506</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.55789e-05</td>\n",
       "      <td>0.910708</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>1.0573</td>\n",
       "      <td>0.303832</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000485679</td>\n",
       "      <td>0.802923</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.122117</td>\n",
       "      <td>0.726750</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHR    POS        varid REF ALT          MAF      INFO       N TEST  \\\n",
       "0   1  13259  rs562993331   G   A  0.000240613  0.844357  251433  ADD   \n",
       "1   1  17569  rs535086049   C   A  1.34679e-05  0.894182  251433  ADD   \n",
       "2   1  17641  rs578081284   G   A  0.000794684  0.852345  251433  ADD   \n",
       "3   1  30741  rs558169846   C   A  2.55789e-05  0.910708  251433  ADD   \n",
       "4   1  52144  rs190291950   T   A  0.000485679  0.802923  251433  ADD   \n",
       "\n",
       "        BETA        SE     CHISQ         P EXTRA SNP_original          SNP  \n",
       "0   0.444965  0.405872   1.20192  0.272939   NaN  1:13259:G:A  1:13259:G:A  \n",
       "1   -1.19541   1.86985  0.408719  0.522621   NaN  1:17569:C:A  1:17569:C:A  \n",
       "2  -0.435933  0.233139    3.4963  0.061506   NaN  1:17641:G:A  1:17641:G:A  \n",
       "3   -1.28043   1.24525    1.0573  0.303832   NaN  1:30741:C:A  1:30741:C:A  \n",
       "4   0.117852  0.337249  0.122117  0.726750   NaN  1:52144:T:A  1:52144:T:A  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393['SNP']=f3393.loc[:, 'SNP_original']\n",
    "f3393.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "95151cd9-f216-44a2-9361-cea1e7ba6c2a",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# To get the variant ID of the alleles that need to be flipped\n",
    "cd ~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393\n",
    "cat 071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.clumped | grep \"not found in dataset\" | awk '{print $1}' > f3393_allele_flip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "5dfc553d-f602-4c11-bf40-5a078ff563a7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f3393/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats_original_columns.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "stat.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','LOG10P':'P'}, inplace=True)\n",
    "stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393/f3393_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "f250d4ed-969e-4968-ba84-5c57c140a5a9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "eef44f68-51bb-4339-94fc-0eaadde38ab5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "033b48e6-c7e2-4e1e-a6cb-2374d9c6ed3b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>CHR_POS</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13259</td>\n",
       "      <td>rs562993331</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000240613</td>\n",
       "      <td>0.844357</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.444965</td>\n",
       "      <td>0.405872</td>\n",
       "      <td>1.20192</td>\n",
       "      <td>2.729393e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "      <td>1:13259</td>\n",
       "      <td>1:13259:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>17569</td>\n",
       "      <td>rs535086049</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>1.34679e-05</td>\n",
       "      <td>0.894182</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.19541</td>\n",
       "      <td>1.86985</td>\n",
       "      <td>0.408719</td>\n",
       "      <td>5.226212e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "      <td>1:17569</td>\n",
       "      <td>1:17569:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>17641</td>\n",
       "      <td>rs578081284</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000794684</td>\n",
       "      <td>0.852345</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.435933</td>\n",
       "      <td>0.233139</td>\n",
       "      <td>3.4963</td>\n",
       "      <td>6.150636e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "      <td>1:17641</td>\n",
       "      <td>1:17641:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>30741</td>\n",
       "      <td>rs558169846</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>2.55789e-05</td>\n",
       "      <td>0.910708</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-1.28043</td>\n",
       "      <td>1.24525</td>\n",
       "      <td>1.0573</td>\n",
       "      <td>3.038323e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "      <td>1:30741</td>\n",
       "      <td>1:30741:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>52144</td>\n",
       "      <td>rs190291950</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000485679</td>\n",
       "      <td>0.802923</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.117852</td>\n",
       "      <td>0.337249</td>\n",
       "      <td>0.122117</td>\n",
       "      <td>7.267499e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "      <td>1:52144</td>\n",
       "      <td>1:52144:T:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735888</th>\n",
       "      <td>22</td>\n",
       "      <td>51232581</td>\n",
       "      <td>rs5771020</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.297602</td>\n",
       "      <td>0.801896</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.00162598</td>\n",
       "      <td>0.0153923</td>\n",
       "      <td>0.011159</td>\n",
       "      <td>9.158710e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51232581:T:C</td>\n",
       "      <td>22:51232581</td>\n",
       "      <td>22:51232581:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735889</th>\n",
       "      <td>22</td>\n",
       "      <td>51236013</td>\n",
       "      <td>rs200507571</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>0.252029</td>\n",
       "      <td>0.805492</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.00270919</td>\n",
       "      <td>0.01616</td>\n",
       "      <td>0.0281057</td>\n",
       "      <td>8.668607e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51236013:A:AT</td>\n",
       "      <td>22:51236013</td>\n",
       "      <td>22:51236013:A:AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735890</th>\n",
       "      <td>22</td>\n",
       "      <td>51237063</td>\n",
       "      <td>rs3896457</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.298901</td>\n",
       "      <td>0.857807</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.00859535</td>\n",
       "      <td>0.0148427</td>\n",
       "      <td>0.33535</td>\n",
       "      <td>5.625252e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51237063:T:C</td>\n",
       "      <td>22:51237063</td>\n",
       "      <td>22:51237063:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735891</th>\n",
       "      <td>22</td>\n",
       "      <td>51237215</td>\n",
       "      <td>rs536109858</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000413208</td>\n",
       "      <td>0.914193</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>1.34564</td>\n",
       "      <td>0.209808</td>\n",
       "      <td>33.7213</td>\n",
       "      <td>6.360189e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51237215:C:T</td>\n",
       "      <td>22:51237215</td>\n",
       "      <td>22:51237215:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21735892</th>\n",
       "      <td>22</td>\n",
       "      <td>51237712</td>\n",
       "      <td>rs370652263</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.05513</td>\n",
       "      <td>0.864382</td>\n",
       "      <td>251433</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0351139</td>\n",
       "      <td>0.0298536</td>\n",
       "      <td>1.38346</td>\n",
       "      <td>2.395130e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51237712:G:A</td>\n",
       "      <td>22:51237712</td>\n",
       "      <td>22:51237712:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21735893 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS        varid REF ALT          MAF      INFO       N  \\\n",
       "0          1     13259  rs562993331   G   A  0.000240613  0.844357  251433   \n",
       "1          1     17569  rs535086049   C   A  1.34679e-05  0.894182  251433   \n",
       "2          1     17641  rs578081284   G   A  0.000794684  0.852345  251433   \n",
       "3          1     30741  rs558169846   C   A  2.55789e-05  0.910708  251433   \n",
       "4          1     52144  rs190291950   T   A  0.000485679  0.802923  251433   \n",
       "...       ..       ...          ...  ..  ..          ...       ...     ...   \n",
       "21735888  22  51232581    rs5771020   T   C     0.297602  0.801896  251433   \n",
       "21735889  22  51236013  rs200507571   A  AT     0.252029  0.805492  251433   \n",
       "21735890  22  51237063    rs3896457   T   C     0.298901  0.857807  251433   \n",
       "21735891  22  51237215  rs536109858   C   T  0.000413208  0.914193  251433   \n",
       "21735892  22  51237712  rs370652263   G   A      0.05513  0.864382  251433   \n",
       "\n",
       "         TEST         BETA         SE      CHISQ             P EXTRA  \\\n",
       "0         ADD     0.444965   0.405872    1.20192  2.729393e-01   NaN   \n",
       "1         ADD     -1.19541    1.86985   0.408719  5.226212e-01   NaN   \n",
       "2         ADD    -0.435933   0.233139     3.4963  6.150636e-02   NaN   \n",
       "3         ADD     -1.28043    1.24525     1.0573  3.038323e-01   NaN   \n",
       "4         ADD     0.117852   0.337249   0.122117  7.267499e-01   NaN   \n",
       "...       ...          ...        ...        ...           ...   ...   \n",
       "21735888  ADD   0.00162598  0.0153923   0.011159  9.158710e-01   NaN   \n",
       "21735889  ADD  -0.00270919    0.01616  0.0281057  8.668607e-01   NaN   \n",
       "21735890  ADD  -0.00859535  0.0148427    0.33535  5.625252e-01   NaN   \n",
       "21735891  ADD      1.34564   0.209808    33.7213  6.360189e-09   NaN   \n",
       "21735892  ADD   -0.0351139  0.0298536    1.38346  2.395130e-01   NaN   \n",
       "\n",
       "              SNP_original      CHR_POS               SNP  \n",
       "0              1:13259:G:A      1:13259       1:13259:G:A  \n",
       "1              1:17569:C:A      1:17569       1:17569:C:A  \n",
       "2              1:17641:G:A      1:17641       1:17641:G:A  \n",
       "3              1:30741:C:A      1:30741       1:30741:C:A  \n",
       "4              1:52144:T:A      1:52144       1:52144:T:A  \n",
       "...                    ...          ...               ...  \n",
       "21735888   22:51232581:T:C  22:51232581   22:51232581:T:C  \n",
       "21735889  22:51236013:A:AT  22:51236013  22:51236013:A:AT  \n",
       "21735890   22:51237063:T:C  22:51237063   22:51237063:T:C  \n",
       "21735891   22:51237215:C:T  22:51237215   22:51237215:C:T  \n",
       "21735892   22:51237712:G:A  22:51237712   22:51237712:G:A  \n",
       "\n",
       "[21735893 rows x 17 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "be20cdba-2361-4ade-b69d-9ee5dfd7bd02",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"MAF\", \"INFO\", \"BETA\", \"SE\", \"P\", \"varid\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29cf5b28-2321-4644-9f96-00d1c42b7a37",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### After applying this code there are still 22 top variants not present in the reference file \n",
    "\n",
    "```\n",
    "5:73090534:T:C not found in dataset\n",
    "2:54959092:G:C not found in dataset\n",
    "6:32582603:A:G not found in dataset\n",
    "6:32673475:C:T not found in dataset\n",
    "6:32633668:G:A not found in dataset\n",
    "17:43853922:A:G not found in dataset\n",
    "17:44138377:GA:G not found in dataset\n",
    "17:44138377:GAA:GA not found in dataset\n",
    "17:44129949:G:C not found in dataset\n",
    "17:44063723:C:A not found in dataset\n",
    "17:43756458:A:G not found in dataset\n",
    "17:43950976:T:G not found in dataset\n",
    "17:44293020:A:G not found in dataset\n",
    "14:81846115:C:T not found in dataset\n",
    "3:181975568:CTT:C not found in dataset\n",
    "17:44355634:G:C not found in dataset\n",
    "12:102780879:A:G not found in dataset\n",
    "17:44173502:ATTC:A not found in dataset\n",
    "5:73066154:A:T not found in dataset\n",
    "11:118514022:G:C not found in dataset\n",
    "17:44355634:A:C not found in dataset\n",
    "10:30315633:T:C not found in dataset\n",
    "6:84127089:TTTG:T not found in dataset\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "880dd2a0-a9bc-4484-a717-fb4d12c5e30a",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393/f3393-ldclumping_flip2022-08-05.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wb50975dbec8641f8) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f3393\n",
    "clumping_sbatch=$clumping_dir/f3393-ldclumping_flip$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.gz\n",
    "genoFile=`echo $UKBB_yale/ukb39554_imputeddataset/ukb_imp_chr{1..22}_v3.bgen`\n",
    "sampleFile=$UKBB_yale/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=1\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c6c50a6-7bd5-49a3-9f8f-8709d284da62",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## f2247 clump\n",
    "\n",
    "Asign 80GB memory for the job to run "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1fb412ff-aa72-44e5-ac02-d0d0beaf55f3",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Checking \u001b[32mdefault\u001b[0m: Perform LD-clumping in PLINKv1.9\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/mfs/hgrcgrid/homes/dmc2245/project/UKBB_GWAS_dev/analysis/pleio_AD_ARHI/tmpbxu796k5/singularity_run_18882.sh\n",
      "plink \\\n",
      "--bfile /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.nondup.2000.ref_geno \\\n",
      "--clump /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.gz \\\n",
      "--clump-field P \\\n",
      "--clump-p1 5e-06 \\\n",
      "--clump-p2 1.0 \\\n",
      "--clump-r2 0.04 \\\n",
      "--clump-kb 2000 \\\n",
      "--clump-verbose \\\n",
      "--clump-annotate varid \\\n",
      "--clump-allow-overlap \\\n",
      "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats \\\n",
      "--threads 20 \\\n",
      "&& touch /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.clumped # need to touch and create empty file because some chroms may not have anything significant to clump.\n",
      "grep \"RANGE\" /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.clumped | awk -F \":\" '{print $2, $3}' | sort -V | sed 's/\\../ /g; s/^[[:blank:]]*//g ; s/chr//g' > /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.clumped_region\n",
      "\n",
      "\n",
      "INFO: \u001b[32mdefault\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mdefault\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.clumped /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.clumped_region\u001b[0m\n",
      "INFO: Workflow default (ID=w3dcb6a286447a098) is tested successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Singularity/3.5.3\n",
    "sos dryrun /home/dmc2245/project/bioworkflows/GWAS/LD_Clumping.ipynb \\\n",
    "    default \\\n",
    "    --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247\\\n",
    "    --genoFile /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr1_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr2_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr3_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr4_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr5_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr6_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr7_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr8_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr9_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr10_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr11_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr12_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr13_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr14_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr15_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr16_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr17_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr18_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr19_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr20_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr21_v3.bgen /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_imp_chr22_v3.bgen\\\n",
    "    --sampleFile /home/dmc2245/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\\\n",
    "    --reference_genotype_prefix /home/dmc2245/UKBiobank/results/LD_clumping/ref_files/ukb39554_c1_22_v3.imputed.renamedvar.nondup  \\\n",
    "    --sumstatsFiles /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.gz \\\n",
    "    --ld_sample_size 2000 \\\n",
    "    --clump_field P\\\n",
    "    --clump_p1 5e-06 \\\n",
    "    --clump_p2 1 \\\n",
    "    --clump_r2 0.04 \\\n",
    "    --clump_kb 2000 \\\n",
    "    --clump_annotate varid \\\n",
    "    --numThreads 20 \\\n",
    "    --job_size 1\\\n",
    "    --container /home/dmc2245/containers/lmm.sif"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "14b50820-66b1-4cc9-8e43-162dccf93e81",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats_original_columns.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "stat.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','LOG10P':'P'}, inplace=True)\n",
    "stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/f2247_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "99c728ff-a5e6-4c68-af9d-3c9178141922",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "159d8f43-648a-4340-93a9-759b29251fa2",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "664d2d93-a4a0-44e1-81b6-7a61bce45d22",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>CHR_POS</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>47322</th>\n",
       "      <td>1</td>\n",
       "      <td>6493666</td>\n",
       "      <td>rs569450960</td>\n",
       "      <td>C</td>\n",
       "      <td>CAA</td>\n",
       "      <td>0.0609333</td>\n",
       "      <td>0.96973</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0526317</td>\n",
       "      <td>0.0114712</td>\n",
       "      <td>21.1616</td>\n",
       "      <td>4.221337e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:6493666:C:CAA</td>\n",
       "      <td>1:6493666</td>\n",
       "      <td>1:6493666:CAA:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336176</th>\n",
       "      <td>1</td>\n",
       "      <td>46026397</td>\n",
       "      <td>rs373579463</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.566553</td>\n",
       "      <td>0.969944</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0326015</td>\n",
       "      <td>0.00549209</td>\n",
       "      <td>35.2231</td>\n",
       "      <td>2.940086e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46026397:CT:C</td>\n",
       "      <td>1:46026397</td>\n",
       "      <td>1:46026397:C:CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336267</th>\n",
       "      <td>1</td>\n",
       "      <td>46036960</td>\n",
       "      <td>rs514595</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.840746</td>\n",
       "      <td>0.997817</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0385353</td>\n",
       "      <td>0.00730693</td>\n",
       "      <td>27.7363</td>\n",
       "      <td>1.390273e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46036960:T:C</td>\n",
       "      <td>1:46036960</td>\n",
       "      <td>1:46036960:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336273</th>\n",
       "      <td>1</td>\n",
       "      <td>46037357</td>\n",
       "      <td>rs518216</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.83866</td>\n",
       "      <td>0.997918</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0374882</td>\n",
       "      <td>0.00727</td>\n",
       "      <td>26.5194</td>\n",
       "      <td>2.609036e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46037357:C:T</td>\n",
       "      <td>1:46037357</td>\n",
       "      <td>1:46037357:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336275</th>\n",
       "      <td>1</td>\n",
       "      <td>46037394</td>\n",
       "      <td>rs659437</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.838692</td>\n",
       "      <td>0.997913</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0375429</td>\n",
       "      <td>0.00727049</td>\n",
       "      <td>26.5931</td>\n",
       "      <td>2.511424e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46037394:T:C</td>\n",
       "      <td>1:46037394</td>\n",
       "      <td>1:46037394:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21850668</th>\n",
       "      <td>22</td>\n",
       "      <td>51043278</td>\n",
       "      <td>rs3219201</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.678689</td>\n",
       "      <td>0.991739</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0277026</td>\n",
       "      <td>0.00575655</td>\n",
       "      <td>23.1349</td>\n",
       "      <td>1.510254e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51043278:G:A</td>\n",
       "      <td>22:51043278</td>\n",
       "      <td>22:51043278:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21850695</th>\n",
       "      <td>22</td>\n",
       "      <td>51046096</td>\n",
       "      <td>rs470115</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.819177</td>\n",
       "      <td>0.970074</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0323748</td>\n",
       "      <td>0.00706527</td>\n",
       "      <td>20.9521</td>\n",
       "      <td>4.709231e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51046096:A:G</td>\n",
       "      <td>22:51046096</td>\n",
       "      <td>22:51046096:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21850707</th>\n",
       "      <td>22</td>\n",
       "      <td>51047218</td>\n",
       "      <td>rs5770938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.7617</td>\n",
       "      <td>0.978794</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0353692</td>\n",
       "      <td>0.00634002</td>\n",
       "      <td>31.0625</td>\n",
       "      <td>2.498561e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51047218:G:A</td>\n",
       "      <td>22:51047218</td>\n",
       "      <td>22:51047218:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21850720</th>\n",
       "      <td>22</td>\n",
       "      <td>51048059</td>\n",
       "      <td>rs965618</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.729428</td>\n",
       "      <td>0.995698</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0323653</td>\n",
       "      <td>0.00603233</td>\n",
       "      <td>28.7416</td>\n",
       "      <td>8.270847e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51048059:A:G</td>\n",
       "      <td>22:51048059</td>\n",
       "      <td>22:51048059:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21851000</th>\n",
       "      <td>22</td>\n",
       "      <td>51078547</td>\n",
       "      <td>22:51078547_TG_T</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>0.82304</td>\n",
       "      <td>0.922781</td>\n",
       "      <td>344693</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0377349</td>\n",
       "      <td>0.00728908</td>\n",
       "      <td>26.7332</td>\n",
       "      <td>2.335770e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51078547:TG:T</td>\n",
       "      <td>22:51078547</td>\n",
       "      <td>22:51078547:T:TG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3277 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS             varid REF  ALT        MAF      INFO  \\\n",
       "47322      1   6493666       rs569450960   C  CAA  0.0609333   0.96973   \n",
       "336176     1  46026397       rs373579463  CT    C   0.566553  0.969944   \n",
       "336267     1  46036960          rs514595   T    C   0.840746  0.997817   \n",
       "336273     1  46037357          rs518216   C    T    0.83866  0.997918   \n",
       "336275     1  46037394          rs659437   T    C   0.838692  0.997913   \n",
       "...       ..       ...               ...  ..  ...        ...       ...   \n",
       "21850668  22  51043278         rs3219201   G    A   0.678689  0.991739   \n",
       "21850695  22  51046096          rs470115   A    G   0.819177  0.970074   \n",
       "21850707  22  51047218         rs5770938   G    A     0.7617  0.978794   \n",
       "21850720  22  51048059          rs965618   A    G   0.729428  0.995698   \n",
       "21851000  22  51078547  22:51078547_TG_T  TG    T    0.82304  0.922781   \n",
       "\n",
       "               N TEST        BETA          SE    CHISQ             P EXTRA  \\\n",
       "47322     344693  ADD  -0.0526317   0.0114712  21.1616  4.221337e-06   NaN   \n",
       "336176    344693  ADD  -0.0326015  0.00549209  35.2231  2.940086e-09   NaN   \n",
       "336267    344693  ADD  -0.0385353  0.00730693  27.7363  1.390273e-07   NaN   \n",
       "336273    344693  ADD  -0.0374882     0.00727  26.5194  2.609036e-07   NaN   \n",
       "336275    344693  ADD  -0.0375429  0.00727049  26.5931  2.511424e-07   NaN   \n",
       "...          ...  ...         ...         ...      ...           ...   ...   \n",
       "21850668  344693  ADD  -0.0277026  0.00575655  23.1349  1.510254e-06   NaN   \n",
       "21850695  344693  ADD  -0.0323748  0.00706527  20.9521  4.709231e-06   NaN   \n",
       "21850707  344693  ADD  -0.0353692  0.00634002  31.0625  2.498561e-08   NaN   \n",
       "21850720  344693  ADD  -0.0323653  0.00603233  28.7416  8.270847e-08   NaN   \n",
       "21851000  344693  ADD  -0.0377349  0.00728908  26.7332  2.335770e-07   NaN   \n",
       "\n",
       "              SNP_original      CHR_POS               SNP  \n",
       "47322      1:6493666:C:CAA    1:6493666   1:6493666:CAA:C  \n",
       "336176     1:46026397:CT:C   1:46026397   1:46026397:C:CT  \n",
       "336267      1:46036960:T:C   1:46036960    1:46036960:C:T  \n",
       "336273      1:46037357:C:T   1:46037357    1:46037357:T:C  \n",
       "336275      1:46037394:T:C   1:46037394    1:46037394:C:T  \n",
       "...                    ...          ...               ...  \n",
       "21850668   22:51043278:G:A  22:51043278   22:51043278:A:G  \n",
       "21850695   22:51046096:A:G  22:51046096   22:51046096:G:A  \n",
       "21850707   22:51047218:G:A  22:51047218   22:51047218:A:G  \n",
       "21850720   22:51048059:A:G  22:51048059   22:51048059:G:A  \n",
       "21851000  22:51078547:TG:T  22:51078547  22:51078547:T:TG  \n",
       "\n",
       "[3277 rows x 17 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat[stat['SNP_original'].isin(changelist)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "a8772ec2-359f-4ef7-87c3-5647e7f295d1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"MAF\", \"INFO\", \"BETA\", \"SE\", \"P\", \"varid\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "4a1790d1-08fe-4e6d-a333-fa9267b68fd1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/f2247-ldclumping_flip_2022-08-05.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w8d2ba18b02a6165c) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247\n",
    "clumping_sbatch=$clumping_dir/f2247-ldclumping_flip_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "717badc4-c683-4d0a-ac4e-57de49e9943b",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### After applying the allele_flip code there are still 91 top variants not present in the reference_file\n",
    "\n",
    "These variants can be found at the tail of this file `/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247/071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.clumped`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b05c8dda-f1b7-4b54-a59a-9f4872c94ace",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## f2257 clump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "162a3e5a-0744-4e2f-b3a6-de8f12afa03b",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257/f2257-ldclumping_2022-08-03.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w27f2a3ef13e0301f) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257\n",
    "clumping_sbatch=$clumping_dir/f2257-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=1\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "609db56e-42ca-4165-9b79-0df1852e9ef9",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2257/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats_original_columns.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "stat.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','LOG10P':'P'}, inplace=True)\n",
    "stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257/f2257_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "f078e30a-78d2-4295-8738-bdf66b3f1f0d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "dfbf4006-4115-4f2e-b716-878b24134e83",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "56e1aaf3-a103-4eb9-9a4d-0e5a41005c6d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>varid</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>MAF</th>\n",
       "      <th>INFO</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>P</th>\n",
       "      <th>EXTRA</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>CHR_POS</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>58269</th>\n",
       "      <td>1</td>\n",
       "      <td>7965280</td>\n",
       "      <td>rs6694557</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.864359</td>\n",
       "      <td>0.964317</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0343144</td>\n",
       "      <td>0.00696062</td>\n",
       "      <td>24.2706</td>\n",
       "      <td>8.370858e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:7965280:T:C</td>\n",
       "      <td>1:7965280</td>\n",
       "      <td>1:7965280:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209878</th>\n",
       "      <td>1</td>\n",
       "      <td>28975150</td>\n",
       "      <td>rs35910435</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.0518141</td>\n",
       "      <td>1.0</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>0.0507732</td>\n",
       "      <td>0.0105006</td>\n",
       "      <td>23.3218</td>\n",
       "      <td>1.370377e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:28975150:C:T</td>\n",
       "      <td>1:28975150</td>\n",
       "      <td>1:28975150:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226419</th>\n",
       "      <td>1</td>\n",
       "      <td>31259857</td>\n",
       "      <td>1:31259857_CT_C</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.612566</td>\n",
       "      <td>0.811564</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0249873</td>\n",
       "      <td>0.00534486</td>\n",
       "      <td>21.8505</td>\n",
       "      <td>2.947406e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:31259857:CT:C</td>\n",
       "      <td>1:31259857</td>\n",
       "      <td>1:31259857:C:CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336745</th>\n",
       "      <td>1</td>\n",
       "      <td>46026397</td>\n",
       "      <td>rs373579463</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.566571</td>\n",
       "      <td>0.970091</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0257179</td>\n",
       "      <td>0.00480954</td>\n",
       "      <td>28.5894</td>\n",
       "      <td>8.947259e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46026397:CT:C</td>\n",
       "      <td>1:46026397</td>\n",
       "      <td>1:46026397:C:CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336872</th>\n",
       "      <td>1</td>\n",
       "      <td>46041100</td>\n",
       "      <td>rs1084086</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.703508</td>\n",
       "      <td>1.0</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0280157</td>\n",
       "      <td>0.00513516</td>\n",
       "      <td>29.7464</td>\n",
       "      <td>4.924136e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1:46041100:T:G</td>\n",
       "      <td>1:46041100</td>\n",
       "      <td>1:46041100:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21795261</th>\n",
       "      <td>22</td>\n",
       "      <td>38474852</td>\n",
       "      <td>rs1004764</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.620458</td>\n",
       "      <td>0.997395</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0234469</td>\n",
       "      <td>0.00483889</td>\n",
       "      <td>23.4727</td>\n",
       "      <td>1.267010e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:38474852:G:A</td>\n",
       "      <td>22:38474852</td>\n",
       "      <td>22:38474852:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21801263</th>\n",
       "      <td>22</td>\n",
       "      <td>39286208</td>\n",
       "      <td>rs132502</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.233611</td>\n",
       "      <td>1.0</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.026474</td>\n",
       "      <td>0.00556285</td>\n",
       "      <td>22.6694</td>\n",
       "      <td>1.924110e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:39286208:A:C</td>\n",
       "      <td>22:39286208</td>\n",
       "      <td>22:39286208:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21906441</th>\n",
       "      <td>22</td>\n",
       "      <td>51043278</td>\n",
       "      <td>rs3219201</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.678556</td>\n",
       "      <td>0.991675</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0236357</td>\n",
       "      <td>0.00503915</td>\n",
       "      <td>21.9901</td>\n",
       "      <td>2.740627e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51043278:G:A</td>\n",
       "      <td>22:51043278</td>\n",
       "      <td>22:51043278:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21906480</th>\n",
       "      <td>22</td>\n",
       "      <td>51047218</td>\n",
       "      <td>rs5770938</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.76171</td>\n",
       "      <td>0.978624</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0274931</td>\n",
       "      <td>0.00555585</td>\n",
       "      <td>24.4688</td>\n",
       "      <td>7.552313e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51047218:G:A</td>\n",
       "      <td>22:51047218</td>\n",
       "      <td>22:51047218:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21906493</th>\n",
       "      <td>22</td>\n",
       "      <td>51048059</td>\n",
       "      <td>rs965618</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.729347</td>\n",
       "      <td>0.995682</td>\n",
       "      <td>395732</td>\n",
       "      <td>ADD</td>\n",
       "      <td>-0.0261407</td>\n",
       "      <td>0.00528435</td>\n",
       "      <td>24.4552</td>\n",
       "      <td>7.605714e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22:51048059:A:G</td>\n",
       "      <td>22:51048059</td>\n",
       "      <td>22:51048059:G:A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3480 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR       POS            varid REF ALT        MAF      INFO       N  \\\n",
       "58269      1   7965280        rs6694557   T   C   0.864359  0.964317  395732   \n",
       "209878     1  28975150       rs35910435   C   T  0.0518141       1.0  395732   \n",
       "226419     1  31259857  1:31259857_CT_C  CT   C   0.612566  0.811564  395732   \n",
       "336745     1  46026397      rs373579463  CT   C   0.566571  0.970091  395732   \n",
       "336872     1  46041100        rs1084086   T   G   0.703508       1.0  395732   \n",
       "...       ..       ...              ...  ..  ..        ...       ...     ...   \n",
       "21795261  22  38474852        rs1004764   G   A   0.620458  0.997395  395732   \n",
       "21801263  22  39286208         rs132502   A   C   0.233611       1.0  395732   \n",
       "21906441  22  51043278        rs3219201   G   A   0.678556  0.991675  395732   \n",
       "21906480  22  51047218        rs5770938   G   A    0.76171  0.978624  395732   \n",
       "21906493  22  51048059         rs965618   A   G   0.729347  0.995682  395732   \n",
       "\n",
       "         TEST        BETA          SE    CHISQ             P EXTRA  \\\n",
       "58269     ADD  -0.0343144  0.00696062  24.2706  8.370858e-07   NaN   \n",
       "209878    ADD   0.0507732   0.0105006  23.3218  1.370377e-06   NaN   \n",
       "226419    ADD  -0.0249873  0.00534486  21.8505  2.947406e-06   NaN   \n",
       "336745    ADD  -0.0257179  0.00480954  28.5894  8.947259e-08   NaN   \n",
       "336872    ADD  -0.0280157  0.00513516  29.7464  4.924136e-08   NaN   \n",
       "...       ...         ...         ...      ...           ...   ...   \n",
       "21795261  ADD  -0.0234469  0.00483889  23.4727  1.267010e-06   NaN   \n",
       "21801263  ADD   -0.026474  0.00556285  22.6694  1.924110e-06   NaN   \n",
       "21906441  ADD  -0.0236357  0.00503915  21.9901  2.740627e-06   NaN   \n",
       "21906480  ADD  -0.0274931  0.00555585  24.4688  7.552313e-07   NaN   \n",
       "21906493  ADD  -0.0261407  0.00528435  24.4552  7.605714e-07   NaN   \n",
       "\n",
       "             SNP_original      CHR_POS              SNP  \n",
       "58269       1:7965280:T:C    1:7965280    1:7965280:C:T  \n",
       "209878     1:28975150:C:T   1:28975150   1:28975150:T:C  \n",
       "226419    1:31259857:CT:C   1:31259857  1:31259857:C:CT  \n",
       "336745    1:46026397:CT:C   1:46026397  1:46026397:C:CT  \n",
       "336872     1:46041100:T:G   1:46041100   1:46041100:G:T  \n",
       "...                   ...          ...              ...  \n",
       "21795261  22:38474852:G:A  22:38474852  22:38474852:A:G  \n",
       "21801263  22:39286208:A:C  22:39286208  22:39286208:C:A  \n",
       "21906441  22:51043278:G:A  22:51043278  22:51043278:A:G  \n",
       "21906480  22:51047218:G:A  22:51047218  22:51047218:A:G  \n",
       "21906493  22:51048059:A:G  22:51048059  22:51048059:G:A  \n",
       "\n",
       "[3480 rows x 17 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat[stat['SNP_original'].isin(changelist)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "3ffb5659-cb32-420f-a54f-332d924301d1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"MAF\", \"INFO\", \"BETA\", \"SE\", \"P\", \"varid\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "480a2d65-c67d-4cc8-8ca7-57adbfbcacc9",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### After applying the allele_flip code there are still 76 top variants not present in the reference_file\n",
    "\n",
    "These variants can be found at the tail of this file `~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.clumped`\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "2743dc0a-ed37-4dc3-8001-dc1e9bd346cf",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m (index=0) is \u001b[32mignored\u001b[0m due to saved signature\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257/f2257-ldclumping_flip_2022-08-05.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w16814b7e66737f2c) is ignored with 1 ignored step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2257\n",
    "clumping_sbatch=$clumping_dir/f2257-ldclumping_flip_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=varid\n",
    "numThreads=1\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c88214b-f515-4986-a2d8-b21e39d75162",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Combined clump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3a85c4c8-ac6d-48c1-aa53-b07782d963b9",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257/combined-ldclumping_2022-08-04.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wa5d11405fe8fa890) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# # Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257\n",
    "clumping_sbatch=$clumping_dir/combined-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bbf76dc0-93a5-4f58-89bb-444990df86cf",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "# To get the variant ID of the alleles that need to be flipped\n",
    "cd ~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257\n",
    "cat 071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.clumped | grep \"not found in dataset\" | awk '{print $1}' > f2247_f2257_allele_flip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "e81cedf7-4026-408b-b3c8-9ba94f346154",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/REGENIE_results/results_imputed_data/071222_500k_without_1415/f2247_f2257/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats_original_columns.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "stat.rename(columns={'CHROM':'CHR', 'GENPOS':'POS', 'ID':'varid', 'ALLELE0':'REF', 'ALLELE1':'ALT', 'A1FREQ':'MAF','LOG10P':'P'}, inplace=True)\n",
    "stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257/f2247_f2257_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "adb4b965-5c97-417d-aed0-4ff89e9a284a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "46f81d35-498b-493e-b8d5-3eaa1202288b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "e81e6753-9d15-41c3-8178-1b0db6cbb3de",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"MAF\", \"INFO\", \"BETA\", \"SE\", \"P\", \"varid\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "70d75e91-a13f-40ac-8423-186d7d710163",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257/combined-ldclumping_flip_2022-08-05.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w9838dfc6f95d510f) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_f2247_f2257\n",
    "clumping_sbatch=$clumping_dir/combined-ldclumping_flip_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9dd3963-5301-42b2-8f30-ee335c9fe8b6",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### After applying the allele_flip code there are still 76 top variants not present in the reference_file\n",
    "\n",
    "These variants can be found at the tail of this file `~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3d06f9f-8ee6-4d95-9861-fd3677870836",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## AD_meta clump"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6ea356aa-442d-42a6-9d2f-594a69e98671",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta/AD_meta-ldclumping_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w5c3d9f868a13359b) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta\n",
    "clumping_sbatch=$clumping_dir/AD_meta-ldclumping_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "476bed5b-c7ca-41f5-b847-f3e37c051b42",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# To get the variant ID of the alleles that need to be flipped\n",
    "cd ~/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta\n",
    "cat GCST90027158_buildGRCh37_reheader.tsv.clumped| grep \"not found in dataset\" | awk '{print $1}' > ADmeta_allele_flip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "181f9220-9d0d-41f9-a4fd-6cddfc7915cd",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "stat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.tsv.gz', dtype=str, compression='gzip',sep=\"\\t\", header=0)\n",
    "#stat[['P']] = stat[['P']].apply(pd.to_numeric)\n",
    "#stat['P'] = stat['P'].apply(lambda row: 10**-row)\n",
    "stat['SNP_original'] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"],stat['REF'],stat['ALT']], sep=':')\n",
    "stat[\"CHR_POS\"] = stat[\"CHR\"].str.cat(others=[stat[\"POS\"]], sep=':')\n",
    "flip_list= pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta/ADmeta_allele_flip', dtype=str , header=None , names=['SNP'])\n",
    "changelist = flip_list['SNP'].tolist()\n",
    "stat['SNP']=stat.loc[:, 'SNP_original']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d19cc221-a7bb-4261-882c-e45a45c2ce06",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "def snp_flip(row):\n",
    "    row[\"SNP\"] = str(row[\"CHR\"])+\":\"+str(row[\"POS\"])+\":\"+row[\"ALT\"]+\":\"+row[\"REF\"]\n",
    "    return row"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "72617973-8efc-4599-a828-559515b1b552",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "stat[stat['SNP'].isin(changelist)] = stat[stat['SNP'].isin(changelist)].apply(snp_flip,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "10738f12-6cfe-407f-88eb-59d7f8ace114",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>SNP</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>P</th>\n",
       "      <th>SNP_original</th>\n",
       "      <th>CHR_POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7474</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>0.0052</td>\n",
       "      <td>0.0173</td>\n",
       "      <td>0.7631</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7476</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>-0.0071</td>\n",
       "      <td>0.0331</td>\n",
       "      <td>0.8292</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7477</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>-0.0015</td>\n",
       "      <td>0.0141</td>\n",
       "      <td>0.916</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7478</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>0.0024</td>\n",
       "      <td>0.0151</td>\n",
       "      <td>0.8725</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7479</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0:0:T:C</td>\n",
       "      <td>-0.3168</td>\n",
       "      <td>0.3158</td>\n",
       "      <td>0.3157</td>\n",
       "      <td>0:0:C:T</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21093486</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0:0:T:C</td>\n",
       "      <td>0.2512</td>\n",
       "      <td>0.4173</td>\n",
       "      <td>0.5472</td>\n",
       "      <td>0:0:C:T</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21093487</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>0.0186</td>\n",
       "      <td>0.0108</td>\n",
       "      <td>0.08605</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21093488</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0:0:T:C</td>\n",
       "      <td>-0.0327</td>\n",
       "      <td>0.0142</td>\n",
       "      <td>0.02135</td>\n",
       "      <td>0:0:C:T</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21093489</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:C</td>\n",
       "      <td>-0.1226</td>\n",
       "      <td>0.2286</td>\n",
       "      <td>0.5916</td>\n",
       "      <td>0:0:C:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21093490</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0:0:A:G</td>\n",
       "      <td>0.0011</td>\n",
       "      <td>0.0119</td>\n",
       "      <td>0.9266</td>\n",
       "      <td>0:0:G:A</td>\n",
       "      <td>0:0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34111 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHR POS REF ALT      SNP     BETA      SE        P SNP_original  \\\n",
       "7474       0   0   G   A  0:0:A:G   0.0052  0.0173   0.7631      0:0:G:A   \n",
       "7476       0   0   G   A  0:0:A:G  -0.0071  0.0331   0.8292      0:0:G:A   \n",
       "7477       0   0   G   A  0:0:A:G  -0.0015  0.0141    0.916      0:0:G:A   \n",
       "7478       0   0   G   A  0:0:A:G   0.0024  0.0151   0.8725      0:0:G:A   \n",
       "7479       0   0   C   T  0:0:T:C  -0.3168  0.3158   0.3157      0:0:C:T   \n",
       "...       ..  ..  ..  ..      ...      ...     ...      ...          ...   \n",
       "21093486   0   0   C   T  0:0:T:C   0.2512  0.4173   0.5472      0:0:C:T   \n",
       "21093487   0   0   G   A  0:0:A:G   0.0186  0.0108  0.08605      0:0:G:A   \n",
       "21093488   0   0   C   T  0:0:T:C  -0.0327  0.0142  0.02135      0:0:C:T   \n",
       "21093489   0   0   C   A  0:0:A:C  -0.1226  0.2286   0.5916      0:0:C:A   \n",
       "21093490   0   0   G   A  0:0:A:G   0.0011  0.0119   0.9266      0:0:G:A   \n",
       "\n",
       "         CHR_POS  \n",
       "7474         0:0  \n",
       "7476         0:0  \n",
       "7477         0:0  \n",
       "7478         0:0  \n",
       "7479         0:0  \n",
       "...          ...  \n",
       "21093486     0:0  \n",
       "21093487     0:0  \n",
       "21093488     0:0  \n",
       "21093489     0:0  \n",
       "21093490     0:0  \n",
       "\n",
       "[34111 rows x 10 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stat[stat['SNP_original'].isin(changelist)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "10ed6f40-0178-4ec9-9b81-6e092ddd0948",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "header = [\"CHR\", \"POS\", \"REF\", \"ALT\", \"BETA\", \"SE\", \"P\", \"SNP_original\", \"SNP\"]\n",
    "stat.to_csv('~/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz', index=False, compression='gzip',sep=\"\\t\", columns = header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "11c3761a-c502-4d97-a846-eed72dedfc3a",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta/AD_meta-ldclumping_flip_2022-08-08.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wa96242296e322948) is executed successfully with 1 completed step.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## Set the bash variables \n",
    "clumping_dir=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/072822_AD_meta\n",
    "clumping_sbatch=$clumping_dir/AD_meta-ldclumping_flip_$(date +\"%Y-%m-%d\").sbatch\n",
    "sumstatsFiles=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump.gz\n",
    "ld_sample_size=2000\n",
    "clump_field=P\n",
    "clump_p1=5e-06\n",
    "clump_p2=1\n",
    "clump_r2=0.04\n",
    "clump_kb=2000\n",
    "clump_annotate=BP\n",
    "numThreads=20\n",
    "clump_job_size=1\n",
    "\n",
    "clumping_args=\"\"\"default \n",
    "    --cwd $clumping_dir\n",
    "    --genoFile $genoFile\n",
    "    --sampleFile $sampleFile\n",
    "    --reference_genotype_prefix $reference_genotype_prefix  \n",
    "    --sumstatsFiles $sumstatsFiles \n",
    "    --ld_sample_size $ld_sample_size \n",
    "    --clump_field $clump_field\n",
    "    --clump_p1 $clump_p1 \n",
    "    --clump_p2 $clump_p2 \n",
    "    --clump_r2 $clump_r2 \n",
    "    --clump_kb $clump_kb \n",
    "    --clump_annotate $clump_annotate \n",
    "    --numThreads $numThreads \n",
    "    --job_size $clump_job_size\n",
    "    --container $container\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg \\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $clumping_sos \\\n",
    "    --to-script $clumping_sbatch \\\n",
    "    --args \"$clumping_args\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3246456a-e7d4-4b30-99c7-a077b9d19f85",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Find intersection between LD clumping regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "16dfd2f4-921e-4364-93e2-39ddadf545c2",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from functools import reduce\n",
    "import glob\n",
    "filenames = glob.glob(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822*/*.clumped_region\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f08a6c91-6766-4016-8648-07985f833805",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "dfs = [pd.read_csv(filename, dtype=str , sep=' ', header=None) for filename in filenames]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8fe0b052-62f5-4bd6-8c53-e8d1ad095b72",
   "metadata": {
    "kernel": "Python3",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[      0         1         2\n",
       " 0     1   6315125   6631106\n",
       " 1     1  12169337  16161606\n",
       " 2     1  20269718  22239026\n",
       " 3     1  42273297  43365565\n",
       " 4     1  45167429  47068339\n",
       " ..   ..       ...       ...\n",
       " 560  22  37977713  38626556\n",
       " 561  22  46608847  49887600\n",
       " 562  22  48993774  51146132\n",
       " 563  22  50436376  51229805\n",
       " 564  22  51032153  51173041\n",
       " \n",
       " [565 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6315125   6631106\n",
       " 1     1  12197464  16161606\n",
       " 2     1  45167429  47068339\n",
       " 3     1  80995695  80995695\n",
       " 4     1  82120001  82781655\n",
       " ..   ..       ...       ...\n",
       " 450  22  48993774  51146132\n",
       " 451  22  49049741  50616646\n",
       " 452  22  50436376  51229805\n",
       " 453  22  51032153  51173041\n",
       " 454  22  51237215  51237215\n",
       " \n",
       " [455 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6315125   6631106\n",
       " 1     1  12169337  16161606\n",
       " 2     1  20269718  22239026\n",
       " 3     1  42273297  43365565\n",
       " 4     1  45100431  46895641\n",
       " ..   ..       ...       ...\n",
       " 562  22  37976422  38673790\n",
       " 563  22  37977713  38626556\n",
       " 564  22  48993774  51146132\n",
       " 565  22  50436376  51229805\n",
       " 566  22  51032153  51173041\n",
       " \n",
       " [567 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6315125   6631106\n",
       " 1     1  12197464  16161606\n",
       " 2     1  45100431  46886953\n",
       " 3     1  80995695  80995695\n",
       " 4     1  82120001  82781655\n",
       " ..   ..       ...       ...\n",
       " 432  22  48993774  51146132\n",
       " 433  22  49049741  50616646\n",
       " 434  22  50436376  51229805\n",
       " 435  22  51032153  51173041\n",
       " 436  22  51237215  51237215\n",
       " \n",
       " [437 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6317585   6631106\n",
       " 1     1   7631420   8192898\n",
       " 2     1  12169337  16161606\n",
       " 3     1  20269718  22239026\n",
       " 4     1  24690676  25183119\n",
       " ..   ..       ...       ...\n",
       " 582  22  38093414  39346720\n",
       " 583  22  38865590  39306080\n",
       " 584  22  49606728  51146132\n",
       " 585  22  50436376  51229805\n",
       " 586  22  50618958  51176164\n",
       " \n",
       " [587 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6317585   6631106\n",
       " 1     1   7631420   8192898\n",
       " 2     1  12197464  16161606\n",
       " 3     1  24690676  25183119\n",
       " 4     1  27805160  29637318\n",
       " ..   ..       ...       ...\n",
       " 452  22  49049741  50616646\n",
       " 453  22  49606728  51146132\n",
       " 454  22  50436376  51229805\n",
       " 455  22  50618958  51176164\n",
       " 456  22  51237215  51237215\n",
       " \n",
       " [457 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1  20269718  22239026\n",
       " 1     1  42273297  43365565\n",
       " 2     1  50649929  52830530\n",
       " 3     1  62814858  63491421\n",
       " 4     1  63381854  67224373\n",
       " ..   ..       ...       ...\n",
       " 422  21  38925356  42309248\n",
       " 423  21  41743586  45396035\n",
       " 424  21  43747236  44021573\n",
       " 425  22  37978750  38626278\n",
       " 426  22  48993774  51146132\n",
       " \n",
       " [427 rows x 3 columns],\n",
       "       0         1         2\n",
       " 0     1   6036864   6036864\n",
       " 1     1   6043137   6043137\n",
       " 2     1   6048498   6048498\n",
       " 3     1   6049935   6049935\n",
       " 4     1   6050663   6050663\n",
       " ..   ..       ...       ...\n",
       " 310  21  43747236  44021573\n",
       " 311  22  37978750  38626278\n",
       " 312  22  48993774  51146132\n",
       " 313  22  49049741  50616646\n",
       " 314  22  51237215  51237215\n",
       " \n",
       " [315 rows x 3 columns]]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1e8fe8de-bce7-4b14-95ef-0fef25d4650b",
   "metadata": {
    "kernel": "Python3",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>207302548</td>\n",
       "      <td>208001215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>127773524</td>\n",
       "      <td>127898467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>120807217</td>\n",
       "      <td>121829279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>10864183</td>\n",
       "      <td>11050683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>17421637</td>\n",
       "      <td>18282225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>72819206</td>\n",
       "      <td>73187503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>38980511</td>\n",
       "      <td>42216621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>47312531</td>\n",
       "      <td>47875319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7</td>\n",
       "      <td>138432792</td>\n",
       "      <td>138552312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>11424595</td>\n",
       "      <td>11835147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>72300753</td>\n",
       "      <td>74080805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10</td>\n",
       "      <td>94478104</td>\n",
       "      <td>95067066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10</td>\n",
       "      <td>126754527</td>\n",
       "      <td>126865062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>11</td>\n",
       "      <td>7910740</td>\n",
       "      <td>8256690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>11</td>\n",
       "      <td>119524157</td>\n",
       "      <td>123096745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>58829451</td>\n",
       "      <td>59516875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>19</td>\n",
       "      <td>777560</td>\n",
       "      <td>3061573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>19</td>\n",
       "      <td>43339752</td>\n",
       "      <td>47268323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>43423669</td>\n",
       "      <td>47295731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>44078282</td>\n",
       "      <td>47394223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>19</td>\n",
       "      <td>44280645</td>\n",
       "      <td>46123098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>19</td>\n",
       "      <td>44294330</td>\n",
       "      <td>45923337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>19</td>\n",
       "      <td>45065068</td>\n",
       "      <td>45740340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>19</td>\n",
       "      <td>45121726</td>\n",
       "      <td>45351516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>19</td>\n",
       "      <td>45192912</td>\n",
       "      <td>45436247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>19</td>\n",
       "      <td>45248856</td>\n",
       "      <td>45913704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>19</td>\n",
       "      <td>45411110</td>\n",
       "      <td>45688601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>19</td>\n",
       "      <td>45449166</td>\n",
       "      <td>45828980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>19</td>\n",
       "      <td>45516125</td>\n",
       "      <td>45516125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>19</td>\n",
       "      <td>45554058</td>\n",
       "      <td>45891241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     0          1          2\n",
       "0    1  207302548  208001215\n",
       "1    2  127773524  127898467\n",
       "2    3  120807217  121829279\n",
       "3    4   10864183   11050683\n",
       "4    4   17421637   18282225\n",
       "5    5   72819206   73187503\n",
       "6    6   38980511   42216621\n",
       "7    6   47312531   47875319\n",
       "8    7  138432792  138552312\n",
       "9   10   11424595   11835147\n",
       "10  10   72300753   74080805\n",
       "11  10   94478104   95067066\n",
       "12  10  126754527  126865062\n",
       "13  11    7910740    8256690\n",
       "14  11  119524157  123096745\n",
       "15  15   58829451   59516875\n",
       "16  19     777560    3061573\n",
       "17  19   43339752   47268323\n",
       "18  19   43423669   47295731\n",
       "19  19   44078282   47394223\n",
       "20  19   44280645   46123098\n",
       "21  19   44294330   45923337\n",
       "22  19   45065068   45740340\n",
       "23  19   45121726   45351516\n",
       "24  19   45192912   45436247\n",
       "25  19   45248856   45913704\n",
       "26  19   45411110   45688601\n",
       "27  19   45449166   45828980\n",
       "28  19   45516125   45516125\n",
       "29  19   45554058   45891241"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduce(lambda left, right: pd.merge(left, right, how=\"inner\", on=[0,1,2]), dfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "fbd73f4c-865e-46b4-b996-2bc8120da7cf",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "f3393_ADmeta=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f3393_ADmeta/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump_071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.clumped_region', dtype=str , sep=' ', header=None )\n",
    "f2247_ADmeta=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADmeta/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump_071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.clumped_region', dtype=str , sep=' ', header=None )\n",
    "f2257_ADmeta=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADmeta/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump_071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.clumped_region', dtype=str ,sep=' ', header=None )\n",
    "combined_ADmeta=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADmeta/GCST90027158_buildGRCh37_reheader.snp_stats.ldclump_071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.clumped_region', dtype=str , sep=' ', header=None)\n",
    "\n",
    "f3393_ADproxy=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f3393_ADproxy/AD_step2_BT_all.txt.snp_stats.ldclump_071222_UKBB_Hearing_aid_f3393_expandedwhite_14734cases_236699ctrl_500k_PC1_PC2_f3393.regenie.snp_stats.ldclump.clumped_region', dtype=str ,  sep=' ', header=None)\n",
    "f2247_ADproxy=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2247_ADproxy/AD_step2_BT_all.txt.snp_stats.ldclump_071222_UKBB_Hearing_difficulty_f2247_expandedwhite_107994cases_236699ctrl_500k_PC1_PC2_f2247.regenie.snp_stats.ldclump.clumped_region', dtype=str , sep=' ', header=None)\n",
    "f2257_ADproxy=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_f2257_ADproxy/AD_step2_BT_all.txt.snp_stats.ldclump_071222_UKBB_Hearing_noise_f2257_expandedwhite_159033cases_236699ctrl_500k_PC1_PC2_f2257.regenie.snp_stats.ldclump.clumped_region', dtype=str ,  sep=' ', header=None)\n",
    "combined_ADproxy=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/LD_clumping/080822_combined_ADproxy/AD_step2_BT_all.txt.snp_stats.ldclump_071222_UKBB_Combined_f2247_f2257_expandedwhite_91092cases_236699ctrl_500k_PC1_PC2_f2247_f2257.regenie.snp_stats.ldclump.clumped_region', dtype=str , sep=' ', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "35e25208-7021-4b3f-9c6e-691c2c42c6aa",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>20269718</td>\n",
       "      <td>22239026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>42273297</td>\n",
       "      <td>43365565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>50649929</td>\n",
       "      <td>52830530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>62814858</td>\n",
       "      <td>63491421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>63381854</td>\n",
       "      <td>67224373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>21</td>\n",
       "      <td>38925356</td>\n",
       "      <td>42309248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>21</td>\n",
       "      <td>41743586</td>\n",
       "      <td>45396035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>21</td>\n",
       "      <td>43747236</td>\n",
       "      <td>44021573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>22</td>\n",
       "      <td>37978750</td>\n",
       "      <td>38626278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>22</td>\n",
       "      <td>48993774</td>\n",
       "      <td>51146132</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>427 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0         1         2\n",
       "0     1  20269718  22239026\n",
       "1     1  42273297  43365565\n",
       "2     1  50649929  52830530\n",
       "3     1  62814858  63491421\n",
       "4     1  63381854  67224373\n",
       "..   ..       ...       ...\n",
       "422  21  38925356  42309248\n",
       "423  21  41743586  45396035\n",
       "424  21  43747236  44021573\n",
       "425  22  37978750  38626278\n",
       "426  22  48993774  51146132\n",
       "\n",
       "[427 rows x 3 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f3393_ADmeta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ba252722-8905-4823-a363-35a9e8382d1a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>6315125</td>\n",
       "      <td>6631106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>12197464</td>\n",
       "      <td>16161606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>45167429</td>\n",
       "      <td>47068339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>80995695</td>\n",
       "      <td>80995695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>82120001</td>\n",
       "      <td>82781655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>22</td>\n",
       "      <td>48993774</td>\n",
       "      <td>51146132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451</th>\n",
       "      <td>22</td>\n",
       "      <td>49049741</td>\n",
       "      <td>50616646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>22</td>\n",
       "      <td>50436376</td>\n",
       "      <td>51229805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>22</td>\n",
       "      <td>51032153</td>\n",
       "      <td>51173041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>22</td>\n",
       "      <td>51237215</td>\n",
       "      <td>51237215</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>455 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0         1         2\n",
       "0     1   6315125   6631106\n",
       "1     1  12197464  16161606\n",
       "2     1  45167429  47068339\n",
       "3     1  80995695  80995695\n",
       "4     1  82120001  82781655\n",
       "..   ..       ...       ...\n",
       "450  22  48993774  51146132\n",
       "451  22  49049741  50616646\n",
       "452  22  50436376  51229805\n",
       "453  22  51032153  51173041\n",
       "454  22  51237215  51237215\n",
       "\n",
       "[455 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_ADproxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "461ec58d-56a1-4e8c-8576-0f13420bb8d6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>20269718</td>\n",
       "      <td>22239026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>42273297</td>\n",
       "      <td>43365565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>50649929</td>\n",
       "      <td>52830530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>62814858</td>\n",
       "      <td>63491421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>63381854</td>\n",
       "      <td>67224373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329</th>\n",
       "      <td>21</td>\n",
       "      <td>27344189</td>\n",
       "      <td>27765689</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>21</td>\n",
       "      <td>28095992</td>\n",
       "      <td>28209494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>21</td>\n",
       "      <td>33951180</td>\n",
       "      <td>37891455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>21</td>\n",
       "      <td>37948100</td>\n",
       "      <td>38914616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>21</td>\n",
       "      <td>38925356</td>\n",
       "      <td>42309248</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>334 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0         1         2\n",
       "0     1  20269718  22239026\n",
       "1     1  42273297  43365565\n",
       "2     1  50649929  52830530\n",
       "3     1  62814858  63491421\n",
       "4     1  63381854  67224373\n",
       "..   ..       ...       ...\n",
       "329  21  27344189  27765689\n",
       "330  21  28095992  28209494\n",
       "331  21  33951180  37891455\n",
       "332  21  37948100  38914616\n",
       "333  21  38925356  42309248\n",
       "\n",
       "[334 rows x 3 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduce(lambda left, right: pd.merge(left, right, how=\"inner\", on=[0,1,2]), [f3393_ADmeta,f2247_ADmeta,f2257_ADmeta,combined_ADmeta])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c6152393-f640-4912-a725-1ff3608ffa68",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>80995695</td>\n",
       "      <td>80995695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>105458976</td>\n",
       "      <td>107422312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>158531011</td>\n",
       "      <td>158531011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>160834616</td>\n",
       "      <td>161383559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>169267498</td>\n",
       "      <td>172944529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>20</td>\n",
       "      <td>776747</td>\n",
       "      <td>894121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>20</td>\n",
       "      <td>43358848</td>\n",
       "      <td>44214164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>20</td>\n",
       "      <td>59195437</td>\n",
       "      <td>62721760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>22</td>\n",
       "      <td>49049741</td>\n",
       "      <td>50616646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>22</td>\n",
       "      <td>51237215</td>\n",
       "      <td>51237215</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>181 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0          1          2\n",
       "0     1   80995695   80995695\n",
       "1     1  105458976  107422312\n",
       "2     1  158531011  158531011\n",
       "3     1  160834616  161383559\n",
       "4     1  169267498  172944529\n",
       "..   ..        ...        ...\n",
       "176  20     776747     894121\n",
       "177  20   43358848   44214164\n",
       "178  20   59195437   62721760\n",
       "179  22   49049741   50616646\n",
       "180  22   51237215   51237215\n",
       "\n",
       "[181 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduce(lambda left, right: pd.merge(left, right, how=\"inner\", on=[0,1,2]), [f3393_ADproxy,f2247_ADproxy,f2257_ADproxy,combined_ADproxy])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "de5d3e58-2e9e-4dcc-9981-5e6b71a1d4bc",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>207302548</td>\n",
       "      <td>208001215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>127773524</td>\n",
       "      <td>127898467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>120807217</td>\n",
       "      <td>121829279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>10864183</td>\n",
       "      <td>11050683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>17421637</td>\n",
       "      <td>18282225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>72819206</td>\n",
       "      <td>73187503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>38980511</td>\n",
       "      <td>42216621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>47312531</td>\n",
       "      <td>47875319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7</td>\n",
       "      <td>138432792</td>\n",
       "      <td>138552312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>11424595</td>\n",
       "      <td>11835147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>72300753</td>\n",
       "      <td>74080805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10</td>\n",
       "      <td>94478104</td>\n",
       "      <td>95067066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10</td>\n",
       "      <td>126754527</td>\n",
       "      <td>126865062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>11</td>\n",
       "      <td>7910740</td>\n",
       "      <td>8256690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>11</td>\n",
       "      <td>119524157</td>\n",
       "      <td>123096745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>58829451</td>\n",
       "      <td>59516875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>19</td>\n",
       "      <td>777560</td>\n",
       "      <td>3061573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>19</td>\n",
       "      <td>43339752</td>\n",
       "      <td>47268323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>43423669</td>\n",
       "      <td>47295731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>44078282</td>\n",
       "      <td>47394223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>19</td>\n",
       "      <td>44280645</td>\n",
       "      <td>46123098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>19</td>\n",
       "      <td>44294330</td>\n",
       "      <td>45923337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>19</td>\n",
       "      <td>45065068</td>\n",
       "      <td>45740340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>19</td>\n",
       "      <td>45121726</td>\n",
       "      <td>45351516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>19</td>\n",
       "      <td>45192912</td>\n",
       "      <td>45436247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>19</td>\n",
       "      <td>45248856</td>\n",
       "      <td>45913704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>19</td>\n",
       "      <td>45411110</td>\n",
       "      <td>45688601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>19</td>\n",
       "      <td>45449166</td>\n",
       "      <td>45828980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>19</td>\n",
       "      <td>45516125</td>\n",
       "      <td>45516125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>19</td>\n",
       "      <td>45554058</td>\n",
       "      <td>45891241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     0          1          2\n",
       "0    1  207302548  208001215\n",
       "1    2  127773524  127898467\n",
       "2    3  120807217  121829279\n",
       "3    4   10864183   11050683\n",
       "4    4   17421637   18282225\n",
       "5    5   72819206   73187503\n",
       "6    6   38980511   42216621\n",
       "7    6   47312531   47875319\n",
       "8    7  138432792  138552312\n",
       "9   10   11424595   11835147\n",
       "10  10   72300753   74080805\n",
       "11  10   94478104   95067066\n",
       "12  10  126754527  126865062\n",
       "13  11    7910740    8256690\n",
       "14  11  119524157  123096745\n",
       "15  15   58829451   59516875\n",
       "16  19     777560    3061573\n",
       "17  19   43339752   47268323\n",
       "18  19   43423669   47295731\n",
       "19  19   44078282   47394223\n",
       "20  19   44280645   46123098\n",
       "21  19   44294330   45923337\n",
       "22  19   45065068   45740340\n",
       "23  19   45121726   45351516\n",
       "24  19   45192912   45436247\n",
       "25  19   45248856   45913704\n",
       "26  19   45411110   45688601\n",
       "27  19   45449166   45828980\n",
       "28  19   45516125   45516125\n",
       "29  19   45554058   45891241"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reduce(lambda left, right: pd.merge(left, right, how=\"inner\", on=[0,1,2]), [f3393_ADmeta,f2247_ADmeta,f2257_ADmeta,combined_ADmeta,f3393_ADproxy,f2247_ADproxy,f2257_ADproxy,combined_ADproxy])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85f9ce6b-03a5-4f7f-b315-3b2a66e823c8",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "#To find the name of the file where the region was located\n",
    "grep -RH \"1 42273297 43365565\" 080822_*/*.clumped_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44985851-7606-4979-9f34-11d364035546",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "sort -m 080822_*/*.clumped_region | uniq -d > shared_regions.txt\n",
    "grep -Fx -f shared_regions.txt 080822_*/*.clumped_region"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a174a204-a213-40f5-b3e6-0e8b619622c5",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Find the shared significant variants between AD_meta and AD_proxy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d028880-fd77-42ee-924f-8a3a433e48ff",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "4bfed576-0432-4d0a-9936-875531a8e713",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "# Step 4: run region extraction to get some of the regions to test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6607335d-e5f7-445c-9699-6a21846da17b",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "Markdown",
     "markdown",
     "markdown",
     "",
     ""
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     {
      "name": "ipython",
      "version": 3
     }
    ]
   ],
   "version": "0.22.6"
  },
  "toc-autonumbering": true,
  "toc-showcode": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
