{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9e6b0fc1-3b30-4ce4-a6ee-5c2601e05ca8",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "## Create a MWE for susie_rss"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a03b17a-9bed-4f80-b821-dd97d447f023",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Old way of creating the z file from mfi files not longer used"
   ]
  },
  {
   "cell_type": "raw",
   "id": "c79b393b-6e38-4007-8a85-d0b6b4b88b36",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd\n",
    "log_file= open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/regions1_22.log', \"w\")\n",
    "regions=glob.glob('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/**/*.variants', recursive=True)\n",
    "mfi_files=glob.glob('/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr*[0-9]_v3.txt', recursive=True)\n",
    "for filename in regions:\n",
    "    print(filename)\n",
    "    basename=os.path.basename(filename)\n",
    "    sp = basename.split(\"_\")[0]\n",
    "    chrom= sp.lstrip(\"0\")\n",
    "    print(basename)\n",
    "    fullregion = pd.read_csv(filename,header=0,sep=\"\\t\", skiprows=1, skipfooter=1, names=['alternate_ids', 'rsid', 'chromosome', 'position', 'number_of_alleles', 'allele1', 'allele2'])\n",
    "    print('The number of variants in the region is:',fullregion.shape[0])\n",
    "    mfi_file = [file for file in mfi_files if \"chr\" + chrom + \"_\" in file][0]\n",
    "    chrom_file = pd.read_csv(mfi_file,header=None,sep=\"\\t\", names=[\"alternate_id\", \"rsid\", \"position\", \"allele1\", \"allele2\", \"maf\", \"minor_allele\", \"info_score\"])\n",
    "    print('The number of variants in',mfi_file,'is:',chrom_file.shape[0])\n",
    "    subset_region=chrom_file.join(fullregion.set_index(['alternate_ids', 'rsid', 'position', 'allele1', 'allele2']), on=['alternate_id', 'rsid', 'position', 'allele1', 'allele2'], how='right')\n",
    "    maf=0.001\n",
    "    subset_maf=subset_region[subset_region['maf'] > maf]\n",
    "    print('The number of variants in the maf',maf,'filtered z_file is:',subset_maf.shape[0])\n",
    "    ## Need to rename the rsid and switch it with the alternate_id because this is how variants are named in the bgen file\n",
    "    subset_maf.rename(columns={'rsid': 'rsid_old', 'alternate_id': 'rsid'}, inplace=True)\n",
    "    subset_maf[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\", \"alternate_id\"]].to_csv(filename + '_' + str(maf) + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "21a6e37e-ab42-414d-8e20-da2658c489b8",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd\n",
    "log_file= open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/region06.log', \"w\")\n",
    "regions=glob.glob('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/*.variants', recursive=True)\n",
    "mfi_files=glob.glob('/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr06_v3.txt', recursive=True)\n",
    "for filename in regions:\n",
    "    print(filename)\n",
    "    basename=os.path.basename(filename)\n",
    "    sp = basename.split(\"_\")[0]\n",
    "    chrom= sp.lstrip(\"0\")\n",
    "    print(basename)\n",
    "    fullregion = pd.read_csv(filename,header=0,sep=\"\\t\", skiprows=1, skipfooter=1, names=['alternate_ids', 'rsid', 'chromosome', 'position', 'number_of_alleles', 'allele1', 'allele2'])\n",
    "    print('The number of variants in the region is:',fullregion.shape[0])\n",
    "    mfi_file = [file for file in mfi_files if \"chr\" + chrom + \"_\" in file][0]\n",
    "    chrom_file = pd.read_csv(mfi_file,header=None,sep=\"\\t\", names=[\"alternate_id\", \"rsid\", \"position\", \"allele1\", \"allele2\", \"maf\", \"minor_allele\", \"info_score\"])\n",
    "    print('The number of variants in',mfi_file,'is:',chrom_file.shape[0])\n",
    "    subset_region=chrom_file.join(fullregion.set_index(['alternate_ids', 'rsid', 'position', 'allele1', 'allele2']), on=['alternate_id', 'rsid', 'position', 'allele1', 'allele2'], how='right')\n",
    "    maf=0.001\n",
    "    subset_maf=subset_region[subset_region['maf'] > maf]\n",
    "    print('The number of variants in the maf',maf,'filtered z_file is:',subset_maf.shape[0])\n",
    "    ## If chromosome if from 1 to 9 then you need to add a 0 before the chr number\n",
    "    subset_maf['chromosome'] = subset_maf.chromosome.astype(str).str.zfill(2)\n",
    "    subset_maf[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\", \"rsid_old\"]].to_csv(filename + '_' + str(maf) + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0c17177-b997-45b6-ba90-f74dc805bcb3",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "The idea of this notebook is to create a rough idea of the file handling and formatting that has to be done to streamline the fine-mapping process"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "72317a64-b2ef-4fd2-b9b5-120e12a6658a",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## 1. Rename bgen variants to chr:pos:ref:alt\n",
    "\n",
    "One major issue is that the UK Biobank released the genotype and imputed data using rsid as the variant indentification, whereas for WES data we used the convention of CHR:POS:REF:ALT in our variant id's given the multiple known problems of the rsids. \n",
    "\n",
    "Manipulating the bgen files and editing the variant ids is not easy with available python libraries. Therefore, my preferred method is to use a combination of bgenix and plink to do format and file manipulation"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fe7d9ee3-8fe2-4b64-bd23-aeb99e899f0e",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### First, get the region from the UKB imputed data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56993a16-ab0a-4aa3-82f7-807dbf74c17c",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "Use this pipeline `113022_bgenix_ldblocks.ipynb` to get the bgen and bgi files for each LD independent region of the genome based on UKB imputed data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfec027d-7f69-49dc-8db1-e3b965597886",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "```\n",
    "sos run ~/UKBB_GWAS_dev/workflow/113022_bgenix_ldblocks.ipynb \\\n",
    "    bgenix\\\n",
    "    --cwd test\\\n",
    "    --genofile_prefix test/ukb_imp_chr\\\n",
    "    --genofile_suffix _v3.bgen\\\n",
    "    --region_file data/ldblocks/EUR/fourier_ls-all.bed\\\n",
    "    --job_size 10\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cb30b796-d447-4015-afd9-e34893c381bd",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "source": [
    "## Using the original bgen without liftover"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "bf36a978-492a-4cc3-9549-d800b30c9287",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/test_ldstore/01_44969183_46899501_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.01\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/test_ldstore/01_44969183_46899501_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 15:24:37 2023\n",
      "257481 MiB RAM detected, ~209264 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 59885 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/test_ldstore/01_44969183_46899501_hg19-temporary.pgen +\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501_hg19-temporary.psam.\n",
      "59885 variants loaded from\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "54809 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "5076 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.snplist .\n",
      "Writing /home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bim ... done.\n",
      "Writing /home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen ... 14304456698294done.\n",
      "Writing /home/dmc2245/test_ldstore/01_44969183_46899501_hg19.sample ... done.\n",
      "End time: Wed Dec 13 15:24:52 2023\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/test_ldstore/01_44969183_46899501_hg19 \\\n",
    "--maf 0.01 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "98fe9e28-ecf2-4497-a0ad-ba85c9c86da3",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "snplist = pd.read_csv('~/test_ldstore/01_44969183_46899501_hg19.snplist', header=None, sep='\\t', names=[\"rsid\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "319a56da-531d-4a43-8234-5788bff8a2e6",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:44969590:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:44969691:G:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:44969939:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:44970214:TG:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:44970950:G:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              rsid\n",
       "0   1:44969590:C:T\n",
       "1   1:44969691:G:T\n",
       "2   1:44969939:C:T\n",
       "3  1:44970214:TG:T\n",
       "4   1:44970950:G:T"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "63526837-232d-4282-bef6-9d0c9b3affcc",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist[['chromosome', 'position', 'allele1', 'allele2']] = snplist['rsid'].str.split(':', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "74b24caa-71ad-4937-9edc-2a07b8f78f2a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "#snplist['chromosome'] = snplist.chromosome.astype(str).str.zfill(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "74961c2c-0533-46c4-815f-e8024e9181e3",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "#snplist['rsid'] = snplist['rsid'].str.replace(r'^(\\d:)', lambda x: 'chr0' + x.group(1), regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "838120d0-c906-4b02-8d99-1489d5f3de0d",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "#snplist['rsid'] = snplist['rsid'].str.extract(r'^(\\d+:\\d+)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "90580b14-823f-487b-a3ed-e0139a804604",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:44969590:C:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969590</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:44969691:G:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969691</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:44969939:C:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969939</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:44970214:TG:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44970214</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:44970950:G:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44970950</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              rsid chromosome  position allele1 allele2\n",
       "0   1:44969590:C:T          1  44969590       C       T\n",
       "1   1:44969691:G:T          1  44969691       G       T\n",
       "2   1:44969939:C:T          1  44969939       C       T\n",
       "3  1:44970214:TG:T          1  44970214      TG       T\n",
       "4   1:44970950:G:T          1  44970950       G       T"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "8eee0595-2116-419a-8508-98184fc6c524",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\"]].to_csv('~/test_ldstore/01_44969183_46899501_hg19.z', header=True, index=False, sep= ' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "228b392f-93bc-48fc-b8c1-ab1ac2dc82ee",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "40f7152f-66cc-42f3-975c-27d0e05efb44",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen\" in \"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen\" with 5076 variants...\n",
      "Building BGEN index                                         : [******************************] (5076/5076,0.8s,6570.8/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "bgenix -g ~/test_ldstore/01_44969183_46899501_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "618412b5-1b50-49cc-8b2e-2851d0b113bb",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "Building query                                              :  (3/?,0.0s,22193.8/s)\n",
      "# bgenix: started 2023-12-13 14:14:15\n",
      "alternate_ids\trsid\tchromosome\tposition\tnumber_of_alleles\tfirst_allele\talternative_alleles\n",
      ".\t01:44969310:A:G\t1\t44969310\t2\tA\tG\n",
      ".\t01:44969328:C:T\t1\t44969328\t2\tC\tT\n",
      ".\t01:44969330:C:G\t1\t44969330\t2\tC\tG\n",
      "# bgenix: success, total 3 variants.\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g /home/dmc2245/test_ldstore/01_44969183_46899501.bgen -list  -incl-range 1:44969183-44969383"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e686036f-50a8-4704-ac22-99e73cc06cb6",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "Building query                                              :  (3/?,0.0s,20837.7/s)\n",
      "# bgenix: started 2023-12-13 13:47:22\n",
      "alternate_ids\trsid\tchromosome\tposition\tnumber_of_alleles\tfirst_allele\talternative_alleles\n",
      "1:44969310_A_G\trs777318309\t01\t44969310\t2\tA\tG\n",
      "1:44969328_C_T\trs141298448\t01\t44969328\t2\tC\tT\n",
      "1:44969330_C_G\trs181919981\t01\t44969330\t2\tC\tG\n",
      "# bgenix: success, total 3 variants.\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "bgenix -g /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen -list -incl-range 01:44969183-44969383 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "599dcc52-4356-48b0-a016-7a41e070296b",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "bgen=glob.glob(r\"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen\")\n",
    "masterfile = pd.DataFrame({\n",
    "    \"z\" : glob.glob(r\"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.z\"),\n",
    "    \"bgen\" : glob.glob(r\"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen\"),\n",
    "    \"bgi\" : glob.glob(r\"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.bgen.bgi\"),\n",
    "    \"bcor\" : [i.replace('bgen','bcor') for i in bgen],\n",
    "    \"ld\" : [i.replace('bgen','ld') for i in bgen],\n",
    "    \"n_samples\" : 351430,\n",
    "    \"sample\": f\"/home/dmc2245/test_ldstore/01_44969183_46899501_hg19.sample\",\n",
    "    \"incl\" : f\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl\"\n",
    "          })\n",
    "masterfile = masterfile[['z', 'bgen', 'bgi', 'bcor', 'ld','n_samples', 'sample', 'incl']]\n",
    "\n",
    "masterfile.to_csv(\"/home/dmc2245/test_ldstore/mastefile_hg19\", sep=\";\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "c2c82836-8b0a-4206-aa84-5b29969fcafa",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mbcor\u001b[0m: Create bcor files\n",
      "INFO: \u001b[32mbcor\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: Workflow bcor (ID=wd947528e99aa805d) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "## this needs at least 83GB of memory\n",
    "\n",
    "module load Singularity\n",
    "sos run /home/dmc2245/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    bcor\\\n",
    "    --cwd ~/test_ldstore \\\n",
    "    --masterfile /home/dmc2245/test_ldstore/mastefile_hg19 \\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "b38bf4d4-b4a8-4d7d-a1c7-38a2cd5bdff1",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg_mamba\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg_mamba\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg_mamba\u001b[0m output:   \u001b[32m/home/dmc2245/test_ldstore/ldstore_test_2023-12-13.sbatch\u001b[0m\n",
      "INFO: Workflow csg_mamba (ID=w0662ea312dd5cbcc) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "## Set the bash variables \n",
    "cwd=~/test_ldstore\n",
    "ldstore_sbatch=$cwd/ldstore_test_$(date +\"%Y-%m-%d\").sbatch\n",
    "masterfile=/home/dmc2245/test_ldstore/mastefile_hg19\n",
    "jobsize=1\n",
    "ldstore_sos=~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb\n",
    "tpl_file=~/project/bioworkflows/admin/csg.yml\n",
    "mem='100G'\n",
    "job_size=1\n",
    "numThreads=10\n",
    "\n",
    "ldstore_args=\"\"\"bcor\n",
    " --cwd $cwd\n",
    " --masterfile $masterfile\n",
    " --numThreads $numThreads \n",
    " --mem $mem \n",
    " --job_size $job_size\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg_mamba \\\n",
    " --template-file $tpl_file \\\n",
    " --workflow-file $ldstore_sos \\\n",
    " --to-script $ldstore_sbatch \\\n",
    " --args \"$ldstore_args\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "c21e8830-f1b9-45e0-aa88-e3b4de41bc14",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mld_1\u001b[0m: Calculate LD\n",
      "INFO: \u001b[32mld_1\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: Workflow ld (ID=w1b8257c32b61fc0e) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "## this needs at least 83GB of memory\n",
    "\n",
    "module load Singularity\n",
    "sos run /home/dmc2245/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    ld:1\\\n",
    "    --cwd ~/test_ldstore \\\n",
    "    --masterfile /home/dmc2245/test_ldstore/mastefile_hg19 \\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3e2fbdf-e97b-4a99-915e-08031f0f6aa3",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### Now, use plink to rename the variant ID and output as VCF"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c836ded-104a-4f0c-9eb7-3fdcae178a28",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "FIXME: we really do not need Dosage data because we don't care about it. Will this impact LD calculations???"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "903aa0cc-16ba-4af9-9e9d-207c9adc19fb",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/test_ldstore/01_44969183_46899501.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen ref-first\n",
      "  --export vcf bgz id-paste=iid\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/test_ldstore/01_44969183_46899501\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "\n",
      "Start time: Tue Dec 12 14:24:16 2023\n",
      "257481 MiB RAM detected, ~221495 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 59885 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501-temporary.psam .\n",
      "--bgen: /home/dmc2245/test_ldstore/01_44969183_46899501-temporary.pgen +\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/test_ldstore/01_44969183_46899501-temporary.psam.\n",
      "59885 variants loaded from\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "--export vcf bgz to /home/dmc2245/test_ldstore/01_44969183_46899501.vcf.gz ...\n",
      "10101111121213131414151516161717181819202021212222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "End time: Tue Dec 12 14:25:36 2023\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--export vcf bgz id-paste=iid \\\n",
    "--set-all-var-ids @:#:\\$r:\\$a \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--out /home/dmc2245/test_ldstore/01_44969183_46899501"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c3ebd96-4fca-4f91-b8bc-d7999123134d",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "### A very critical step is to do LiftOver if the downstream analysis will use hg38 genomic coordinates"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecab6775-b504-4d3a-b551-e21b023ea978",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Get all the necessary programs and reference files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e7839891-b14d-4d2a-982b-64c91f7d923b",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-12-12 14:31:54--  https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar\n",
      "Resolving menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)... 10.147.211.93\n",
      "Connecting to menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)|10.147.211.93|:8080... connected.\n",
      "Proxy request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231212%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231212T193154Z&X-Amz-Expires=300&X-Amz-Signature=d66030f0278cda0b061fed812f8f3de9bfa3c976d9c829383e1512baa2749c41&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=18225913&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream [following]\n",
      "--2023-12-12 14:31:54--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/839cd9dd-e7dc-4c29-ab0d-6ab4bdcded4e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231212%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231212T193154Z&X-Amz-Expires=300&X-Amz-Signature=d66030f0278cda0b061fed812f8f3de9bfa3c976d9c829383e1512baa2749c41&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=18225913&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream\n",
      "Connecting to menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)|10.147.211.93|:8080... connected.\n",
      "Proxy request sent, awaiting response... 200 OK\n",
      "Length: 17544754 (17M) [application/octet-stream]\n",
      "Saving to: ‘picard.jar’\n",
      "\n",
      "picard.jar          100%[===================>]  16.73M  15.9MB/s    in 1.1s    \n",
      "\n",
      "2023-12-12 14:31:56 (15.9 MB/s) - ‘picard.jar’ saved [17544754/17544754]\n",
      "\n",
      "--2023-12-12 14:31:56--  https://raw.githubusercontent.com/broadinstitute/gatk/master/scripts/funcotator/data_sources/gnomAD/b37ToHg38.over.chain\n",
      "Resolving menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)... 10.147.211.93\n",
      "Connecting to menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)|10.147.211.93|:8080... connected.\n",
      "Proxy request sent, awaiting response... 200 OK\n",
      "Length: 602613 (588K) [text/plain]\n",
      "Saving to: ‘b37ToHg38.over.chain’\n",
      "\n",
      "b37ToHg38.over.chai 100%[===================>] 588.49K  --.-KB/s    in 0.06s   \n",
      "\n",
      "2023-12-12 14:31:56 (10.3 MB/s) - ‘b37ToHg38.over.chain’ saved [602613/602613]\n",
      "\n",
      "--2023-12-12 14:31:56--  https://ilmn-dragen-giab-samples.s3.amazonaws.com/FASTA/hg38.fa\n",
      "Resolving menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)... 10.147.211.93\n",
      "Connecting to menloproxy.cumc.columbia.edu (menloproxy.cumc.columbia.edu)|10.147.211.93|:8080... connected.\n",
      "Proxy request sent, awaiting response... 200 OK\n",
      "Length: 3263683042 (3.0G) [binary/octet-stream]\n",
      "Saving to: ‘hg38.fa’\n",
      "\n",
      "hg38.fa             100%[===================>]   3.04G  16.5MB/s    in 3m 31s  \n",
      "\n",
      "2023-12-12 14:35:29 (14.8 MB/s) - ‘hg38.fa’ saved [3263683042/3263683042]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cd /home/dmc2245/test_ldstore && \\\n",
    "wget https://github.com/broadinstitute/picard/releases/download/2.27.4/picard.jar && \\\n",
    "wget https://raw.githubusercontent.com/broadinstitute/gatk/master/scripts/funcotator/data_sources/gnomAD/b37ToHg38.over.chain && \\\n",
    "wget https://ilmn-dragen-giab-samples.s3.amazonaws.com/FASTA/hg38.fa\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39607a48-d655-48bb-9726-199d1abf8918",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "source": [
    "Run the liftover on the `vcf.gz`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c675de9a-c327-4ce3-be10-b56654450811",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "14:56:19.306 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/vast/hpc/homes/dmc2245/test_ldstore/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Dec 12 14:56:19 EST 2023] LiftoverVcf --INPUT /home/dmc2245/test_ldstore/01_44969183_46899501.vcf.gz --OUTPUT /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.vcf --CHAIN b37ToHg38.over.chain --REJECT /home/dmc2245/test_ldstore/rejected_variants.vcf --RECOVER_SWAPPED_REF_ALT true --DISABLE_SORT true --REFERENCE_SEQUENCE /home/dmc2245/test_ldstore/GRCh38_full_analysis_set_plus_decoy_hla.fa --WARN_ON_MISSING_CONTIG false --LOG_FAILED_INTERVALS true --WRITE_ORIGINAL_POSITION false --WRITE_ORIGINAL_ALLELES false --LIFTOVER_MIN_MATCH 1.0 --ALLOW_MISSING_FIELDS_IN_HEADER false --TAGS_TO_REVERSE AF --TAGS_TO_DROP MAX_AF --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 5 --MAX_RECORDS_IN_RAM 500000 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false\n",
      "[Tue Dec 12 14:56:19 EST 2023] Executing as dmc2245@node43 on Linux 4.19.0-20-amd64 amd64; OpenJDK 64-Bit Server VM 11.0.14+9-post-Debian-1deb10u1; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:2.27.4-SNAPSHOT\n",
      "INFO\t2023-12-12 14:56:20\tLiftoverVcf\tLoading up the target reference genome.\n",
      "INFO\t2023-12-12 14:56:33\tLiftoverVcf\tLifting variants over and writing the output file. Variants will not be sorted.\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\tProcessed 59885 variants.\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\t0 variants failed to liftover.\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\t0 variants lifted over but had mismatching reference alleles after lift over.\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\t0.0000% of variants were not successfully lifted over and written to the output.\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\tliftover success by source contig:\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\t1: 59885 / 59885 (100.0000%)\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\tlifted variants by target contig:\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\tchr1: 59885\n",
      "INFO\t2023-12-12 22:11:22\tLiftoverVcf\t1 variants were lifted by swapping REF/ALT alleles.\n",
      "[Tue Dec 12 22:11:22 EST 2023] picard.vcf.LiftoverVcf done. Elapsed time: 435.05 minutes.\n",
      "Runtime.totalMemory()=12469665792\n"
     ]
    }
   ],
   "source": [
    "## hg38 coordinates will be 44503511-46433829\n",
    "java -jar ~/test_ldstore/picard.jar LiftoverVcf -I ~/test_ldstore/01_44969183_46899501.vcf.gz -O ~/test_ldstore/01_44969183_46899501.lo38.vcf \\\n",
    "   -C b37ToHg38.over.chain --REJECT ~/test_ldstore/rejected_variants.vcf -R ~/test_ldstore/GRCh38_full_analysis_set_plus_decoy_hla.fa \\\n",
    "   --RECOVER_SWAPPED_REF_ALT true --DISABLE_SORT true"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75d541b9-3c5f-4d40-bfd3-1bfe4994d0b8",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Sort the file using bcftools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c2d07330-9e68-4410-9236-f46173a9b02a",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing to /tmp/6958056.1.csg.q/bcftools.dFBgzY\n",
      "Merging 85 temporary files\n",
      "Cleaning\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "module load BCFTOOLS/1.18\n",
    "bcftools sort -o ~/test_ldstore/01_44969183_46899501.lo38.vcf.gz -O z ~/test_ldstore/01_44969183_46899501.lo38.vcf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dad76d72-e38f-40aa-8dba-b21d9273836f",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Convert to bgen format again to be able to run bcor in ldstore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cfe40342-d667-4ba1-beaa-6587a716da46",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.log.\n",
      "Options in effect:\n",
      "  --export bgen-1.2 bits=8 sample-v2 ref-first\n",
      "  --memory 10000\n",
      "  --out /home/dmc2245/test_ldstore/01_44969183_46899501.lo38\n",
      "  --threads 8\n",
      "  --vcf /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.vcf.gz\n",
      "\n",
      "Start time: Wed Dec 13 11:02:28 2023\n",
      "257481 MiB RAM detected, ~221246 available; reserving 10000 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "--vcf: 59885 variants scanned.\n",
      "--vcf: /home/dmc2245/test_ldstore/01_44969183_46899501.lo38-temporary.pgen +\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501.lo38-temporary.pvar.zst +\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501.lo38-temporary.psam written.\n",
      "487409 samples (0 females, 0 males, 487409 ambiguous; 487409 founders) loaded\n",
      "from /home/dmc2245/test_ldstore/01_44969183_46899501.lo38-temporary.psam.\n",
      "59885 variants loaded from\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501.lo38-temporary.pvar.zst.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.bgen ... 1011131517182022232527293032343537394142444647495153545658596163646668707173757678808283858788909294959799done.\n",
      "Writing /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.sample ... done.\n",
      "End time: Wed Dec 13 11:06:32 2023\n"
     ]
    }
   ],
   "source": [
    "plink2 --threads 8 --memory 10000 \\\n",
    "        --vcf ~/test_ldstore/01_44969183_46899501.lo38.vcf.gz \\\n",
    "        --export bgen-1.2 'bits=8' 'sample-v2' 'ref-first' \\\n",
    "        --out ~/test_ldstore/01_44969183_46899501.lo38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b3e23f97-0830-4b11-8086-ac4e072ead17",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/test_ldstore/01_44969183_46899501.lo38.bgen -index"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88f0485b-7840-46e7-9825-6e7926386455",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Create the z file using plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e6d7b0c4-60a1-432d-bd57-eb8c42b14121",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.log.\n",
      "Options in effect:\n",
      "  --memory 10000\n",
      "  --out /home/dmc2245/test_ldstore/01_44969183_46899501.lo38\n",
      "  --threads 8\n",
      "  --vcf /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.vcf.gz\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 11:19:29 2023\n",
      "257481 MiB RAM detected, ~221215 available; reserving 10000 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "59885 variants loaded from\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.vcf.gz.\n",
      "Note: No phenotype data present.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.snplist .\n",
      "End time: Wed Dec 13 11:20:56 2023\n"
     ]
    }
   ],
   "source": [
    "plink2 --threads 8 --memory 10000 \\\n",
    "        --vcf ~/test_ldstore/01_44969183_46899501.lo38.vcf.gz \\\n",
    "        -write-snplist \\\n",
    "        --out ~/test_ldstore/01_44969183_46899501.lo38"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b45a9656-e05c-4c9f-8fa5-15ce99f2bd3c",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Use python to finish getting the formatting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ac03021a-4a6a-4ccc-91c4-585c4e1da314",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "snplist = pd.read_csv('~/test_ldstore/01_44969183_46899501.lo38.snplist', header=None, names=[\"rsid_ori\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c2eab1c8-70b0-463d-a614-bfd61a12d1c7",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid_ori</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:44969310:A:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:44969328:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:44969330:C:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:44969417:A:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:44969509:C:T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         rsid_ori\n",
       "0  1:44969310:A:G\n",
       "1  1:44969328:C:T\n",
       "2  1:44969330:C:G\n",
       "3  1:44969417:A:T\n",
       "4  1:44969509:C:T"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2b8860db-dfdc-4c5e-8ad3-aaac4b587662",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist[['chromosome', 'position', 'allele1', 'allele2']] = snplist['rsid_ori'].str.split(':', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "96022e94-cd89-4fa4-ba10-ee84c5d0a81f",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist['chromosome'] = snplist.chromosome.astype(str).str.zfill(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2d7b33b5-7e15-4ab9-a955-6bde12b8b692",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "#snplist['rsid'] = snplist['rsid'].str.replace(r'^(\\d:)', lambda x: 'chr0' + x.group(1), regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "64153372-9707-4b4d-afaf-6adf8b6022a0",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist['rsid'] = snplist['rsid_ori'].str.extract(r'^(\\d+:\\d+)')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "8c6b05aa-b05f-4c9b-8df4-c4eec237e8be",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid_ori</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:44969310:A:G</td>\n",
       "      <td>1</td>\n",
       "      <td>44969310</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1:44969310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:44969328:C:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969328</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:44969328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:44969330:C:G</td>\n",
       "      <td>1</td>\n",
       "      <td>44969330</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1:44969330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:44969417:A:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969417</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1:44969417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:44969509:C:T</td>\n",
       "      <td>1</td>\n",
       "      <td>44969509</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1:44969509</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         rsid_ori chromosome  position allele1 allele2        rsid\n",
       "0  1:44969310:A:G          1  44969310       A       G  1:44969310\n",
       "1  1:44969328:C:T          1  44969328       C       T  1:44969328\n",
       "2  1:44969330:C:G          1  44969330       C       G  1:44969330\n",
       "3  1:44969417:A:T          1  44969417       A       T  1:44969417\n",
       "4  1:44969509:C:T          1  44969509       C       T  1:44969509"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "081c6bed-0177-4f08-99e7-340187e654f5",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "snplist[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\"]].to_csv('~/test_ldstore/01_44969183_46899501.lo38.z', header=True, index=False, sep= ' ')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc7369c4-88e5-4df3-8bd4-8962f0af8510",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "### Create the master file for LD store"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bc1c53ee-3b25-4397-b777-1a8c6791888a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "bgen=glob.glob(r\"/home/dmc2245/test_ldstore/*.bgen\")\n",
    "masterfile = pd.DataFrame({\n",
    "    \"z\" : glob.glob(r\"/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.z\"),\n",
    "    \"bgen\" : glob.glob(r\"/home/dmc2245/test_ldstore/*.bgen\"),\n",
    "    \"bgi\" : glob.glob(r\"/home/dmc2245/test_ldstore/*.bgi\"),\n",
    "    \"bcor\" : [i.replace('bgen','bcor') for i in bgen],\n",
    "    \"ld\" : [i.replace('bgen','ld') for i in bgen],\n",
    "    \"n_samples\" : 351430,\n",
    "    \"sample\": f\"/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\",\n",
    "    \"incl\" : f\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl\"\n",
    "          })\n",
    "masterfile = masterfile[['z', 'bgen', 'bgi', 'bcor', 'ld','n_samples', 'sample', 'incl']]\n",
    "\n",
    "masterfile.to_csv(\"/home/dmc2245/test_ldstore/mastefile\", sep=\";\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7be1639e-00c8-4605-b283-58c7352f2141",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Compute LD using LDstore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "0304992c-79d8-4d3b-af52-d57ea35bfd93",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mbcor\u001b[0m: Create bcor files\n",
      "INFO: \u001b[32mbcor\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: Workflow bcor (ID=we1b00b592f51ddac) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "module load Singularity\n",
    "sos run /home/dmc2245/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    bcor\\\n",
    "    --cwd ~/test_ldstore \\\n",
    "    --masterfile /home/dmc2245/test_ldstore/mastefile \\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ba9db6d3-07f3-46f3-ad42-a997a6673fe3",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "Building query                                              :  (0/?,0.0s,0.0/s)\n",
      "# bgenix: started 2023-12-13 12:51:30\n",
      "alternate_ids\trsid\tchromosome\tposition\tnumber_of_alleles\tfirst_allele\talternative_alleles\n",
      "# bgenix: success, total 0 variants.\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "bgenix -g /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.bgen -list -incl-range 1:44503511-44503520"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cabc86e6-19ca-4045-9fad-8a1bf5df8516",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rsid chromosome position allele1 allele2\n",
      "01:44969310:A:G 01 44969310 A G\n",
      "01:44969328:C:T 01 44969328 C T\n",
      "01:44969330:C:G 01 44969330 C G\n",
      "01:44969417:A:T 01 44969417 A T\n",
      "01:44969509:C:T 01 44969509 C T\n",
      "01:44969513:C:G 01 44969513 C G\n",
      "01:44969515:A:G 01 44969515 A G\n",
      "01:44969518:C:A 01 44969518 C A\n",
      "01:44969527:T:C 01 44969527 T C\n"
     ]
    }
   ],
   "source": [
    "cat /home/dmc2245/test_ldstore/01_44969183_46899501.lo38.z | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e94f74f-a29a-41dc-b9e5-83c59c213be6",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "Error : SNP with rsID '1:46899482:G:A' in file '/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.z' could not be matched with SNP in BGEN file '/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.bgen'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acaa30b3-e1f4-47ee-8428-9637d8afe3a6",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "In the terminal install bgen-reader. Read [tutorial](https://bgen-reader.readthedocs.io/en/latest/install.html)\n",
    "\n",
    "```\n",
    "micromamba install bgen-reader\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "b4c56397-934e-40e0-94fe-00d0b0f61a64",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'bgen_reader'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[26], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m# Read into bgen file\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mbgen_reader\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m open_bgen\n\u001b[1;32m      3\u001b[0m bgen_file \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.bgen\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m      4\u001b[0m bgen \u001b[38;5;241m=\u001b[39m open_bgen(bgen_file, verbose\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'bgen_reader'"
     ]
    }
   ],
   "source": [
    "# Read into bgen file\n",
    "from bgen_reader import open_bgen\n",
    "bgen_file = '/home/dmc2245/test_ldstore/01_44969183_46899501.lo38.bgen'\n",
    "bgen = open_bgen(bgen_file, verbose=False)\n",
    "\n",
    "# Print first 5 samples\n",
    "print(bgen.samples[:5])\n",
    "# ['sample_0' 'sample_1' 'sample_2' 'sample_3' 'sample_4']\n",
    "# Print first 5 variants\n",
    "print(bgen.ids[:5]) #first 5\n",
    "# ['6:31571218_C_T' '6:31571228_A_C' '6:31571296_A_G' '6:31571308_C_T' '6:31571330_C_A']\n",
    "print(bgen.nvariants)\n",
    "# 43053\n",
    "\n",
    "## Read the probabilities of the first variant\n",
    "probs = bgen.read(0)\n",
    "print(probs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "cda1316d-f929-4dd3-afd4-33e1ff48a50a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'bgen'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[25], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mbgen\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m BgenReader, BgenWriter\n\u001b[1;32m      3\u001b[0m bfile \u001b[38;5;241m=\u001b[39m BgenReader(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m/home/dmc2245/01_44969183_46899501.lo38.bgen\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m      4\u001b[0m rsids \u001b[38;5;241m=\u001b[39m bfile\u001b[38;5;241m.\u001b[39mrsids()\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'bgen'"
     ]
    }
   ],
   "source": [
    "from bgen import BgenReader, BgenWriter\n",
    "\n",
    "bfile = BgenReader('/home/dmc2245/01_44969183_46899501.lo38.bgen')\n",
    "rsids = bfile.rsids()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "5eff621a-f9b2-4562-aa6b-929fabe6190d",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "z;bgen;bgi;bcor;ld;n_samples;sample;incl\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664_0.01.z;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen.bgi;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bcor;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld;351430;/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl\n"
     ]
    }
   ],
   "source": [
    "cat /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "228ceb06-36b5-4c2b-971e-f45cec7dbf2c",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rsid chromosome position allele1 allele2\n",
      "chr6:31571218:C:T 6 31571218 C T\n",
      "chr6:31571228:A:C 6 31571228 A C\n",
      "chr6:31571296:A:G 6 31571296 A G\n",
      "chr6:31571308:C:T 6 31571308 C T\n",
      "chr6:31571330:C:A 6 31571330 C A\n",
      "chr6:31571337:C:T 6 31571337 C T\n",
      "chr6:31571362:C:T 6 31571362 C T\n",
      "chr6:31571375:A:G 6 31571375 A G\n",
      "chr6:31571384:A:G 6 31571384 A G\n"
     ]
    }
   ],
   "source": [
    "cat /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664_0.01.z | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbbc6a48-f244-4ba5-b237-973c188b8264",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "### Create MWE for susie_rss"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad4b298f-563f-4bb2-8dcd-0ec5da5d40b9",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    " Create a bgen_list_file for regions of interest\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "5ec55864-37b7-4d73-9001-b61670687c21",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "\n",
    "bgen=glob.glob('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/'+'/01/*.bgen')\n",
    "df = pd.DataFrame({'bgen':bgen})\n",
    "df['sample'] = df.apply(lambda x:'/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "5a3f7266-06ff-4bdd-8c21-226d7e383513",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "df.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/chr1_bgen_list.txt',sep=';', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dc7833f-deda-4a60-bd6f-26ab983aa296",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "bgi=glob.glob(${cwd:r}+'/*/*.bgen.bgi')\n",
    "df2 = pd.DataFrame({'bgi':bgi})\n",
    "df3=pd.concat([df,df1,df2], axis=1)\n",
    "bcor=[i.replace('bgen','bcor') for i in bgen]\n",
    "df4 = pd.DataFrame({'bcor':bcor})\n",
    "ld=[i.replace('bgen','ld') for i in bgen]\n",
    "df5 = pd.DataFrame({'ld':ld})\n",
    "df_final=pd.concat([df3,df4,df5], axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c7b644c-d0ed-4363-b766-48f9b667dbfa",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Regions to generate\n",
    "```\n",
    "01_159913048_162346721.bgen (index pos 161155392)\n",
    "01_206073265_208410364.bgen (index pos 207786828)\n",
    "02_127373764_128034347.bgen (index pos 127891427)\n",
    "02_233550003_235150987.bgen (index pos 233981912) \n",
    "03_56433907_58157519.bgen (index pos 57226150)\n",
    "```\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "89ac9cd9-6d13-4a61-84e5-bebca5c74971",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/susie_mwe/01_159913048_162346721_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_159913048_162346721.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.001\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/susie_mwe/01_159913048_162346721_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 17:19:56 2023\n",
      "257481 MiB RAM detected, ~217198 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 79539 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/susie_mwe/01_159913048_162346721_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/susie_mwe/01_159913048_162346721_hg19-temporary.pgen +\n",
      "/home/dmc2245/susie_mwe/01_159913048_162346721_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/susie_mwe/01_159913048_162346721_hg19-temporary.psam.\n",
      "79539 variants loaded from\n",
      "/home/dmc2245/susie_mwe/01_159913048_162346721_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 82done.\n",
      "61823 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "17716 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.snplist .\n",
      "Writing /home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bim ... done.\n",
      "Writing /home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bgen ... 1828384859718192done.\n",
      "Writing /home/dmc2245/susie_mwe/01_159913048_162346721_hg19.sample ... done.\n",
      "End time: Wed Dec 13 17:20:22 2023\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_159913048_162346721.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/susie_mwe/01_159913048_162346721_hg19 \\\n",
    "--maf 0.001 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "f49fec43-5617-42c6-9101-4d435cf59876",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bgen\" in \"/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bgen\" with 17716 variants...\n",
      "Building BGEN index                                         : [******************************] (17716/17716,1.9s,9524.0/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/susie_mwe/01_159913048_162346721_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "e1204b5c-67cc-4706-a669-b0f44ba90863",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_206073265_208410364.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.001\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/susie_mwe/01_206073265_208410364_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 17:20:31 2023\n",
      "257481 MiB RAM detected, ~217180 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 65942 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/susie_mwe/01_206073265_208410364_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/susie_mwe/01_206073265_208410364_hg19-temporary.pgen +\n",
      "/home/dmc2245/susie_mwe/01_206073265_208410364_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/susie_mwe/01_206073265_208410364_hg19-temporary.psam.\n",
      "65942 variants loaded from\n",
      "/home/dmc2245/susie_mwe/01_206073265_208410364_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 99done.\n",
      "52858 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "13084 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.snplist .\n",
      "Writing /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bim ... done.\n",
      "Writing /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bgen ... 1325374960748799done.\n",
      "Writing /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.sample ... done.\n",
      "End time: Wed Dec 13 17:20:52 2023\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_206073265_208410364.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/susie_mwe/01_206073265_208410364_hg19 \\\n",
    "--maf 0.001 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "ee9fb371-e750-4f88-9505-0c16e9cd8d46",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bgen\" in \"/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bgen\" with 13084 variants...\n",
      "Building BGEN index                                         : [******************************] (13084/13084,1.4s,9460.9/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/susie_mwe/01_206073265_208410364_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "41e058f7-f82b-4fab-8592-72b5ead32389",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/susie_mwe/02_127373764_128034347_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/02/02_127373764_128034347.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.001\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/susie_mwe/02_127373764_128034347_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 17:21:24 2023\n",
      "257481 MiB RAM detected, ~217167 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 24508 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/susie_mwe/02_127373764_128034347_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/susie_mwe/02_127373764_128034347_hg19-temporary.pgen +\n",
      "/home/dmc2245/susie_mwe/02_127373764_128034347_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/susie_mwe/02_127373764_128034347_hg19-temporary.psam.\n",
      "24508 variants loaded from\n",
      "/home/dmc2245/susie_mwe/02_127373764_128034347_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "19099 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "5409 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/susie_mwe/02_127373764_128034347_hg19.snplist .\n",
      "Writing /home/dmc2245/susie_mwe/02_127373764_128034347_hg19.bim ... done.\n",
      "Writing /home/dmc2245/susie_mwe/02_127373764_128034347_hg19.bgen ... 3266done.\n",
      "Writing /home/dmc2245/susie_mwe/02_127373764_128034347_hg19.sample ... done.\n",
      "End time: Wed Dec 13 17:21:33 2023\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/02/02_127373764_128034347.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/susie_mwe/02_127373764_128034347_hg19 \\\n",
    "--maf 0.001 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "271513d8-d45b-48a0-aca8-2230b2917f4e",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/susie_mwe/02_127373764_128034347_hg19.bgen\" in \"/home/dmc2245/susie_mwe/02_127373764_128034347_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/susie_mwe/02_127373764_128034347_hg19.bgen\" with 5409 variants...\n",
      "Building BGEN index                                         : [******************************] (5409/5409,0.6s,9233.6/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/susie_mwe/02_127373764_128034347_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "88f9ad61-1cd1-4f90-948d-31defb1f0345",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/susie_mwe/02_233550003_235150987_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/02/02_233550003_235150987.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.001\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/susie_mwe/02_233550003_235150987_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 17:22:02 2023\n",
      "257481 MiB RAM detected, ~217179 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 57170 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/susie_mwe/02_233550003_235150987_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/susie_mwe/02_233550003_235150987_hg19-temporary.pgen +\n",
      "/home/dmc2245/susie_mwe/02_233550003_235150987_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/susie_mwe/02_233550003_235150987_hg19-temporary.psam.\n",
      "57170 variants loaded from\n",
      "/home/dmc2245/susie_mwe/02_233550003_235150987_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "45332 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "11838 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/susie_mwe/02_233550003_235150987_hg19.snplist .\n",
      "Writing /home/dmc2245/susie_mwe/02_233550003_235150987_hg19.bim ... done.\n",
      "Writing /home/dmc2245/susie_mwe/02_233550003_235150987_hg19.bgen ... 132943567086done.\n",
      "Writing /home/dmc2245/susie_mwe/02_233550003_235150987_hg19.sample ... done.\n",
      "End time: Wed Dec 13 17:22:21 2023\n"
     ]
    }
   ],
   "source": [
    "\n",
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/02/02_233550003_235150987.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/susie_mwe/02_233550003_235150987_hg19 \\\n",
    "--maf 0.001 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "fb70b737-e425-4176-93b2-b208b02a8d43",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/susie_mwe/02_233550003_235150987_hg19.bgen\" in \"/home/dmc2245/susie_mwe/02_233550003_235150987_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/susie_mwe/02_233550003_235150987_hg19.bgen\" with 11838 variants...\n",
      "Building BGEN index                                         : [******************************] (11838/11838,1.3s,9218.0/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/susie_mwe/02_233550003_235150987_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "6c379553-be3b-4d63-bd91-ddd4deec2cf0",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/dmc2245/susie_mwe/03_56433907_58157519_hg19.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/03/03_56433907_58157519.bgen ref-first\n",
      "  --export bgen-1.2 bits=8 ref-first\n",
      "  --maf 0.001\n",
      "  --make-just-bim\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /home/dmc2245/susie_mwe/03_56433907_58157519_hg19\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids @:#:$r:$a\n",
      "  --write-snplist\n",
      "\n",
      "Start time: Wed Dec 13 17:22:46 2023\n",
      "257481 MiB RAM detected, ~217163 available; reserving 128740 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 56772 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/home/dmc2245/susie_mwe/03_56433907_58157519_hg19-temporary.psam .\n",
      "--bgen: /home/dmc2245/susie_mwe/03_56433907_58157519_hg19-temporary.pgen +\n",
      "/home/dmc2245/susie_mwe/03_56433907_58157519_hg19-temporary.pvar written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from /home/dmc2245/susie_mwe/03_56433907_58157519_hg19-temporary.psam.\n",
      "56772 variants loaded from\n",
      "/home/dmc2245/susie_mwe/03_56433907_58157519_hg19-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "45355 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "11417 variants remaining after main filters.\n",
      "--write-snplist: Variant IDs written to\n",
      "/home/dmc2245/susie_mwe/03_56433907_58157519_hg19.snplist .\n",
      "Writing /home/dmc2245/susie_mwe/03_56433907_58157519_hg19.bim ... done.\n",
      "Writing /home/dmc2245/susie_mwe/03_56433907_58157519_hg19.bgen ... 153146597587done.\n",
      "Writing /home/dmc2245/susie_mwe/03_56433907_58157519_hg19.sample ... done.\n",
      "End time: Wed Dec 13 17:23:03 2023\n"
     ]
    }
   ],
   "source": [
    "\n",
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/03/03_56433907_58157519.bgen 'ref-first' \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--write-snplist \\\n",
    "--out /home/dmc2245/susie_mwe/03_56433907_58157519_hg19 \\\n",
    "--maf 0.001 \\\n",
    "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
    "--set-all-var-ids '@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--make-just-bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "72872d1d-070d-4d2f-9e71-9cb4b8abf1a7",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Welcome to bgenix\n",
      "(version: 1.1.8, revision )\n",
      "\n",
      "(C) 2009-2017 University of Oxford\n",
      "\n",
      "bgenix: creating index for \"/home/dmc2245/susie_mwe/03_56433907_58157519_hg19.bgen\" in \"/home/dmc2245/susie_mwe/03_56433907_58157519_hg19.bgen.bgi\"...\n",
      "bgenix: Opened \"/home/dmc2245/susie_mwe/03_56433907_58157519_hg19.bgen\" with 11417 variants...\n",
      "Building BGEN index                                         : [******************************] (11417/11417,1.1s,10799.4/s)\n",
      "\n",
      "Thank you for using bgenix.\n"
     ]
    }
   ],
   "source": [
    "module load  Bgenix/1.1.8\n",
    "bgenix -g ~/susie_mwe/03_56433907_58157519_hg19.bgen -index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "5717019c-1315-4fc2-9d04-d54e2226b8a1",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Checking \u001b[32msubset_bgen\u001b[0m: Subset bgen files to a specific maf, change the variant id, write snplist to create *z file and output bim files for downstream liftover\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp88ikv6_d/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_100826405_102041016.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_100826405_102041016.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_100826405_102041016.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpte2jyhi4/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_102041016_102898745.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_102041016_102898745.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_102041016_102898745.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpged2m3jr/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_102898745_103914211.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_102898745_103914211.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_102898745_103914211.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpr3yfmhp_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_103914211_106087842.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_103914211_106087842.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_103914211_106087842.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp__opcman/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_10583_1892607.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_10583_1892607.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_10583_1892607.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpb_kmfa7a/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_106087842_108409665.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_106087842_108409665.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_106087842_108409665.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=5) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpdkls4nwb/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_10806984_11777841.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_10806984_11777841.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_10806984_11777841.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=6) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpm5cjd38s/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_108409665_110303931.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_108409665_110303931.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_108409665_110303931.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=7) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpps24z8jj/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_110303931_111741737.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_110303931_111741737.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_110303931_111741737.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=8) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp7xd4v05y/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_111741737_113273306.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_111741737_113273306.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_111741737_113273306.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=9) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpittl0l6q/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_113273306_114873845.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_113273306_114873845.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_113273306_114873845.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=10) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmphzsfwd6t/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_114873845_115880593.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_114873845_115880593.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_114873845_115880593.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=11) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp513xjcfc/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_115880593_118839067.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_115880593_118839067.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_115880593_118839067.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=12) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpn7hsa6r2/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_11777841_12779466.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_11777841_12779466.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_11777841_12779466.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=13) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpuqcquf8c/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_118839067_144977494.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_118839067_144977494.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_118839067_144977494.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=14) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpct8dbphb/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_12779466_14891511.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_12779466_14891511.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_12779466_14891511.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=15) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpcje693qj/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_144977494_148361253.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_144977494_148361253.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_144977494_148361253.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=16) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp4g39zwbk/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_148361253_151538881.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_148361253_151538881.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_148361253_151538881.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=17) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpvicy77ki/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_14891511_16897730.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_14891511_16897730.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_14891511_16897730.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=18) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpaiho_eee/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_151538881_153180829.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_151538881_153180829.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_151538881_153180829.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=19) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp995h5o07/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_153180829_154770403.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_153180829_154770403.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_153180829_154770403.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=20) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpamjkj0n4/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_154770403_156336133.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_154770403_156336133.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_154770403_156336133.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=21) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpdlddkxl3/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_156336133_158027412.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_156336133_158027412.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_156336133_158027412.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=22) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0436qldo/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_158027412_159913048.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_158027412_159913048.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_158027412_159913048.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=23) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpqy7sazv3/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_159913048_162346721.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_159913048_162346721.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_159913048_162346721.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=24) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpyry0nz8c/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_162346721_165191702.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_162346721_165191702.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_162346721_165191702.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=25) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp2ltaf3um/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_165191702_166460517.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_165191702_166460517.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_165191702_166460517.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=26) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpx7ieqd67/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_166460517_169086324.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_166460517_169086324.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_166460517_169086324.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=27) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpjl9nxu3j/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_16897730_18662899.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_16897730_18662899.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_16897730_18662899.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=28) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp4jfe1e_b/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_169086324_170557776.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_169086324_170557776.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_169086324_170557776.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=29) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpma1ntxp3/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_170557776_173097907.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_170557776_173097907.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_170557776_173097907.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=30) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpzv9vvbcw/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_173097907_175089768.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_173097907_175089768.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_173097907_175089768.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=31) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpcv0gucbj/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_175089768_177433381.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_175089768_177433381.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_175089768_177433381.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=32) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpr7jlxwiz/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_177433381_178944309.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_177433381_178944309.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_177433381_178944309.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=33) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp1o3j7b5w/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_178944309_178954470.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_178944309_178954470.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_178944309_178954470.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=34) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpozpq_4pa/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_178954470_181144121.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_178954470_181144121.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_178954470_181144121.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=35) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmplsplpxx8/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_181144121_182755356.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_181144121_182755356.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_181144121_182755356.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=36) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0gw7ny3y/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_182755356_184595513.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_182755356_184595513.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_182755356_184595513.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=37) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmphkvgop05/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_184595513_186810023.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_184595513_186810023.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_184595513_186810023.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=38) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp4rz7ijbm/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_18662899_20469149.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_18662899_20469149.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_18662899_20469149.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=39) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp3blrzu6n/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_186810023_188759945.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_186810023_188759945.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_186810023_188759945.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=40) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp8_c7fqp5/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_188759945_189904130.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_188759945_189904130.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_188759945_189904130.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=41) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpzn7yktzg/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_1892607_3582736.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_1892607_3582736.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_1892607_3582736.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=42) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpnt1rxaxz/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_189904130_191868930.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_189904130_191868930.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_189904130_191868930.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=43) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp55of4uuy/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_191868930_194107442.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_191868930_194107442.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_191868930_194107442.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=44) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpdp9snz46/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_194107442_196176201.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_194107442_196176201.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_194107442_196176201.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=45) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpk__4n_rt/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_196176201_197311514.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_196176201_197311514.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_196176201_197311514.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=46) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpigy7p8hq/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_197311514_199239884.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_197311514_199239884.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_197311514_199239884.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=47) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmplhmoa8j0/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_199239884_200137649.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_199239884_200137649.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_199239884_200137649.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=48) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp3jcex3mn/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_200137649_201589975.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_200137649_201589975.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_200137649_201589975.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=49) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpong4_xsv/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_201589975_203334734.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_201589975_203334734.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_201589975_203334734.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=50) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp_414ygs6/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_203334734_204681068.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_203334734_204681068.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_203334734_204681068.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=51) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpdtw_pe5a/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_204681068_206073265.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_204681068_206073265.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_204681068_206073265.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=52) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpgmuodlgq/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_20469149_21736588.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_20469149_21736588.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_20469149_21736588.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=53) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpfhfwnsv7/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_206073265_208410364.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_206073265_208410364.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_206073265_208410364.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=54) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpp1k23vce/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_208410364_210787750.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_208410364_210787750.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_208410364_210787750.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=55) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpg5nb7lil/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_210787750_212842330.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_210787750_212842330.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_210787750_212842330.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=56) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpgr6_nxn_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_212842330_214578266.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_212842330_214578266.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_212842330_214578266.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=57) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpfjpq0f4i/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_214578266_216243634.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_214578266_216243634.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_214578266_216243634.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=58) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp_mmcybza/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_216243634_218705513.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_216243634_218705513.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_216243634_218705513.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=59) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp302gx3ew/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_21736588_23086883.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_21736588_23086883.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_21736588_23086883.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=60) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpnbresbj9/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_218705513_219590571.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_218705513_219590571.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_218705513_219590571.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=61) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0f821z4x/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_219590571_221858231.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_219590571_221858231.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_219590571_221858231.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=62) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpu_syx1re/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_221858231_222230220.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_221858231_222230220.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_221858231_222230220.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=63) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpg4zrr3m4/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_222230220_224938520.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_222230220_224938520.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_222230220_224938520.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=64) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp_4el4yus/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_224938520_226810860.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_224938520_226810860.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_224938520_226810860.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=65) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp2ie0ah4u/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_226810860_229156248.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_226810860_229156248.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_226810860_229156248.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=66) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp_irthg8a/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_229156248_230685255.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_229156248_230685255.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_229156248_230685255.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=67) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpqhgde02f/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_230685255_232090252.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_230685255_232090252.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_230685255_232090252.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=68) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmprhix5hkk/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_23086883_23920590.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_23086883_23920590.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_23086883_23920590.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=69) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0avu5gzx/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_232090252_233429284.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_232090252_233429284.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_232090252_233429284.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=70) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp8_a6vhij/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_233429284_234599434.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_233429284_234599434.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_233429284_234599434.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=71) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp3mu4m4uv/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_234599434_235819436.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_234599434_235819436.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_234599434_235819436.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=72) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpv0bvd88r/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_235819436_237555628.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_235819436_237555628.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_235819436_237555628.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=73) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp1ei5g9hy/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_237555628_238567807.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_237555628_238567807.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_237555628_238567807.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=74) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0p7tn83m/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_238567807_239925314.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_238567807_239925314.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_238567807_239925314.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=75) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp758y67wj/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_23920590_25516845.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_23920590_25516845.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_23920590_25516845.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=76) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpps5pse6b/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_239925314_240564091.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_239925314_240564091.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_239925314_240564091.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=77) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpg28uglx0/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_240564091_241582220.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_240564091_241582220.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_240564091_241582220.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=78) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpna1gazrf/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_241582220_242071602.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_241582220_242071602.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_241582220_242071602.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=79) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpi1i3jeb_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_242071602_244109499.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_242071602_244109499.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_242071602_244109499.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=80) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp1cb_e49x/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_244109499_245506746.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_244109499_245506746.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_244109499_245506746.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=81) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpao40wj7_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_245506746_247344518.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_245506746_247344518.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_245506746_247344518.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=82) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpzv0e528k/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_247344518_249239466.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_247344518_249239466.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_247344518_249239466.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=83) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpvsr8pur4/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_25516845_27401867.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_25516845_27401867.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_25516845_27401867.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=84) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp3yby9799/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_27401867_30161881.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_27401867_30161881.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_27401867_30161881.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=85) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp9rupfy0z/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_30161881_32438685.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_30161881_32438685.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_30161881_32438685.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=86) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmptp196s48/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_32438685_34799758.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_32438685_34799758.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_32438685_34799758.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=87) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpos93re0_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_34799758_37549183.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_34799758_37549183.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_34799758_37549183.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=88) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpltvk62kl/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_3582736_4380811.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_3582736_4380811.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_3582736_4380811.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=89) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpirrk6trl/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_37549183_38731847.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_37549183_38731847.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_37549183_38731847.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=90) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp172eshj6/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_38731847_40200567.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_38731847_40200567.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_38731847_40200567.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=91) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpvfg48_61/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_40200567_41975327.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_40200567_41975327.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_40200567_41975327.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=92) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpiprc55x4/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_41975327_43758457.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_41975327_43758457.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_41975327_43758457.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=93) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpz8g6pmol/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_43758457_44969183.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_43758457_44969183.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_43758457_44969183.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=94) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp8binvu2s/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_4380811_5913893.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_4380811_5913893.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_4380811_5913893.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=95) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp92zrrm4r/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_44969183_46899501.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_44969183_46899501.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_44969183_46899501.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=96) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp9jobtqj8/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_46899501_48002576.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_46899501_48002576.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_46899501_48002576.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=97) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpebt6nulr/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_48002576_48978188.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_48002576_48978188.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_48002576_48978188.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=98) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpka7gfhwm/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_48978188_49894177.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_48978188_49894177.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_48978188_49894177.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=99) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp12pxuw09/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_49894177_51713726.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_49894177_51713726.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_49894177_51713726.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=100) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp0bzi203g/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_51713726_54226262.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_51713726_54226262.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_51713726_54226262.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=101) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmprtvcpvmq/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_54226262_56413117.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_54226262_56413117.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_54226262_56413117.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=102) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpe4altx30/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_56413117_57021728.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_56413117_57021728.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_56413117_57021728.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=103) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpmqux092i/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_57021728_58865399.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_57021728_58865399.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_57021728_58865399.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=104) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmps2e_c3lq/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_58865399_59890409.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_58865399_59890409.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_58865399_59890409.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=105) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpaon4in2v/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_5913893_7247335.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_5913893_7247335.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_5913893_7247335.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=106) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpe7oci5p3/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_59890409_61922365.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_59890409_61922365.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_59890409_61922365.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=107) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpycwi2egf/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_61922365_63455089.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_61922365_63455089.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_61922365_63455089.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=108) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp83szmncs/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_63455089_65041704.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_63455089_65041704.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_63455089_65041704.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=109) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpg2eswqb8/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_65041704_66939404.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_65041704_66939404.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_65041704_66939404.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=110) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpvskuk_3j/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_66939404_68477895.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_66939404_68477895.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_66939404_68477895.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=111) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp_z2xgpqi/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_68477895_69687616.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_68477895_69687616.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_68477895_69687616.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=112) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpdn_p2y0f/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_69687616_71372470.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_69687616_71372470.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_69687616_71372470.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=113) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp6fb79c3b/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_71372470_71684405.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_71372470_71684405.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_71372470_71684405.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=114) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmprkg36cbe/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_71684405_74326907.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_71684405_74326907.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_71684405_74326907.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=115) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmphk4sebgi/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_7247335_9365199.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_7247335_9365199.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_7247335_9365199.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=116) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpce75jfzt/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_74326907_76728135.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_74326907_76728135.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_74326907_76728135.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=117) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmprcethlij/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_76728135_79661198.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_76728135_79661198.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_76728135_79661198.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=118) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmps0dgrhaw/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_79661198_81354191.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_79661198_81354191.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_79661198_81354191.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=119) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpd3_2mzoy/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_81354191_83991748.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_81354191_83991748.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_81354191_83991748.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=120) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp8euxajmu/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_83991748_84844495.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_83991748_84844495.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_83991748_84844495.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=121) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpnlxq140_/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_84844495_86454751.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_84844495_86454751.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_84844495_86454751.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=122) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp1g64re0m/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_86454751_88128631.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_86454751_88128631.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_86454751_88128631.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=123) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpeyqmht5w/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_88128631_90066303.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_88128631_90066303.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_88128631_90066303.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=124) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpi9ozukth/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_90066303_91886317.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_90066303_91886317.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_90066303_91886317.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=125) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpg3762kzl/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_91886317_93536459.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_91886317_93536459.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_91886317_93536459.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=126) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp4uwxqu20/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_93536459_94506191.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_93536459_94506191.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_93536459_94506191.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=127) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp946esh0l/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_9365199_10806984.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_9365199_10806984.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_9365199_10806984.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=128) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpuft4jo19/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_94506191_96150893.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_94506191_96150893.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_94506191_96150893.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=129) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp6vcirvyu/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_96150893_97885249.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_96150893_97885249.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_96150893_97885249.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=130) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmp88j3cc4z/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_97885249_99800604.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_97885249_99800604.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_97885249_99800604.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=131) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: singularity exec  /home/dmc2245/containers/lmm.sif /bin/bash /mnt/vast/hpc/homes/dmc2245/project/UKBB_GWAS_dev/code/python/tmpq2f4di73/singularity_run_64496.sh\n",
      "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/01_99800604_100826405.bgen 'ref-first' \\\n",
      "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
      "--write-snplist \\\n",
      "--out /home/dmc2245/susie_mwe/01_99800604_100826405.0.001.subset \\\n",
      "--maf 0.001 \\\n",
      "--export bgen-1.2 'bits=8' 'ref-first' \\\n",
      "--set-all-var-ids '@:#:$r:$a' \\\n",
      "--new-id-max-allele-len 100 \\\n",
      "--make-just-bim\n",
      "\n",
      "bgenix -g /home/dmc2245/susie_mwe/01_99800604_100826405.0.001.subset.bgen -index\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m (index=132) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32msubset_bgen\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/01_100826405_102041016.0.001.subset.bgen /home/dmc2245/susie_mwe/01_100826405_102041016.0.001.subset.sample... (665 items in 133 groups)\u001b[0m\n",
      "INFO: Workflow subset_bgen (ID=w4a51dc018de9455a) is tested successfully with 1 completed step and 133 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "# Step 2 generate subset bgen file\n",
    "sos dryrun ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    subset_bgen \\\n",
    "    --cwd ~/susie_mwe \\\n",
    "    --masterfile  \\\n",
    "    --bgen_list_file /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/01/chr1_bgen_list.txt \\\n",
    "    --maf_filter 0.001 \\\n",
    "    --numThreads 10 \\\n",
    "    --mem 10G \\\n",
    "    --job_size 1 \\\n",
    "    --container ~/containers/lmm.sif"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "80c8b301-ab23-4851-a490-d93911d36ff2",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mCreation of the master file LDStore2\u001b[0m: \n",
      "INFO: \u001b[32mz_file\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mz_file\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mz_file\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mz_file\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mz_file\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mCreation of the master file LDStore2\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.z /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.z... (5 items in 5 groups)\u001b[0m\n",
      "INFO: Workflow z_file (ID=wc5cc0b79f48825c8) is executed successfully with 1 completed step and 5 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "# Step 3 create z file\n",
    "sos run ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    z_file \\\n",
    "    --cwd ~/susie_mwe \\\n",
    "    --masterfile ~/susie_mwe/mastefile_hg19\\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "e3bbacf2-eb51-4fe7-a8bf-a8096577569e",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mCreation of the masterfile\u001b[0m: Creation of the masterfile\n",
      "INFO: \u001b[32mCreation of the masterfile\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mCreation of the masterfile\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/susie_mwe.masterfile\u001b[0m\n",
      "INFO: Workflow masterfile (ID=wd57c21927186d3ef) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "# Step 4 Create the masterfile with the new subsetted files \n",
    "sos run ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    masterfile \\\n",
    "    --cwd ~/susie_mwe \\\n",
    "    --masterfile 'susie_mwe'\\\n",
    "    --number_of_samples 351430 \\\n",
    "    --incl_samples /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl \\\n",
    "    --numThreads 10 \\\n",
    "    --mem 1G \\\n",
    "    --job_size 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "31a0e41a-b2af-481e-a8a4-95391696acb3",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Checking \u001b[32mbcor\u001b[0m: Create bcor files\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files f'/home/dmc2245/susie_mwe/susie_mwe.masterfile'\\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \\\n",
      "--compression 'high'\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files f'/home/dmc2245/susie_mwe/susie_mwe.masterfile'\\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \\\n",
      "--compression 'high'\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files f'/home/dmc2245/susie_mwe/susie_mwe.masterfile'\\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \\\n",
      "--compression 'high'\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files f'/home/dmc2245/susie_mwe/susie_mwe.masterfile'\\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \\\n",
      "--compression 'high'\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files f'/home/dmc2245/susie_mwe/susie_mwe.masterfile'\\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \\\n",
      "--compression 'high'\n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mbcor\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.bcor /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bcor... (5 items in 5 groups)\u001b[0m\n",
      "INFO: Workflow bcor (ID=w136253b996f770ee) is tested successfully with 1 completed step and 5 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "# Step 5 Run bcor in LDstore using previously generated masterfile\n",
    "sos dryrun ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    bcor \\\n",
    "    --cwd ~/susie_mwe \\\n",
    "    --masterfile 'susie_mwe'\\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "a1082e8d-002b-4605-8164-eaf137e4ea6d",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mld_1\u001b[0m: Calculate LD\n",
      "INFO: Step \u001b[32mld_1\u001b[0m (index=0) is \u001b[32mignored\u001b[0m with signature constructed\n",
      "INFO: Step \u001b[32mld_1\u001b[0m (index=2) is \u001b[32mignored\u001b[0m with signature constructed\n",
      "INFO: Step \u001b[32mld_1\u001b[0m (index=3) is \u001b[32mignored\u001b[0m with signature constructed\n",
      "INFO: Step \u001b[32mld_1\u001b[0m (index=4) is \u001b[32mignored\u001b[0m with signature constructed\n",
      "INFO: Step \u001b[32mld_1\u001b[0m (index=1) is \u001b[32mignored\u001b[0m with signature constructed\n",
      "INFO: \u001b[32mld_1\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.ld /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.ld... (5 items in 5 groups)\u001b[0m\n",
      "INFO: Running \u001b[32mld_2\u001b[0m: Output LD matriz as compressed xz format\n",
      "INFO: \u001b[32mld_2\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mld_2\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mld_2\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mld_2\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mld_2\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mld_2\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.xz /home/dmc2245/susie_mwe/01_206073265_208410364_hg19.xz... (5 items in 5 groups)\u001b[0m\n",
      "INFO: Workflow ld (ID=wd6287cc78f2a91b1) is executed successfully with 1 completed step, 5 completed substeps, 1 ignored step and 5 ignored substeps.\n"
     ]
    }
   ],
   "source": [
    "# Step 6 Generate ld files and save as xz format\n",
    "sos run ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    ld \\\n",
    "    --cwd ~/susie_mwe \\\n",
    "    --masterfile 'susie_mwe'\\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1 -s build"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "5a70adb4-fb12-46d0-a33b-427f063bbf10",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Welcome, dcornejo88!\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "UPGRADE AVAILABLE\n",
      "\n",
      "A more recent version of the Synapse Client (3.2.0) is available. Your version (2.7.2) can be upgraded by typing:\n",
      "    pip install --upgrade synapseclient\n",
      "\n",
      "Python Synapse Client version 3.2.0 release notes\n",
      "\n",
      "https://python-docs.synapse.org/build/html/news.html\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Upload to synapse\n",
    "\n",
    "import synapseclient\n",
    "from synapseclient import File\n",
    "syn=synapseclient.login(authToken=\"eyJ0eXAiOiJKV1QiLCJraWQiOiJXN05OOldMSlQ6SjVSSzpMN1RMOlQ3TDc6M1ZYNjpKRU9VOjY0NFI6VTNJWDo1S1oyOjdaQ0s6RlBUSCIsImFsZyI6IlJTMjU2In0.eyJhY2Nlc3MiOnsic2NvcGUiOlsidmlldyJdLCJvaWRjX2NsYWltcyI6e319LCJ0b2tlbl90eXBlIjoiUEVSU09OQUxfQUNDRVNTX1RPS0VOIiwiaXNzIjoiaHR0cHM6Ly9yZXBvLXByb2QucHJvZC5zYWdlYmFzZS5vcmcvYXV0aC92MSIsImF1ZCI6IjAiLCJuYmYiOjE3MDI5MzUyMTYsImlhdCI6MTcwMjkzNTIxNiwianRpIjoiNDYxOCIsInN1YiI6IjM0ODY1ODEifQ.VcKauz2h1b2rgRBzwZDXyq8sdM9ljnmLYlk_cWoS72xsMgv1Iq0s-IXZfrgAY0jUd6bN4A4jbHDYjEPatK2z-re46kJpVBHacusLlAU6mDNCXKx5iBWp5SyLInT2olzoiJJ7dVolbhGhXTM5-y7XxcmhXoDTlNKvki2VlbRJsRFKrk3bPZbdGVthdBZIdIWXmI2sK12OXMp_JUhK8Ws-9rnIRrwRyQrz_84PQNy8wUql5Os5Anf4v9h-GZiOF1s3z0JkF5Q5v4vQF6GQZ7rd_rRNq0tlqEZ0foO4ZJVabEnxIgKRr-I4JoZubUbk5TtHdooImGX6Agpgwr1wIFqeCw\")\n",
    "\n",
    "# Add a local file to an existing project (syn12345) on Synapse\n",
    "#file = File(path='/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.xz', parent='syn53163876')\n",
    "#file = syn.store(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f024783b-7d36-420c-98bd-d173f2c023f0",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## Save upper diagonal matrix\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import lzma\n",
    "\n",
    "np_ld = np.loadtxt('/home/dmc2245/susie_mwe/02_233550003_235150987_hg19.ld', dtype = \"float16\"\n",
    "\n",
    "with xz.open(args.out, \"w+\", preset=9) as f:\n",
    "    for r in range(np_ld.shape[0]):\n",
    "        f.write(\" \".join([\"{:.6f}\".format(x) for x in np_ld[r, :]]).encode())\n",
    "        f.write(b\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2f19df21-3b61-4318-b5c9-195f76117802",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import lzma\n",
    "#import xz\n",
    "\n",
    "#z_file=pd.read_csv('/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.z', sep=\" \", skiprows=1, header=None)[0].to_numpy()\n",
    "np_ld = np.loadtxt('/home/dmc2245/susie_mwe/01_159913048_162346721_hg19.ld', dtype = \"float16\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f7f546fa-8d1e-4c94-be13-696bf3eef8a4",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "np.set_printoptions(formatter={'float': lambda x: \"{0:0.6f}\".format(x)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5d57cea2-e430-4896-967b-e7fcb1e7738a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>17707</th>\n",
       "      <th>17708</th>\n",
       "      <th>17709</th>\n",
       "      <th>17710</th>\n",
       "      <th>17711</th>\n",
       "      <th>17712</th>\n",
       "      <th>17713</th>\n",
       "      <th>17714</th>\n",
       "      <th>17715</th>\n",
       "      <th>17716</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.039520</td>\n",
       "      <td>0.076936</td>\n",
       "      <td>-0.010260</td>\n",
       "      <td>-0.011528</td>\n",
       "      <td>-0.007244</td>\n",
       "      <td>0.025617</td>\n",
       "      <td>0.349776</td>\n",
       "      <td>-0.040015</td>\n",
       "      <td>-0.003163</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003292</td>\n",
       "      <td>0.001617</td>\n",
       "      <td>-0.002359</td>\n",
       "      <td>0.004602</td>\n",
       "      <td>-0.001753</td>\n",
       "      <td>-0.000437</td>\n",
       "      <td>0.004527</td>\n",
       "      <td>0.003477</td>\n",
       "      <td>-0.001143</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.039520</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.199267</td>\n",
       "      <td>-0.043424</td>\n",
       "      <td>-0.050493</td>\n",
       "      <td>-0.029090</td>\n",
       "      <td>0.034237</td>\n",
       "      <td>-0.014709</td>\n",
       "      <td>0.577217</td>\n",
       "      <td>-0.015779</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.001471</td>\n",
       "      <td>0.001519</td>\n",
       "      <td>-0.004886</td>\n",
       "      <td>0.001469</td>\n",
       "      <td>0.000761</td>\n",
       "      <td>0.002153</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.005014</td>\n",
       "      <td>-0.001248</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.076936</td>\n",
       "      <td>-0.199267</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.087308</td>\n",
       "      <td>-0.149868</td>\n",
       "      <td>-0.081060</td>\n",
       "      <td>0.316285</td>\n",
       "      <td>0.036852</td>\n",
       "      <td>-0.458906</td>\n",
       "      <td>-0.043338</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000466</td>\n",
       "      <td>-0.000174</td>\n",
       "      <td>0.003233</td>\n",
       "      <td>0.004464</td>\n",
       "      <td>0.003842</td>\n",
       "      <td>-0.005526</td>\n",
       "      <td>-0.001074</td>\n",
       "      <td>-0.002356</td>\n",
       "      <td>0.002305</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>-0.010260</td>\n",
       "      <td>-0.043424</td>\n",
       "      <td>0.087308</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.013239</td>\n",
       "      <td>-0.007831</td>\n",
       "      <td>0.028841</td>\n",
       "      <td>-0.003291</td>\n",
       "      <td>-0.041080</td>\n",
       "      <td>-0.004297</td>\n",
       "      <td>...</td>\n",
       "      <td>0.004205</td>\n",
       "      <td>0.001773</td>\n",
       "      <td>-0.002052</td>\n",
       "      <td>-0.000001</td>\n",
       "      <td>-0.000024</td>\n",
       "      <td>-0.002142</td>\n",
       "      <td>-0.001593</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>-0.000206</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-0.011528</td>\n",
       "      <td>-0.050493</td>\n",
       "      <td>-0.149868</td>\n",
       "      <td>-0.013239</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.009549</td>\n",
       "      <td>0.030593</td>\n",
       "      <td>-0.005738</td>\n",
       "      <td>-0.045597</td>\n",
       "      <td>-0.001204</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000477</td>\n",
       "      <td>-0.001368</td>\n",
       "      <td>0.003686</td>\n",
       "      <td>0.000412</td>\n",
       "      <td>-0.002626</td>\n",
       "      <td>0.001747</td>\n",
       "      <td>-0.004787</td>\n",
       "      <td>-0.004930</td>\n",
       "      <td>-0.002133</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17711</th>\n",
       "      <td>-0.001753</td>\n",
       "      <td>0.000761</td>\n",
       "      <td>0.003842</td>\n",
       "      <td>-0.000024</td>\n",
       "      <td>-0.002626</td>\n",
       "      <td>-0.001269</td>\n",
       "      <td>0.002409</td>\n",
       "      <td>-0.002987</td>\n",
       "      <td>0.001626</td>\n",
       "      <td>-0.002262</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.029220</td>\n",
       "      <td>-0.904367</td>\n",
       "      <td>-0.005875</td>\n",
       "      <td>0.063842</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.094399</td>\n",
       "      <td>-0.006545</td>\n",
       "      <td>-0.013314</td>\n",
       "      <td>-0.007536</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17712</th>\n",
       "      <td>-0.000437</td>\n",
       "      <td>0.002153</td>\n",
       "      <td>-0.005526</td>\n",
       "      <td>-0.002142</td>\n",
       "      <td>0.001747</td>\n",
       "      <td>-0.003843</td>\n",
       "      <td>-0.001777</td>\n",
       "      <td>-0.002409</td>\n",
       "      <td>0.003086</td>\n",
       "      <td>-0.002267</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.146516</td>\n",
       "      <td>0.103868</td>\n",
       "      <td>0.067467</td>\n",
       "      <td>-0.201952</td>\n",
       "      <td>-0.094399</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.036214</td>\n",
       "      <td>0.138500</td>\n",
       "      <td>-0.048430</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17713</th>\n",
       "      <td>0.004527</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>-0.001074</td>\n",
       "      <td>-0.001593</td>\n",
       "      <td>-0.004787</td>\n",
       "      <td>-0.001417</td>\n",
       "      <td>-0.005010</td>\n",
       "      <td>0.003239</td>\n",
       "      <td>-0.000946</td>\n",
       "      <td>0.001803</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.012423</td>\n",
       "      <td>0.006909</td>\n",
       "      <td>-0.000102</td>\n",
       "      <td>0.048063</td>\n",
       "      <td>-0.006545</td>\n",
       "      <td>-0.036214</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.000973</td>\n",
       "      <td>-0.003410</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17714</th>\n",
       "      <td>0.003477</td>\n",
       "      <td>0.005014</td>\n",
       "      <td>-0.002356</td>\n",
       "      <td>0.000161</td>\n",
       "      <td>-0.004930</td>\n",
       "      <td>-0.003532</td>\n",
       "      <td>-0.001426</td>\n",
       "      <td>0.000944</td>\n",
       "      <td>0.009759</td>\n",
       "      <td>-0.001598</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.021886</td>\n",
       "      <td>0.015060</td>\n",
       "      <td>-0.004916</td>\n",
       "      <td>-0.076375</td>\n",
       "      <td>-0.013314</td>\n",
       "      <td>0.138500</td>\n",
       "      <td>-0.000973</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.005541</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17715</th>\n",
       "      <td>-0.001143</td>\n",
       "      <td>-0.001248</td>\n",
       "      <td>0.002305</td>\n",
       "      <td>-0.000206</td>\n",
       "      <td>-0.002133</td>\n",
       "      <td>-0.001377</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000175</td>\n",
       "      <td>0.001248</td>\n",
       "      <td>-0.000621</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.012153</td>\n",
       "      <td>0.007855</td>\n",
       "      <td>-0.003097</td>\n",
       "      <td>0.043374</td>\n",
       "      <td>-0.007536</td>\n",
       "      <td>-0.048430</td>\n",
       "      <td>-0.003410</td>\n",
       "      <td>-0.005541</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17716 rows × 17717 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          0         1         2         3         4         5         6      \\\n",
       "0      1.000000 -0.039520  0.076936 -0.010260 -0.011528 -0.007244  0.025617   \n",
       "1     -0.039520  1.000000 -0.199267 -0.043424 -0.050493 -0.029090  0.034237   \n",
       "2      0.076936 -0.199267  1.000000  0.087308 -0.149868 -0.081060  0.316285   \n",
       "3     -0.010260 -0.043424  0.087308  1.000000 -0.013239 -0.007831  0.028841   \n",
       "4     -0.011528 -0.050493 -0.149868 -0.013239  1.000000 -0.009549  0.030593   \n",
       "...         ...       ...       ...       ...       ...       ...       ...   \n",
       "17711 -0.001753  0.000761  0.003842 -0.000024 -0.002626 -0.001269  0.002409   \n",
       "17712 -0.000437  0.002153 -0.005526 -0.002142  0.001747 -0.003843 -0.001777   \n",
       "17713  0.004527  0.000074 -0.001074 -0.001593 -0.004787 -0.001417 -0.005010   \n",
       "17714  0.003477  0.005014 -0.002356  0.000161 -0.004930 -0.003532 -0.001426   \n",
       "17715 -0.001143 -0.001248  0.002305 -0.000206 -0.002133 -0.001377  0.000268   \n",
       "\n",
       "          7         8         9      ...     17707     17708     17709  \\\n",
       "0      0.349776 -0.040015 -0.003163  ...  0.003292  0.001617 -0.002359   \n",
       "1     -0.014709  0.577217 -0.015779  ... -0.001471  0.001519 -0.004886   \n",
       "2      0.036852 -0.458906 -0.043338  ...  0.000466 -0.000174  0.003233   \n",
       "3     -0.003291 -0.041080 -0.004297  ...  0.004205  0.001773 -0.002052   \n",
       "4     -0.005738 -0.045597 -0.001204  ...  0.000477 -0.001368  0.003686   \n",
       "...         ...       ...       ...  ...       ...       ...       ...   \n",
       "17711 -0.002987  0.001626 -0.002262  ... -0.029220 -0.904367 -0.005875   \n",
       "17712 -0.002409  0.003086 -0.002267  ... -0.146516  0.103868  0.067467   \n",
       "17713  0.003239 -0.000946  0.001803  ... -0.012423  0.006909 -0.000102   \n",
       "17714  0.000944  0.009759 -0.001598  ... -0.021886  0.015060 -0.004916   \n",
       "17715  0.000175  0.001248 -0.000621  ... -0.012153  0.007855 -0.003097   \n",
       "\n",
       "          17710     17711     17712     17713     17714     17715  17716  \n",
       "0      0.004602 -0.001753 -0.000437  0.004527  0.003477 -0.001143    NaN  \n",
       "1      0.001469  0.000761  0.002153  0.000074  0.005014 -0.001248    NaN  \n",
       "2      0.004464  0.003842 -0.005526 -0.001074 -0.002356  0.002305    NaN  \n",
       "3     -0.000001 -0.000024 -0.002142 -0.001593  0.000161 -0.000206    NaN  \n",
       "4      0.000412 -0.002626  0.001747 -0.004787 -0.004930 -0.002133    NaN  \n",
       "...         ...       ...       ...       ...       ...       ...    ...  \n",
       "17711  0.063842  1.000000 -0.094399 -0.006545 -0.013314 -0.007536    NaN  \n",
       "17712 -0.201952 -0.094399  1.000000 -0.036214  0.138500 -0.048430    NaN  \n",
       "17713  0.048063 -0.006545 -0.036214  1.000000 -0.000973 -0.003410    NaN  \n",
       "17714 -0.076375 -0.013314  0.138500 -0.000973  1.000000 -0.005541    NaN  \n",
       "17715  0.043374 -0.007536 -0.048430 -0.003410 -0.005541  1.000000    NaN  \n",
       "\n",
       "[17716 rows x 17717 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ld_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "4980c096-f069-4b74-ab2a-f19ed1794d2f",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1.000000, -0.039520, 0.076965, ..., 0.004528, 0.003477,\n",
       "        -0.001143],\n",
       "       [-0.039520, 1.000000, -0.199219, ..., 0.000074, 0.005013,\n",
       "        -0.001247],\n",
       "       [0.076965, -0.199219, 1.000000, ..., -0.001074, -0.002356,\n",
       "        0.002304],\n",
       "       ...,\n",
       "       [0.004528, 0.000074, -0.001074, ..., 1.000000, -0.000973,\n",
       "        -0.003410],\n",
       "       [0.003477, 0.005013, -0.002356, ..., -0.000973, 1.000000,\n",
       "        -0.005539],\n",
       "       [-0.001143, -0.001247, 0.002304, ..., -0.003410, -0.005539,\n",
       "        1.000000]], dtype=float16)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np_ld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "71992a05-11da-40b6-96aa-ad1593cce099",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "tri_upper_diag = np.triu(np_ld, k=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c3dea4ea-d22a-410c-8ba4-6afa1766d08f",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1.000000, -0.039520, 0.076965, ..., 0.004528, 0.003477,\n",
       "        -0.001143],\n",
       "       [0.000000, 1.000000, -0.199219, ..., 0.000074, 0.005013,\n",
       "        -0.001247],\n",
       "       [0.000000, 0.000000, 1.000000, ..., -0.001074, -0.002356,\n",
       "        0.002304],\n",
       "       ...,\n",
       "       [0.000000, 0.000000, 0.000000, ..., 1.000000, -0.000973,\n",
       "        -0.003410],\n",
       "       [0.000000, 0.000000, 0.000000, ..., 0.000000, 1.000000, -0.005539],\n",
       "       [0.000000, 0.000000, 0.000000, ..., 0.000000, 0.000000, 1.000000]],\n",
       "      dtype=float16)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tri_upper_diag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d12026a1-c7ea-4f04-b8ae-88f8f4b35cb0",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "tri_lower_diag = np.tril(np_ld, k=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "1eeed6ba-69cd-4063-8b3d-a9ee4debd3e9",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1.000000, 0.000000, 0.000000, ..., 0.000000, 0.000000, 0.000000],\n",
       "       [-0.039520, 1.000000, 0.000000, ..., 0.000000, 0.000000, 0.000000],\n",
       "       [0.076965, -0.199219, 1.000000, ..., 0.000000, 0.000000, 0.000000],\n",
       "       ...,\n",
       "       [0.004528, 0.000074, -0.001074, ..., 1.000000, 0.000000, 0.000000],\n",
       "       [0.003477, 0.005013, -0.002356, ..., -0.000973, 1.000000,\n",
       "        0.000000],\n",
       "       [-0.001143, -0.001247, 0.002304, ..., -0.003410, -0.005539,\n",
       "        1.000000]], dtype=float16)"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tri_lower_diag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a85a050-ab08-43d1-af6d-1a6bbc3011a1",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import xz\n",
    "with xz.open('/home/dmc2245/susie_mwe/01_159913048_162346721_hg19_test_lower_diag.xz', \"w+\", preset=9) as f:\n",
    "        for r in range(tri_lower_diag.shape[0]):\n",
    "            f.write(\" \".join([\"{:.6f}\".format(x) for x in tri_lower_diag[r, :]]).encode())\n",
    "            f.write(b\"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16b3932c-626a-45d8-9310-768b8c58dbf3",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## Run liftover"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "20ae2267-beb5-4c4d-909a-81c7df47ef8e",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mliftover_1\u001b[0m: Run liftover\n",
      "INFO: \u001b[32mliftover_1\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_1\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_1\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_1\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_1\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_1\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/liftover/01_159913048_162346721_hg19.bed /home/dmc2245/susie_mwe/liftover/01_159913048_162346721_hg19.hg38.bed... (15 items in 5 groups)\u001b[0m\n",
      "INFO: Running \u001b[32mliftover_2\u001b[0m: Organize bim file to account for unmapped variants\n",
      "INFO: \u001b[32mliftover_2\u001b[0m (index=2) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_2\u001b[0m (index=3) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_2\u001b[0m (index=4) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_2\u001b[0m (index=1) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_2\u001b[0m (index=0) is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mliftover_2\u001b[0m output:   \u001b[32m/home/dmc2245/susie_mwe/liftover/01_159913048_162346721_hg19.hg38.bim /home/dmc2245/susie_mwe/liftover/01_206073265_208410364_hg19.hg38.bim... (5 items in 5 groups)\u001b[0m\n",
      "INFO: Workflow liftover (ID=wc14e823629a61d88) is executed successfully with 2 completed steps and 10 completed substeps.\n"
     ]
    }
   ],
   "source": [
    "# Step 7 Run liftover on the bim files\n",
    "sos run ~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "     liftover \\\n",
    "    --cwd ~/susie_mwe/liftover \\\n",
    "    --masterfile 'susie_mwe'\\\n",
    "    --bim_name `echo /home/dmc2245/susie_mwe/*_hg19.bim` \\\n",
    "    --to_build hg38 \\\n",
    "    --chain_file ~/liftover_ucsc/hg19ToHg38.over.chain.gz \\\n",
    "    -s build"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "829833b0-80b3-4ed6-a4a1-d25827039939",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Initial bim file in hg19\n",
    "import pandas as pd\n",
    "hg19_bim=pd.read_csv('/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.bim',sep='\\t', header=None, names=[\"chr\",\"id\", \"cm\", \"pos_hg19\",\"minor_allele\",\"major_allele\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "60315ef2-e00b-43c5-a849-13579d270c43",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>id</th>\n",
       "      <th>cm</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>major_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>206073267</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206073386</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074010</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074070:T:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074070</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074127:G:C</td>\n",
       "      <td>0</td>\n",
       "      <td>206074127</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13079</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410043:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410043</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13080</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410130:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410130</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13081</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410160:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>208410160</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13082</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410337:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410337</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13083</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>208410364</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13084 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr               id  cm   pos_hg19 minor_allele major_allele\n",
       "0        1  1:206073267:A:T   0  206073267            T            A\n",
       "1        1  1:206073386:A:G   0  206073386            G            A\n",
       "2        1  1:206074010:A:G   0  206074010            G            A\n",
       "3        1  1:206074070:T:G   0  206074070            G            T\n",
       "4        1  1:206074127:G:C   0  206074127            C            G\n",
       "...    ...              ...  ..        ...          ...          ...\n",
       "13079    1  1:208410043:A:C   0  208410043            C            A\n",
       "13080    1  1:208410130:T:C   0  208410130            C            T\n",
       "13081    1  1:208410160:G:A   0  208410160            A            G\n",
       "13082    1  1:208410337:A:C   0  208410337            C            A\n",
       "13083    1  1:208410364:A:T   0  208410364            T            A\n",
       "\n",
       "[13084 rows x 6 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg19_bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "d26d9f8c-7c53-4c69-8756-98d654198cbe",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Final file in hg38 after running liftover\n",
    "hg38_bed=pd.read_csv('/home/dmc2245/susie_mwe/01_206073265_208410364_hg19.hg38.bed',sep='\\t', header=None, names=[\"chr\",\"start\", \"end\", \"id\", \"cm\",\"minor_allele\",\"major_allele\"])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "d8c19a13-d24a-4e5f-84ed-622aeb496d47",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>id</th>\n",
       "      <th>cm</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>major_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>206268083</td>\n",
       "      <td>206268083</td>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>206267964</td>\n",
       "      <td>206267964</td>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>206267340</td>\n",
       "      <td>206267340</td>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>206267280</td>\n",
       "      <td>206267280</td>\n",
       "      <td>1:206074070:T:G</td>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>206267223</td>\n",
       "      <td>206267223</td>\n",
       "      <td>1:206074127:G:C</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13057</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208236698</td>\n",
       "      <td>208236698</td>\n",
       "      <td>1:208410043:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13058</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208236785</td>\n",
       "      <td>208236785</td>\n",
       "      <td>1:208410130:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13059</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208236815</td>\n",
       "      <td>208236815</td>\n",
       "      <td>1:208410160:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13060</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208236992</td>\n",
       "      <td>208236992</td>\n",
       "      <td>1:208410337:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13061</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208237019</td>\n",
       "      <td>208237019</td>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13062 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr      start        end               id  cm minor_allele  \\\n",
       "0      chr1  206268083  206268083  1:206073267:A:T   0            T   \n",
       "1      chr1  206267964  206267964  1:206073386:A:G   0            G   \n",
       "2      chr1  206267340  206267340  1:206074010:A:G   0            G   \n",
       "3      chr1  206267280  206267280  1:206074070:T:G   0            G   \n",
       "4      chr1  206267223  206267223  1:206074127:G:C   0            C   \n",
       "...     ...        ...        ...              ...  ..          ...   \n",
       "13057  chr1  208236698  208236698  1:208410043:A:C   0            C   \n",
       "13058  chr1  208236785  208236785  1:208410130:T:C   0            C   \n",
       "13059  chr1  208236815  208236815  1:208410160:G:A   0            A   \n",
       "13060  chr1  208236992  208236992  1:208410337:A:C   0            C   \n",
       "13061  chr1  208237019  208237019  1:208410364:A:T   0            T   \n",
       "\n",
       "      major_allele  \n",
       "0                A  \n",
       "1                A  \n",
       "2                A  \n",
       "3                T  \n",
       "4                G  \n",
       "...            ...  \n",
       "13057            A  \n",
       "13058            T  \n",
       "13059            G  \n",
       "13060            A  \n",
       "13061            A  \n",
       "\n",
       "[13062 rows x 7 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg38_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "0df6c77c-7553-4bee-9313-8e33a79722e8",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "merged_df = pd.merge(hg19_bim,hg38_bed, on='id', how='outer' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "86323195-1015-4e29-9c43-b6fb5b61d6cf",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      chr_x                   id  cm_x   pos_hg19 minor_allele_x  \\\n",
      "471       1     1:206207649:CT:C     0  206207649              C   \n",
      "1474      1     1:206508203:AT:A     0  206508203              A   \n",
      "1487      1      1:206512352:T:G     0  206512352              G   \n",
      "1495      1    1:206513621:CCT:C     0  206513621              C   \n",
      "1590      1  1:206545086:AAAAG:A     0  206545086              A   \n",
      "1754      1      1:206592248:G:C     0  206592248              C   \n",
      "1755      1      1:206592250:G:C     0  206592250              C   \n",
      "1756      1      1:206592252:G:C     0  206592252              C   \n",
      "1757      1      1:206592254:G:C     0  206592254              C   \n",
      "1955      1    1:206622631:ATT:A     0  206622631              A   \n",
      "1970      1     1:206624458:TA:T     0  206624458              T   \n",
      "2050      1     1:206639880:CA:C     0  206639880              C   \n",
      "2056      1  1:206640435:AGAAT:A     0  206640435              A   \n",
      "2813      1     1:206750305:AT:A     0  206750305              A   \n",
      "2827      1      1:206751563:T:A     0  206751563              A   \n",
      "2895      1      1:206760686:T:A     0  206760686              A   \n",
      "2915      1     1:206766442:CA:C     0  206766442              C   \n",
      "2919      1    1:206766855:TGC:T     0  206766855              T   \n",
      "2942      1     1:206770283:AC:A     0  206770283              A   \n",
      "2956      1    1:206771988:ACG:A     0  206771988              A   \n",
      "2959      1    1:206772194:ACT:A     0  206772194              A   \n",
      "3052      1     1:206786589:CA:C     0  206786589              C   \n",
      "\n",
      "     major_allele_x chr_y  start  end  cm_y minor_allele_y major_allele_y  \n",
      "471              CT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1474             AT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1487              T   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1495            CCT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1590          AAAAG   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1754              G   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1755              G   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1756              G   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1757              G   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1955            ATT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "1970             TA   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2050             CA   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2056          AGAAT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2813             AT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2827              T   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2895              T   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2915             CA   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2919            TGC   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2942             AC   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2956            ACG   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "2959            ACT   NaN    NaN  NaN   NaN            NaN            NaN  \n",
      "3052             CA   NaN    NaN  NaN   NaN            NaN            NaN  \n"
     ]
    }
   ],
   "source": [
    "merged_df_2 = pd.merge(hg19_bim,hg38_bed, on='id', how='left', indicator=True)\n",
    "rows_not_in_df2 = merged_df_2[merged_df_2['_merge'] == 'left_only'].drop(columns=['_merge'])\n",
    "\n",
    "print(rows_not_in_df2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c542ff6f-8f81-452c-8436-076c62507cdd",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_x</th>\n",
       "      <th>id</th>\n",
       "      <th>cm_x</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>minor_allele_x</th>\n",
       "      <th>major_allele_x</th>\n",
       "      <th>chr_y</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>cm_y</th>\n",
       "      <th>minor_allele_y</th>\n",
       "      <th>major_allele_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>206073267</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>206268083.0</td>\n",
       "      <td>206268083.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206073386</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>206267964.0</td>\n",
       "      <td>206267964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074010</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>206267340.0</td>\n",
       "      <td>206267340.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074070:T:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074070</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1</td>\n",
       "      <td>206267280.0</td>\n",
       "      <td>206267280.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074127:G:C</td>\n",
       "      <td>0</td>\n",
       "      <td>206074127</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>206267223.0</td>\n",
       "      <td>206267223.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13079</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410043:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410043</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>208236698.0</td>\n",
       "      <td>208236698.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13080</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410130:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410130</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr1</td>\n",
       "      <td>208236785.0</td>\n",
       "      <td>208236785.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13081</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410160:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>208410160</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>208236815.0</td>\n",
       "      <td>208236815.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13082</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410337:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>208410337</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>208236992.0</td>\n",
       "      <td>208236992.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13083</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>208410364</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>chr1</td>\n",
       "      <td>208237019.0</td>\n",
       "      <td>208237019.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13084 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr_x               id  cm_x   pos_hg19 minor_allele_x major_allele_x  \\\n",
       "0          1  1:206073267:A:T     0  206073267              T              A   \n",
       "1          1  1:206073386:A:G     0  206073386              G              A   \n",
       "2          1  1:206074010:A:G     0  206074010              G              A   \n",
       "3          1  1:206074070:T:G     0  206074070              G              T   \n",
       "4          1  1:206074127:G:C     0  206074127              C              G   \n",
       "...      ...              ...   ...        ...            ...            ...   \n",
       "13079      1  1:208410043:A:C     0  208410043              C              A   \n",
       "13080      1  1:208410130:T:C     0  208410130              C              T   \n",
       "13081      1  1:208410160:G:A     0  208410160              A              G   \n",
       "13082      1  1:208410337:A:C     0  208410337              C              A   \n",
       "13083      1  1:208410364:A:T     0  208410364              T              A   \n",
       "\n",
       "      chr_y        start          end  cm_y minor_allele_y major_allele_y  \n",
       "0      chr1  206268083.0  206268083.0   0.0              T              A  \n",
       "1      chr1  206267964.0  206267964.0   0.0              G              A  \n",
       "2      chr1  206267340.0  206267340.0   0.0              G              A  \n",
       "3      chr1  206267280.0  206267280.0   0.0              G              T  \n",
       "4      chr1  206267223.0  206267223.0   0.0              C              G  \n",
       "...     ...          ...          ...   ...            ...            ...  \n",
       "13079  chr1  208236698.0  208236698.0   0.0              C              A  \n",
       "13080  chr1  208236785.0  208236785.0   0.0              C              T  \n",
       "13081  chr1  208236815.0  208236815.0   0.0              A              G  \n",
       "13082  chr1  208236992.0  208236992.0   0.0              C              A  \n",
       "13083  chr1  208237019.0  208237019.0   0.0              T              A  \n",
       "\n",
       "[13084 rows x 12 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "8438140b-543e-4cd5-bfb9-864904f00bd5",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Example usage:\n",
    "df1 = pd.DataFrame({'chr': [1, 1, 1,1,1,1], 'id': ['1:206073267:A:T','1:206073386:A:G','1:206074010:A:G','1:206207649:CT:C', '1:206508203:AT:A','1:208410364:A:T'], 'cm': [0, 0, 0,0,0,0], 'pos_hg19': [206073267, 206073386, 206074010,206207649,206508203,208410364],\n",
    "                    'minor_allele': ['T', 'G', 'G','C','A','A'], 'major_allele': ['A', 'A', 'A','CT','AT','T']})\n",
    "\n",
    "df2= pd.DataFrame({'id': ['1:206073267:A:T', '1:206073386:A:G', '1:206074010:A:G','1:208410364:A:T'], 'chr': [1, 1, 1,1], 'cm': [0, 0, 0,0], 'pos_hg38': [206268083, 206267964, 206267340,208237019],\n",
    "                    'minor_allele': ['T', 'G', 'G','T'], 'major_allele': ['A', 'A', 'A','A']})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b20c8173-d6bf-428b-90b7-7fd50efebcd8",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>id</th>\n",
       "      <th>cm</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>major_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>206073267</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206073386</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074010</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206207649:CT:C</td>\n",
       "      <td>0</td>\n",
       "      <td>206207649</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206508203:AT:A</td>\n",
       "      <td>0</td>\n",
       "      <td>206508203</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>208410364</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr                id  cm   pos_hg19 minor_allele major_allele\n",
       "0    1   1:206073267:A:T   0  206073267            T            A\n",
       "1    1   1:206073386:A:G   0  206073386            G            A\n",
       "2    1   1:206074010:A:G   0  206074010            G            A\n",
       "3    1  1:206207649:CT:C   0  206207649            C           CT\n",
       "4    1  1:206508203:AT:A   0  206508203            A           AT\n",
       "5    1   1:208410364:A:T   0  208410364            A            T"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c3c21569-c538-4d6f-b12a-a42a3b6de1d9",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>cm</th>\n",
       "      <th>pos_hg38</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>major_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>206268083</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>206267964</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>206267340</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>208237019</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                id  chr  cm   pos_hg38 minor_allele major_allele\n",
       "0  1:206073267:A:T    1   0  206268083            T            A\n",
       "1  1:206073386:A:G    1   0  206267964            G            A\n",
       "2  1:206074010:A:G    1   0  206267340            G            A\n",
       "3  1:208410364:A:T    1   0  208237019            T            A"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "7e32f036-d269-4c9e-a238-1ecac421397d",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "def merge_dataframes(df1, df2):\n",
    "    # Merge DataFrames using a left join on the 'id' column\n",
    "    merged_df = pd.merge(df1, df2, on='id', how='left', suffixes=('_df1', '_df2'))\n",
    "    # Create a list to store dataframes\n",
    "    result_dfs = []\n",
    "    # Iterate through rows\n",
    "    for index, row in merged_df.iterrows():\n",
    "        if not pd.isna(row['pos_hg38']):  # If 'pos_hg38' is present in df2\n",
    "            id_string = f\"{int(row['chr_df2'])}:{row['pos_hg38']}:{row['major_allele_df2']}:{row['minor_allele_df2']}\"\n",
    "            result_row = pd.Series([int(row['chr_df2']), id_string, int(row['cm_df2']), row['pos_hg38'], row['minor_allele_df2'], row['major_allele_df2']],\n",
    "                                   index=['chr_df2', 'id', 'cm_df2', 'pos_hg38', 'minor_allele_df2', 'major_allele_df2'])\n",
    "        else:\n",
    "             result_row = pd.Series([0, row['id'], int(row['cm_df1']), row['pos_hg19'], row['minor_allele_df1'], row['major_allele_df1']],\n",
    "                                   index=['chr_df2', 'id', 'cm_df2', 'pos_hg38', 'minor_allele_df2', 'major_allele_df2'])\n",
    "\n",
    "        result_dfs.append(result_row)\n",
    "    # Concatenate the list of DataFrames into the final result DataFrame\n",
    "    result_df = pd.DataFrame(result_dfs)\n",
    "    # Sort the result DataFrame based on the original order in df1\n",
    "    #result_df = result_df.sort_values(by='id').reset_index(drop=True)\n",
    "    return result_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "26fa548f-40cc-47b2-9fdc-1099b1dca5b8",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_df1</th>\n",
       "      <th>id</th>\n",
       "      <th>cm_df1</th>\n",
       "      <th>pos_hg19</th>\n",
       "      <th>minor_allele_df1</th>\n",
       "      <th>major_allele_df1</th>\n",
       "      <th>chr_df2</th>\n",
       "      <th>cm_df2</th>\n",
       "      <th>pos_hg38</th>\n",
       "      <th>minor_allele_df2</th>\n",
       "      <th>major_allele_df2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073267:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>206073267</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>206268083.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206073386:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206073386</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>206267964.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206074010:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206074010</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>206267340.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206207649:CT:C</td>\n",
       "      <td>0</td>\n",
       "      <td>206207649</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206508203:AT:A</td>\n",
       "      <td>0</td>\n",
       "      <td>206508203</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208410364:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>208410364</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>208237019.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr_df1                id  cm_df1   pos_hg19 minor_allele_df1  \\\n",
       "0        1   1:206073267:A:T       0  206073267                T   \n",
       "1        1   1:206073386:A:G       0  206073386                G   \n",
       "2        1   1:206074010:A:G       0  206074010                G   \n",
       "3        1  1:206207649:CT:C       0  206207649                C   \n",
       "4        1  1:206508203:AT:A       0  206508203                A   \n",
       "5        1   1:208410364:A:T       0  208410364                A   \n",
       "\n",
       "  major_allele_df1  chr_df2  cm_df2     pos_hg38 minor_allele_df2  \\\n",
       "0                A      1.0     0.0  206268083.0                T   \n",
       "1                A      1.0     0.0  206267964.0                G   \n",
       "2                A      1.0     0.0  206267340.0                G   \n",
       "3               CT      NaN     NaN          NaN              NaN   \n",
       "4               AT      NaN     NaN          NaN              NaN   \n",
       "5                T      1.0     0.0  208237019.0                T   \n",
       "\n",
       "  major_allele_df2  \n",
       "0                A  \n",
       "1                A  \n",
       "2                A  \n",
       "3              NaN  \n",
       "4              NaN  \n",
       "5                A  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "d5fad9f8-76a0-4a7c-a1f9-7a5075158867",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "res= merge_dataframes(df1,df2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "46d0164e-9398-497c-82be-6252a40495c0",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr_df2</th>\n",
       "      <th>id</th>\n",
       "      <th>cm_df2</th>\n",
       "      <th>pos_hg38</th>\n",
       "      <th>minor_allele_df2</th>\n",
       "      <th>major_allele_df2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206268083.0:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>206268083.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206267964.0:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206267964.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1:206267340.0:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>206267340.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>1:206207649:CT:C</td>\n",
       "      <td>0</td>\n",
       "      <td>206207649.0</td>\n",
       "      <td>C</td>\n",
       "      <td>CT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>1:206508203:AT:A</td>\n",
       "      <td>0</td>\n",
       "      <td>206508203.0</td>\n",
       "      <td>A</td>\n",
       "      <td>AT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>1:208237019.0:A:T</td>\n",
       "      <td>0</td>\n",
       "      <td>208237019.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr_df2                 id  cm_df2     pos_hg38 minor_allele_df2  \\\n",
       "0        1  1:206268083.0:A:T       0  206268083.0                T   \n",
       "1        1  1:206267964.0:A:G       0  206267964.0                G   \n",
       "2        1  1:206267340.0:A:G       0  206267340.0                G   \n",
       "3        0   1:206207649:CT:C       0  206207649.0                C   \n",
       "4        0   1:206508203:AT:A       0  206508203.0                A   \n",
       "5        1  1:208237019.0:A:T       0  208237019.0                T   \n",
       "\n",
       "  major_allele_df2  \n",
       "0                A  \n",
       "1                A  \n",
       "2                A  \n",
       "3               CT  \n",
       "4               AT  \n",
       "5                A  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ad8938e2-dbed-43c0-bd2a-4e6a0a6026bf",
   "metadata": {
    "kernel": "Markdown",
    "tags": []
   },
   "source": [
    "## Read the xz files using R"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3ce804c-4690-4def-b2a4-2f14c07c1743",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "source": [
    "Need to `install.packages('data.table')`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c393c5dc-da81-4ae8-8ad7-124fd718fbc2",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "library('data.table')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "ee4339bf-d9f5-4872-9394-1d6841df543d",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "file <- fread(cmd = paste(\"xzcat\", \"/home/dmc2245/susie_mwe/01_159913048_162346721_hg19_test.xz\"), header = FALSE, sep = \" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "bc365e02-32f9-4d65-ac57-2da78c8af112",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 17716 × 17716</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>⋯</th><th scope=col>V17707</th><th scope=col>V17708</th><th scope=col>V17709</th><th scope=col>V17710</th><th scope=col>V17711</th><th scope=col>V17712</th><th scope=col>V17713</th><th scope=col>V17714</th><th scope=col>V17715</th><th scope=col>V17716</th></tr>\n",
       "\t<tr><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1</td><td>-0.03952</td><td> 0.076965</td><td>-0.010262</td><td>-0.011528</td><td>-0.007244</td><td>0.025620</td><td> 0.349854</td><td>-0.040009</td><td>-0.003162</td><td>⋯</td><td>-0.000100</td><td> 0.003292</td><td> 0.001616</td><td>-0.002359</td><td> 0.004601</td><td>-0.001753</td><td>-0.000437</td><td> 0.004528</td><td> 0.003477</td><td>-0.001143</td></tr>\n",
       "\t<tr><td>0</td><td> 1.00000</td><td>-0.199219</td><td>-0.043427</td><td>-0.050507</td><td>-0.029083</td><td>0.034241</td><td>-0.014709</td><td> 0.577148</td><td>-0.015778</td><td>⋯</td><td> 0.001030</td><td>-0.001472</td><td> 0.001519</td><td>-0.004887</td><td> 0.001469</td><td> 0.000762</td><td> 0.002153</td><td> 0.000074</td><td> 0.005013</td><td>-0.001247</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 1.000000</td><td> 0.087280</td><td>-0.149902</td><td>-0.081055</td><td>0.316406</td><td> 0.036865</td><td>-0.458984</td><td>-0.043335</td><td>⋯</td><td>-0.003998</td><td> 0.000465</td><td>-0.000174</td><td> 0.003233</td><td> 0.004463</td><td> 0.003841</td><td>-0.005527</td><td>-0.001074</td><td>-0.002356</td><td> 0.002304</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 1.000000</td><td>-0.013237</td><td>-0.007828</td><td>0.028839</td><td>-0.003292</td><td>-0.041077</td><td>-0.004299</td><td>⋯</td><td> 0.000781</td><td> 0.004204</td><td> 0.001773</td><td>-0.002052</td><td>-0.000001</td><td>-0.000024</td><td>-0.002142</td><td>-0.001593</td><td> 0.000161</td><td>-0.000206</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.009552</td><td>0.030594</td><td>-0.005737</td><td>-0.045593</td><td>-0.001204</td><td>⋯</td><td> 0.005833</td><td> 0.000477</td><td>-0.001369</td><td> 0.003685</td><td> 0.000412</td><td>-0.002626</td><td> 0.001747</td><td>-0.004787</td><td>-0.004929</td><td>-0.002132</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>0.005970</td><td>-0.004547</td><td>-0.024643</td><td>-0.002275</td><td>⋯</td><td>-0.003075</td><td> 0.001641</td><td> 0.001968</td><td>-0.000655</td><td> 0.002501</td><td>-0.001268</td><td>-0.003843</td><td>-0.001417</td><td>-0.003532</td><td>-0.001376</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>1.000000</td><td> 0.012573</td><td> 0.095459</td><td>-0.129761</td><td>⋯</td><td> 0.000926</td><td> 0.001922</td><td> 0.002653</td><td> 0.002949</td><td> 0.004833</td><td> 0.002409</td><td>-0.001778</td><td>-0.005009</td><td>-0.001427</td><td> 0.000268</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 1.000000</td><td>-0.016174</td><td>-0.002609</td><td>⋯</td><td>-0.000933</td><td> 0.003387</td><td> 0.003214</td><td>-0.000992</td><td> 0.002140</td><td>-0.002987</td><td>-0.002409</td><td> 0.003239</td><td> 0.000944</td><td> 0.000175</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.012650</td><td>⋯</td><td> 0.001830</td><td>-0.002825</td><td>-0.001021</td><td>-0.005630</td><td>-0.000737</td><td> 0.001626</td><td> 0.003086</td><td>-0.000947</td><td> 0.009758</td><td> 0.001248</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>⋯</td><td> 0.003416</td><td>-0.001208</td><td> 0.002728</td><td> 0.001072</td><td>-0.001986</td><td>-0.002262</td><td>-0.002268</td><td> 0.001803</td><td>-0.001597</td><td>-0.000621</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.000408</td><td> 0.000751</td><td> 0.000011</td><td> 0.001480</td><td>-0.002943</td><td> 0.001328</td><td> 0.000930</td><td>-0.002052</td><td> 0.000879</td><td>-0.005005</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.001634</td><td>-0.002693</td><td>-0.001399</td><td>-0.005199</td><td>-0.000607</td><td> 0.002066</td><td> 0.003286</td><td>-0.001204</td><td> 0.009796</td><td> 0.001442</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.003178</td><td> 0.002230</td><td> 0.003824</td><td> 0.005157</td><td> 0.003963</td><td> 0.000956</td><td>-0.003025</td><td> 0.000073</td><td>-0.005714</td><td> 0.000440</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.002131</td><td> 0.004147</td><td> 0.002497</td><td> 0.000448</td><td>-0.001878</td><td>-0.000697</td><td> 0.000125</td><td>-0.002186</td><td>-0.000058</td><td> 0.000815</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002150</td><td> 0.002237</td><td> 0.004013</td><td> 0.003674</td><td> 0.004833</td><td>-0.001129</td><td>-0.001676</td><td>-0.004311</td><td>-0.000638</td><td> 0.000141</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.003214</td><td> 0.002443</td><td> 0.003559</td><td> 0.005058</td><td> 0.003836</td><td> 0.001032</td><td>-0.003391</td><td> 0.000064</td><td>-0.005825</td><td> 0.000550</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002977</td><td>-0.002562</td><td> 0.003391</td><td>-0.000667</td><td>-0.003534</td><td>-0.003256</td><td> 0.005707</td><td>-0.000011</td><td>-0.000241</td><td> 0.000664</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002012</td><td>-0.002876</td><td>-0.001159</td><td>-0.005474</td><td>-0.000380</td><td> 0.001699</td><td> 0.003292</td><td>-0.001380</td><td> 0.009491</td><td> 0.001838</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.005165</td><td>-0.001267</td><td>-0.007256</td><td>-0.003458</td><td>-0.005848</td><td> 0.007759</td><td> 0.000376</td><td> 0.004910</td><td>-0.001518</td><td>-0.000483</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.000907</td><td> 0.001980</td><td>-0.004456</td><td>-0.002598</td><td> 0.004963</td><td> 0.002831</td><td>-0.003077</td><td> 0.001796</td><td>-0.002714</td><td> 0.000148</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002151</td><td> 0.002262</td><td> 0.004021</td><td> 0.003677</td><td> 0.004768</td><td>-0.001153</td><td>-0.001672</td><td>-0.004330</td><td>-0.000648</td><td> 0.000168</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002136</td><td>-0.002308</td><td>-0.000544</td><td>-0.005749</td><td> 0.000390</td><td> 0.001315</td><td> 0.004803</td><td>-0.000593</td><td> 0.008240</td><td> 0.000856</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.001023</td><td> 0.002226</td><td> 0.003578</td><td> 0.005356</td><td> 0.004047</td><td>-0.000945</td><td> 0.000290</td><td>-0.003910</td><td> 0.002016</td><td>-0.000914</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.000234</td><td>-0.001267</td><td>-0.001380</td><td>-0.000499</td><td> 0.003063</td><td> 0.002371</td><td> 0.000489</td><td> 0.003508</td><td>-0.001062</td><td>-0.002787</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.002131</td><td>-0.002415</td><td>-0.000894</td><td>-0.005707</td><td> 0.000339</td><td> 0.001283</td><td> 0.004761</td><td>-0.000663</td><td> 0.008217</td><td> 0.000826</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.005180</td><td>-0.001430</td><td> 0.000865</td><td>-0.003817</td><td> 0.000124</td><td>-0.000494</td><td>-0.002331</td><td> 0.001020</td><td>-0.004375</td><td> 0.000978</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td>-0.000216</td><td> 0.003359</td><td> 0.000936</td><td>-0.002672</td><td>-0.006420</td><td>-0.003790</td><td> 0.003767</td><td> 0.000622</td><td> 0.001378</td><td> 0.000960</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.000403</td><td>-0.001416</td><td> 0.001929</td><td> 0.002844</td><td>-0.001254</td><td>-0.001178</td><td>-0.001458</td><td>-0.001413</td><td>-0.002264</td><td> 0.001591</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.000155</td><td>-0.000913</td><td> 0.001951</td><td> 0.003183</td><td>-0.001164</td><td>-0.001325</td><td>-0.001625</td><td>-0.001315</td><td>-0.003014</td><td> 0.000762</td></tr>\n",
       "\t<tr><td>0</td><td> 0.00000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td>⋯</td><td> 0.005592</td><td>-0.000209</td><td>-0.000628</td><td>-0.004028</td><td>-0.000814</td><td> 0.000950</td><td>-0.005909</td><td>-0.001210</td><td>-0.006847</td><td> 0.000751</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋱</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.010933</td><td> 0.935059</td><td> 0.032623</td><td>-0.011162</td><td> 0.104675</td><td>-0.030396</td><td>-0.154907</td><td>-0.013107</td><td>-0.023239</td><td>-0.012566</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.057037</td><td>-0.158325</td><td> 0.105835</td><td> 0.063477</td><td>-0.153931</td><td>-0.095703</td><td> 0.898926</td><td> 0.065063</td><td> 0.135254</td><td>-0.050659</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.025223</td><td>-0.113770</td><td> 0.078979</td><td> 0.086853</td><td> 0.001963</td><td>-0.071899</td><td>-0.381592</td><td>-0.027603</td><td>-0.062042</td><td>-0.037872</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.054840</td><td>-0.166260</td><td>-0.126831</td><td>-0.037689</td><td>-0.135986</td><td> 0.156860</td><td> 0.876465</td><td> 0.062195</td><td> 0.129639</td><td>-0.052582</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.004322</td><td>-0.010605</td><td> 0.007133</td><td>-0.003828</td><td> 0.052856</td><td>-0.007687</td><td>-0.040649</td><td> 0.006851</td><td>-0.004059</td><td>-0.003998</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.011734</td><td> 0.842773</td><td> 0.035400</td><td>-0.012283</td><td> 0.133423</td><td>-0.033752</td><td>-0.172974</td><td>-0.014435</td><td>-0.025391</td><td>-0.015839</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.003172</td><td>-0.010902</td><td> 0.006474</td><td>-0.001335</td><td> 0.051117</td><td>-0.005539</td><td>-0.040802</td><td>-0.004128</td><td>-0.006432</td><td>-0.002035</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.003738</td><td>-0.014763</td><td> 0.010216</td><td>-0.003300</td><td>-0.084534</td><td>-0.009338</td><td> 0.092957</td><td> 0.004913</td><td> 0.675781</td><td>-0.003143</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.002186</td><td> 0.008362</td><td> 0.384033</td><td> 0.004158</td><td> 0.073364</td><td> 0.007736</td><td> 0.042145</td><td> 0.003010</td><td> 0.006969</td><td> 0.002356</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.003878</td><td>-0.014587</td><td> 0.010170</td><td>-0.005234</td><td> 0.056854</td><td>-0.008888</td><td>-0.056671</td><td>-0.000908</td><td>-0.008812</td><td>-0.005516</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.002249</td><td> 0.007401</td><td> 0.381592</td><td> 0.004272</td><td> 0.072266</td><td> 0.007229</td><td> 0.041870</td><td> 0.003029</td><td> 0.007084</td><td> 0.001984</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.002703</td><td>-0.009216</td><td> 0.006172</td><td>-0.002552</td><td>-0.041565</td><td>-0.005405</td><td>-0.034760</td><td>-0.000909</td><td>-0.005432</td><td>-0.003893</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.016525</td><td>-0.076233</td><td> 0.054291</td><td>-0.017334</td><td> 0.050049</td><td>-0.049622</td><td>-0.269775</td><td>-0.020203</td><td>-0.042297</td><td>-0.025299</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.002586</td><td>-0.012581</td><td> 0.007622</td><td> 0.000579</td><td> 0.042023</td><td>-0.006882</td><td>-0.031128</td><td> 0.707520</td><td>-0.005772</td><td>-0.004761</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.027283</td><td>-0.132446</td><td> 0.091125</td><td>-0.032623</td><td> 0.099792</td><td>-0.081055</td><td>-0.474365</td><td>-0.032196</td><td>-0.063904</td><td> 0.101624</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.002802</td><td> 0.007664</td><td> 0.375000</td><td> 0.004139</td><td> 0.070801</td><td> 0.008072</td><td> 0.041687</td><td> 0.002821</td><td> 0.006973</td><td> 0.002094</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.002758</td><td> 0.007797</td><td> 0.378418</td><td> 0.004131</td><td> 0.070251</td><td> 0.008003</td><td> 0.041565</td><td> 0.002779</td><td> 0.006931</td><td> 0.002037</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.001348</td><td>-0.004429</td><td>-0.008324</td><td>-0.001637</td><td> 0.007542</td><td>-0.003332</td><td>-0.017395</td><td> 0.000688</td><td>-0.002024</td><td>-0.002193</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>-0.001383</td><td>-0.004250</td><td>-0.009079</td><td>-0.001659</td><td> 0.007153</td><td>-0.003307</td><td>-0.017456</td><td>-0.001461</td><td>-0.002131</td><td>-0.002245</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.002775</td><td> 0.008362</td><td> 0.379883</td><td> 0.004131</td><td> 0.070801</td><td> 0.007988</td><td> 0.041656</td><td> 0.002731</td><td> 0.006882</td><td> 0.002134</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 1.000000</td><td>-0.010612</td><td> 0.004982</td><td>-0.002924</td><td> 0.026581</td><td>-0.004398</td><td> 0.058655</td><td>-0.002159</td><td>-0.004002</td><td>-0.001085</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 1.000000</td><td> 0.030930</td><td>-0.010536</td><td> 0.092468</td><td>-0.029221</td><td>-0.146484</td><td>-0.012421</td><td>-0.021881</td><td>-0.012154</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td> 0.006088</td><td>-0.029404</td><td>-0.904297</td><td> 0.103882</td><td> 0.006908</td><td> 0.015060</td><td> 0.007858</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td> 0.013580</td><td>-0.005875</td><td> 0.067444</td><td>-0.000102</td><td>-0.004917</td><td>-0.003098</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td> 0.063843</td><td>-0.201904</td><td> 0.048065</td><td>-0.076355</td><td> 0.043365</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.094421</td><td>-0.006546</td><td>-0.013313</td><td>-0.007534</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.036224</td><td> 0.138550</td><td>-0.048431</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.000973</td><td>-0.003410</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td><td>-0.005539</td></tr>\n",
       "\t<tr><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 0.000000</td><td> 1.000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 17716 × 17716\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & ⋯ & V17707 & V17708 & V17709 & V17710 & V17711 & V17712 & V17713 & V17714 & V17715 & V17716\\\\\n",
       " <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 1 & -0.03952 &  0.076965 & -0.010262 & -0.011528 & -0.007244 & 0.025620 &  0.349854 & -0.040009 & -0.003162 & ⋯ & -0.000100 &  0.003292 &  0.001616 & -0.002359 &  0.004601 & -0.001753 & -0.000437 &  0.004528 &  0.003477 & -0.001143\\\\\n",
       "\t 0 &  1.00000 & -0.199219 & -0.043427 & -0.050507 & -0.029083 & 0.034241 & -0.014709 &  0.577148 & -0.015778 & ⋯ &  0.001030 & -0.001472 &  0.001519 & -0.004887 &  0.001469 &  0.000762 &  0.002153 &  0.000074 &  0.005013 & -0.001247\\\\\n",
       "\t 0 &  0.00000 &  1.000000 &  0.087280 & -0.149902 & -0.081055 & 0.316406 &  0.036865 & -0.458984 & -0.043335 & ⋯ & -0.003998 &  0.000465 & -0.000174 &  0.003233 &  0.004463 &  0.003841 & -0.005527 & -0.001074 & -0.002356 &  0.002304\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  1.000000 & -0.013237 & -0.007828 & 0.028839 & -0.003292 & -0.041077 & -0.004299 & ⋯ &  0.000781 &  0.004204 &  0.001773 & -0.002052 & -0.000001 & -0.000024 & -0.002142 & -0.001593 &  0.000161 & -0.000206\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  1.000000 & -0.009552 & 0.030594 & -0.005737 & -0.045593 & -0.001204 & ⋯ &  0.005833 &  0.000477 & -0.001369 &  0.003685 &  0.000412 & -0.002626 &  0.001747 & -0.004787 & -0.004929 & -0.002132\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 & 0.005970 & -0.004547 & -0.024643 & -0.002275 & ⋯ & -0.003075 &  0.001641 &  0.001968 & -0.000655 &  0.002501 & -0.001268 & -0.003843 & -0.001417 & -0.003532 & -0.001376\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 1.000000 &  0.012573 &  0.095459 & -0.129761 & ⋯ &  0.000926 &  0.001922 &  0.002653 &  0.002949 &  0.004833 &  0.002409 & -0.001778 & -0.005009 & -0.001427 &  0.000268\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  1.000000 & -0.016174 & -0.002609 & ⋯ & -0.000933 &  0.003387 &  0.003214 & -0.000992 &  0.002140 & -0.002987 & -0.002409 &  0.003239 &  0.000944 &  0.000175\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  1.000000 & -0.012650 & ⋯ &  0.001830 & -0.002825 & -0.001021 & -0.005630 & -0.000737 &  0.001626 &  0.003086 & -0.000947 &  0.009758 &  0.001248\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  1.000000 & ⋯ &  0.003416 & -0.001208 &  0.002728 &  0.001072 & -0.001986 & -0.002262 & -0.002268 &  0.001803 & -0.001597 & -0.000621\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.000408 &  0.000751 &  0.000011 &  0.001480 & -0.002943 &  0.001328 &  0.000930 & -0.002052 &  0.000879 & -0.005005\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.001634 & -0.002693 & -0.001399 & -0.005199 & -0.000607 &  0.002066 &  0.003286 & -0.001204 &  0.009796 &  0.001442\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.003178 &  0.002230 &  0.003824 &  0.005157 &  0.003963 &  0.000956 & -0.003025 &  0.000073 & -0.005714 &  0.000440\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.002131 &  0.004147 &  0.002497 &  0.000448 & -0.001878 & -0.000697 &  0.000125 & -0.002186 & -0.000058 &  0.000815\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002150 &  0.002237 &  0.004013 &  0.003674 &  0.004833 & -0.001129 & -0.001676 & -0.004311 & -0.000638 &  0.000141\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.003214 &  0.002443 &  0.003559 &  0.005058 &  0.003836 &  0.001032 & -0.003391 &  0.000064 & -0.005825 &  0.000550\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002977 & -0.002562 &  0.003391 & -0.000667 & -0.003534 & -0.003256 &  0.005707 & -0.000011 & -0.000241 &  0.000664\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002012 & -0.002876 & -0.001159 & -0.005474 & -0.000380 &  0.001699 &  0.003292 & -0.001380 &  0.009491 &  0.001838\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.005165 & -0.001267 & -0.007256 & -0.003458 & -0.005848 &  0.007759 &  0.000376 &  0.004910 & -0.001518 & -0.000483\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.000907 &  0.001980 & -0.004456 & -0.002598 &  0.004963 &  0.002831 & -0.003077 &  0.001796 & -0.002714 &  0.000148\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002151 &  0.002262 &  0.004021 &  0.003677 &  0.004768 & -0.001153 & -0.001672 & -0.004330 & -0.000648 &  0.000168\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002136 & -0.002308 & -0.000544 & -0.005749 &  0.000390 &  0.001315 &  0.004803 & -0.000593 &  0.008240 &  0.000856\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.001023 &  0.002226 &  0.003578 &  0.005356 &  0.004047 & -0.000945 &  0.000290 & -0.003910 &  0.002016 & -0.000914\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.000234 & -0.001267 & -0.001380 & -0.000499 &  0.003063 &  0.002371 &  0.000489 &  0.003508 & -0.001062 & -0.002787\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.002131 & -0.002415 & -0.000894 & -0.005707 &  0.000339 &  0.001283 &  0.004761 & -0.000663 &  0.008217 &  0.000826\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.005180 & -0.001430 &  0.000865 & -0.003817 &  0.000124 & -0.000494 & -0.002331 &  0.001020 & -0.004375 &  0.000978\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ & -0.000216 &  0.003359 &  0.000936 & -0.002672 & -0.006420 & -0.003790 &  0.003767 &  0.000622 &  0.001378 &  0.000960\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.000403 & -0.001416 &  0.001929 &  0.002844 & -0.001254 & -0.001178 & -0.001458 & -0.001413 & -0.002264 &  0.001591\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.000155 & -0.000913 &  0.001951 &  0.003183 & -0.001164 & -0.001325 & -0.001625 & -0.001315 & -0.003014 &  0.000762\\\\\n",
       "\t 0 &  0.00000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 & 0.000000 &  0.000000 &  0.000000 &  0.000000 & ⋯ &  0.005592 & -0.000209 & -0.000628 & -0.004028 & -0.000814 &  0.000950 & -0.005909 & -0.001210 & -0.006847 &  0.000751\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋱ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.010933 &  0.935059 &  0.032623 & -0.011162 &  0.104675 & -0.030396 & -0.154907 & -0.013107 & -0.023239 & -0.012566\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.057037 & -0.158325 &  0.105835 &  0.063477 & -0.153931 & -0.095703 &  0.898926 &  0.065063 &  0.135254 & -0.050659\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.025223 & -0.113770 &  0.078979 &  0.086853 &  0.001963 & -0.071899 & -0.381592 & -0.027603 & -0.062042 & -0.037872\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.054840 & -0.166260 & -0.126831 & -0.037689 & -0.135986 &  0.156860 &  0.876465 &  0.062195 &  0.129639 & -0.052582\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.004322 & -0.010605 &  0.007133 & -0.003828 &  0.052856 & -0.007687 & -0.040649 &  0.006851 & -0.004059 & -0.003998\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.011734 &  0.842773 &  0.035400 & -0.012283 &  0.133423 & -0.033752 & -0.172974 & -0.014435 & -0.025391 & -0.015839\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.003172 & -0.010902 &  0.006474 & -0.001335 &  0.051117 & -0.005539 & -0.040802 & -0.004128 & -0.006432 & -0.002035\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.003738 & -0.014763 &  0.010216 & -0.003300 & -0.084534 & -0.009338 &  0.092957 &  0.004913 &  0.675781 & -0.003143\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.002186 &  0.008362 &  0.384033 &  0.004158 &  0.073364 &  0.007736 &  0.042145 &  0.003010 &  0.006969 &  0.002356\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.003878 & -0.014587 &  0.010170 & -0.005234 &  0.056854 & -0.008888 & -0.056671 & -0.000908 & -0.008812 & -0.005516\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.002249 &  0.007401 &  0.381592 &  0.004272 &  0.072266 &  0.007229 &  0.041870 &  0.003029 &  0.007084 &  0.001984\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.002703 & -0.009216 &  0.006172 & -0.002552 & -0.041565 & -0.005405 & -0.034760 & -0.000909 & -0.005432 & -0.003893\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.016525 & -0.076233 &  0.054291 & -0.017334 &  0.050049 & -0.049622 & -0.269775 & -0.020203 & -0.042297 & -0.025299\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.002586 & -0.012581 &  0.007622 &  0.000579 &  0.042023 & -0.006882 & -0.031128 &  0.707520 & -0.005772 & -0.004761\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.027283 & -0.132446 &  0.091125 & -0.032623 &  0.099792 & -0.081055 & -0.474365 & -0.032196 & -0.063904 &  0.101624\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.002802 &  0.007664 &  0.375000 &  0.004139 &  0.070801 &  0.008072 &  0.041687 &  0.002821 &  0.006973 &  0.002094\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.002758 &  0.007797 &  0.378418 &  0.004131 &  0.070251 &  0.008003 &  0.041565 &  0.002779 &  0.006931 &  0.002037\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.001348 & -0.004429 & -0.008324 & -0.001637 &  0.007542 & -0.003332 & -0.017395 &  0.000688 & -0.002024 & -0.002193\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & -0.001383 & -0.004250 & -0.009079 & -0.001659 &  0.007153 & -0.003307 & -0.017456 & -0.001461 & -0.002131 & -0.002245\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.002775 &  0.008362 &  0.379883 &  0.004131 &  0.070801 &  0.007988 &  0.041656 &  0.002731 &  0.006882 &  0.002134\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  1.000000 & -0.010612 &  0.004982 & -0.002924 &  0.026581 & -0.004398 &  0.058655 & -0.002159 & -0.004002 & -0.001085\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  1.000000 &  0.030930 & -0.010536 &  0.092468 & -0.029221 & -0.146484 & -0.012421 & -0.021881 & -0.012154\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  1.000000 &  0.006088 & -0.029404 & -0.904297 &  0.103882 &  0.006908 &  0.015060 &  0.007858\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  1.000000 &  0.013580 & -0.005875 &  0.067444 & -0.000102 & -0.004917 & -0.003098\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 &  0.063843 & -0.201904 &  0.048065 & -0.076355 &  0.043365\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 & -0.094421 & -0.006546 & -0.013313 & -0.007534\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 & -0.036224 &  0.138550 & -0.048431\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 & -0.000973 & -0.003410\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000 & -0.005539\\\\\n",
       "\t 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  0.000000 &  1.000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 17716 × 17716\n",
       "\n",
       "| V1 &lt;dbl&gt; | V2 &lt;dbl&gt; | V3 &lt;dbl&gt; | V4 &lt;dbl&gt; | V5 &lt;dbl&gt; | V6 &lt;dbl&gt; | V7 &lt;dbl&gt; | V8 &lt;dbl&gt; | V9 &lt;dbl&gt; | V10 &lt;dbl&gt; | ⋯ ⋯ | V17707 &lt;dbl&gt; | V17708 &lt;dbl&gt; | V17709 &lt;dbl&gt; | V17710 &lt;dbl&gt; | V17711 &lt;dbl&gt; | V17712 &lt;dbl&gt; | V17713 &lt;dbl&gt; | V17714 &lt;dbl&gt; | V17715 &lt;dbl&gt; | V17716 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | -0.03952 |  0.076965 | -0.010262 | -0.011528 | -0.007244 | 0.025620 |  0.349854 | -0.040009 | -0.003162 | ⋯ | -0.000100 |  0.003292 |  0.001616 | -0.002359 |  0.004601 | -0.001753 | -0.000437 |  0.004528 |  0.003477 | -0.001143 |\n",
       "| 0 |  1.00000 | -0.199219 | -0.043427 | -0.050507 | -0.029083 | 0.034241 | -0.014709 |  0.577148 | -0.015778 | ⋯ |  0.001030 | -0.001472 |  0.001519 | -0.004887 |  0.001469 |  0.000762 |  0.002153 |  0.000074 |  0.005013 | -0.001247 |\n",
       "| 0 |  0.00000 |  1.000000 |  0.087280 | -0.149902 | -0.081055 | 0.316406 |  0.036865 | -0.458984 | -0.043335 | ⋯ | -0.003998 |  0.000465 | -0.000174 |  0.003233 |  0.004463 |  0.003841 | -0.005527 | -0.001074 | -0.002356 |  0.002304 |\n",
       "| 0 |  0.00000 |  0.000000 |  1.000000 | -0.013237 | -0.007828 | 0.028839 | -0.003292 | -0.041077 | -0.004299 | ⋯ |  0.000781 |  0.004204 |  0.001773 | -0.002052 | -0.000001 | -0.000024 | -0.002142 | -0.001593 |  0.000161 | -0.000206 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  1.000000 | -0.009552 | 0.030594 | -0.005737 | -0.045593 | -0.001204 | ⋯ |  0.005833 |  0.000477 | -0.001369 |  0.003685 |  0.000412 | -0.002626 |  0.001747 | -0.004787 | -0.004929 | -0.002132 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 | 0.005970 | -0.004547 | -0.024643 | -0.002275 | ⋯ | -0.003075 |  0.001641 |  0.001968 | -0.000655 |  0.002501 | -0.001268 | -0.003843 | -0.001417 | -0.003532 | -0.001376 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 1.000000 |  0.012573 |  0.095459 | -0.129761 | ⋯ |  0.000926 |  0.001922 |  0.002653 |  0.002949 |  0.004833 |  0.002409 | -0.001778 | -0.005009 | -0.001427 |  0.000268 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  1.000000 | -0.016174 | -0.002609 | ⋯ | -0.000933 |  0.003387 |  0.003214 | -0.000992 |  0.002140 | -0.002987 | -0.002409 |  0.003239 |  0.000944 |  0.000175 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  1.000000 | -0.012650 | ⋯ |  0.001830 | -0.002825 | -0.001021 | -0.005630 | -0.000737 |  0.001626 |  0.003086 | -0.000947 |  0.009758 |  0.001248 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  1.000000 | ⋯ |  0.003416 | -0.001208 |  0.002728 |  0.001072 | -0.001986 | -0.002262 | -0.002268 |  0.001803 | -0.001597 | -0.000621 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.000408 |  0.000751 |  0.000011 |  0.001480 | -0.002943 |  0.001328 |  0.000930 | -0.002052 |  0.000879 | -0.005005 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.001634 | -0.002693 | -0.001399 | -0.005199 | -0.000607 |  0.002066 |  0.003286 | -0.001204 |  0.009796 |  0.001442 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.003178 |  0.002230 |  0.003824 |  0.005157 |  0.003963 |  0.000956 | -0.003025 |  0.000073 | -0.005714 |  0.000440 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.002131 |  0.004147 |  0.002497 |  0.000448 | -0.001878 | -0.000697 |  0.000125 | -0.002186 | -0.000058 |  0.000815 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002150 |  0.002237 |  0.004013 |  0.003674 |  0.004833 | -0.001129 | -0.001676 | -0.004311 | -0.000638 |  0.000141 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.003214 |  0.002443 |  0.003559 |  0.005058 |  0.003836 |  0.001032 | -0.003391 |  0.000064 | -0.005825 |  0.000550 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002977 | -0.002562 |  0.003391 | -0.000667 | -0.003534 | -0.003256 |  0.005707 | -0.000011 | -0.000241 |  0.000664 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002012 | -0.002876 | -0.001159 | -0.005474 | -0.000380 |  0.001699 |  0.003292 | -0.001380 |  0.009491 |  0.001838 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.005165 | -0.001267 | -0.007256 | -0.003458 | -0.005848 |  0.007759 |  0.000376 |  0.004910 | -0.001518 | -0.000483 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.000907 |  0.001980 | -0.004456 | -0.002598 |  0.004963 |  0.002831 | -0.003077 |  0.001796 | -0.002714 |  0.000148 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002151 |  0.002262 |  0.004021 |  0.003677 |  0.004768 | -0.001153 | -0.001672 | -0.004330 | -0.000648 |  0.000168 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002136 | -0.002308 | -0.000544 | -0.005749 |  0.000390 |  0.001315 |  0.004803 | -0.000593 |  0.008240 |  0.000856 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.001023 |  0.002226 |  0.003578 |  0.005356 |  0.004047 | -0.000945 |  0.000290 | -0.003910 |  0.002016 | -0.000914 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.000234 | -0.001267 | -0.001380 | -0.000499 |  0.003063 |  0.002371 |  0.000489 |  0.003508 | -0.001062 | -0.002787 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.002131 | -0.002415 | -0.000894 | -0.005707 |  0.000339 |  0.001283 |  0.004761 | -0.000663 |  0.008217 |  0.000826 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.005180 | -0.001430 |  0.000865 | -0.003817 |  0.000124 | -0.000494 | -0.002331 |  0.001020 | -0.004375 |  0.000978 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ | -0.000216 |  0.003359 |  0.000936 | -0.002672 | -0.006420 | -0.003790 |  0.003767 |  0.000622 |  0.001378 |  0.000960 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.000403 | -0.001416 |  0.001929 |  0.002844 | -0.001254 | -0.001178 | -0.001458 | -0.001413 | -0.002264 |  0.001591 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.000155 | -0.000913 |  0.001951 |  0.003183 | -0.001164 | -0.001325 | -0.001625 | -0.001315 | -0.003014 |  0.000762 |\n",
       "| 0 |  0.00000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 | 0.000000 |  0.000000 |  0.000000 |  0.000000 | ⋯ |  0.005592 | -0.000209 | -0.000628 | -0.004028 | -0.000814 |  0.000950 | -0.005909 | -0.001210 | -0.006847 |  0.000751 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋱ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.010933 |  0.935059 |  0.032623 | -0.011162 |  0.104675 | -0.030396 | -0.154907 | -0.013107 | -0.023239 | -0.012566 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.057037 | -0.158325 |  0.105835 |  0.063477 | -0.153931 | -0.095703 |  0.898926 |  0.065063 |  0.135254 | -0.050659 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.025223 | -0.113770 |  0.078979 |  0.086853 |  0.001963 | -0.071899 | -0.381592 | -0.027603 | -0.062042 | -0.037872 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.054840 | -0.166260 | -0.126831 | -0.037689 | -0.135986 |  0.156860 |  0.876465 |  0.062195 |  0.129639 | -0.052582 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.004322 | -0.010605 |  0.007133 | -0.003828 |  0.052856 | -0.007687 | -0.040649 |  0.006851 | -0.004059 | -0.003998 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.011734 |  0.842773 |  0.035400 | -0.012283 |  0.133423 | -0.033752 | -0.172974 | -0.014435 | -0.025391 | -0.015839 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.003172 | -0.010902 |  0.006474 | -0.001335 |  0.051117 | -0.005539 | -0.040802 | -0.004128 | -0.006432 | -0.002035 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.003738 | -0.014763 |  0.010216 | -0.003300 | -0.084534 | -0.009338 |  0.092957 |  0.004913 |  0.675781 | -0.003143 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.002186 |  0.008362 |  0.384033 |  0.004158 |  0.073364 |  0.007736 |  0.042145 |  0.003010 |  0.006969 |  0.002356 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.003878 | -0.014587 |  0.010170 | -0.005234 |  0.056854 | -0.008888 | -0.056671 | -0.000908 | -0.008812 | -0.005516 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.002249 |  0.007401 |  0.381592 |  0.004272 |  0.072266 |  0.007229 |  0.041870 |  0.003029 |  0.007084 |  0.001984 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.002703 | -0.009216 |  0.006172 | -0.002552 | -0.041565 | -0.005405 | -0.034760 | -0.000909 | -0.005432 | -0.003893 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.016525 | -0.076233 |  0.054291 | -0.017334 |  0.050049 | -0.049622 | -0.269775 | -0.020203 | -0.042297 | -0.025299 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.002586 | -0.012581 |  0.007622 |  0.000579 |  0.042023 | -0.006882 | -0.031128 |  0.707520 | -0.005772 | -0.004761 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.027283 | -0.132446 |  0.091125 | -0.032623 |  0.099792 | -0.081055 | -0.474365 | -0.032196 | -0.063904 |  0.101624 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.002802 |  0.007664 |  0.375000 |  0.004139 |  0.070801 |  0.008072 |  0.041687 |  0.002821 |  0.006973 |  0.002094 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.002758 |  0.007797 |  0.378418 |  0.004131 |  0.070251 |  0.008003 |  0.041565 |  0.002779 |  0.006931 |  0.002037 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.001348 | -0.004429 | -0.008324 | -0.001637 |  0.007542 | -0.003332 | -0.017395 |  0.000688 | -0.002024 | -0.002193 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | -0.001383 | -0.004250 | -0.009079 | -0.001659 |  0.007153 | -0.003307 | -0.017456 | -0.001461 | -0.002131 | -0.002245 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.002775 |  0.008362 |  0.379883 |  0.004131 |  0.070801 |  0.007988 |  0.041656 |  0.002731 |  0.006882 |  0.002134 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  1.000000 | -0.010612 |  0.004982 | -0.002924 |  0.026581 | -0.004398 |  0.058655 | -0.002159 | -0.004002 | -0.001085 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  1.000000 |  0.030930 | -0.010536 |  0.092468 | -0.029221 | -0.146484 | -0.012421 | -0.021881 | -0.012154 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  1.000000 |  0.006088 | -0.029404 | -0.904297 |  0.103882 |  0.006908 |  0.015060 |  0.007858 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  1.000000 |  0.013580 | -0.005875 |  0.067444 | -0.000102 | -0.004917 | -0.003098 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 |  0.063843 | -0.201904 |  0.048065 | -0.076355 |  0.043365 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 | -0.094421 | -0.006546 | -0.013313 | -0.007534 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 | -0.036224 |  0.138550 | -0.048431 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 | -0.000973 | -0.003410 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 | -0.005539 |\n",
       "| 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  0.000000 |  1.000000 |\n",
       "\n"
      ],
      "text/plain": [
       "      V1 V2       V3        V4        V5        V6        V7       V8       \n",
       "1     1  -0.03952  0.076965 -0.010262 -0.011528 -0.007244 0.025620  0.349854\n",
       "2     0   1.00000 -0.199219 -0.043427 -0.050507 -0.029083 0.034241 -0.014709\n",
       "3     0   0.00000  1.000000  0.087280 -0.149902 -0.081055 0.316406  0.036865\n",
       "4     0   0.00000  0.000000  1.000000 -0.013237 -0.007828 0.028839 -0.003292\n",
       "5     0   0.00000  0.000000  0.000000  1.000000 -0.009552 0.030594 -0.005737\n",
       "6     0   0.00000  0.000000  0.000000  0.000000  1.000000 0.005970 -0.004547\n",
       "7     0   0.00000  0.000000  0.000000  0.000000  0.000000 1.000000  0.012573\n",
       "8     0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  1.000000\n",
       "9     0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "10    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "11    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "12    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "13    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "14    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "15    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "16    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "17    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "18    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "19    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "20    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "21    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "22    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "23    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "24    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "25    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "26    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "27    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "28    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "29    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "30    0   0.00000  0.000000  0.000000  0.000000  0.000000 0.000000  0.000000\n",
       "⋮     ⋮  ⋮        ⋮         ⋮         ⋮         ⋮         ⋮        ⋮        \n",
       "17687 0  0        0         0         0         0         0        0        \n",
       "17688 0  0        0         0         0         0         0        0        \n",
       "17689 0  0        0         0         0         0         0        0        \n",
       "17690 0  0        0         0         0         0         0        0        \n",
       "17691 0  0        0         0         0         0         0        0        \n",
       "17692 0  0        0         0         0         0         0        0        \n",
       "17693 0  0        0         0         0         0         0        0        \n",
       "17694 0  0        0         0         0         0         0        0        \n",
       "17695 0  0        0         0         0         0         0        0        \n",
       "17696 0  0        0         0         0         0         0        0        \n",
       "17697 0  0        0         0         0         0         0        0        \n",
       "17698 0  0        0         0         0         0         0        0        \n",
       "17699 0  0        0         0         0         0         0        0        \n",
       "17700 0  0        0         0         0         0         0        0        \n",
       "17701 0  0        0         0         0         0         0        0        \n",
       "17702 0  0        0         0         0         0         0        0        \n",
       "17703 0  0        0         0         0         0         0        0        \n",
       "17704 0  0        0         0         0         0         0        0        \n",
       "17705 0  0        0         0         0         0         0        0        \n",
       "17706 0  0        0         0         0         0         0        0        \n",
       "17707 0  0        0         0         0         0         0        0        \n",
       "17708 0  0        0         0         0         0         0        0        \n",
       "17709 0  0        0         0         0         0         0        0        \n",
       "17710 0  0        0         0         0         0         0        0        \n",
       "17711 0  0        0         0         0         0         0        0        \n",
       "17712 0  0        0         0         0         0         0        0        \n",
       "17713 0  0        0         0         0         0         0        0        \n",
       "17714 0  0        0         0         0         0         0        0        \n",
       "17715 0  0        0         0         0         0         0        0        \n",
       "17716 0  0        0         0         0         0         0        0        \n",
       "      V9        V10       ⋯ V17707    V17708    V17709    V17710    V17711   \n",
       "1     -0.040009 -0.003162 ⋯ -0.000100  0.003292  0.001616 -0.002359  0.004601\n",
       "2      0.577148 -0.015778 ⋯  0.001030 -0.001472  0.001519 -0.004887  0.001469\n",
       "3     -0.458984 -0.043335 ⋯ -0.003998  0.000465 -0.000174  0.003233  0.004463\n",
       "4     -0.041077 -0.004299 ⋯  0.000781  0.004204  0.001773 -0.002052 -0.000001\n",
       "5     -0.045593 -0.001204 ⋯  0.005833  0.000477 -0.001369  0.003685  0.000412\n",
       "6     -0.024643 -0.002275 ⋯ -0.003075  0.001641  0.001968 -0.000655  0.002501\n",
       "7      0.095459 -0.129761 ⋯  0.000926  0.001922  0.002653  0.002949  0.004833\n",
       "8     -0.016174 -0.002609 ⋯ -0.000933  0.003387  0.003214 -0.000992  0.002140\n",
       "9      1.000000 -0.012650 ⋯  0.001830 -0.002825 -0.001021 -0.005630 -0.000737\n",
       "10     0.000000  1.000000 ⋯  0.003416 -0.001208  0.002728  0.001072 -0.001986\n",
       "11     0.000000  0.000000 ⋯  0.000408  0.000751  0.000011  0.001480 -0.002943\n",
       "12     0.000000  0.000000 ⋯  0.001634 -0.002693 -0.001399 -0.005199 -0.000607\n",
       "13     0.000000  0.000000 ⋯ -0.003178  0.002230  0.003824  0.005157  0.003963\n",
       "14     0.000000  0.000000 ⋯ -0.002131  0.004147  0.002497  0.000448 -0.001878\n",
       "15     0.000000  0.000000 ⋯  0.002150  0.002237  0.004013  0.003674  0.004833\n",
       "16     0.000000  0.000000 ⋯ -0.003214  0.002443  0.003559  0.005058  0.003836\n",
       "17     0.000000  0.000000 ⋯  0.002977 -0.002562  0.003391 -0.000667 -0.003534\n",
       "18     0.000000  0.000000 ⋯  0.002012 -0.002876 -0.001159 -0.005474 -0.000380\n",
       "19     0.000000  0.000000 ⋯ -0.005165 -0.001267 -0.007256 -0.003458 -0.005848\n",
       "20     0.000000  0.000000 ⋯  0.000907  0.001980 -0.004456 -0.002598  0.004963\n",
       "21     0.000000  0.000000 ⋯  0.002151  0.002262  0.004021  0.003677  0.004768\n",
       "22     0.000000  0.000000 ⋯  0.002136 -0.002308 -0.000544 -0.005749  0.000390\n",
       "23     0.000000  0.000000 ⋯ -0.001023  0.002226  0.003578  0.005356  0.004047\n",
       "24     0.000000  0.000000 ⋯ -0.000234 -0.001267 -0.001380 -0.000499  0.003063\n",
       "25     0.000000  0.000000 ⋯  0.002131 -0.002415 -0.000894 -0.005707  0.000339\n",
       "26     0.000000  0.000000 ⋯  0.005180 -0.001430  0.000865 -0.003817  0.000124\n",
       "27     0.000000  0.000000 ⋯ -0.000216  0.003359  0.000936 -0.002672 -0.006420\n",
       "28     0.000000  0.000000 ⋯  0.000403 -0.001416  0.001929  0.002844 -0.001254\n",
       "29     0.000000  0.000000 ⋯  0.000155 -0.000913  0.001951  0.003183 -0.001164\n",
       "30     0.000000  0.000000 ⋯  0.005592 -0.000209 -0.000628 -0.004028 -0.000814\n",
       "⋮     ⋮         ⋮         ⋱ ⋮         ⋮         ⋮         ⋮         ⋮        \n",
       "17687 0         0         ⋯ -0.010933  0.935059  0.032623 -0.011162  0.104675\n",
       "17688 0         0         ⋯  0.057037 -0.158325  0.105835  0.063477 -0.153931\n",
       "17689 0         0         ⋯ -0.025223 -0.113770  0.078979  0.086853  0.001963\n",
       "17690 0         0         ⋯  0.054840 -0.166260 -0.126831 -0.037689 -0.135986\n",
       "17691 0         0         ⋯ -0.004322 -0.010605  0.007133 -0.003828  0.052856\n",
       "17692 0         0         ⋯ -0.011734  0.842773  0.035400 -0.012283  0.133423\n",
       "17693 0         0         ⋯ -0.003172 -0.010902  0.006474 -0.001335  0.051117\n",
       "17694 0         0         ⋯ -0.003738 -0.014763  0.010216 -0.003300 -0.084534\n",
       "17695 0         0         ⋯  0.002186  0.008362  0.384033  0.004158  0.073364\n",
       "17696 0         0         ⋯ -0.003878 -0.014587  0.010170 -0.005234  0.056854\n",
       "17697 0         0         ⋯  0.002249  0.007401  0.381592  0.004272  0.072266\n",
       "17698 0         0         ⋯ -0.002703 -0.009216  0.006172 -0.002552 -0.041565\n",
       "17699 0         0         ⋯ -0.016525 -0.076233  0.054291 -0.017334  0.050049\n",
       "17700 0         0         ⋯ -0.002586 -0.012581  0.007622  0.000579  0.042023\n",
       "17701 0         0         ⋯ -0.027283 -0.132446  0.091125 -0.032623  0.099792\n",
       "17702 0         0         ⋯  0.002802  0.007664  0.375000  0.004139  0.070801\n",
       "17703 0         0         ⋯  0.002758  0.007797  0.378418  0.004131  0.070251\n",
       "17704 0         0         ⋯ -0.001348 -0.004429 -0.008324 -0.001637  0.007542\n",
       "17705 0         0         ⋯ -0.001383 -0.004250 -0.009079 -0.001659  0.007153\n",
       "17706 0         0         ⋯  0.002775  0.008362  0.379883  0.004131  0.070801\n",
       "17707 0         0         ⋯  1.000000 -0.010612  0.004982 -0.002924  0.026581\n",
       "17708 0         0         ⋯  0.000000  1.000000  0.030930 -0.010536  0.092468\n",
       "17709 0         0         ⋯  0.000000  0.000000  1.000000  0.006088 -0.029404\n",
       "17710 0         0         ⋯  0.000000  0.000000  0.000000  1.000000  0.013580\n",
       "17711 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  1.000000\n",
       "17712 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  0.000000\n",
       "17713 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  0.000000\n",
       "17714 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  0.000000\n",
       "17715 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  0.000000\n",
       "17716 0         0         ⋯  0.000000  0.000000  0.000000  0.000000  0.000000\n",
       "      V17712    V17713    V17714    V17715    V17716   \n",
       "1     -0.001753 -0.000437  0.004528  0.003477 -0.001143\n",
       "2      0.000762  0.002153  0.000074  0.005013 -0.001247\n",
       "3      0.003841 -0.005527 -0.001074 -0.002356  0.002304\n",
       "4     -0.000024 -0.002142 -0.001593  0.000161 -0.000206\n",
       "5     -0.002626  0.001747 -0.004787 -0.004929 -0.002132\n",
       "6     -0.001268 -0.003843 -0.001417 -0.003532 -0.001376\n",
       "7      0.002409 -0.001778 -0.005009 -0.001427  0.000268\n",
       "8     -0.002987 -0.002409  0.003239  0.000944  0.000175\n",
       "9      0.001626  0.003086 -0.000947  0.009758  0.001248\n",
       "10    -0.002262 -0.002268  0.001803 -0.001597 -0.000621\n",
       "11     0.001328  0.000930 -0.002052  0.000879 -0.005005\n",
       "12     0.002066  0.003286 -0.001204  0.009796  0.001442\n",
       "13     0.000956 -0.003025  0.000073 -0.005714  0.000440\n",
       "14    -0.000697  0.000125 -0.002186 -0.000058  0.000815\n",
       "15    -0.001129 -0.001676 -0.004311 -0.000638  0.000141\n",
       "16     0.001032 -0.003391  0.000064 -0.005825  0.000550\n",
       "17    -0.003256  0.005707 -0.000011 -0.000241  0.000664\n",
       "18     0.001699  0.003292 -0.001380  0.009491  0.001838\n",
       "19     0.007759  0.000376  0.004910 -0.001518 -0.000483\n",
       "20     0.002831 -0.003077  0.001796 -0.002714  0.000148\n",
       "21    -0.001153 -0.001672 -0.004330 -0.000648  0.000168\n",
       "22     0.001315  0.004803 -0.000593  0.008240  0.000856\n",
       "23    -0.000945  0.000290 -0.003910  0.002016 -0.000914\n",
       "24     0.002371  0.000489  0.003508 -0.001062 -0.002787\n",
       "25     0.001283  0.004761 -0.000663  0.008217  0.000826\n",
       "26    -0.000494 -0.002331  0.001020 -0.004375  0.000978\n",
       "27    -0.003790  0.003767  0.000622  0.001378  0.000960\n",
       "28    -0.001178 -0.001458 -0.001413 -0.002264  0.001591\n",
       "29    -0.001325 -0.001625 -0.001315 -0.003014  0.000762\n",
       "30     0.000950 -0.005909 -0.001210 -0.006847  0.000751\n",
       "⋮     ⋮         ⋮         ⋮         ⋮         ⋮        \n",
       "17687 -0.030396 -0.154907 -0.013107 -0.023239 -0.012566\n",
       "17688 -0.095703  0.898926  0.065063  0.135254 -0.050659\n",
       "17689 -0.071899 -0.381592 -0.027603 -0.062042 -0.037872\n",
       "17690  0.156860  0.876465  0.062195  0.129639 -0.052582\n",
       "17691 -0.007687 -0.040649  0.006851 -0.004059 -0.003998\n",
       "17692 -0.033752 -0.172974 -0.014435 -0.025391 -0.015839\n",
       "17693 -0.005539 -0.040802 -0.004128 -0.006432 -0.002035\n",
       "17694 -0.009338  0.092957  0.004913  0.675781 -0.003143\n",
       "17695  0.007736  0.042145  0.003010  0.006969  0.002356\n",
       "17696 -0.008888 -0.056671 -0.000908 -0.008812 -0.005516\n",
       "17697  0.007229  0.041870  0.003029  0.007084  0.001984\n",
       "17698 -0.005405 -0.034760 -0.000909 -0.005432 -0.003893\n",
       "17699 -0.049622 -0.269775 -0.020203 -0.042297 -0.025299\n",
       "17700 -0.006882 -0.031128  0.707520 -0.005772 -0.004761\n",
       "17701 -0.081055 -0.474365 -0.032196 -0.063904  0.101624\n",
       "17702  0.008072  0.041687  0.002821  0.006973  0.002094\n",
       "17703  0.008003  0.041565  0.002779  0.006931  0.002037\n",
       "17704 -0.003332 -0.017395  0.000688 -0.002024 -0.002193\n",
       "17705 -0.003307 -0.017456 -0.001461 -0.002131 -0.002245\n",
       "17706  0.007988  0.041656  0.002731  0.006882  0.002134\n",
       "17707 -0.004398  0.058655 -0.002159 -0.004002 -0.001085\n",
       "17708 -0.029221 -0.146484 -0.012421 -0.021881 -0.012154\n",
       "17709 -0.904297  0.103882  0.006908  0.015060  0.007858\n",
       "17710 -0.005875  0.067444 -0.000102 -0.004917 -0.003098\n",
       "17711  0.063843 -0.201904  0.048065 -0.076355  0.043365\n",
       "17712  1.000000 -0.094421 -0.006546 -0.013313 -0.007534\n",
       "17713  0.000000  1.000000 -0.036224  0.138550 -0.048431\n",
       "17714  0.000000  0.000000  1.000000 -0.000973 -0.003410\n",
       "17715  0.000000  0.000000  0.000000  1.000000 -0.005539\n",
       "17716  0.000000  0.000000  0.000000  0.000000  1.000000"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "dc4dcc95-763b-4b64-abbb-7c2519fdce03",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in fread(temp_file, header = TRUE, sep = \"\\t\"): could not find function \"fread\"\n",
     "output_type": "error",
     "traceback": [
      "Error in fread(temp_file, header = TRUE, sep = \"\\t\"): could not find function \"fread\"\nTraceback:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "xzdec: /home/dmc2245/susie_mwe/01_159913048_162346721_hg19_test.xz: No such file or directory\n"
     ]
    }
   ],
   "source": [
    " temp_file <- tempfile()\n",
    "  system(paste(\"xzdec\", \"/home/dmc2245/susie_mwe/01_159913048_162346721_hg19_test.xz\", \">\", temp_file))\n",
    "  LD.matrix <- fread(temp_file, header = TRUE, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a50c7c30-fb3a-4050-8bd9-a5f7f0f293c4",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     {
      "name": "ipython",
      "version": 3
     }
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.24.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
